--- 
title: "Epiregulon documentation"
author: "Xiaosai Yao, Tomasz WÅ‚odarczyk"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography:
- book.bib
- packages.bib
description: "This book presents the Epiregulon package. We provide the tutorials
  showing different\nscenarions in which the package can be usd to analyse single-cell
  data. \n"
link-citations: yes
github-repo: "rstudio/bookdown-demo"
---

# Introduction {-}

Placeholder



<!--chapter:end:index.Rmd-->

# Installation

All the epiregulon components are available on [github](https://github.com/xiaosaiyao?tab=repositories). 
Apart of the basic version the user may also choose to use `epiregulon.archr`, which allows
for the seamless integration with `ArchR` package through accepting its output to be used
in the downstream workflow. 

```{r installation, message=FALSE, eval=FALSE}
# install devtools
if(!require(devtools)) install.packages("devtools")

# install basic epiregulon package
devtools::install_github(repo='xiaosaiyao/epiregulon')

# install extended version of epiregulon
devtools::install_github(repo='xiaosaiyao/epiregulon.archr')
```
 
Moreover, we provide a suite of tools for the enrichment analysis, visualization, and network 
analysis which can be run on the `epireglon` or `epiregulon.archr` output.

```{r message=FALSE, eval=FALSE}
# install extended version of epiregulon
devtools::install_github(repo='xiaosaiyao/epiregulon.extra')
```

<!--chapter:end:installation.Rmd-->


# Multiome - ArchR workflow

Placeholder


## Data preparation
## Quick start
### Retrieve bulk TF ChIP-seq binding sites 
### Link ATAC-seq peaks to target genes
### Add TF motif binding to peaks
### Generate regulons
### Network pruning (highly recommended)
### Add Weights
### (Optional) Annotate with TF motifs
### Calculate TF activity 
### Perform differential activity
### Visualize the results
### Geneset enrichment
### Network analysis
### Differential networks
## Session Info

<!--chapter:end:multiome.mae.Rmd-->


# Hematopoeisis tutorial - MAE

Placeholder


## Data preparation
## Quick start
### Retrieve bulk TF ChIP-seq binding sites 
### Link ATACseq peaks to target genes
### Add TF motif binding to peaks
### Generate regulons
### Prune network
### Add Weights
### Calculate TF activity 
### Differential TF activity test
### Visualizing TF activities
### Geneset enrichment
## Differential Network analysis
## Session Info

<!--chapter:end:hematopoeisis.mae.Rmd-->


# Prostate cancer cells

Placeholder


## Data preparation
## Quick start
### Retrieve bulk TF ChIP-seq binding sites 
### Link ATAC-seq peaks to target genes
### Add TF motif binding to peaks
### Generate regulons
### Calculate TF activity 
### Perform differential activity
### Visualize the results
### Geneset enrichment
### Network analysis
## Session Info

<!--chapter:end:prostate.Rmd-->

# Single modality: gene expression

Epiregulon also supports transcription factor activity inference when users only have scRNA-seq. After all, multiome or scATAC-seq data is still relatively rare. To enable TF activity inference on scRNA-seq, users can supply a pre-constructed gene regulatory network. [Dorothea](https://saezlab.github.io/dorothea/articles/dorothea.html) provides both human and mouse pre-constructed gene regulatory networks based on curated experimental and computational data. In this vignette, we bypass the regulon construction step and go straight to calculate TF activity from a Dorothea GRN.


## Load regulon

Dorothea assigns confidence level to its regulons with A being the most confident (i.e. supported by multiple lines of evidence) and E being the least confident. 

```{r load_regulon_singlemodality}
library(dorothea)
data(dorothea_mm, package = "dorothea")
regulon <- dorothea_mm 

#known tfs
genes_to_plot <- c("Foxa1", "Neurod1","Pdx1","Arx")
```


## Load scRNA-seq data

We download the raw counts of a mouse pancreas data set from [scRNAseq](https://bioconductor.org/packages/release/data/experiment/html/scRNAseq.html). We add normalized logcounts, perform dimension reduction and visualize the embeddings using [scater](https://bioconductor.org/packages/release/bioc/html/scater.html). 

```{r load_scRNAseq_singlemodality, message = FALSE}

library(scRNAseq)
library(scater)

sce <- BaronPancreasData('mouse')
sce <- logNormCounts(sce)
sce <- runPCA(sce)
sce <- runUMAP(sce)

plotUMAP(sce, colour_by = "label", text_by = "label")

```

## Calculate activity

Even though Dorothea provides weights under the mor column, we achieved superior performance if we recompute the weights based on the correlation between tf and target gene expression based on our own data. We performed 2 steps, the first step is to add weights to the Dorothea regulons and the second step is to estimate the TF activity by taking the weighted average of the target gene expression.

``` {r activity_singlemodality, message=FALSE, warning=FALSE}

library(epiregulon)


#Add weights to regulon
regulon.ms <- addWeights(regulon = regulon,
                         expMatrix = sce,
                         clusters = sce$label,
                         BPPARAM = BiocParallel::MulticoreParam(),
                         method="corr")

#Calculate activity
score.combine <- calculateActivity(sce, 
                                   regulon = regulon.ms, 
                                   mode = "weight", 
                                   method = "weightedMean")

```

## Perform differential activity
```{r differential_singlemodality}

markers <- findDifferentialActivity(activity_matrix = score.combine, 
                                    clusters = sce$label, 
                                    pval.type = "some", 
                                    direction = "up", 
                                    test.type = "t")
```

Take the top TFs
```{r message=FALSE}
markers.sig <- getSigGenes(markers, topgenes = 5 )

```


## Visualize activity

Finally we visualize the TF activity by either UMAP, violin plots or bubble plots.
We confirm the activity of known lineage factors Pdx1 and Neurod1 in beta cells, 
Arx in alpha cells and Foxa1 in ductal cells.

```{r visualization_singlemodality}

# plot umap
plotActivityDim(sce = sce, 
                activity_matrix = score.combine,
                tf = genes_to_plot, 
                legend.label = "score",
                point_size = 0.1,
                dimtype = "UMAP", 
                label = "label", 
                combine = TRUE,
                text_size = 2)

# plot violin plot
plotActivityViolin(score.combine, 
                   tf = genes_to_plot,
                   clusters = sce$label)

# plot bubble plot
plotBubble(score.combine, 
           tf = genes_to_plot, 
           clusters = sce$label)
```
Plot bubble plot of differential TFs

```{r, fig.height=10}
plotBubble(score.combine, 
           tf = markers.sig$tf, 
           clusters = sce$label)
```
We can adapt the epiregulon package to plot gene expression. When compared against TF activity, gene expression of Foxa1 and Arx has noisy signals and high dropout rates. 
Epiregulon enhances the signal to noise ratio of TF activity and better resolves lineage differences.

```{r scRNAseq}

# plot umap
plotActivityDim(sce = sce, 
                activity_matrix = logcounts(sce),
                tf = genes_to_plot, 
                legend.label = "score",
                point_size = 0.1,
                dimtype = "UMAP", 
                label = "label", 
                combine = TRUE,
                text_size = 2,
                colors = c("gray","blue"),
                limit = c(0,2))

# plot violin plot
plotActivityViolin(logcounts(sce), 
                   tf = genes_to_plot,
                   clusters = sce$label)
```

```{r, fig.height=10}
# plot Bubble plot
plotBubble(logcounts(sce), 
           tf = markers.sig$tf, 
           clusters = sce$label)
```

We can visualize the target genes for transcription factors of interest
```{r, fig.width=12}
plotHeatmapRegulon(sce=sce,
                   tfs=genes_to_plot,
                   regulon=regulon.ms,
                   regulon_cutoff=0.5,
                   downsample=1000,
                   cell_attributes="label",
                   col_gap="label",
                   exprs_values="logcounts",
                   name="regulon heatmap",
                   column_title_rot = 45)

plotHeatmapActivity(activity_matrix = score.combine,
                    sce=sce,
                    tfs=genes_to_plot,
                    downsample=1000,
                    cell_attributes="label",
                    col_gap="label",
                    name="regulon heatmap",
                    column_title_rot = 45)
```
## Pathway enrichment

Sometimes it is useful to understand what pathways are enriched in the regulons. We take the highly correlated target genes of a regulon and perform geneset enrichment using the enricher function from [clusterProfiler](http://yulab-smu.top/biomedical-knowledge-mining-book/universal-api.html).

```{r enrichment_singlemodality, warning=FALSE, fig.height = 14}
#retrieve genesets
H <- EnrichmentBrowser::getGenesets(org = "mmu", 
                                    db = "msigdb", 
                                    cat = "H", 
                                    gene.id.type = "SYMBOL", 
                                    cache = FALSE)
C6 <- EnrichmentBrowser::getGenesets(org = "mmu", 
                                     db = "msigdb", 
                                     cat = "C6", 
                                     gene.id.type = "SYMBOL",  
                                     cache = FALSE)

#combine genesets and convert genesets to be compatible with enricher
gs <- c(H,C6)
gs.list <- do.call(rbind,lapply(names(gs), function(x) 
  {data.frame(gs = x, genes = gs[[x]])}))

enrichresults <- regulonEnrich(genes_to_plot, 
                               regulon = regulon.ms, 
                               weight = "weight",
                               weight_cutoff = 0.5, 
                               genesets = gs.list)

#plot results
enrichPlot(results = enrichresults, ncol = 1)

```

## Session Info

```{r}
sessionInfo()
```

<!--chapter:end:scRNAseq.only.Rmd-->

