```{r echo=FALSE}
library(BiocStyle)
```
# Quick start

In this chapter, we illustrate the main steps of the epiregulon workflow. 

## Prepare data

We need the following components to perform GRN analysis: <br>
1. Peak matrix from scATAC-seq <br>
2. Paired gene expression matrix from scRNA-seq <br>
3. Dimensionality reduction matrix <br>


We download an example PBMC dataset from the `r Biocpkg("scMultiome")` package.  
```{r, message=FALSE}
# load the MAE object
library(scMultiome)
library(epiregulon)

# Download the example dataset
mae <- scMultiome::PBMC_10x()

# Load peak matrix
PeakMatrix <- mae[["PeakMatrix"]]

# Load expression matrix
GeneExpressionMatrix <- mae[["GeneExpressionMatrix"]]
```
Visualize singleCellExperiment by UMAP

```{r, warning=FALSE, message=FALSE}
scater::plotReducedDim(GeneExpressionMatrix, 
                       dimred = "UMAP_RNA", 
                       text_by = "cell_type", 
                       colour_by = "cell_type",
                       point_size = 0.3,
                       point_alpha = 0.3)
```

## Retrieve bulk TF ChIP-seq binding sites 

We retrieve a `GRangesList` object containing the binding sites of transcription factors and co-regulators. These binding sites are derived from bulk ChIP-seq data in the [ChIP-Atlas](https://chip-atlas.org/) and [ENCODE](https://www.encodeproject.org/) databases. For further information on how these peaks are derived, please refer to `?epiregulon::getTFMotifInfo`. Currently, only human genomes hg19 and hg38 and mouse mm10 are supported. 

```{r getTFMotifInfo, warning = FALSE, message = FALSE}
grl <- getTFMotifInfo(genome = "hg38")
grl
```

## Link ATAC-seq peaks to target genes

Next, we link ATAC-seq peaks to their putative target genes. We assign a peak to genes within a size window (default Â±250kb) if the chromatin accessibility of the peak and expression of the target genes are highly correlated. To compute correlation, we first create cell aggregates by performing k-means clustering on the reduced dimensionality matrix. To estimate the optimal number of cells per cluster we run `optimizeMetacellNumber`.

```{r, warning=FALSE, message=FALSE}
set.seed(1010)
cellNum <- optimizeMetacellNumber(expMatrix = GeneExpressionMatrix,
                                  peakMatrix = PeakMatrix,
                                  exp_assay = "normalizedCounts",
                                  peak_assay = "counts",
                                  reducedDim = reducedDim(GeneExpressionMatrix, "LSI_RNA"),
                                  BPPARAM = BiocParallel::SerialParam(progressbar = FALSE))

```

`cellNum` object is then provided as an argument to `calculateP2G`.

```{r calculateP2G, message = FALSE, warning=FALSE}
set.seed(1010)
p2g <- calculateP2G(peakMatrix = PeakMatrix, 
                    expMatrix = GeneExpressionMatrix, 
                    reducedDim = reducedDim(GeneExpressionMatrix, "LSI_RNA"),
                    exp_assay = "normalizedCounts",
                    peak_assay = "counts")

p2g
```

## Add TF motif binding to peaks

We next add the TF binding information by overlapping ATAC peaks with ChIP-seq peaks. The output is a data frame object with three columns:

1. `idxATAC` - index in the peak matrix
2. `idxTF` - index in the gene expression matrix corresponding to the transcription factor
3. `tf` - name of the transcription factor


```{r addTFMotifInfo, message = FALSE, warning=FALSE}
overlap <- addTFMotifInfo(grl = grl, p2g = p2g, peakMatrix = PeakMatrix)
head(overlap)
```

## Generate regulons

We then generate a DataFrame object representing the inferred regulons, indicating target genes, transcription factors, and regulatory elements.

```{r, warning=FALSE, getRegulon}
regulon <- getRegulon(p2g = p2g, overlap = overlap)
regulon
```

## Prune network (recommended)

Since our regulons could contain false connections, we apply the pruning function.

```{r message = FALSE}

pruned.regulon <- pruneRegulon(expMatrix = GeneExpressionMatrix,
                               exp_assay = "normalizedCounts",
                               peakMatrix = PeakMatrix,
                               peak_assay = "counts",
                               test = "chi.sq",
                               regulon,
                               prune_value = "pval",
                               regulon_cutoff = 0.05
                               )

pruned.regulon
```

## Annotate regulon

Regulons can be annotated for further refinement. One such metric is the log fold change of gene expression in specified groups. Before running `addLogFC` function, we need to log-transform gene expression counts. In this example, we are interested in target genes differentially expressed in Naive CD4+ T cells.

```{r message = FALSE}

# add 'logcounts' assay
GeneExpressionMatrix <- scuttle::logNormCounts(GeneExpressionMatrix)

regulon.FC <- addLogFC(expMatrix = GeneExpressionMatrix,
                       clusters = GeneExpressionMatrix$cell_type,
                       regulon = pruned.regulon,
                       pval.type = "any",
                       sig_type = "FDR",
                       assay.type = "logcounts",
                       logFC_condition=c('Naive CD4+ T')
)

regulon.FC
```


We can also annotate regulons for the presence of motifs at each regulatory element. 

```{r message = FALSE}
regulon.motif <- addMotifScore(regulon = pruned.regulon,
                                peaks = rowRanges(PeakMatrix),
                                species = "human",
                                genome = "hg38")
regulon.motif
```

## Calculate TF activity 

Finally, activities for a specific TF in each cell are computed by averaging expressions of target genes weighted by the test statistics of choice.
$$y=\frac{1}{n}\sum_{i=1}^{n} x_i * weights_i$$
where $y$ is the activity of a TF for a cell, <br>
$n$ is the total number of targets for a TF,<br>
$x_i$ is the log count expression of target $i$ where $i$ in {1,2,...,n} and<br>
$weights_i$ is the weight of TF - target $i$, calculated by running `addWeights` function.


```{r calculateActivity, message = FALSE}
regulon.w <- addWeights(regulon = pruned.regulon,
                        expMatrix  = GeneExpressionMatrix,
                        exp_assay  = "normalizedCounts",
                        peakMatrix = PeakMatrix,
                        peak_assay = "counts")


score.combine <- calculateActivity(expMatrix = GeneExpressionMatrix,
                                   regulon = regulon.w, 
                                   mode = "weight", 
                                   exp_assay = "normalizedCounts",
                                   normalize = FALSE)

score.combine[1:5,1:5]
```
