```{r echo=FALSE}
library(BiocStyle)
```
# Quick start

In this chapter, we illustrate the epiregulon workflow, using the PBMC dataset which can be downloaded from the `r Biocpkg("scMultiome")` package.  

## Prepare data

Prior to using `epiregulon`, single cell preprocessing needs to be performed by user's favorite methods. The following components are required: <br>
1. Peak matrix from scATAC-seq containing the chromatin accessibility information <br>
2. Gene expression matrix from either paired or unpaired scRNA-seq. RNA-seq integration needs to be performed for unpaired dataset. <br>
3. Dimensionality reduction matrix from with either single modalities or joint scRNA-seq and scATAC-seq <br>

```{r, message=FALSE}
# load the MAE object
library(scMultiome)
library(epiregulon)

# Download the example dataset
mae <- scMultiome::PBMC_10x()

# Load peak matrix
PeakMatrix <- mae[["PeakMatrix"]]

# Load expression matrix
GeneExpressionMatrix <- mae[["GeneExpressionMatrix"]]
```
Visualize singleCellExperiment by UMAP

```{r}
scater::plotReducedDim(GeneExpressionMatrix, 
                       dimred = "UMAP_RNA", 
                       text_by = "cell_type", 
                       colour_by = "cell_type",
                       point_size = 0.3,
                       point_alpha = 0.3)
```

## Retrieve bulk TF ChIP-seq binding sites 

First, we retrieve a `GRangesList` object containing the binding sites of all the transcription factors and co-regulators. These binding sites are derived from bulk ChIP-seq data in the [ChIP-Atlas](https://chip-atlas.org/) and [ENCODE](https://www.encodeproject.org/) databases. For the same transcription factor, multiple ChIP-seq files from different cell lines or tissues are merged. For further information on how these peaks are derived, please refer to `?epiregulon::getTFMotifInfo`. Currently, human genomes hg19 and hg38 and mouse mm10 are supported. 

```{r getTFMotifInfo, warning = FALSE, message = FALSE}
grl <- getTFMotifInfo(genome = "hg38")
grl
```

## Link ATAC-seq peaks to target genes

Next, we try to link ATAC-seq peaks to their putative target genes. We assign a peak to a gene within a size window (default Â±250kb) if the chromatin accessibility of the peak and expression of the target genes are highly correlated (default threshold 0.5). To compute correlations, we first create cell aggregates by performing k-means clustering on the reduced dimensionality matrix. To estimate the optimal number of cells per cluster we will run the function `optimizeMetacellNumber`.

```{r}
set.seed(1010)
cellNum <- optimizeMetacellNumber(expMatrix = GeneExpressionMatrix,
                                  peakMatrix = PeakMatrix,
                                  exp_assay = "normalizedCounts",
                                  peak_assay = "counts",
                                  reducedDim = reducedDim(GeneExpressionMatrix, "LSI_RNA"),
                                  BPPARAM = BiocParallel::SerialParam(progressbar = FALSE))

```

`cellNum` object is now provided as an argument to `calculateP2G`.

```{r calculateP2G}
set.seed(1010)
p2g <- calculateP2G(peakMatrix = PeakMatrix, 
                    expMatrix = GeneExpressionMatrix, 
                    reducedDim = reducedDim(GeneExpressionMatrix, "LSI_RNA"),
                    exp_assay = "normalizedCounts",
                    peak_assay = "counts")

p2g
```

## Add TF motif binding to peaks

The next step is to add the TF binding information by overlapping regions of the peak matrix with the bulk chip-seq database. The output is a data frame object with three columns:

1. `idxATAC` - index of the peak in the peak matrix
2. `idxTF` - index in the gene expression matrix corresponding to the transcription factor
3. `tf` - name of the transcription factor


```{r addTFMotifInfo}
overlap <- addTFMotifInfo(grl = grl, p2g = p2g, peakMatrix = PeakMatrix)
head(overlap)
```

## Generate regulons

A data frame representing the inferred regulons is then generated. The data frame consists of several columns, including those indicating target genes, transcription factors, and regulatory elements indexed by their corresponding rows in the `PeakMatrix`.

```{r, warning=FALSE, getRegulon}
regulon <- getRegulon(p2g = p2g, overlap = overlap, aggregate = FALSE)
regulon
```

## Prune network (recommended)

Since our regulon is quite large, and likely contains many false connections, we will pass it through the pruning function.

```{r results = "hide", message = FALSE}

pruned.regulon <- pruneRegulon(expMatrix = GeneExpressionMatrix,
                               exp_assay = "normalizedCounts",
                               peakMatrix = PeakMatrix,
                               peak_assay = "counts",
                               test = "chi.sq",
                               regulon,
                               prune_value = "pval",
                               regulon_cutoff = 0.05
                               )

pruned.regulon
```

## Annotate regulon

Apart from pruning, the regulon can be annotated with metrics useful for further refinement. One such metric is the log fold change of gene expression in a specified group of cells compared to all other or a selected group. Before running `addLogFC` function we need to log-transform gene expression counts.

```{r results = "hide", message = FALSE}

# add 'logcounts' assay
GeneExpressionMatrix <- scuttle::logNormCounts(GeneExpressionMatrix)

regulon.FC <- addLogFC(expMatrix = GeneExpressionMatrix,
                       clusters = GeneExpressionMatrix$cell_type,
                       regulon = pruned.regulon,
                       pval.type = "any",
                       sig_type = "FDR",
                       assay.type = "logcounts",
                       logFC_condition=c('Naive CD4+ T')
)

regulon.FC
```


Another way to annotate the regulon is to add information about the presence of a motif at the regulatory element. By default, a new column named `motif` will be appended to the regulon, with 1s in rows where a motif has been found that matches the declared transcription factor.

```{r results = "hide", message = FALSE}
regulon.motif <- addMotifScore(regulon = pruned.regulon,
                                peaks = rowRanges(PeakMatrix),
                                species = "human",
                                genome = "hg38")
regulon.motif
```

## Calculate TF activity 

Finally, the activities for a specific TF in each cell are computed by averaging expressions of target genes linked to the TF weighted by the test statistics of choice. Using the `method` argument to the `addWeights` function, the user may choose among correlation, mutual information, or Wilcoxon as the test statistic.
$$y=\frac{1}{n}\sum_{i=1}^{n} x_i * weights_i$$
where $y$ is the activity of a TF for a cell, <br>
$n$ is the total number of targets for a TF,<br>
$x_i$ is the log count expression of target $i$ where $i$ in {1,2,...,n} and<br>
$weights_i$ is the weight of TF - target $i$, calculated by running `addWeights` function.


```{r calculateActivity}
regulon.w <- addWeights(regulon = pruned.regulon,
                        expMatrix  = GeneExpressionMatrix,
                        exp_assay  = "normalizedCounts",
                        peakMatrix = PeakMatrix,
                        peak_assay = "counts")


score.combine <- calculateActivity(expMatrix = GeneExpressionMatrix,
                                   regulon = regulon.w, 
                                   mode = "weight", 
                                   method = "weightedMean", 
                                   exp_assay = "normalizedCounts",
                                   normalize = FALSE)

score.combine[1:5,1:5]
```
