# Basic workflow with MultiAssayExperiment

In this chapter, we illustrate the epiregulon workflow starting from data in the form of `SingleCellExperiment` objects using the correlation weight estimation method.

This is a dataset generated by infecting LNCaP cells with NKX2-1, GATA6 and FOXA2 to examine the effects of these TFs on AR activity using a technique called [reprogram-seq](https://www.sciencedirect.com/science/article/pii/S2211124719307089).

## Data preparation

Single cell preprocessing needs to performed by user's favorite methods prior to using Epiregulon. The following components are required: <br>
1. Peak matrix from scATAC-seq <br>
2. Gene expression matrix from either paired or unpaired scRNA-seq. RNA-seq integration needs to be performed for unpaired dataset. <br>
3. Dimensionality reduction matrix from with either single modalities or joint scRNA-seq and scATAC-seq <br>

In this case, we have preprocessed the data and converted the archR project into a MAE object using the function `epiregulon.archr::archr2MAE` and made the data available through the `r BiocStyle::Biocpkg("scMultiome")` package 


```{r, results = "hide", warning = FALSE, message = FALSE}
# load the MAE object
library(scMultiome)

mae <- scMultiome::reprogramSeq()

# peak matrix
PeakMatrix <- mae[["PeakMatrix"]]

# expression matrix
GeneExpressionMatrix <- mae[["GeneExpressionMatrix"]]
rownames(GeneExpressionMatrix) <- rowData(GeneExpressionMatrix)$name

# dimensional reduction matrix
reducedDimMatrix <- reducedDim(mae[['TileMatrix500']], "LSI_ATAC")

```

Visualize the SingleCellExperiment by UMAP
```{r}
# transfer UMAP_combined from TileMatrix to GeneExpressionMatrix
reducedDim(GeneExpressionMatrix, "UMAP_Combined") <- reducedDim(mae[['TileMatrix500']], "UMAP_Combined")
scater::plotReducedDim(GeneExpressionMatrix, 
                       dimred = "UMAP_Combined", 
                       text_by = "Clusters", 
                       colour_by = "Clusters")

```

## Retrieve bulk TF ChIP-seq binding sites 

First, we retrieve the information of TF binding sites collected from Cistrome and ENCODE ChIP-seq, which are available through the `r BiocStyle::Biocpkg("scMultiome")` package . Currently, human genomes hg19 and hg38 and mouse mm10 are available. 

```{r getTFMotifInfo_multiome}
library(epiregulon)
grl <- getTFMotifInfo(genome = "hg38")
grl
```

## Link ATAC-seq peaks to target genes

Next, we compute peak to gene correlations using a custom algorithm that has similar performance to ArchR's P2G function.

```{r calculateP2G_multiome}
set.seed(1010)
p2g <- calculateP2G(peakMatrix = PeakMatrix, 
                    expMatrix = GeneExpressionMatrix, 
                    reducedDim = reducedDimMatrix,
                    exp_assay = "normalizedCounts")

p2g

```

## Add TF motif binding to peaks

The next step is to add the TF binding information by overlapping regions of the peak matrix with the bulk chip-seq database loaded in 2. The user can supply either an archR project path and this function will retrieve the peak matrix, or a peakMatrix in the form of a Granges object or RangedSummarizedExperiment.

```{r addTFMotifInfo_multiome}

overlap <- addTFMotifInfo(grl = grl, p2g = p2g, peakMatrix = PeakMatrix)
head(overlap)
```

## Generate regulons

A long format data frame, representing the inferred regulons, is then generated. The data frame consists of three important columns:

* tf (transcription factor)
* target gene
* peak to gene correlation between tf and target gene

```{r, warning=FALSE, getRegulon_multiome}

regulon <- getRegulon(p2g = p2g, overlap = overlap, aggregate = FALSE)
regulon

```

## Network pruning (highly recommended)

Epiregulon prunes the network by performing tests of independence on the observed number of cells jointly expressing transcription factor (TF), regulatory element (RE) and target gene (TG) vs the expected number of cells if TF/RE and TG are independently expressed. We implement two tests, the binomial test and the chi-square test. In the binomial test, the expected probability is P(TF, RE) * P(TG), and the number of trials is the total number of cells, and the observed successes is the number of cells jointly expressing all three elements. In the chi-square test, the expected probability for having all 3 elements active is also P(TF, RE) * P(TG) and the probability otherwise is 1- P(TF, RE) * P(TG). The observed cell count for the active category is the number of cells jointly expressing all three elements, and the cell count for the inactive category is n - n_triple.

We calculate cluster-specific p-values if users supply cluster labels. This is useful if we are interested in cluster-specific networks. The pruned regulons can then be used to visualize differential networks for transcription factors of interest. See section on differential networks.

If you supply your cluster label make sure that the gene expression SingleCellExperiment contains matrix representing reduced dimension coordinates for cells. The name of this matrix should be passed in `useDim` parameter. It is also necessary if you want cells to be aggregated into pseudobulks, which can be done by specifying `aggregateCells=TRUE`.  

```{r network_pruning_multiome}

pruned.regulon <- pruneRegulon(expMatrix = GeneExpressionMatrix,
                               exp_assay = "normalizedCounts",
                               peakMatrix = PeakMatrix,
                               peak_assay = "counts",
                               test = "chi.sq",
                               regulon,
                               clusters = GeneExpressionMatrix$Clusters,
                               prune_value = "pval",
                               regulon_cutoff = 0.05
                               )

pruned.regulon
```

## Add Weights

While the `pruneRegulon' function provides statistics on the joint occurrence of TF-RE-TG, we would like to further estimate the strength of regulation. Biologically, this can be interpreted as the magnitude of gene expression changes induced by transcription factor activity. Epiregulon estimates the regulatory potential using one of the three measures: 1) correlation between TG and TF or between TG and the product of TF and RE, 2) mutual information between TG and TF expression or between TG and the product of TF and RE, or 3) Wilcoxon test statistics of target gene expression in cells jointly expressing all 3 elements vs cells that do not.

Two measures (correlation and Wilcoxon) give both the magnitude and directionality of changes whereas mutational information is always positive. The correlation and mutual information statistics are computed on grouped pseudobulks by user-supplied cluster labels and yield a single weight across all clusters per each TF-RE-target triplet. In contrast, the Wilcoxon method group cells based on the joint expression of TF, RE and TG in each single cell or in cell aggregates. If cell labels are provided, we calculate cluster-specific weights in addition to estimating weights from all the cells. Cell aggregation uses a default value of 10 cells and can help overcome sparsity and speed up computation. If cluster labels are provided, we can obtain weights of individual clusters and all cells combined. In this example, we apply Wilcoxon test on cell aggregates of 10 cells. 

In this example, we illustrate weight estimation using the correlation method. We chose correlation as the weight estimation because we reasoned that the increase in TF activity is driven by the overexpression of the TF, and thus its target genes should see a corresponding change in their expression. 

When `tf_re.merge` is set to `TRUE`, we compute weight based on the correlation between TG gene expression and the product of the TF gene expression and RE chromatin accessibility. If `tf_re.merge` is set to `FALSE`, we do not take into account the RE chromatin accessibility and the correlation is computed only based on the correlation between TG gene expression and TF gene expression. Here, we set `tf_re.merge` to `TRUE` because we are interested in target genes whose gene expression and chromatin accessibility at the regulatory elements change concordantly with the expression of their putative transcription factors.

```{r addWeights_multiome, results = "hide", warning = FALSE, message = FALSE}

regulon.w <- addWeights(regulon = pruned.regulon,
                        expMatrix  = GeneExpressionMatrix,
                        exp_assay  = "normalizedCounts",
                        peakMatrix = PeakMatrix,
                        peak_assay = "counts",
                        clusters = GeneExpressionMatrix$Clusters,
                        block_factor = NULL,
                        tf_re.merge = TRUE,
                        method = "corr")

regulon.w

```

## (Optional) Annotate with TF motifs

So far the gene regulatory network was constructed from TF ChIP-seq exclusively. Some users would prefer to further annotate the regulatory elements with the presence of motifs. We provide an option to annotate peaks with motifs from the Cisbp database. If motifs are available for a factor and the RE contains a motif, we return 1. If motifs are available and the RE does not contain a motif, we return 0. If no motifs are known for this particular factor (as in the case of co-factors or chromatin modifiers), we return NA.

In this example, while we illustrate the possibility of annotating regulons with motifs, we contain the remaining workflow with the full regulons `regulon.w`. It is important to note that filtering for the presence of motifs removes a large fraction of the target genes because motifs are often present in a small subset of the ChIP-seq peaks (can be as low as 10%). Second, indirect TF binding, possibly through its interaction partners, may have a true biological function. However, if the user would prefer to retain only target genes with REs containing the motif of the TFs, they should proceed with `regulon.w.motif`.

```{r addMotifScore_multiome}

regulon.w.motif <- addMotifScore(regulon = regulon.w,
                                 peaks = rowRanges(PeakMatrix),
                                 species = "human",
                                 genome = "hg38")

# if desired, set weight to 0 if no motif is found
regulon.w.motif$weight[regulon.w.motif$motif == 0] <- 0
```

## Calculate TF activity 

Finally, the activities for a specific TF in each cell are computed by averaging expressions of target genes linked to the TF weighted by the test statistics of choice, chosen from either correlation, mutual information or Wilcoxon test statistics. 
$$y=\frac{1}{n}\sum_{i=1}^{n} x_i * weights_i$$
where $y$ is the activity of a TF for a cell
$n$ is the total number of targets for a TF
$x_i$ is the log count expression of target i where i in {1,2,...,n}
$weights_i$ is the weight of TF and target i

```{r calculateActivity_multiome, results = "hide"}
score.combine <- calculateActivity(expMatrix = GeneExpressionMatrix,
                                   regulon = regulon.w, 
                                   mode = "weight", 
                                   method = "weightedMean", 
                                   exp_assay = "normalizedCounts",
                                   normalize = FALSE)

```

## Perform differential activity

```{r differential_multiome}
library(epiregulon.extra)
markers <- findDifferentialActivity(activity_matrix = score.combine, 
                                    groups = GeneExpressionMatrix$hash_assignment, 
                                    pval.type = "some", 
                                    direction = "up", 
                                    test.type = "t")
```

Take the top TFs
```{r}
markers.sig <- getSigGenes(markers, topgenes = 5 )

```

## Visualize the results

First visualize the known differential TFs by bubble plot
```{r visualization_multiome}
plotBubble(activity_matrix = score.combine, 
           tf = c("NKX2-1","GATA6","FOXA1","FOXA2", "AR"), 
           clusters = GeneExpressionMatrix$hash_assignment)
```

Then visualize the most differential TFs by clusters
```{r}

plotBubble(activity_matrix = score.combine, 
           tf = markers.sig$tf, 
           clusters = GeneExpressionMatrix$hash_assignment)

```

Visualize the known differential TFs by violin plot. 
```{r}
plotActivityViolin(activity_matrix = score.combine, 
                   tf = c("NKX2-1","GATA6","FOXA1","FOXA2", "AR"), 
                   clusters = GeneExpressionMatrix$hash_assignment)
```

Visualize the known differential TFs by UMAP
```{r, fig.height = 8, fig.width = 12}
plotActivityDim(sce = GeneExpressionMatrix, 
                activity_matrix = score.combine, 
                tf = c("NKX2-1","GATA6","FOXA1","FOXA2", "AR"), 
                dimtype = "UMAP_Combined", 
                label = "Clusters", 
                point_size = 1, 
                ncol = 3)

```

In contrast, the gene expression of the TFs is very sparse
```{r, fig.height = 8, fig.width = 12}

plotActivityDim(sce = GeneExpressionMatrix, 
                activity_matrix = counts(GeneExpressionMatrix), 
                tf = c("NKX2-1","GATA6","FOXA1","FOXA2", "AR"), 
                dimtype = "UMAP_Combined", 
                label = "Clusters", 
                point_size = 1, 
                ncol = 3,
                limit = c(0,2),
                colors = c("grey","blue"),
                legend.label = "GEX")
```

Visualize the gene expression of the regulons by heatmap
```{r}
plotHeatmapRegulon(sce=GeneExpressionMatrix,
                   tfs=c("GATA6","NKX2-1"),
                   regulon=regulon.w,
                   regulon_cutoff=0.1,
                   downsample=1000,
                   cell_attributes="Clusters",
                   col_gap="Clusters",
                   exprs_values="counts",
                   name="regulon heatmap")
```

```{r}
plotHeatmapActivity(activity=score.combine,
                    sce=GeneExpressionMatrix,
                    tfs=rownames(score.combine),
                    downsample=5000,
                    cell_attributes="Clusters",
                    col_gap="Clusters",
                    name = "Activity")
```

## Geneset enrichment

Sometimes we are interested to know what pathways are enriched in the regulon of a particular TF. We can perform geneset enrichment using the enricher function from [clusterProfiler](http://yulab-smu.top/biomedical-knowledge-mining-book/universal-api.html).

```{r enrichment_multiome, fig.height = 10, fig.width = 12}
#retrieve genesets
H <- EnrichmentBrowser::getGenesets(org = "hsa", 
                                    db = "msigdb", 
                                    cat = "H", 
                                    gene.id.type = "SYMBOL" )
C2 <- EnrichmentBrowser::getGenesets(org = "hsa", 
                                     db = "msigdb", 
                                     cat = "C2",
                                     gene.id.type = "SYMBOL" )
C6 <- EnrichmentBrowser::getGenesets(org = "hsa", 
                                     db = "msigdb", 
                                     cat = "C6",
                                     gene.id.type = "SYMBOL" )

#combine genesets and convert genesets to be compatible with enricher
gs <- c(H, C2, C6)
gs.list <- do.call(rbind,lapply(names(gs), function(x) 
  {data.frame(gs=x, genes=gs[[x]])}))

enrichresults <- regulonEnrich(TF = c("GATA6","NKX2-1"), 
                               regulon = regulon.w, 
                               weight = "weight",
                               weight_cutoff = 0, 
                               genesets = gs.list)

#plot results
enrichPlot(results = enrichresults )
```

## Network analysis

We can visualize the genesets as a network
```{r plot_gsea_network_multiome}

plotGseaNetwork(tf = names(enrichresults), 
                enrichresults = enrichresults,
                p.adj_cutoff = 0.1,
                ntop_pathways = 10)


```

## Session Info

```{r}
sessionInfo()
```
