---
output: BiocStyle::html_document
---

# Construction of the gene regulatory network {#construction}
This chapter outlines the basic steps for constructing a gene regulatory network, which in epiregulon takes the form of a tripartite graph: TF → RE → TG. In brief, TF occupancy data—derived from either ChIP-seq or motif information—are used to map transcription factors onto genomic regions (here, ATAC-seq peaks). These regions are then linked to their target genes by leveraging the assumption that chromatin accessibility and gene expression are highly correlated.

## Downloading and formatting data set
We will make use of the PBMC data set from 10x Genomics. The MAE object was uploaded to `r BiocStyle::Biocpkg("scMultiome")` for full reproducibility.


```{r, results = "hide", warning = FALSE, message = FALSE}
# Download the example dataset
mae <- scMultiome::PBMC_10x()

# Load peak matrix
PeakMatrix <- mae[["PeakMatrix"]]

# Load expression matrix
GeneExpressionMatrix <- mae[["GeneExpressionMatrix"]]
```

Visualize the data.
```{r}

scater::plotReducedDim(GeneExpressionMatrix, 
                       dimred = "UMAP_RNA", 
                       text_by = "cell_type", 
                       colour_by = "cell_type",
                       point_size = 0.3,
                       point_alpha = 0.3)
```


## Retrieve TF binding sites
We are now ready to infer the gene regulatory network. We recommend using ChIP-seq data—rather than motifs—as the proxy for TF occupancy. The gold standard in the field is to use sample-matched ChIP-seq for TF binding. To support this, we provide both sample-specific and tissue-specific ChIP-seq peak sets, compiled from [ChIP-Atlas](https://chip-atlas.org/). However, sample-matched or tissue-matched ChIP-seq data are not available for many factors. To address this, we include a set of pan-tissue ChIP-seq peaks, generated by merging data across multiple samples and lineages; this serves as the default TF occupancy source. Benchmarking in our [manuscript](https://www.nature.com/articles/s41467-025-62252-5) demonstrates that pan-tissue ChIP-seq still outperforms motif-based annotations for TF activity inference and target-gene recovery. In addition, ChIP-seq offers a key advantage over motifs in enabling GRN inference for transcriptional co-regulators that lack known motifs.

One limitation of ChIP-seq–based occupancy is that its accuracy depends on the underlying experimental quality. We applied several quality-control filters to the ChIP-seq data included (see Methods in the [manuscript](https://www.nature.com/articles/s41467-025-62252-5)), but some factors still exhibit very large numbers of binding sites (>100k). This may reflect true biology or simply false positives. In such cases, filtering TF occupancy based on the intersection of ChIP-seq peaks and motif sites (see next [chapter](#refinement)) can be helpful.


### ChIP-seq data
We retrieve a `GRangesList` object containing the binding sites of all the transcription factors and co-regulators. These binding sites are derived from bulk ChIP-seq data in the [ChIP-Atlas](https://chip-atlas.org/) and [ENCODE](https://www.encodeproject.org/) databases, and merged across multiple cell lines or tissues. Currently, human genomes hg19 and hg38 and mouse mm10 are supported. All ChIP-seq and motif binding sites are hosted on ExperimentHub. The initial download may take some time, but the data are cached for subsequent use.


```{r, message=FALSE}
library(epiregulon)
grl <- getTFMotifInfo(genome = "hg38")
grl
```

For both [ChIP-Atlas](https://chip-atlas.org/) and [ENCODE](https://www.encodeproject.org/) data, we also provide sample-specific TF binding sites. They are retrieved as a list of `GrangesList` objects, by specifying `atlas.sample` or `encode.sample` in the `source` argument. 

```{r, message=FALSE}
grl_list_sample_atlas <- getTFMotifInfo(genome = "hg38", source = "atlas.sample")
head(names(grl_list_sample_atlas))
```

Then we need to access a specific list element by providing the sample of interest. Here we choose `"Peripheral blood mononuclear cells"`.
```{r}
grl_list_sample_atlas[["Peripheral blood mononuclear cells"]]
```
Alternatively, we might retrieve chip-seq data for `"K562"` cells present in the [ENCODE](https://www.encodeproject.org/) database. 

```{r, message=FALSE}
grl_list_sample_encode <- getTFMotifInfo(genome = "hg38", source = "encode.sample")
grl_list_sample_encode[["K562"]]
```

The tissue-specific chip-seq data are also available. They were compiled from the [ChIP-Atlas](https://chip-atlas.org/) database. 

```{r, message=FALSE}
grl_list_tissue <- getTFMotifInfo(genome = "hg38", source = "atlas.tissue")
grl_list_tissue[["Blood"]]
```

The advantage of using tissue- or sample-specific ChIP-seq data is that they accurately reflect TF binding sites that are characteristic of a given cell type or cell line. The downside is the small number of factors and peaks due to the limited number of experiments used to compile the data. 

```{r}
# calculate the total number of factors contained in each grl
# bulk ChIP-seq
length(grl)
# blood tissue
length(grl_list_tissue[["Blood"]])
# Peripheral blood mononuclear cells sample
length(grl_list_sample_atlas[["Peripheral blood mononuclear cells"]])
# K562 cell line sample
length(grl_list_sample_encode[["K562"]])
```

```{r}
# calculate the total number of regions contained in each grl
# bulk ChIP-seq
format(sum(unlist(lapply(grl, length))), big.mark=",")
# blood tissue
format(sum(unlist(lapply(grl_list_tissue[["Blood"]], length))), big.mark=",")
# Peripheral blood mononuclear cells sample
format(sum(unlist(lapply(grl_list_sample_atlas[["Peripheral blood mononuclear cells"]], length))), big.mark=",")
# K562 cell line sample
format(sum(unlist(lapply(grl_list_sample_encode[["K562"]], length))), big.mark=",")
```
Lastly, users can always supply their custom ChIP-seq data and convert them into `GrangesList`.


### Motif information
While ChIP-seq data provide experimentally determined TF biding sites, some users might prefer to start from motif sequences as positions of TF occupancy. If `motif` is chosen for `mode`, `getTFMotifInfo` retrieves the motif information from the [CIS_BP](https://cisbp.ccbr.utoronto.ca/) database and scans the peak regions for presence of motifs. 

```{r, eval=FALSE}
grl_motif <- getTFMotifInfo(genome = "hg38", mode = "motif", peaks = rowRanges(PeakMatrix))
```

## Link ATACseq peaks to target genes

Next, we link ATAC-seq peaks to their putative target genes. We assign a peak to a gene if it falls within a size window (default ±250kb) and if the chromatin accessibility of the peak and expression of the target gene is highly correlated across cell aggregates. 

To calculate correlations, we create cell aggregates (or metacells) by performing deterministic k-means clustering on the reduced dimensionality matrix (for description of the algorithm description, see [https://ltla.github.io/CppKmeans/classkmeans_1_1InitializeVariancePartition.html](https://ltla.github.io/CppKmeans/classkmeans_1_1InitializeVariancePartition.html). Then we aggregate the counts for the gene expression and peak matrices and normalize them by the number of cells. Correlations are thus computed on the averaged gene expression and chromatin accessibility. 

### Optimization of the number of metacells (optional)

It is important to pick the correct value for the `cellNum` parameter, which controls the number of metacells used for aggregation. Too few metacells can obscure biologically meaningful variation, while overly fine-grained aggregation may fail to sufficiently stabilize the signal, leaving substantial noise arising from data sparsity, technical bias, or biologically irrelevant fluctuations. In practice, we aim to strike a balance between these extremes and identify an optimal metacell number. This is the motivation behind the introduction of the new function `optimizeMetacellNumber`, released with `epiregulon v2.0.0`. 

The function computes mean empirical p-values across a range of metacell numbers, then fits a quadratic model that regresses these p-values against the square root of the mean number of cells per metacell. Squaring the value of $x_{min}$ yields the estimated optimal mean number of cells per metacell.

```{r}
set.seed(1010)
cellNum <- optimizeMetacellNumber(expMatrix = GeneExpressionMatrix,
                                  peakMatrix = PeakMatrix,
                                  exp_assay = "normalizedCounts",
                                  peak_assay = "counts",
                                  reducedDim = reducedDim(GeneExpressionMatrix, "LSI_RNA"),
                                  BPPARAM = BiocParallel::SerialParam(progressbar = FALSE))

cellNum
```

We can check specifically the estimated optimal number of cells per cluster (mean metacell size) by computing the square of the `solution` slot.
```{r}
cellNum@solution^2
```

The `plot` function allows for visualization of the function performance at different evaluation points. The estimated optimum is marked by a red dashed line.

```{r}
plot(cellNum)
```

The object returned by `optimizeMetacellNumber` can be passed to `calculateP2G` as the `cellNum` parameter. User might also provide a numeric to directly set the mean number of cells per cluster.

### Finding peak-target gene links within a defined distance {#distance_window}

The default method for linking regulatory elements to target genes is to compute correlations at the metacell level. A null distribution is generated by correlating randomly selected peak–gene pairs located on different chromosomes. Correlations are then calculated for all peak–gene pairs within a specified genomic distance (default ±250 kb). By default, only links with a p-value < 0.05 are retained, although this behavior can be modified via the `cutoff_sig` argument.

Although the algorithm for identifying metacell clusters is deterministic, reproducibility still requires setting a random seed because the null distribution is estimated from randomly sampled peak–gene pairs. To reduce the stochasticity introduced by this random sampling, one may increase the `nRandConn` parameter, which controls the number of random connections drawn.

If cluster labels are provided, peak–gene correlations are also computed within each cluster. A peak–gene link is retained if it passes the p-value threshold in any cluster. The resulting expanded list of links captures both inter- and intra-cluster variation.

```{r}
set.seed(1010)
p2g <- calculateP2G(peakMatrix = PeakMatrix, 
                    expMatrix = GeneExpressionMatrix, 
                    exp_assay = "normalizedCounts",
                    reducedDim = reducedDim(GeneExpressionMatrix, "LSI_RNA"),
                    clusters = GeneExpressionMatrix$cell_type,
                    cellNum = cellNum,
                    BPPARAM = BiocParallel::SerialParam(progressbar = FALSE))

p2g
```

### Linking peaks to the nearest genes

Alternatively, we can identify the downstream targets of each regulatory region by assigning them to their nearest gene. The `maxDist` argument specifies the maximum allowed distance between a regulatory element and a gene for a link to be made. As before, correlations are computed and only those exceeding the threshold are retained.

```{r}
set.seed(1010)
p2g_nearest <- calculateP2G(peakMatrix = PeakMatrix, 
                            expMatrix = GeneExpressionMatrix, 
                            exp_assay = "normalizedCounts",
                            reducedDim = reducedDim(GeneExpressionMatrix, "LSI_RNA"),
                            clusters = GeneExpressionMatrix$cell_type,
                            assignment_method = "nearest",
                            BPPARAM = BiocParallel::SerialParam(progressbar = FALSE)
)
```

### Filtering

The number of possible links between target genes and regulatory elements can be large. Therefore, we need to reduce this number, keeping only those links for which we have support in our data. By default, `calculateP2G` uses a p-value threshold to pinpoint putative connections between target genes and regulatory regions. However, users may also choose to use FDR or rely directly on the correlation. This behavior is controlled by the `cutoff_stat` argument. If `Correlation` is specified, the default threshold of 0.5 is used, as defined by `cor_cutoff`. Note that only links with correlation greater than this threshold will be kept, potentially ignoring meaningful highly negative correlations. Additionally, the correlation distribution might vary with dataset size, resulting in different FDRs for the same correlation threshold. Threshold for `p_val` and `FDR` is set by the `cutoff_sig` argument.

## Add TF binding sites to peaks

The next step is to add the TF binding information by overlapping the regions of the peak matrix with the bulk ChIP-seq database.
```{r}
overlap <- addTFMotifInfo(grl = grl, p2g = p2g, peakMatrix = PeakMatrix)
head(overlap)
```

## Generate regulons

A DataFrame, representing the inferred regulons, is then generated. The `DataFrame` object consists of the following columns:

1. `idxATAC` - index of the peak in the peak matrix
2. `chr` - chromosome number
3. `start` - start position of the peak
4. `end` - end position of the peak
5. `idxRNA` - index in the gene expression matrix corresponding to the target gene
6. `target` - name of the target gene
7. `corr` - correlation between target gene expression and the chromatin accessibility at the peak. If cluster labels are provided, 
this field is a matrix with columns names corresponding to correlation across all cells and for each of the clusters.
8. `p_val_peak_gene` - empirical p-value of the correlation between peak and target gene. If cluster labels are provided, 
this field is a matrix with columns names corresponding to p-value for all cells and for each of the clusters.
9. `FDR_peak_gene` - false discovery rates calculated based on the empirical p-value of the correlations between peaks and target genes. If cluster labels are provided, 
this field is a matrix with columns names corresponding to FDR for all cells and for each of the clusters.
10. `distance` - distance between the transcription start site of the target gene and the middle of the peak
11. `idxTF` - index in the gene expression matrix corresponding to the transcription factor
12. `tf` - name of the transcription factor

```{r}
regulon <- getRegulon(p2g, overlap, aggregate=FALSE)
regulon
```
