# Calculation of the TF activity

Once we define the target genes of TFs by constructing the GRN, we can calculate the per cell TF activity by summing up the expression of the individual target genes. While it is reasonable to assume equal contribution from each target genes, epiregulon tries to estimate the weights of each target gene. Biologically, this can be interpreted as the magnitude of gene expression changes induced by transcription factor activity.


## Add weights{#addWeights}

Epiregulon estimates the regulatory potential using one of the three measures: 

1. Wilcoxon test statistics of target gene expression in cells jointly expressing all 3 elements vs. cells that do not. 
2. Correlation between TG and TF, or between TG and the product of TF and RE. 
3. Mutual information between TG and TF expression, or between TG and the product of TF and RE.

Wilcoxon and correlation methods give both the magnitude and directionality of changes whereas mutual information is always positive. The correlation and mutual information statistics are computed on grouped pseudobulks following user-supplied cluster labels and yield a single weight across all clusters per each TF-RE-target triplet. In contrast, the Wilcoxon method can give a single weight across all cells or give cluster-specific weights if the users provide cluster labels. In general, we recommend computing a single weight across all cells because the data in totality provides the variability of target gene expression necessary for weights estimation. For example, we want to include both untreated and treated cells to gauge the effects of the treatment. Cluster-specific weights can be useful for inferring differential network connectivity (See section on [Differential Network Analysis](#diffNetwork)).

The choice of weights method can have a profound effect on the estimate of TF activity. When the TF activity is directly correlated with TF expression, all methods tend to work quite well. However, there are scenarios such as drug treatment or CRISPR knockout when TF continues to be expressed despite post-translational inhibition of TF activity.  Epiregulon was designed to handle scenarios where TF expression is decoupled from TF activity. In such cases, `wilcox` is the preferred choice. For a specific use case, see an [example](#weights_prostate) on comparing `"wilcox"` and `"corr"` methods. Refer to our [manuscript](https://www.nature.com/articles/s41467-025-62252-5) for a more in-depth discussion of the biological scenarios compatible with each weight method. 


```{r}
regulon.w <- addWeights(regulon = pruned.regulon.filtered,
                        expMatrix  = GeneExpressionMatrix,
                        exp_assay  = "normalizedCounts",
                        peakMatrix = PeakMatrix,
                        peak_assay = "counts",
                        method = "wilcox",
                        clusters = GeneExpressionMatrix$cell_type,
                        useDim = "LSI_RNA")
regulon.w
```

### Aggregate cells
As with the `pruneRegulon function`, cells can be aggregated into metacells using the `aggregateCells` argument in `addWeights`. Use this option only with the Wilcoxon method, since the other two methods already incorporate cell aggregation. By default, the function generates aggregates containing an average of 10 cells. However, the exact number may vary across metacells, depending on how single cells are distributed in the reduced dimensionality space. Refer to the cell aggregation [section]({#cell_aggregation}) in Regulation Refinement for additional information.
```{r eval=FALSE}
regulon.w.aggr <- addWeights(regulon = pruned.regulon,
                             expMatrix  = GeneExpressionMatrix,
                             exp_assay  = "normalizedCounts",
                             peakMatrix = PeakMatrix,
                             peak_assay = "counts",
                             method = "wilcox",
                             clusters = GeneExpressionMatrix$cell_type,
                             aggregateCells = TRUE,
                             useDim = "LSI_RNA",
                             exp_cutoff=NULL,
                             peak_cutoff=NULL)
```

### Using chromatin accessibility data
The `corr` and `MI` methods for calculating weights by default ignore information about chromatin accessibility. However, it is possible to incorporate this modality into the calculations by setting the `tf_re.merge` argument to `TRUE`. This adds an additional step to the procedure, in which transcription factor expression is multiplied by chromatin accessibility. Therefore, expression is reduced to zero if the corresponding element is not accessible. The TF-RE product is calculated at the level of aggregated cells, mitigating the effect of sparsity, which affects ATAC-seq data more than gene expression. See an [example](#weights_prostate) using the `tf_re.merge` argument.

## Calculate TF activity 

Finally, the activity for a specific TF in each cell is computed by averaging the expression of target genes weighted by the test statistics of choice as explained in the [add weights](#addWeights) section. 
$$y=\frac{1}{n}\sum_{i=1}^{n} x_i * weights_i$$
where $y$ is the activity of a TF for a cell,
$n$ is the total number of targets for a TF,
$x_i$ is the log count expression of target $i$ where $i$ in {1,2,...,n} and
$weights_i$ is the weight of TF - target $i$

```{r warning=FALSE, message=FALSE}
score.combine <- calculateActivity(expMatrix = GeneExpressionMatrix,
                                   regulon = regulon.w, 
                                   mode = "weight", 
                                   exp_assay = "normalizedCounts")

score.combine[1:5,1:5]
```

