# Calculation of the TF activity

Once the GRN has been constructed, we can calculate the activity of each transcription factor in each cell. The first step in this process is running `addWeights`. This function estimates the potential of transcription factors to regulate each target gene based on gene expression. Depending on the configuration, peak data from accessible genomic regions may also be used. Following this step, transcription factor activity is calculated as a weighted mean of the target gene expression values. In this section, we present the details of the epiregulon approach to activity inference, and show how the process can be tailored to the user's needs by modifying function parameters.

## Add Weights

While the `pruneRegulon` function provides statistics on the joint occurrence of TF-RE-TG (Section \@ref(pruning)), we would like to further estimate the strength of regulation. Biologically, this can be interpreted as the magnitude of gene expression changes induced by transcription factor activity. Epiregulon estimates the regulatory potential using one of the three measures: 

1. Wilcoxon test statistics of target gene expression in cells jointly expressing all 3 elements vs cells that do not. 
2. Correlation between TG and TF or between TG and the product of TF and RE. 
3. Mutual information between TG and TF expression or between TG and the product of TF and RE.

Two measures (Wilcoxon and correlation) give both the magnitude and directionality of changes whereas mutational information (MI) is always positive. The correlation and mutual information statistics are computed on grouped pseudobulks following user-supplied cluster labels and yield a single weight across all clusters per each TF-RE-target triplet. In contrast, the Wilcoxon method counts the cells in groups generated by possible outcomes of expression thresholds applied jointly to TF and RE, and separately to TG. Providing cluster labels with this method results in the calculation of cluster-specific weights, in addition to estimating weights from all cells. Thus, cluster labels result in the computation of statistics for cell groups, much as is the case with `calculateP2G` (Section \@ref(#distance_window)). Cell aggregation uses a default value of 10 cells per group and can help overcome sparsity and speed up computation.

We use the Wilcoxon weight method because we are interested in computing cell type-specific weights. The user might also want to use `"corr"` or `"MI"` method instead.

For the use case showing the difference between `"wilcox"` and `"corr"` methods see Section \@ref(weights_prostate).

```{r}
regulon.w <- addWeights(regulon = pruned.regulon,
                        expMatrix  = GeneExpressionMatrix,
                        exp_assay  = "normalizedCounts",
                        peakMatrix = PeakMatrix,
                        peak_assay = "counts",
                        method = "wilcox",
                        clusters = GeneExpressionMatrix$cell_type,
                        useDim = "LSI_RNA")
regulon.w
```

### Cell aggregation
As with the `pruneRegulon` function, we can also aggregate cells into metacells using the `addWeights` function. It is sensible to use this option only with the Wilcoxon method, since the other two methods already incorporate cell aggregation. All the remarks made previously regarding cell aggregation (Section \@ref(cell_aggregation)) apply here as well. By default, the function creates cell aggregates of 10 cells on average. However, this value may vary from metacell to metacell, depending on the distribution of single cells in the reduced dimensionality space.  
```{r}
regulon.w.aggr <- addWeights(regulon = pruned.regulon,
                        expMatrix  = GeneExpressionMatrix,
                        exp_assay  = "normalizedCounts",
                        peakMatrix = PeakMatrix,
                        peak_assay = "counts",
                        method = "wilcox",
                        clusters = GeneExpressionMatrix$cell_type,
                        aggregateCells = TRUE,
                        useDim = "LSI_RNA",
                        exp_cutoff=NULL,
                        peak_cutoff=NULL)
```

## Calculate TF activity 

Finally, the activities for a specific TF in each cell are computed by averaging expressions of target genes linked to the TF weighted by the test statistics of choice, chosen from either correlation, mutual information or the Wilcoxon test statistics. 
$$y=\frac{1}{n}\sum_{i=1}^{n} x_i * weights_i$$
where $y$ is the activity of a TF for a cell,
$n$ is the total number of targets for a TF,
$x_i$ is the log count expression of target $i$ where $i$ in {1,2,...,n} and
$weights_i$ is the weight of TF - target $i$

```{r}
score.combine <- calculateActivity(expMatrix = GeneExpressionMatrix,
                                   regulon = regulon.w, 
                                   mode = "weight", 
                                   method = "weightedMean", 
                                   exp_assay = "normalizedCounts",
                                   normalize = FALSE)

score.combine[1:5,1:5]
```

