<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Hematopoeisis tutorial - MAE | A Minimal Book Example</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
set in the _output.yml file.
The HTML output format for this example is bookdown::gitbook,</p>" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Hematopoeisis tutorial - MAE | A Minimal Book Example" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
set in the _output.yml file.
The HTML output format for this example is bookdown::gitbook,</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Hematopoeisis tutorial - MAE | A Minimal Book Example" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
set in the _output.yml file.
The HTML output format for this example is bookdown::gitbook,</p>" />
  

<meta name="author" content="John Doe" />


<meta name="date" content="2023-11-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="hello-bookdown.html"/>
<link rel="next" href="blocks.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="hello-bookdown.html"><a href="hello-bookdown.html"><i class="fa fa-check"></i><b>2</b> Hello bookdown</a>
<ul>
<li class="chapter" data-level="2.1" data-path="hello-bookdown.html"><a href="hello-bookdown.html#a-section"><i class="fa fa-check"></i><b>2.1</b> A section</a>
<ul>
<li class="chapter" data-level="" data-path="hello-bookdown.html"><a href="hello-bookdown.html#an-unnumbered-section"><i class="fa fa-check"></i>An unnumbered section</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html"><i class="fa fa-check"></i><b>3</b> Hematopoeisis tutorial - MAE</a>
<ul>
<li class="chapter" data-level="3.1" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#installation"><i class="fa fa-check"></i><b>3.1</b> Installation</a></li>
<li class="chapter" data-level="3.2" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#data-preparation"><i class="fa fa-check"></i><b>3.2</b> Data preparation</a></li>
<li class="chapter" data-level="3.3" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#quick-start"><i class="fa fa-check"></i><b>3.3</b> Quick start</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#retrieve-bulk-tf-chip-seq-binding-sites"><i class="fa fa-check"></i><b>3.3.1</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="3.3.2" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#link-atacseq-peaks-to-target-genes"><i class="fa fa-check"></i><b>3.3.2</b> Link ATACseq peaks to target genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="blocks.html"><a href="blocks.html"><i class="fa fa-check"></i><b>4</b> Blocks</a>
<ul>
<li class="chapter" data-level="4.1" data-path="blocks.html"><a href="blocks.html#equations"><i class="fa fa-check"></i><b>4.1</b> Equations</a></li>
<li class="chapter" data-level="4.2" data-path="blocks.html"><a href="blocks.html#theorems-and-proofs"><i class="fa fa-check"></i><b>4.2</b> Theorems and proofs</a></li>
<li class="chapter" data-level="4.3" data-path="blocks.html"><a href="blocks.html#callout-blocks"><i class="fa fa-check"></i><b>4.3</b> Callout blocks</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="sharing-your-book.html"><a href="sharing-your-book.html"><i class="fa fa-check"></i><b>5</b> Sharing your book</a>
<ul>
<li class="chapter" data-level="5.1" data-path="sharing-your-book.html"><a href="sharing-your-book.html#publishing"><i class="fa fa-check"></i><b>5.1</b> Publishing</a></li>
<li class="chapter" data-level="5.2" data-path="sharing-your-book.html"><a href="sharing-your-book.html#pages"><i class="fa fa-check"></i><b>5.2</b> 404 pages</a></li>
<li class="chapter" data-level="5.3" data-path="sharing-your-book.html"><a href="sharing-your-book.html#metadata-for-sharing"><i class="fa fa-check"></i><b>5.3</b> Metadata for sharing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Minimal Book Example</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="hematopoeisis-tutorial---mae" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Hematopoeisis tutorial - MAE<a href="hematopoeisis-tutorial---mae.html#hematopoeisis-tutorial---mae" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In this vignette, we used a dataset from the ArchR <a href="https://www.archrproject.com/bookdown/index.html">tutorial</a>. Prior to using epiregulon, this dataset has been fully preprocessed in ArchR, and converted to a <code>MultiAssayExperiment</code> using <code>epireglon::archr2MAE</code>. The MAE object was uploaded to <code>scMultiome</code> for full reproducibility. In this dataset, scRNAseq and scATACseq were unpaired and integrated by the <code>ArchR::addGeneIntegrationMatrix</code> function.</p>
<div id="installation" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Installation<a href="hematopoeisis-tutorial---mae.html#installation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Epiregulon is currently available on R/dev</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="hematopoeisis-tutorial---mae.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(epiregulon)</span></code></pre></div>
<pre><code>## Warning: replacing previous import &#39;GenomicRanges::union&#39; by &#39;igraph::union&#39;
## when loading &#39;epiregulon&#39;</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="hematopoeisis-tutorial---mae.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#devtools::load_all(&quot;/gstore/project/lineage/xiaosai/epiregulon&quot;)</span></span></code></pre></div>
<p>If you would like to install from gitlab,</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="hematopoeisis-tutorial---mae.html#cb4-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="at">repo=</span><span class="st">&#39;xiaosaiyao/epiregulon&#39;</span>)</span>
<span id="cb4-2"><a href="hematopoeisis-tutorial---mae.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(epiregulon)</span></code></pre></div>
</div>
<div id="data-preparation" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Data preparation<a href="hematopoeisis-tutorial---mae.html#data-preparation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Download the example dataset from <em><a href="https://bioconductor.org/packages/3.18/scMultiome">scMultiome</a></em> package</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="hematopoeisis-tutorial---mae.html#cb5-1" aria-hidden="true" tabindex="-1"></a>mae <span class="ot">&lt;-</span> scMultiome<span class="sc">::</span><span class="fu">hematopoiesis</span>()</span></code></pre></div>
<pre><code>## see ?scMultiome and browseVignettes(&#39;scMultiome&#39;) for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="hematopoeisis-tutorial---mae.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load peak matrix</span></span>
<span id="cb8-2"><a href="hematopoeisis-tutorial---mae.html#cb8-2" aria-hidden="true" tabindex="-1"></a>PeakMatrix <span class="ot">&lt;-</span> mae[[<span class="st">&quot;PeakMatrix&quot;</span>]]</span>
<span id="cb8-3"><a href="hematopoeisis-tutorial---mae.html#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="hematopoeisis-tutorial---mae.html#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Load expression matrix</span></span>
<span id="cb8-5"><a href="hematopoeisis-tutorial---mae.html#cb8-5" aria-hidden="true" tabindex="-1"></a>GeneExpressionMatrix <span class="ot">&lt;-</span> mae[[<span class="st">&quot;GeneIntegrationMatrix&quot;</span>]]</span>
<span id="cb8-6"><a href="hematopoeisis-tutorial---mae.html#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="hematopoeisis-tutorial---mae.html#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Add gene symbols to rownames</span></span>
<span id="cb8-8"><a href="hematopoeisis-tutorial---mae.html#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(GeneExpressionMatrix) <span class="ot">&lt;-</span> <span class="fu">rowData</span>(GeneExpressionMatrix)<span class="sc">$</span>name</span>
<span id="cb8-9"><a href="hematopoeisis-tutorial---mae.html#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="hematopoeisis-tutorial---mae.html#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Transfer dimensionality reduction matrix to GeneExpression</span></span>
<span id="cb8-11"><a href="hematopoeisis-tutorial---mae.html#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="fu">reducedDim</span>(GeneExpressionMatrix, <span class="st">&quot;IterativeLSI&quot;</span>) <span class="ot">&lt;-</span> </span>
<span id="cb8-12"><a href="hematopoeisis-tutorial---mae.html#cb8-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reducedDim</span>(mae[[<span class="st">&#39;TileMatrix500&#39;</span>]], <span class="st">&quot;IterativeLSI&quot;</span>)</span>
<span id="cb8-13"><a href="hematopoeisis-tutorial---mae.html#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="fu">reducedDim</span>(GeneExpressionMatrix, <span class="st">&quot;UMAP&quot;</span>) <span class="ot">&lt;-</span> </span>
<span id="cb8-14"><a href="hematopoeisis-tutorial---mae.html#cb8-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reducedDim</span>(mae[[<span class="st">&#39;TileMatrix500&#39;</span>]], <span class="st">&quot;UMAP&quot;</span>)</span></code></pre></div>
<p>Visualize the data</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="hematopoeisis-tutorial---mae.html#cb9-1" aria-hidden="true" tabindex="-1"></a>scater<span class="sc">::</span><span class="fu">plotReducedDim</span>(GeneExpressionMatrix, </span>
<span id="cb9-2"><a href="hematopoeisis-tutorial---mae.html#cb9-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">dimred =</span> <span class="st">&quot;UMAP&quot;</span>, </span>
<span id="cb9-3"><a href="hematopoeisis-tutorial---mae.html#cb9-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">text_by =</span> <span class="st">&quot;Clusters2&quot;</span>, </span>
<span id="cb9-4"><a href="hematopoeisis-tutorial---mae.html#cb9-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">colour_by =</span> <span class="st">&quot;Clusters2&quot;</span>,</span>
<span id="cb9-5"><a href="hematopoeisis-tutorial---mae.html#cb9-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">point_size =</span> <span class="fl">0.3</span>,</span>
<span id="cb9-6"><a href="hematopoeisis-tutorial---mae.html#cb9-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">point_alpha =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
</div>
<div id="quick-start" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Quick start<a href="hematopoeisis-tutorial---mae.html#quick-start" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="retrieve-bulk-tf-chip-seq-binding-sites" class="section level3 hasAnchor" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Retrieve bulk TF ChIP-seq binding sites<a href="hematopoeisis-tutorial---mae.html#retrieve-bulk-tf-chip-seq-binding-sites" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First, we retrieve the information of TF binding sites collected from Cistrome and ENCODE ChIP-seq. Currently, human genomes hg19 and hg38 and mouse genome mm10 are available</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="hematopoeisis-tutorial---mae.html#cb10-1" aria-hidden="true" tabindex="-1"></a>grl <span class="ot">&lt;-</span> <span class="fu">getTFMotifInfo</span>(<span class="at">genome =</span> <span class="st">&quot;hg19&quot;</span>)</span></code></pre></div>
<pre><code>## see ?scMultiome and browseVignettes(&#39;scMultiome&#39;) for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="hematopoeisis-tutorial---mae.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(grl)</span></code></pre></div>
<pre><code>## GRangesList object of length 6:
## $`5-hmC`
## GRanges object with 22860 ranges and 0 metadata columns:
##           seqnames        ranges strand
##              &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt;
##       [1]     chr1   10001-10685      *
##       [2]     chr1   13362-13694      *
##       [3]     chr1   29631-29989      *
##       [4]     chr1   40454-40754      *
##       [5]     chr1 135395-135871      *
##       ...      ...           ...    ...
##   [22856]     chrM   15303-15326      *
##   [22857]     chrM   15328-16172      *
##   [22858]     chrM   16174-16183      *
##   [22859]     chrM   16186-16224      *
##   [22860]     chrM   16226-16492      *
##   -------
##   seqinfo: 25 sequences from an unspecified genome; no seqlengths
## 
## ...
## &lt;5 more elements&gt;</code></pre>
</div>
<div id="link-atacseq-peaks-to-target-genes" class="section level3 hasAnchor" number="3.3.2">
<h3><span class="header-section-number">3.3.2</span> Link ATACseq peaks to target genes<a href="hematopoeisis-tutorial---mae.html#link-atacseq-peaks-to-target-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Next, we compute peak to gene correlations using a custom algorithm that has similar performance to ArchR’s P2G function. Wherever possible, use a multidimensional dimensionality reduction matrix such as LSI or PCA instead of UMAP or TSNE since the former provides a more accurate estimate of cell similarity.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="hematopoeisis-tutorial---mae.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(epiregulon<span class="sc">:::</span>rowSums)</span></code></pre></div>
<pre><code>## new(&quot;standardGeneric&quot;, .Data = function (x, ...) 
## standardGeneric(&quot;rowSums&quot;), generic = &quot;rowSums&quot;, package = &quot;epiregulon&quot;, 
##     group = list(), valueClass = character(0), signature = &quot;x&quot;, 
##     default = new(&quot;derivedDefaultMethod&quot;, .Data = function (x, 
##         ...) 
##     Matrix::rowSums(x), target = new(&quot;signature&quot;, .Data = &quot;ANY&quot;, 
##         names = &quot;x&quot;, package = &quot;methods&quot;), defined = new(&quot;signature&quot;, 
##         .Data = &quot;ANY&quot;, names = &quot;x&quot;, package = &quot;methods&quot;), generic = &quot;rowSums&quot;), 
##     skeleton = (new(&quot;derivedDefaultMethod&quot;, .Data = function (x, 
##         ...) 
##     Matrix::rowSums(x), target = new(&quot;signature&quot;, .Data = &quot;ANY&quot;, 
##         names = &quot;x&quot;, package = &quot;methods&quot;), defined = new(&quot;signature&quot;, 
##         .Data = &quot;ANY&quot;, names = &quot;x&quot;, package = &quot;methods&quot;), generic = &quot;rowSums&quot;))(x, 
##         ...))
## &lt;bytecode: 0x55558667a238&gt;
## &lt;environment: 0x55558666c440&gt;
## attr(,&quot;generic&quot;)
## [1] &quot;rowSums&quot;
## attr(,&quot;generic&quot;)attr(,&quot;package&quot;)
## [1] &quot;epiregulon&quot;
## attr(,&quot;package&quot;)
## [1] &quot;epiregulon&quot;
## attr(,&quot;group&quot;)
## list()
## attr(,&quot;valueClass&quot;)
## character(0)
## attr(,&quot;signature&quot;)
## [1] &quot;x&quot;
## attr(,&quot;default&quot;)
## Method Definition (Class &quot;derivedDefaultMethod&quot;):
## 
## function (x, ...) 
## Matrix::rowSums(x)
## &lt;environment: namespace:epiregulon&gt;
## 
## Signatures:
##         x    
## target  &quot;ANY&quot;
## defined &quot;ANY&quot;
## attr(,&quot;skeleton&quot;)
## (new(&quot;derivedDefaultMethod&quot;, .Data = function (x, ...) 
## Matrix::rowSums(x), target = new(&quot;signature&quot;, .Data = &quot;ANY&quot;, 
##     names = &quot;x&quot;, package = &quot;methods&quot;), defined = new(&quot;signature&quot;, 
##     .Data = &quot;ANY&quot;, names = &quot;x&quot;, package = &quot;methods&quot;), generic = &quot;rowSums&quot;))(x, 
##     ...)
## attr(,&quot;class&quot;)
## [1] &quot;standardGeneric&quot;
## attr(,&quot;class&quot;)attr(,&quot;package&quot;)
## [1] &quot;methods&quot;</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="hematopoeisis-tutorial---mae.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1010</span>)</span>
<span id="cb17-2"><a href="hematopoeisis-tutorial---mae.html#cb17-2" aria-hidden="true" tabindex="-1"></a>p2g <span class="ot">&lt;-</span> <span class="fu">calculateP2G</span>(<span class="at">peakMatrix =</span> PeakMatrix, </span>
<span id="cb17-3"><a href="hematopoeisis-tutorial---mae.html#cb17-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">expMatrix =</span> GeneExpressionMatrix, </span>
<span id="cb17-4"><a href="hematopoeisis-tutorial---mae.html#cb17-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">exp_assay =</span> <span class="st">&quot;normalizedCounts&quot;</span>,</span>
<span id="cb17-5"><a href="hematopoeisis-tutorial---mae.html#cb17-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">reducedDim =</span> <span class="fu">reducedDim</span>(GeneExpressionMatrix, <span class="st">&quot;IterativeLSI&quot;</span>))</span></code></pre></div>
<pre><code>## Using epiregulon to compute peak to gene links...</code></pre>
<pre><code>## performing k means clustering to form metacells</code></pre>
<pre><code>## Computing correlation</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="hematopoeisis-tutorial---mae.html#cb21-1" aria-hidden="true" tabindex="-1"></a>p2g</span></code></pre></div>
<pre><code>## DataFrame with 23711 rows and 8 columns
##         idxATAC         chr     start       end    idxRNA    target Correlation
##       &lt;integer&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;   &lt;array&gt;    &lt;matrix&gt;
## 1             7        chr1    801002    801502         2 LINC00115    0.864244
## 2             8        chr1    805039    805539         6    KLHL17    0.625471
## 3             9        chr1    845326    845826        10      AGRN    0.545993
## 4            10        chr1    846428    846928        10      AGRN    0.646209
## 5            13        chr1    856263    856763        10      AGRN    0.549411
## ...         ...         ...       ...       ...       ...       ...         ...
## 23707    146403       chr22  51021154  51021654     12089      ARSA    0.655996
## 23708    146412       chr22  51110826  51111326     12090    SHANK3    0.560404
## 23709    146417       chr22  51143606  51144106     12090    SHANK3    0.500026
## 23710    146421       chr22  51213512  51214012     12090    SHANK3    0.504567
## 23711    146421       chr22  51213512  51214012     12091       ACR    0.557113
##        distance
##       &lt;integer&gt;
## 1         36099
## 2         88427
## 3        107676
## 4        106574
## 5         96739
## ...         ...
## 23707     44747
## 23708         0
## 23709     30336
## 23710    100242
## 23711     36660</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="hello-bookdown.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="blocks.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/04-test.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
