<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Prostate cancer cells | Epiregulon documentation</title>
  <meta name="description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Prostate cancer cells | Epiregulon documentation" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Prostate cancer cells | Epiregulon documentation" />
  
  <meta name="twitter:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  

<meta name="author" content="Xiaosai Yao" />


<meta name="date" content="2023-11-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="hematopoeisis-tutorial---mae.html"/>
<link rel="next" href="single-modality-gene-expression.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html"><i class="fa fa-check"></i><b>2</b> Multiome - ArchR workflow (#multiome-archr)</a>
<ul>
<li class="chapter" data-level="2.1" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#installation-1"><i class="fa fa-check"></i><b>2.1</b> Installation</a></li>
<li class="chapter" data-level="2.2" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#data-preparation"><i class="fa fa-check"></i><b>2.2</b> Data preparation</a></li>
<li class="chapter" data-level="2.3" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#quick-start"><i class="fa fa-check"></i><b>2.3</b> Quick start</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#retrieve-bulk-tf-chip-seq-binding-sites"><i class="fa fa-check"></i><b>2.3.1</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="2.3.2" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#link-atac-seq-peaks-to-target-genes"><i class="fa fa-check"></i><b>2.3.2</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="2.3.3" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#add-tf-motif-binding-to-peaks"><i class="fa fa-check"></i><b>2.3.3</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="2.3.4" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#generate-regulons"><i class="fa fa-check"></i><b>2.3.4</b> Generate regulons</a></li>
<li class="chapter" data-level="2.3.5" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#network-pruning-highly-recommended"><i class="fa fa-check"></i><b>2.3.5</b> Network pruning (highly recommended)</a></li>
<li class="chapter" data-level="2.3.6" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#add-weights"><i class="fa fa-check"></i><b>2.3.6</b> Add Weights</a></li>
<li class="chapter" data-level="2.3.7" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#optional-annotate-with-tf-motifs"><i class="fa fa-check"></i><b>2.3.7</b> (Optional) Annotate with TF motifs</a></li>
<li class="chapter" data-level="2.3.8" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#calculate-tf-activity"><i class="fa fa-check"></i><b>2.3.8</b> Calculate TF activity</a></li>
<li class="chapter" data-level="2.3.9" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#perform-differential-activity"><i class="fa fa-check"></i><b>2.3.9</b> Perform differential activity</a></li>
<li class="chapter" data-level="2.3.10" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#visualize-the-results"><i class="fa fa-check"></i><b>2.3.10</b> Visualize the results</a></li>
<li class="chapter" data-level="2.3.11" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#geneset-enrichment"><i class="fa fa-check"></i><b>2.3.11</b> Geneset enrichment</a></li>
<li class="chapter" data-level="2.3.12" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#network-analysis"><i class="fa fa-check"></i><b>2.3.12</b> Network analysis</a></li>
<li class="chapter" data-level="2.3.13" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#differential-networks"><i class="fa fa-check"></i><b>2.3.13</b> Differential networks</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="multiome---archr-workflow-multiome-archr.html"><a href="multiome---archr-workflow-multiome-archr.html#session-info"><i class="fa fa-check"></i><b>2.4</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html"><i class="fa fa-check"></i><b>3</b> Hematopoeisis tutorial - MAE</a>
<ul>
<li class="chapter" data-level="3.1" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#installation-2"><i class="fa fa-check"></i><b>3.1</b> Installation</a></li>
<li class="chapter" data-level="3.2" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#data-preparation-1"><i class="fa fa-check"></i><b>3.2</b> Data preparation</a></li>
<li class="chapter" data-level="3.3" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#quick-start-1"><i class="fa fa-check"></i><b>3.3</b> Quick start</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#retrieve-bulk-tf-chip-seq-binding-sites-1"><i class="fa fa-check"></i><b>3.3.1</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="3.3.2" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#link-atacseq-peaks-to-target-genes"><i class="fa fa-check"></i><b>3.3.2</b> Link ATACseq peaks to target genes</a></li>
<li class="chapter" data-level="3.3.3" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#add-tf-motif-binding-to-peaks-1"><i class="fa fa-check"></i><b>3.3.3</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="3.3.4" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#generate-regulons-1"><i class="fa fa-check"></i><b>3.3.4</b> Generate regulons</a></li>
<li class="chapter" data-level="3.3.5" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#prune-network"><i class="fa fa-check"></i><b>3.3.5</b> Prune network</a></li>
<li class="chapter" data-level="3.3.6" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#add-weights-1"><i class="fa fa-check"></i><b>3.3.6</b> Add Weights</a></li>
<li class="chapter" data-level="3.3.7" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#calculate-tf-activity-1"><i class="fa fa-check"></i><b>3.3.7</b> Calculate TF activity</a></li>
<li class="chapter" data-level="3.3.8" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#differential-tf-activity-test"><i class="fa fa-check"></i><b>3.3.8</b> Differential TF activity test</a></li>
<li class="chapter" data-level="3.3.9" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#visualizing-tf-activities"><i class="fa fa-check"></i><b>3.3.9</b> Visualizing TF activities</a></li>
<li class="chapter" data-level="3.3.10" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#geneset-enrichment-1"><i class="fa fa-check"></i><b>3.3.10</b> Geneset enrichment</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#differential-network-analysis"><i class="fa fa-check"></i><b>3.4</b> Differential Network analysis</a></li>
<li class="chapter" data-level="3.5" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#session-info-1"><i class="fa fa-check"></i><b>3.5</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html"><i class="fa fa-check"></i><b>4</b> Prostate cancer cells</a>
<ul>
<li class="chapter" data-level="4.1" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#data-preparation-2"><i class="fa fa-check"></i><b>4.1</b> Data preparation</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#load-archr-project"><i class="fa fa-check"></i><b>4.1.1</b> Load ArchR project</a></li>
<li class="chapter" data-level="4.1.2" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#retrieve-matrices-from-archr-project"><i class="fa fa-check"></i><b>4.1.2</b> Retrieve matrices from ArchR project</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#quick-start-2"><i class="fa fa-check"></i><b>4.2</b> Quick start</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#retrieve-bulk-tf-chip-seq-binding-sites-2"><i class="fa fa-check"></i><b>4.2.1</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="4.2.2" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#link-atac-seq-peaks-to-target-genes-1"><i class="fa fa-check"></i><b>4.2.2</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="4.2.3" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#add-tf-motif-binding-to-peaks-2"><i class="fa fa-check"></i><b>4.2.3</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="4.2.4" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#generate-regulons-2"><i class="fa fa-check"></i><b>4.2.4</b> Generate regulons</a></li>
<li class="chapter" data-level="4.2.5" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#calculate-tf-activity-2"><i class="fa fa-check"></i><b>4.2.5</b> Calculate TF activity</a></li>
<li class="chapter" data-level="4.2.6" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#perform-differential-activity-1"><i class="fa fa-check"></i><b>4.2.6</b> Perform differential activity</a></li>
<li class="chapter" data-level="4.2.7" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#visualize-the-results-1"><i class="fa fa-check"></i><b>4.2.7</b> Visualize the results</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#session-info-2"><i class="fa fa-check"></i><b>4.3</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html"><i class="fa fa-check"></i><b>5</b> Single modality: gene expression</a>
<ul>
<li class="chapter" data-level="5.1" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#load-regulon"><i class="fa fa-check"></i><b>5.1</b> Load regulon</a></li>
<li class="chapter" data-level="5.2" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#load-scrna-seq-data"><i class="fa fa-check"></i><b>5.2</b> Load scRNA-seq data</a></li>
<li class="chapter" data-level="5.3" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#calculate-activity"><i class="fa fa-check"></i><b>5.3</b> Calculate activity</a></li>
<li class="chapter" data-level="5.4" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#perform-differential-activity-2"><i class="fa fa-check"></i><b>5.4</b> Perform differential activity</a></li>
<li class="chapter" data-level="5.5" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#visualize-activity"><i class="fa fa-check"></i><b>5.5</b> Visualize activity</a></li>
<li class="chapter" data-level="5.6" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#pathway-enrichment"><i class="fa fa-check"></i><b>5.6</b> Pathway enrichment</a></li>
<li class="chapter" data-level="5.7" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#session-info-3"><i class="fa fa-check"></i><b>5.7</b> Session Info</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Epiregulon documentation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="prostate-cancer-cells" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Prostate cancer cells<a href="prostate-cancer-cells.html#prostate-cancer-cells" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This tutorial walks through an example of TF activity inference in unpaired scATACseq/scRNAseq of parental LNCaP cells treated with DMSO, Enzalutamide and Enza resistant cells. The dataset was taken from <a href="https://www.nature.com/articles/s41467-021-25624-1">Taavitsainen et al</a> <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE168667">GSE168667</a> and <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE168668">GSE168668</a>.</p>
<div id="data-preparation-2" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Data preparation<a href="prostate-cancer-cells.html#data-preparation-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Please refer to the full ArchR <a href="https://www.archrproject.com/bookdown/index.html">manual</a> for instructions</p>
<p>Before running Epiregulon, the following analyses need to be completed:
1. Obtain a peak matrix on scATACseq by using addGroupCoverages &gt; addReproduciblePeakSet &gt; addPeakMatrix. See chapter <a href="https://www.archrproject.com/bookdown/calling-peaks-with-archr.html">10</a> from ArchR manual
2. RNA-seq integration.
a. For unpaired scATAC-seq, use addGeneIntegrationMatrix. See chapter <a href="https://www.archrproject.com/bookdown/defining-cluster-identity-with-scrna-seq.html">8</a> from ArchR manual
b. For multiome data, use addGeneExpressionMatrix. See <a href="https://greenleaflab.github.io/ArchR_2020/Ex-Analyze-Multiome.html">multiome</a> tutorial
3. Perform dimensionality reduction from with either single modalities or joint scRNAseq and scATACseq using <a href="https://www.archrproject.com/reference/addCombinedDims.html">addCombinedDims</a></p>
<div id="load-archr-project" class="section level3 hasAnchor" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> Load ArchR project<a href="prostate-cancer-cells.html#load-archr-project" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Copy this ArchR project into your own directory</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="prostate-cancer-cells.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ArchR, <span class="at">quietly =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-2"><a href="prostate-cancer-cells.html#cb1-2" aria-hidden="true" tabindex="-1"></a>archR_project_path <span class="ot">&lt;-</span> <span class="st">&quot;/gstore/project/lineage/prostate/GSE168667/OUTPUT/multiome/&quot;</span></span>
<span id="cb1-3"><a href="prostate-cancer-cells.html#cb1-3" aria-hidden="true" tabindex="-1"></a>proj <span class="ot">&lt;-</span> <span class="fu">loadArchRProject</span>(<span class="at">path =</span> archR_project_path, <span class="at">showLogo =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>We verify that “GeneExpressionMatrix” and “PeakMatrix” are present for this tutorial.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="prostate-cancer-cells.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getAvailableMatrices</span>(proj)</span></code></pre></div>
<pre><code>## [1] &quot;GeneIntegrationMatrix&quot; &quot;GeneScoreMatrix&quot;       &quot;MotifMatrix&quot;          
## [4] &quot;PeakMatrix&quot;            &quot;TileMatrix&quot;</code></pre>
<p>We will use the joint reducedDims - “LSI_Combined” and joint embeddings - “UMAP_Combined”</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="prostate-cancer-cells.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">getReducedDims</span>(proj, <span class="at">reducedDims =</span> <span class="st">&quot;iLSI_Combined&quot;</span>)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code></pre></div>
<pre><code>##                                     LSI1       LSI2       LSI3        LSI4
## SRR13927735#TTATGTCTCCAGGTAT-1 -2.713935 -0.3677949 -0.4484238 -0.30645138
## SRR13927735#TATTGCTCATCAGAAA-1 -2.642781 -0.2767556 -0.9142714 -0.19675812
## SRR13927735#TTCGATTGTAGGGTTG-1 -2.322865 -0.1543080 -1.4106049 -0.08891276
## SRR13927735#CATTCATTCGGATGTT-1 -2.572976 -0.1917188 -1.0464294 -0.12660121
## SRR13927735#ACGTTAGGTCAACTGT-1 -2.478552 -0.1776639 -1.1037295 -0.22976613
## SRR13927735#AAATGCCCAGCAATGG-1 -2.595352 -0.3803464 -0.7770309 -0.52431765
##                                        LSI5
## SRR13927735#TTATGTCTCCAGGTAT-1 -0.046845365
## SRR13927735#TATTGCTCATCAGAAA-1  0.075746940
## SRR13927735#TTCGATTGTAGGGTTG-1  0.019873276
## SRR13927735#CATTCATTCGGATGTT-1  0.009947438
## SRR13927735#ACGTTAGGTCAACTGT-1 -0.150097539
## SRR13927735#AAATGCCCAGCAATGG-1 -0.243074591</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="prostate-cancer-cells.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">getEmbedding</span>(proj, <span class="at">embedding =</span> <span class="st">&quot;UMAP_Combined&quot;</span>))</span></code></pre></div>
<pre><code>##                                iLSI_Combined#UMAP_Dimension_1
## SRR13927735#TTATGTCTCCAGGTAT-1                      -9.622903
## SRR13927735#TATTGCTCATCAGAAA-1                      -9.360211
## SRR13927735#TTCGATTGTAGGGTTG-1                      -8.617347
## SRR13927735#CATTCATTCGGATGTT-1                      -9.285448
## SRR13927735#ACGTTAGGTCAACTGT-1                      -8.809260
## SRR13927735#AAATGCCCAGCAATGG-1                      -9.261216
##                                iLSI_Combined#UMAP_Dimension_2
## SRR13927735#TTATGTCTCCAGGTAT-1                     -0.2908237
## SRR13927735#TATTGCTCATCAGAAA-1                     -0.2892935
## SRR13927735#TTCGATTGTAGGGTTG-1                     -0.2154103
## SRR13927735#CATTCATTCGGATGTT-1                     -0.3267481
## SRR13927735#ACGTTAGGTCAACTGT-1                     -0.2168703
## SRR13927735#AAATGCCCAGCAATGG-1                      0.3200356</code></pre>
</div>
<div id="retrieve-matrices-from-archr-project" class="section level3 hasAnchor" number="4.1.2">
<h3><span class="header-section-number">4.1.2</span> Retrieve matrices from ArchR project<a href="prostate-cancer-cells.html#retrieve-matrices-from-archr-project" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Retrieve gene expression and peak matrix from the ArchR project</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="prostate-cancer-cells.html#cb8-1" aria-hidden="true" tabindex="-1"></a>GeneExpressionMatrix <span class="ot">&lt;-</span> <span class="fu">getMatrixFromProject</span>(</span>
<span id="cb8-2"><a href="prostate-cancer-cells.html#cb8-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">ArchRProj =</span> proj,</span>
<span id="cb8-3"><a href="prostate-cancer-cells.html#cb8-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">useMatrix =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>,</span>
<span id="cb8-4"><a href="prostate-cancer-cells.html#cb8-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">useSeqnames =</span> <span class="cn">NULL</span>,</span>
<span id="cb8-5"><a href="prostate-cancer-cells.html#cb8-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb8-6"><a href="prostate-cancer-cells.html#cb8-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">binarize =</span> <span class="cn">FALSE</span>,</span>
<span id="cb8-7"><a href="prostate-cancer-cells.html#cb8-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">threads =</span> <span class="dv">1</span>,</span>
<span id="cb8-8"><a href="prostate-cancer-cells.html#cb8-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">logFile =</span> <span class="st">&quot;x&quot;</span></span>
<span id="cb8-9"><a href="prostate-cancer-cells.html#cb8-9" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## 2023-11-14 06:59:35.49956 : Organizing colData, 1.476 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 06:59:35.639859 : Organizing rowData, 1.478 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 06:59:35.643214 : Organizing rowRanges, 1.478 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 06:59:35.649084 : Organizing Assays (1 of 1), 1.478 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 06:59:41.919702 : Constructing SummarizedExperiment, 1.583 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 06:59:42.634202 : Finished Matrix Creation, 1.595 mins elapsed.</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="prostate-cancer-cells.html#cb15-1" aria-hidden="true" tabindex="-1"></a>PeakMatrix <span class="ot">&lt;-</span> <span class="fu">getMatrixFromProject</span>(</span>
<span id="cb15-2"><a href="prostate-cancer-cells.html#cb15-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">ArchRProj =</span> proj,</span>
<span id="cb15-3"><a href="prostate-cancer-cells.html#cb15-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">useMatrix =</span> <span class="st">&quot;PeakMatrix&quot;</span>,</span>
<span id="cb15-4"><a href="prostate-cancer-cells.html#cb15-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">useSeqnames =</span> <span class="cn">NULL</span>,</span>
<span id="cb15-5"><a href="prostate-cancer-cells.html#cb15-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb15-6"><a href="prostate-cancer-cells.html#cb15-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">binarize =</span> <span class="cn">FALSE</span>,</span>
<span id="cb15-7"><a href="prostate-cancer-cells.html#cb15-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">threads =</span> <span class="dv">1</span>,</span>
<span id="cb15-8"><a href="prostate-cancer-cells.html#cb15-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">logFile =</span> <span class="st">&quot;x&quot;</span></span>
<span id="cb15-9"><a href="prostate-cancer-cells.html#cb15-9" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## 2023-11-14 07:00:30.221794 : Organizing colData, 0.793 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:00:30.358536 : Organizing rowData, 0.795 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:00:30.366956 : Organizing rowRanges, 0.795 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:00:30.377451 : Organizing Assays (1 of 1), 0.796 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:00:33.562546 : Constructing SummarizedExperiment, 0.849 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:00:46.584914 : Finished Matrix Creation, 1.066 mins elapsed.</code></pre>
<p>If we extract the gene expression from matrix, it will be in the form of RangedSummarizedExperiment.
We can make use of <code>ArchRMatrix2SCE</code> to convert gene expression matrix to SingleCellExperiment object.
It’s also important to note that gene expression from ArchR is library size normalized. We further transform it to logcounts by adding a pseudocount of 1 and taking log2.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="prostate-cancer-cells.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(epiregulon)</span></code></pre></div>
<pre><code>## Loading required package: SingleCellExperiment</code></pre>
<pre><code>## Warning: replacing previous import &#39;GenomicRanges::union&#39; by &#39;igraph::union&#39;
## when loading &#39;epiregulon&#39;</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="prostate-cancer-cells.html#cb25-1" aria-hidden="true" tabindex="-1"></a>GeneExpressionMatrix <span class="ot">&lt;-</span> <span class="fu">ArchRMatrix2SCE</span>(GeneExpressionMatrix)</span>
<span id="cb25-2"><a href="prostate-cancer-cells.html#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(GeneExpressionMatrix) <span class="ot">&lt;-</span> <span class="fu">rowData</span>(GeneExpressionMatrix)<span class="sc">$</span>name</span>
<span id="cb25-3"><a href="prostate-cancer-cells.html#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(GeneExpressionMatrix) <span class="ot">&lt;-</span> <span class="fu">as</span>(<span class="fu">log2</span>(<span class="fu">assay</span>(GeneExpressionMatrix)<span class="sc">+</span><span class="dv">1</span>), <span class="st">&quot;CsparseMatrix&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in asMethod(object): sparse-&gt;dense coercion: allocating vector of size
## 2.7 GiB</code></pre>
<p>We rename the assay name of the PeakMatrix as counts</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="prostate-cancer-cells.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">assayNames</span>(PeakMatrix)  <span class="ot">&lt;-</span>  <span class="st">&quot;counts&quot;</span></span></code></pre></div>
<p>Transfer embeddings from ArchR project to singleCellExperiment</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="prostate-cancer-cells.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">reducedDim</span>(GeneExpressionMatrix, <span class="st">&quot;UMAP_Combined&quot;</span>) <span class="ot">&lt;-</span> <span class="fu">getEmbedding</span>(<span class="at">ArchRProj =</span> proj, </span>
<span id="cb28-2"><a href="prostate-cancer-cells.html#cb28-2" aria-hidden="true" tabindex="-1"></a>                                                                  <span class="at">embedding =</span> <span class="st">&quot;UMAP_Combined&quot;</span>, </span>
<span id="cb28-3"><a href="prostate-cancer-cells.html#cb28-3" aria-hidden="true" tabindex="-1"></a>                                                                  <span class="at">returnDF =</span> <span class="cn">TRUE</span>)[<span class="fu">colnames</span>(GeneExpressionMatrix),]</span>
<span id="cb28-4"><a href="prostate-cancer-cells.html#cb28-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="prostate-cancer-cells.html#cb28-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-6"><a href="prostate-cancer-cells.html#cb28-6" aria-hidden="true" tabindex="-1"></a><span class="co"># add cell label</span></span>
<span id="cb28-7"><a href="prostate-cancer-cells.html#cb28-7" aria-hidden="true" tabindex="-1"></a>GeneExpressionMatrix<span class="sc">$</span>label <span class="ot">&lt;-</span> GeneExpressionMatrix<span class="sc">$</span>Cells</span>
<span id="cb28-8"><a href="prostate-cancer-cells.html#cb28-8" aria-hidden="true" tabindex="-1"></a>GeneExpressionMatrix<span class="sc">$</span>label[GeneExpressionMatrix<span class="sc">$</span>Treatment <span class="sc">==</span> <span class="st">&quot;enzalutamide 48h&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;LNCaP–ENZ48&quot;</span></span>
<span id="cb28-9"><a href="prostate-cancer-cells.html#cb28-9" aria-hidden="true" tabindex="-1"></a>GeneExpressionMatrix<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">factor</span>(GeneExpressionMatrix<span class="sc">$</span>label, </span>
<span id="cb28-10"><a href="prostate-cancer-cells.html#cb28-10" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;LNCaP&quot;</span>, <span class="st">&quot;LNCaP–ENZ48&quot;</span>, <span class="st">&quot;LNCaP RES-A&quot;</span>, <span class="st">&quot;LNCaP RES-B&quot;</span>))</span></code></pre></div>
<p>Visualize singleCellExperiment by UMAP</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="prostate-cancer-cells.html#cb29-1" aria-hidden="true" tabindex="-1"></a>scater<span class="sc">::</span><span class="fu">plotReducedDim</span>(GeneExpressionMatrix, </span>
<span id="cb29-2"><a href="prostate-cancer-cells.html#cb29-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">dimred =</span> <span class="st">&quot;UMAP_Combined&quot;</span>, </span>
<span id="cb29-3"><a href="prostate-cancer-cells.html#cb29-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">text_by =</span> <span class="st">&quot;label&quot;</span>, </span>
<span id="cb29-4"><a href="prostate-cancer-cells.html#cb29-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">colour_by =</span> <span class="st">&quot;label&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
</div>
</div>
<div id="quick-start-2" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Quick start<a href="prostate-cancer-cells.html#quick-start-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="retrieve-bulk-tf-chip-seq-binding-sites-2" class="section level3 hasAnchor" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Retrieve bulk TF ChIP-seq binding sites<a href="prostate-cancer-cells.html#retrieve-bulk-tf-chip-seq-binding-sites-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First, we retrieve the information of TF binding sites collected from Cistrome and ENCODE ChIP-seq, which are hosted on Genomitory. Currently, human genomes HG19 and HG38 and mouse mm10 are available.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="prostate-cancer-cells.html#cb30-1" aria-hidden="true" tabindex="-1"></a>grl <span class="ot">&lt;-</span> <span class="fu">getTFMotifInfo</span>(<span class="at">genome =</span> <span class="st">&quot;hg38&quot;</span>)</span></code></pre></div>
<pre><code>## see ?scMultiome and browseVignettes(&#39;scMultiome&#39;) for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="prostate-cancer-cells.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(grl)</span></code></pre></div>
<pre><code>## GRangesList object of length 6:
## $`5-hmC`
## GRanges object with 24048 ranges and 0 metadata columns:
##           seqnames            ranges strand
##              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;
##       [1]     chr1       10000-10685      *
##       [2]     chr1       13362-13694      *
##       [3]     chr1       29631-29989      *
##       [4]     chr1       40454-40754      *
##       [5]     chr1     135395-135871      *
##       ...      ...               ...    ...
##   [24044]     chrY 56864377-56864627      *
##   [24045]     chrY 56876124-56876182      *
##   [24046]     chrM           84-2450      *
##   [24047]     chrM       13613-14955      *
##   [24048]     chrM       15134-16490      *
##   -------
##   seqinfo: 25 sequences from an unspecified genome; no seqlengths
## 
## ...
## &lt;5 more elements&gt;</code></pre>
</div>
<div id="link-atac-seq-peaks-to-target-genes-1" class="section level3 hasAnchor" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> Link ATAC-seq peaks to target genes<a href="prostate-cancer-cells.html#link-atac-seq-peaks-to-target-genes-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Next, we compute peak to gene correlations using the <code>addPeak2GeneLinks</code> function from the ArchR package. The user would need
to supply a path to an ArchR project already containing peak and gene matrices, as well as Latent semantic indexing (LSI) dimensionality reduction.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="prostate-cancer-cells.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># path to ArchR project</span></span>
<span id="cb35-2"><a href="prostate-cancer-cells.html#cb35-2" aria-hidden="true" tabindex="-1"></a>p2g <span class="ot">&lt;-</span> <span class="fu">calculateP2G</span>(<span class="at">ArchR_path =</span> archR_project_path, </span>
<span id="cb35-3"><a href="prostate-cancer-cells.html#cb35-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">useDim =</span> <span class="st">&quot;iLSI_Combined&quot;</span>, </span>
<span id="cb35-4"><a href="prostate-cancer-cells.html#cb35-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">useMatrix =</span> <span class="st">&quot;GeneIntegrationMatrix&quot;</span>,</span>
<span id="cb35-5"><a href="prostate-cancer-cells.html#cb35-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">threads =</span> <span class="dv">1</span>,</span>
<span id="cb35-6"><a href="prostate-cancer-cells.html#cb35-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">exp_assay =</span> <span class="st">&quot;counts&quot;</span>)</span></code></pre></div>
<pre><code>## Setting ArchRLogging = FALSE</code></pre>
<pre><code>## Using ArchR to compute peak to gene links...</code></pre>
<pre><code>## 2023-11-14 07:02:19.222277 : Getting Available Matrices, 0 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:02:20.147558 : Filtered Low Prediction Score Cells (0 of 15522, 0), 0.002 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:02:20.624869 : Computing KNN, 0.01 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:02:21.565718 : Identifying Non-Overlapping KNN pairs, 0.025 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:02:23.892419 : Identified 497 Groupings!, 0.064 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:02:23.947447 : Getting Group RNA Matrix, 0.065 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:04:15.513178 : Getting Group ATAC Matrix, 1.924 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:06:11.84156 : Normalizing Group Matrices, 3.863 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:06:19.029902 : Finding Peak Gene Pairings, 3.983 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:06:19.627915 : Computing Correlations, 3.993 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:06:28.863833 : Completed Peak2Gene Correlations!, 4.147 mins elapsed.</code></pre>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="prostate-cancer-cells.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(p2g)</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 8 columns
##     idxATAC      chr     start       end    idxRNA      target Correlation
##   &lt;integer&gt; &lt;factor&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt;   &lt;numeric&gt;
## 1        15     chr1    912762    913262         7       NOC2L    0.543478
## 2        25     chr1    920261    920761         7       NOC2L    0.602633
## 3        25     chr1    920261    920761         8      KLHL17    0.597510
## 4        32     chr1    927728    928228         7       NOC2L    0.660584
## 5        32     chr1    927728    928228         8      KLHL17    0.532240
## 6        73     chr1   1079335   1079835        23        SDF4    0.598232
##    distance
##   &lt;numeric&gt;
## 1     46297
## 2     38798
## 3     40076
## 4     31331
## 5     32609
## 6    152446</code></pre>
</div>
<div id="add-tf-motif-binding-to-peaks-2" class="section level3 hasAnchor" number="4.2.3">
<h3><span class="header-section-number">4.2.3</span> Add TF motif binding to peaks<a href="prostate-cancer-cells.html#add-tf-motif-binding-to-peaks-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The next step is to add the TF motif binding information by overlapping the regions of the peak matrix with the bulk chip-seq database loaded in 2. The user can supply either an archR project path and this function will retrieve the peak matrix, or a peakMatrix in the form of a Granges object or RangedSummarizedExperiment.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="prostate-cancer-cells.html#cb51-1" aria-hidden="true" tabindex="-1"></a>overlap <span class="ot">&lt;-</span> <span class="fu">addTFMotifInfo</span>(<span class="at">archR_project_path =</span> archR_project_path, <span class="at">grl =</span> grl, <span class="at">p2g =</span> p2g)</span></code></pre></div>
<pre><code>## Successfully loaded ArchRProject!</code></pre>
<pre><code>## Computing overlap...</code></pre>
<pre><code>## Success!</code></pre>
</div>
<div id="generate-regulons-2" class="section level3 hasAnchor" number="4.2.4">
<h3><span class="header-section-number">4.2.4</span> Generate regulons<a href="prostate-cancer-cells.html#generate-regulons-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A long format dataframe, representing the inferred regulons, is then generated. The dataframe consists of three columns:</p>
<ul>
<li>tf (transcription factor)</li>
<li>target gene</li>
<li>peak to gene correlation between tf and target gene</li>
</ul>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="prostate-cancer-cells.html#cb55-1" aria-hidden="true" tabindex="-1"></a>regulon <span class="ot">&lt;-</span> <span class="fu">getRegulon</span>(<span class="at">p2g =</span> p2g, <span class="at">overlap =</span> overlap, <span class="at">aggregate =</span> <span class="cn">FALSE</span>)</span>
<span id="cb55-2"><a href="prostate-cancer-cells.html#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(regulon)</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 10 columns
##     idxATAC      chr     start       end    idxRNA      target     corr
##   &lt;integer&gt; &lt;factor&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt; &lt;matrix&gt;
## 1        15     chr1    912762    913262         7       NOC2L         
## 2        15     chr1    912762    913262         7       NOC2L         
## 3        15     chr1    912762    913262         7       NOC2L         
## 4        15     chr1    912762    913262         7       NOC2L         
## 5        15     chr1    912762    913262         7       NOC2L         
## 6        15     chr1    912762    913262         7       NOC2L         
##    distance     idxTF          tf
##   &lt;numeric&gt; &lt;integer&gt; &lt;character&gt;
## 1     46297        10        AGO1
## 2     46297        22    AML1-ETO
## 3     46297        32      ARID4A
## 4     46297        33      ARID4B
## 5     46297        34      ARID5B
## 6     46297        80        BCOR</code></pre>
<p>Epiregulon computes weights using either correlation, linear regression, mutual information, log fold change or wilcoxon rank sum test. The choice of methods depends on the datasets. Correlation works best when increased TF activity results from increased TF expression, such in the case of normal development. The user has a choice between computing the correlation of TF expression vs target gene expression by setting <code>method = "corr"</code>, or the product of TF expression and chromatin accessibility at TF-bound regulatory elements vs target gene expression by setting <code>method = "corr"</code> and <code>"tf_re.merge = TRUE</code>.</p>
<p>In the case of drug treatment, however, the activity of TF is suppressed often not by downregulation of the TF gene expression, but by direct interference of the TF protein function. In this dataset, the drug enzalutamide blocks the ligand binding domain of the androgen receptor and prevents it from binding to the chromatin. As a result, while the AR gene expression stays the same, the chromatin accessibility of AR, as computed by chromVar in the ArchR package, is greatly reduced by 48 hour treatment of enzalutamide.</p>
<p>First, we visualize the AR expression and observed that enzalutamide did not decrease AR expression.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="prostate-cancer-cells.html#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotActivityDim</span>(<span class="at">sce =</span> GeneExpressionMatrix,</span>
<span id="cb57-2"><a href="prostate-cancer-cells.html#cb57-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">activity_matrix =</span> <span class="fu">assay</span>(GeneExpressionMatrix), </span>
<span id="cb57-3"><a href="prostate-cancer-cells.html#cb57-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">tf =</span> <span class="st">&quot;AR&quot;</span>, </span>
<span id="cb57-4"><a href="prostate-cancer-cells.html#cb57-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">dimtype =</span> <span class="st">&quot;UMAP_Combined&quot;</span>, </span>
<span id="cb57-5"><a href="prostate-cancer-cells.html#cb57-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="st">&quot;label&quot;</span>, </span>
<span id="cb57-6"><a href="prostate-cancer-cells.html#cb57-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">point_size =</span> <span class="dv">1</span>,</span>
<span id="cb57-7"><a href="prostate-cancer-cells.html#cb57-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">legend.label =</span> <span class="st">&quot;gene expression&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<p>Then we extract the chromVarMatrix from ArchR project and then visualize the chromatin accessibility at AR bound sites. We can see that 48 hour of enzalutamide treatment reduced chromatin accessibility at AR bound sites</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="prostate-cancer-cells.html#cb58-1" aria-hidden="true" tabindex="-1"></a>chromVarMatrix <span class="ot">&lt;-</span> <span class="fu">getMatrixFromProject</span>(</span>
<span id="cb58-2"><a href="prostate-cancer-cells.html#cb58-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">ArchRProj =</span> proj,</span>
<span id="cb58-3"><a href="prostate-cancer-cells.html#cb58-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">useMatrix =</span> <span class="st">&quot;MotifMatrix&quot;</span>,</span>
<span id="cb58-4"><a href="prostate-cancer-cells.html#cb58-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">useSeqnames =</span> <span class="cn">NULL</span>,</span>
<span id="cb58-5"><a href="prostate-cancer-cells.html#cb58-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb58-6"><a href="prostate-cancer-cells.html#cb58-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">binarize =</span> <span class="cn">FALSE</span>,</span>
<span id="cb58-7"><a href="prostate-cancer-cells.html#cb58-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">threads =</span> <span class="dv">1</span></span>
<span id="cb58-8"><a href="prostate-cancer-cells.html#cb58-8" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## 2023-11-14 07:07:53.896806 : Organizing colData, 0.179 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:07:54.03119 : Organizing rowData, 0.181 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:07:54.033689 : Organizing rowRanges, 0.181 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:07:54.038551 : Organizing Assays (1 of 2), 0.181 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:07:54.423194 : Organizing Assays (2 of 2), 0.188 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:07:54.802267 : Constructing SummarizedExperiment, 0.194 mins elapsed.</code></pre>
<pre><code>## 2023-11-14 07:07:56.879486 : Finished Matrix Creation, 0.228 mins elapsed.</code></pre>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="prostate-cancer-cells.html#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotActivityDim</span>(<span class="at">sce =</span> GeneExpressionMatrix,</span>
<span id="cb66-2"><a href="prostate-cancer-cells.html#cb66-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">activity_matrix =</span> <span class="fu">assay</span>(chromVarMatrix, <span class="st">&quot;z&quot;</span>), </span>
<span id="cb66-3"><a href="prostate-cancer-cells.html#cb66-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">tf =</span> <span class="st">&quot;AR_689&quot;</span>, </span>
<span id="cb66-4"><a href="prostate-cancer-cells.html#cb66-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">dimtype =</span> <span class="st">&quot;UMAP_Combined&quot;</span>, </span>
<span id="cb66-5"><a href="prostate-cancer-cells.html#cb66-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="st">&quot;label&quot;</span>, </span>
<span id="cb66-6"><a href="prostate-cancer-cells.html#cb66-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">point_size =</span> <span class="dv">1</span>,</span>
<span id="cb66-7"><a href="prostate-cancer-cells.html#cb66-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">legend.label =</span> <span class="st">&quot;chromVar&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<p>Therefore, we consider the choice of the <code>wilcoxon</code> test which compare target gene expression in cells meeting both the TF expression and accessibility cutoffs vs cells failing either the TF expression or/and accessibility cutoffs. We also compare the output of <code>wilcoxon</code> vs <code>corr</code>.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="prostate-cancer-cells.html#cb67-1" aria-hidden="true" tabindex="-1"></a>regulon.w.wilcox <span class="ot">&lt;-</span> <span class="fu">addWeights</span>(<span class="at">regulon =</span> pruned.regulon,</span>
<span id="cb67-2"><a href="prostate-cancer-cells.html#cb67-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">expMatrix =</span> GeneExpressionMatrix,</span>
<span id="cb67-3"><a href="prostate-cancer-cells.html#cb67-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">exp_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb67-4"><a href="prostate-cancer-cells.html#cb67-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">peakMatrix =</span> PeakMatrix,</span>
<span id="cb67-5"><a href="prostate-cancer-cells.html#cb67-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">peak_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb67-6"><a href="prostate-cancer-cells.html#cb67-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>Sample,</span>
<span id="cb67-7"><a href="prostate-cancer-cells.html#cb67-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">method =</span> <span class="st">&quot;wilcoxon&quot;</span>)</span></code></pre></div>
<pre><code>## adding weights using wilcoxon...</code></pre>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="prostate-cancer-cells.html#cb69-1" aria-hidden="true" tabindex="-1"></a>regulon.w.corr <span class="ot">&lt;-</span> <span class="fu">addWeights</span>(<span class="at">regulon =</span> pruned.regulon,</span>
<span id="cb69-2"><a href="prostate-cancer-cells.html#cb69-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">expMatrix =</span> GeneExpressionMatrix,</span>
<span id="cb69-3"><a href="prostate-cancer-cells.html#cb69-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">exp_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb69-4"><a href="prostate-cancer-cells.html#cb69-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">peakMatrix =</span> PeakMatrix,</span>
<span id="cb69-5"><a href="prostate-cancer-cells.html#cb69-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">peak_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb69-6"><a href="prostate-cancer-cells.html#cb69-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>Sample,</span>
<span id="cb69-7"><a href="prostate-cancer-cells.html#cb69-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">method =</span> <span class="st">&quot;corr&quot;</span>)</span></code></pre></div>
<pre><code>## adding weights using corr...</code></pre>
<pre><code>## calculating average expression across clusters...</code></pre>
<pre><code>## computing weights...</code></pre>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="prostate-cancer-cells.html#cb73-1" aria-hidden="true" tabindex="-1"></a>regulon.w.corr.re <span class="ot">&lt;-</span> <span class="fu">addWeights</span>(<span class="at">regulon =</span> pruned.regulon,</span>
<span id="cb73-2"><a href="prostate-cancer-cells.html#cb73-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">expMatrix =</span> GeneExpressionMatrix,</span>
<span id="cb73-3"><a href="prostate-cancer-cells.html#cb73-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">exp_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb73-4"><a href="prostate-cancer-cells.html#cb73-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">peakMatrix =</span> PeakMatrix,</span>
<span id="cb73-5"><a href="prostate-cancer-cells.html#cb73-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">peak_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb73-6"><a href="prostate-cancer-cells.html#cb73-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>Sample,</span>
<span id="cb73-7"><a href="prostate-cancer-cells.html#cb73-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">method =</span> <span class="st">&quot;corr&quot;</span>,</span>
<span id="cb73-8"><a href="prostate-cancer-cells.html#cb73-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">tf_re.merge =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## adding weights using corr...</code></pre>
<pre><code>## calculating average expression across clusters...</code></pre>
<pre><code>## computing weights...</code></pre>
</div>
<div id="calculate-tf-activity-2" class="section level3 hasAnchor" number="4.2.5">
<h3><span class="header-section-number">4.2.5</span> Calculate TF activity<a href="prostate-cancer-cells.html#calculate-tf-activity-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Finally, the activities for a specific TF in each cell are computed by averaging the weighted expressions of target genes linked to the TF.
<span class="math display">\[y=\frac{1}{n}\sum_{i=1}^{n} x_i * weight_i\]</span>
where <span class="math inline">\(y\)</span> is the activity of a TF for a cell
<span class="math inline">\(n\)</span> is the total number of targets for a TF
<span class="math inline">\(x_i\)</span> is the log count expression of target i where i in {1,2,…,n}
<span class="math inline">\(weight_i\)</span> is the weight of TF and target i</p>
<p>We calculate three different activities corresponding to the different weighted regulons</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="prostate-cancer-cells.html#cb77-1" aria-hidden="true" tabindex="-1"></a>score.combine.wilcox <span class="ot">&lt;-</span> <span class="fu">calculateActivity</span>(<span class="at">expMatrix =</span> GeneExpressionMatrix,</span>
<span id="cb77-2"><a href="prostate-cancer-cells.html#cb77-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">exp_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb77-3"><a href="prostate-cancer-cells.html#cb77-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">regulon =</span> regulon.w.wilcox,</span>
<span id="cb77-4"><a href="prostate-cancer-cells.html#cb77-4" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">normalize =</span> <span class="cn">TRUE</span>,</span>
<span id="cb77-5"><a href="prostate-cancer-cells.html#cb77-5" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">mode =</span> <span class="st">&quot;weight&quot;</span>,</span>
<span id="cb77-6"><a href="prostate-cancer-cells.html#cb77-6" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">&quot;weightedMean&quot;</span>)</span></code></pre></div>
<pre><code>## calculating TF activity from regulon using weightedmean</code></pre>
<pre><code>## Warning in calculateActivity(expMatrix = GeneExpressionMatrix, exp_assay =
## &quot;counts&quot;, : The weight column contains multiple subcolumns but no cluster
## information was provided. Using first column to compute activity...</code></pre>
<pre><code>## aggregating regulons...</code></pre>
<pre><code>## creating weight matrix...</code></pre>
<pre><code>## calculating activity scores...</code></pre>
<pre><code>## normalize by mean...</code></pre>
<pre><code>## normalize by the number of targets...</code></pre>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="prostate-cancer-cells.html#cb85-1" aria-hidden="true" tabindex="-1"></a>score.combine.corr <span class="ot">&lt;-</span> <span class="fu">calculateActivity</span>(<span class="at">expMatrix =</span> GeneExpressionMatrix,</span>
<span id="cb85-2"><a href="prostate-cancer-cells.html#cb85-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">exp_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb85-3"><a href="prostate-cancer-cells.html#cb85-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">regulon =</span> regulon.w.corr,</span>
<span id="cb85-4"><a href="prostate-cancer-cells.html#cb85-4" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">normalize =</span> <span class="cn">TRUE</span>,</span>
<span id="cb85-5"><a href="prostate-cancer-cells.html#cb85-5" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">mode =</span> <span class="st">&quot;weight&quot;</span>,</span>
<span id="cb85-6"><a href="prostate-cancer-cells.html#cb85-6" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">&quot;weightedMean&quot;</span>)</span></code></pre></div>
<pre><code>## calculating TF activity from regulon using weightedmean</code></pre>
<pre><code>## aggregating regulons...</code></pre>
<pre><code>## creating weight matrix...</code></pre>
<pre><code>## calculating activity scores...</code></pre>
<pre><code>## normalize by mean...</code></pre>
<pre><code>## normalize by the number of targets...</code></pre>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="prostate-cancer-cells.html#cb92-1" aria-hidden="true" tabindex="-1"></a>score.combine.corr.re <span class="ot">&lt;-</span> <span class="fu">calculateActivity</span>(<span class="at">expMatrix =</span> GeneExpressionMatrix,</span>
<span id="cb92-2"><a href="prostate-cancer-cells.html#cb92-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">exp_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb92-3"><a href="prostate-cancer-cells.html#cb92-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">regulon =</span> regulon.w.corr.re,</span>
<span id="cb92-4"><a href="prostate-cancer-cells.html#cb92-4" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">normalize =</span> <span class="cn">TRUE</span>,</span>
<span id="cb92-5"><a href="prostate-cancer-cells.html#cb92-5" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">mode =</span> <span class="st">&quot;weight&quot;</span>,</span>
<span id="cb92-6"><a href="prostate-cancer-cells.html#cb92-6" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">&quot;weightedMean&quot;</span>)</span></code></pre></div>
<pre><code>## calculating TF activity from regulon using weightedmean</code></pre>
<pre><code>## aggregating regulons...</code></pre>
<pre><code>## creating weight matrix...</code></pre>
<pre><code>## calculating activity scores...</code></pre>
<pre><code>## normalize by mean...</code></pre>
<pre><code>## normalize by the number of targets...</code></pre>
<p>We visualize the different activities side by side</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="prostate-cancer-cells.html#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotActivityViolin</span>(<span class="at">activity_matrix =</span> score.combine.wilcox, </span>
<span id="cb99-2"><a href="prostate-cancer-cells.html#cb99-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">tf =</span> <span class="fu">c</span>( <span class="st">&quot;AR&quot;</span>), </span>
<span id="cb99-3"><a href="prostate-cancer-cells.html#cb99-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>label) <span class="sc">+</span> <span class="fu">ggtitle</span> (<span class="st">&quot;AR activity by wilcoxon&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="prostate-cancer-cells.html#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotActivityViolin</span>(<span class="at">activity_matrix =</span> score.combine.corr, </span>
<span id="cb100-2"><a href="prostate-cancer-cells.html#cb100-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">tf =</span> <span class="fu">c</span>( <span class="st">&quot;AR&quot;</span>), </span>
<span id="cb100-3"><a href="prostate-cancer-cells.html#cb100-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>label) <span class="sc">+</span> <span class="fu">ggtitle</span> (<span class="st">&quot;AR activity by corr TF vs TG&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-11-2.png" width="672" /></p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="prostate-cancer-cells.html#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotActivityViolin</span>(<span class="at">activity_matrix =</span> score.combine.corr.re, </span>
<span id="cb101-2"><a href="prostate-cancer-cells.html#cb101-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">tf =</span> <span class="fu">c</span>( <span class="st">&quot;AR&quot;</span>), </span>
<span id="cb101-3"><a href="prostate-cancer-cells.html#cb101-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>label) <span class="sc">+</span> <span class="fu">ggtitle</span> (<span class="st">&quot;AR activity by corr TF*RE vs TG&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-11-3.png" width="672" /></p>
</div>
<div id="perform-differential-activity-1" class="section level3 hasAnchor" number="4.2.6">
<h3><span class="header-section-number">4.2.6</span> Perform differential activity<a href="prostate-cancer-cells.html#perform-differential-activity-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="prostate-cancer-cells.html#cb102-1" aria-hidden="true" tabindex="-1"></a>markers <span class="ot">&lt;-</span> <span class="fu">findDifferentialActivity</span>(<span class="at">activity_matrix =</span> score.combine.wilcox, </span>
<span id="cb102-2"><a href="prostate-cancer-cells.html#cb102-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">groups =</span> GeneExpressionMatrix<span class="sc">$</span>label, </span>
<span id="cb102-3"><a href="prostate-cancer-cells.html#cb102-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">pval.type =</span> <span class="st">&quot;some&quot;</span>, </span>
<span id="cb102-4"><a href="prostate-cancer-cells.html#cb102-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">direction =</span> <span class="st">&quot;up&quot;</span>, </span>
<span id="cb102-5"><a href="prostate-cancer-cells.html#cb102-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">test.type =</span> <span class="st">&quot;t&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: The `groups` argument of `findDifferentialActivity()` is deprecated as of
## epiregulon 1.0.0.
## ℹ Please use the `clusters` argument instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.</code></pre>
<p>Take the top TFs</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="prostate-cancer-cells.html#cb104-1" aria-hidden="true" tabindex="-1"></a>markers.sig <span class="ot">&lt;-</span> <span class="fu">getSigGenes</span>(markers, <span class="at">topgenes =</span> <span class="dv">8</span> )</span></code></pre></div>
<pre><code>## Using a logFC cutoff of 0 for class LNCaP</code></pre>
<pre><code>## Using a logFC cutoff of 0 for class LNCaP–ENZ48</code></pre>
<pre><code>## Using a logFC cutoff of 0 for class LNCaP RES-A</code></pre>
<pre><code>## Using a logFC cutoff of 0 for class LNCaP RES-B</code></pre>
</div>
<div id="visualize-the-results-1" class="section level3 hasAnchor" number="4.2.7">
<h3><span class="header-section-number">4.2.7</span> Visualize the results<a href="prostate-cancer-cells.html#visualize-the-results-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First visualize the known differential TFs by bubble plot</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="prostate-cancer-cells.html#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotBubble</span>(<span class="at">activity_matrix =</span> score.combine.wilcox, </span>
<span id="cb109-2"><a href="prostate-cancer-cells.html#cb109-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">tf =</span> <span class="fu">c</span>(<span class="st">&quot;AR&quot;</span>,<span class="st">&quot;FOXA1&quot;</span>, <span class="st">&quot;MYC&quot;</span>,<span class="st">&quot;JUN&quot;</span>), </span>
<span id="cb109-3"><a href="prostate-cancer-cells.html#cb109-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>label)</span></code></pre></div>
<p><img src="_main_files/figure-html/visualization_prostate-1.png" width="672" /></p>
<p>Then visualize the most differential TFs by clusters</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="prostate-cancer-cells.html#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotBubble</span>(<span class="at">activity_matrix =</span> score.combine.wilcox, </span>
<span id="cb110-2"><a href="prostate-cancer-cells.html#cb110-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">tf =</span> markers.sig<span class="sc">$</span>tf, </span>
<span id="cb110-3"><a href="prostate-cancer-cells.html#cb110-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>label)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<p>Visualize the known differential TFs by UMAP</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="prostate-cancer-cells.html#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotActivityDim</span>(<span class="at">sce =</span> GeneExpressionMatrix,</span>
<span id="cb111-2"><a href="prostate-cancer-cells.html#cb111-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">activity_matrix =</span> score.combine.wilcox, </span>
<span id="cb111-3"><a href="prostate-cancer-cells.html#cb111-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">tf =</span> <span class="fu">c</span>( <span class="st">&quot;AR&quot;</span>, <span class="st">&quot;FOXA1&quot;</span>, <span class="st">&quot;MYC&quot;</span>, <span class="st">&quot;JUN&quot;</span>), </span>
<span id="cb111-4"><a href="prostate-cancer-cells.html#cb111-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">dimtype =</span> <span class="st">&quot;UMAP_Combined&quot;</span>, </span>
<span id="cb111-5"><a href="prostate-cancer-cells.html#cb111-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="st">&quot;label&quot;</span>, </span>
<span id="cb111-6"><a href="prostate-cancer-cells.html#cb111-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">point_size =</span> <span class="dv">1</span>, </span>
<span id="cb111-7"><a href="prostate-cancer-cells.html#cb111-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb111-8"><a href="prostate-cancer-cells.html#cb111-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">nrow =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p>Visualize the newly discovered differential TFs by UMAP</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="prostate-cancer-cells.html#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotActivityDim</span>(<span class="at">sce =</span> GeneExpressionMatrix,</span>
<span id="cb112-2"><a href="prostate-cancer-cells.html#cb112-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">activity_matrix =</span> score.combine.wilcox, </span>
<span id="cb112-3"><a href="prostate-cancer-cells.html#cb112-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">tf =</span> <span class="fu">c</span>(<span class="st">&quot;AFF4&quot;</span>,<span class="st">&quot;ASH1L&quot;</span>,<span class="st">&quot;DAXX&quot;</span>,<span class="st">&quot;MXD4&quot;</span>), </span>
<span id="cb112-4"><a href="prostate-cancer-cells.html#cb112-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">dimtype =</span> <span class="st">&quot;UMAP_Combined&quot;</span>, </span>
<span id="cb112-5"><a href="prostate-cancer-cells.html#cb112-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="st">&quot;label&quot;</span>, </span>
<span id="cb112-6"><a href="prostate-cancer-cells.html#cb112-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">point_size =</span> <span class="dv">1</span>, </span>
<span id="cb112-7"><a href="prostate-cancer-cells.html#cb112-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb112-8"><a href="prostate-cancer-cells.html#cb112-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">nrow =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<p>Visualize regulons by heatmap</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="prostate-cancer-cells.html#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(GeneExpressionMatrix) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb113-2"><a href="prostate-cancer-cells.html#cb113-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-3"><a href="prostate-cancer-cells.html#cb113-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmapRegulon</span>(<span class="at">sce=</span>GeneExpressionMatrix,</span>
<span id="cb113-4"><a href="prostate-cancer-cells.html#cb113-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">tfs=</span> <span class="fu">c</span>( <span class="st">&quot;AR&quot;</span>, <span class="st">&quot;FOXA1&quot;</span>, <span class="st">&quot;MYC&quot;</span>, <span class="st">&quot;JUN&quot;</span>),</span>
<span id="cb113-5"><a href="prostate-cancer-cells.html#cb113-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">regulon=</span>regulon.w.wilcox,</span>
<span id="cb113-6"><a href="prostate-cancer-cells.html#cb113-6" aria-hidden="true" tabindex="-1"></a>                   <span class="at">regulon_cutoff=</span><span class="fl">0.1</span>,</span>
<span id="cb113-7"><a href="prostate-cancer-cells.html#cb113-7" aria-hidden="true" tabindex="-1"></a>                   <span class="at">downsample=</span><span class="dv">1000</span>,</span>
<span id="cb113-8"><a href="prostate-cancer-cells.html#cb113-8" aria-hidden="true" tabindex="-1"></a>                   <span class="at">cell_attributes=</span><span class="st">&quot;label&quot;</span>,</span>
<span id="cb113-9"><a href="prostate-cancer-cells.html#cb113-9" aria-hidden="true" tabindex="-1"></a>                   <span class="at">col_gap=</span><span class="st">&quot;label&quot;</span>,</span>
<span id="cb113-10"><a href="prostate-cancer-cells.html#cb113-10" aria-hidden="true" tabindex="-1"></a>                   <span class="at">exprs_values=</span><span class="st">&quot;counts&quot;</span>,</span>
<span id="cb113-11"><a href="prostate-cancer-cells.html#cb113-11" aria-hidden="true" tabindex="-1"></a>                   <span class="at">name=</span><span class="st">&quot;regulon heatmap&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;e2&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;80&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;93&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;e2&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;80&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;93&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;e2&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;80&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;93&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;e2&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;80&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;93&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;e2&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;80&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;93&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;e2&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;80&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;93&gt;</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="prostate-cancer-cells.html#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmapActivity</span>(<span class="at">activity=</span>score.combine.wilcox,</span>
<span id="cb132-2"><a href="prostate-cancer-cells.html#cb132-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">sce=</span>GeneExpressionMatrix,</span>
<span id="cb132-3"><a href="prostate-cancer-cells.html#cb132-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">tfs=</span><span class="fu">rownames</span>(score.combine.wilcox),</span>
<span id="cb132-4"><a href="prostate-cancer-cells.html#cb132-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">downsample=</span><span class="dv">1000</span>,</span>
<span id="cb132-5"><a href="prostate-cancer-cells.html#cb132-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">cell_attributes=</span><span class="st">&quot;label&quot;</span>,</span>
<span id="cb132-6"><a href="prostate-cancer-cells.html#cb132-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">col_gap=</span><span class="st">&quot;label&quot;</span>,</span>
<span id="cb132-7"><a href="prostate-cancer-cells.html#cb132-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">name =</span> <span class="st">&quot;transcription factor activity&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;e2&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;80&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;93&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;e2&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;80&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;93&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;e2&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;80&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;93&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;e2&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;80&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;93&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;e2&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;80&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;93&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;e2&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;80&gt;</code></pre>
<pre><code>## Warning in grid.Call(C_textBounds, as.graphicsAnnot(x$label), x$x, x$y, :
## conversion failure on &#39;LNCaP–ENZ48&#39; in &#39;mbcsToSbcs&#39;: dot substituted for &lt;93&gt;</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-17-1.png" width="672" />
### Geneset enrichment
Sometimes we are interested to know what pathways are enriched in the regulon of a particular TF. We can perform geneset enrichment using the enricher function from <a href="http://yulab-smu.top/biomedical-knowledge-mining-book/universal-api.html">clusterProfiler</a>.</p>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="prostate-cancer-cells.html#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="co">#retrieve genesets</span></span>
<span id="cb151-2"><a href="prostate-cancer-cells.html#cb151-2" aria-hidden="true" tabindex="-1"></a>H <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">getGenesets</span>(<span class="at">org =</span> <span class="st">&quot;hsa&quot;</span>, </span>
<span id="cb151-3"><a href="prostate-cancer-cells.html#cb151-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">db =</span> <span class="st">&quot;msigdb&quot;</span>, </span>
<span id="cb151-4"><a href="prostate-cancer-cells.html#cb151-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">cat =</span> <span class="st">&quot;H&quot;</span>, </span>
<span id="cb151-5"><a href="prostate-cancer-cells.html#cb151-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">gene.id.type =</span> <span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb151-6"><a href="prostate-cancer-cells.html#cb151-6" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">cache =</span> <span class="cn">FALSE</span>)</span>
<span id="cb151-7"><a href="prostate-cancer-cells.html#cb151-7" aria-hidden="true" tabindex="-1"></a>C6 <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">getGenesets</span>(<span class="at">org =</span> <span class="st">&quot;hsa&quot;</span>, </span>
<span id="cb151-8"><a href="prostate-cancer-cells.html#cb151-8" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">db =</span> <span class="st">&quot;msigdb&quot;</span>, </span>
<span id="cb151-9"><a href="prostate-cancer-cells.html#cb151-9" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">cat =</span> <span class="st">&quot;C6&quot;</span>, </span>
<span id="cb151-10"><a href="prostate-cancer-cells.html#cb151-10" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">gene.id.type =</span> <span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb151-11"><a href="prostate-cancer-cells.html#cb151-11" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">cache =</span> <span class="cn">FALSE</span>)</span>
<span id="cb151-12"><a href="prostate-cancer-cells.html#cb151-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-13"><a href="prostate-cancer-cells.html#cb151-13" aria-hidden="true" tabindex="-1"></a><span class="co">#combine genesets and convert genesets to be compatible with enricher</span></span>
<span id="cb151-14"><a href="prostate-cancer-cells.html#cb151-14" aria-hidden="true" tabindex="-1"></a>gs <span class="ot">&lt;-</span> <span class="fu">c</span>(H,C6)</span>
<span id="cb151-15"><a href="prostate-cancer-cells.html#cb151-15" aria-hidden="true" tabindex="-1"></a>gs.list <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind,<span class="fu">lapply</span>(<span class="fu">names</span>(gs), </span>
<span id="cb151-16"><a href="prostate-cancer-cells.html#cb151-16" aria-hidden="true" tabindex="-1"></a>                                <span class="cf">function</span>(x) {<span class="fu">data.frame</span>(<span class="at">gs=</span>x, <span class="at">genes=</span>gs[[x]])}))</span>
<span id="cb151-17"><a href="prostate-cancer-cells.html#cb151-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-18"><a href="prostate-cancer-cells.html#cb151-18" aria-hidden="true" tabindex="-1"></a>enrichresults <span class="ot">&lt;-</span> <span class="fu">regulonEnrich</span>(<span class="at">TF =</span> <span class="fu">c</span>(<span class="st">&quot;AR&quot;</span>, <span class="st">&quot;FOXA1&quot;</span>, <span class="st">&quot;MYC&quot;</span>, <span class="st">&quot;JUN&quot;</span>), </span>
<span id="cb151-19"><a href="prostate-cancer-cells.html#cb151-19" aria-hidden="true" tabindex="-1"></a>                               <span class="at">regulon =</span> regulon.w.wilcox, </span>
<span id="cb151-20"><a href="prostate-cancer-cells.html#cb151-20" aria-hidden="true" tabindex="-1"></a>                               <span class="at">weight =</span> <span class="st">&quot;weight&quot;</span>,</span>
<span id="cb151-21"><a href="prostate-cancer-cells.html#cb151-21" aria-hidden="true" tabindex="-1"></a>                               <span class="at">weight_cutoff =</span> <span class="fl">0.1</span>, </span>
<span id="cb151-22"><a href="prostate-cancer-cells.html#cb151-22" aria-hidden="true" tabindex="-1"></a>                               <span class="at">genesets =</span> gs.list)</span></code></pre></div>
<pre><code>## AR</code></pre>
<pre><code>## </code></pre>
<pre><code>## FOXA1</code></pre>
<pre><code>## MYC</code></pre>
<pre><code>## JUN</code></pre>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="prostate-cancer-cells.html#cb157-1" aria-hidden="true" tabindex="-1"></a><span class="co">#plot results</span></span>
<span id="cb157-2"><a href="prostate-cancer-cells.html#cb157-2" aria-hidden="true" tabindex="-1"></a><span class="fu">enrichPlot</span>(<span class="at">results =</span> enrichresults, <span class="at">ncol =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/enrichment_prostate-1.png" width="1152" />
### Network analysis</p>
<p>We can visualize the genesets as a network</p>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="prostate-cancer-cells.html#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotGseaNetwork</span>(<span class="at">tf =</span> <span class="fu">names</span>(enrichresults), </span>
<span id="cb158-2"><a href="prostate-cancer-cells.html#cb158-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">enrichresults =</span> enrichresults, </span>
<span id="cb158-3"><a href="prostate-cancer-cells.html#cb158-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">p.adj_cutoff =</span> <span class="fl">0.1</span>, </span>
<span id="cb158-4"><a href="prostate-cancer-cells.html#cb158-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">ntop_pathways =</span> <span class="dv">10</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="prostate-cancer-cells.html#cb159-1" aria-hidden="true" tabindex="-1"></a>enrichresults <span class="ot">&lt;-</span> <span class="fu">regulonEnrich</span>(<span class="at">TF =</span> <span class="fu">c</span>(<span class="st">&quot;AFF4&quot;</span>,<span class="st">&quot;ASH1L&quot;</span>,<span class="st">&quot;DAXX&quot;</span>,<span class="st">&quot;MXD4&quot;</span>,<span class="st">&quot;CDC73&quot;</span>,<span class="st">&quot;REPIN1&quot;</span>,</span>
<span id="cb159-2"><a href="prostate-cancer-cells.html#cb159-2" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">&quot;BCLAF1&quot;</span>,<span class="st">&quot;RBM34&quot;</span>,<span class="st">&quot;CERS6&quot;</span>,<span class="st">&quot;TOP2B&quot;</span>,<span class="st">&quot;ATF5&quot;</span>,<span class="st">&quot;PITX1&quot;</span>,</span>
<span id="cb159-3"><a href="prostate-cancer-cells.html#cb159-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">&quot;EWSR1&quot;</span>,<span class="st">&quot;TOP2A&quot;</span>), </span>
<span id="cb159-4"><a href="prostate-cancer-cells.html#cb159-4" aria-hidden="true" tabindex="-1"></a>                               <span class="at">regulon =</span> regulon.w.wilcox, </span>
<span id="cb159-5"><a href="prostate-cancer-cells.html#cb159-5" aria-hidden="true" tabindex="-1"></a>                               <span class="at">weight =</span> <span class="st">&quot;weight&quot;</span>,</span>
<span id="cb159-6"><a href="prostate-cancer-cells.html#cb159-6" aria-hidden="true" tabindex="-1"></a>                               <span class="at">weight_cutoff =</span> <span class="fl">0.1</span>, </span>
<span id="cb159-7"><a href="prostate-cancer-cells.html#cb159-7" aria-hidden="true" tabindex="-1"></a>                               <span class="at">genesets =</span> gs.list)</span></code></pre></div>
<pre><code>## AFF4</code></pre>
<pre><code>## ASH1L</code></pre>
<pre><code>## DAXX</code></pre>
<pre><code>## MXD4</code></pre>
<pre><code>## CDC73</code></pre>
<pre><code>## REPIN1</code></pre>
<pre><code>## BCLAF1</code></pre>
<pre><code>## RBM34</code></pre>
<pre><code>## CERS6</code></pre>
<pre><code>## TOP2B</code></pre>
<pre><code>## ATF5</code></pre>
<pre><code>## PITX1</code></pre>
<pre><code>## EWSR1</code></pre>
<pre><code>## TOP2A</code></pre>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="prostate-cancer-cells.html#cb174-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotGseaNetwork</span>(<span class="at">tf =</span> <span class="fu">names</span>(enrichresults), </span>
<span id="cb174-2"><a href="prostate-cancer-cells.html#cb174-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">enrichresults =</span> enrichresults, </span>
<span id="cb174-3"><a href="prostate-cancer-cells.html#cb174-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">p.adj_cutoff =</span> <span class="fl">0.1</span>, </span>
<span id="cb174-4"><a href="prostate-cancer-cells.html#cb174-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">ntop_pathways =</span> <span class="dv">10</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
</div>
</div>
<div id="session-info-2" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Session Info<a href="prostate-cancer-cells.html#session-info-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="prostate-cancer-cells.html#cb175-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.3.0 (2023-04-21)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 18.04.6 LTS
## 
## Matrix products: default
## BLAS:   /usr/local/lib/R/lib/libRblas.so 
## LAPACK: /usr/local/lib/R/lib/libRlapack.so;  LAPACK version 3.11.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
##  [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
##  [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
## [10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 
## 
## time zone: Etc/UTC
## tzcode source: system (glibc)
## 
## attached base packages:
## [1] stats4    grid      stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] org.Hs.eg.db_3.18.0         AnnotationDbi_1.63.2       
##  [3] msigdbr_7.5.1               nabor_0.5.0                
##  [5] scMultiome_1.1.2            MultiAssayExperiment_1.27.5
##  [7] ExperimentHub_2.9.1         AnnotationHub_3.9.2        
##  [9] BiocFileCache_2.9.1         dbplyr_2.3.4               
## [11] epiregulon_1.0.35           SingleCellExperiment_1.23.0
## [13] rhdf5_2.45.1                SummarizedExperiment_1.31.1
## [15] Biobase_2.61.0              RcppArmadillo_0.12.6.4.0   
## [17] Rcpp_1.0.11                 Matrix_1.6-1.1             
## [19] GenomicRanges_1.53.3        GenomeInfoDb_1.37.7        
## [21] IRanges_2.35.3              S4Vectors_0.39.3           
## [23] BiocGenerics_0.47.1         sparseMatrixStats_1.13.4   
## [25] MatrixGenerics_1.13.2       matrixStats_1.0.0          
## [27] data.table_1.14.8           stringr_1.5.0              
## [29] plyr_1.8.9                  magrittr_2.0.3             
## [31] ggplot2_3.4.4               gtable_0.3.4               
## [33] gtools_3.9.4                gridExtra_2.3              
## [35] devtools_2.4.5              usethis_2.1.6              
## [37] ArchR_1.0.3                
## 
## loaded via a namespace (and not attached):
##   [1] fs_1.6.2                           GSVA_1.49.8                       
##   [3] bitops_1.0-7                       enrichplot_1.21.3                 
##   [5] HDO.db_0.99.1                      httr_1.4.6                        
##   [7] RColorBrewer_1.1-3                 doParallel_1.0.17                 
##   [9] Rgraphviz_2.45.0                   profvis_0.3.8                     
##  [11] tools_4.3.0                        backports_1.4.1                   
##  [13] utf8_1.2.3                         R6_2.5.1                          
##  [15] HDF5Array_1.29.3                   lazyeval_0.2.2                    
##  [17] rhdf5filters_1.13.5                GetoptLong_1.0.5                  
##  [19] urlchecker_1.0.1                   withr_2.5.0                       
##  [21] prettyunits_1.1.1                  cli_3.6.1                         
##  [23] Cairo_1.6-1                        scatterpie_0.2.1                  
##  [25] labeling_0.4.3                     sass_0.4.6                        
##  [27] KEGGgraph_1.61.0                   yulab.utils_0.1.0                 
##  [29] Rsamtools_2.17.0                   gson_0.1.0                        
##  [31] DOSE_3.27.3                        scater_1.29.4                     
##  [33] sessioninfo_1.2.2                  BSgenome_1.69.1                   
##  [35] limma_3.57.11                      rstudioapi_0.15.0                 
##  [37] RSQLite_2.3.1                      gridGraphics_0.5-1                
##  [39] generics_0.1.3                     shape_1.4.6                       
##  [41] BiocIO_1.11.0                      dplyr_1.1.3                       
##  [43] GO.db_3.18.0                       ggbeeswarm_0.7.2                  
##  [45] fansi_1.0.4                        abind_1.4-5                       
##  [47] lifecycle_1.0.3                    yaml_2.3.7                        
##  [49] edgeR_3.99.6                       qvalue_2.33.0                     
##  [51] SparseArray_1.1.12                 blob_1.2.4                        
##  [53] promises_1.2.0.1                   dqrng_0.3.1                       
##  [55] crayon_1.5.2                       miniUI_0.1.1.1                    
##  [57] lattice_0.22-5                     beachmat_2.17.17                  
##  [59] cowplot_1.1.1                      annotate_1.79.0                   
##  [61] KEGGREST_1.41.4                    magick_2.8.1                      
##  [63] pillar_1.9.0                       knitr_1.44                        
##  [65] ComplexHeatmap_2.17.0              metapod_1.9.0                     
##  [67] fgsea_1.27.1                       rjson_0.2.21                      
##  [69] codetools_0.2-19                   fastmatch_1.1-4                   
##  [71] glue_1.6.2                         ggfun_0.1.3                       
##  [73] remotes_2.4.2                      treeio_1.25.4                     
##  [75] vctrs_0.6.3                        png_0.1-8                         
##  [77] cachem_1.0.8                       xfun_0.39                         
##  [79] S4Arrays_1.1.6                     mime_0.12                         
##  [81] tidygraph_1.2.3                    iterators_1.0.14                  
##  [83] statmod_1.5.0                      bluster_1.11.4                    
##  [85] interactiveDisplayBase_1.39.0      ellipsis_0.3.2                    
##  [87] nlme_3.1-163                       ggtree_3.9.1                      
##  [89] bit64_4.0.5                        filelock_1.0.2                    
##  [91] bslib_0.5.1                        irlba_2.3.5.1                     
##  [93] vipor_0.4.5                        colorspace_2.1-0                  
##  [95] DBI_1.1.3                          tidyselect_1.2.0                  
##  [97] processx_3.8.1                     bit_4.0.5                         
##  [99] compiler_4.3.0                     curl_5.0.0                        
## [101] graph_1.79.4                       BiocNeighbors_1.19.0              
## [103] BSgenome.Mmusculus.UCSC.mm10_1.4.3 DelayedArray_0.27.10              
## [105] shadowtext_0.1.2                   bookdown_0.36                     
## [107] rtracklayer_1.61.2                 checkmate_2.2.0                   
## [109] scales_1.2.1                       callr_3.7.3                       
## [111] rappdirs_0.3.3                     digest_0.6.31                     
## [113] rmarkdown_2.25                     BSgenome.Hsapiens.UCSC.hg19_1.4.3 
## [115] XVector_0.41.2                     htmltools_0.5.5                   
## [117] pkgconfig_2.0.3                    fastmap_1.1.1                     
## [119] rlang_1.1.1                        GlobalOptions_0.1.2               
## [121] htmlwidgets_1.6.2                  shiny_1.7.5.1                     
## [123] DelayedMatrixStats_1.23.9          farver_2.1.1                      
## [125] jquerylib_0.1.4                    jsonlite_1.8.7                    
## [127] BiocParallel_1.35.4                GOSemSim_2.27.3                   
## [129] BiocSingular_1.17.1                RCurl_1.98-1.12                   
## [131] ggplotify_0.1.2                    scuttle_1.11.3                    
## [133] GenomeInfoDbData_1.2.11            patchwork_1.1.3                   
## [135] Rhdf5lib_1.23.2                    munsell_0.5.0                     
## [137] ape_5.7-1                          babelgene_22.9                    
## [139] viridis_0.6.4                      EnrichmentBrowser_2.31.5          
## [141] stringi_1.7.12                     ggraph_2.1.0                      
## [143] MASS_7.3-60                        zlibbioc_1.47.0                   
## [145] pkgbuild_1.4.0                     HPO.db_0.99.2                     
## [147] parallel_4.3.0                     ggrepel_0.9.4                     
## [149] graphlayouts_1.0.1                 splines_4.3.0                     
## [151] Biostrings_2.69.2                  circlize_0.4.15                   
## [153] locfit_1.5-9.8                     BSgenome.Hsapiens.UCSC.hg38_1.4.5 
## [155] ps_1.7.5                           igraph_1.5.1                      
## [157] reshape2_1.4.4                     ScaledMatrix_1.9.1                
## [159] pkgload_1.3.2                      BiocVersion_3.18.0                
## [161] XML_3.99-0.14                      evaluate_0.21                     
## [163] scran_1.29.3                       BiocManager_1.30.22               
## [165] tweenr_2.0.2                       foreach_1.5.2                     
## [167] httpuv_1.6.11                      polyclip_1.10-6                   
## [169] tidyr_1.3.0                        purrr_1.0.2                       
## [171] clue_0.3-65                        ggforce_0.4.1                     
## [173] rsvd_1.0.5                         xtable_1.8-4                      
## [175] restfulr_0.0.15                    tidytree_0.4.5                    
## [177] MPO.db_0.99.7                      later_1.3.1                       
## [179] viridisLite_0.4.2                  tibble_3.2.1                      
## [181] aplot_0.2.2                        clusterProfiler_4.9.5             
## [183] memoise_2.0.1                      beeswarm_0.4.0                    
## [185] GenomicAlignments_1.37.0           cluster_2.1.4                     
## [187] GSEABase_1.63.0</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="hematopoeisis-tutorial---mae.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="single-modality-gene-expression.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/04-prostate.ENZ.archr.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
