<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Single modality: gene expression | Epiregulon documentation</title>
  <meta name="description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Single modality: gene expression | Epiregulon documentation" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Single modality: gene expression | Epiregulon documentation" />
  
  <meta name="twitter:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  

<meta name="author" content="Xiaosai Yao" />


<meta name="date" content="2023-11-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="prostate-cancer-cells.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#installation"><i class="fa fa-check"></i><b>1.1</b> Installation</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html"><i class="fa fa-check"></i><b>2</b> Multiome - ArchR workflow</a>
<ul>
<li class="chapter" data-level="2.1" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#installation-1"><i class="fa fa-check"></i><b>2.1</b> Installation</a></li>
<li class="chapter" data-level="2.2" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#data-preparation"><i class="fa fa-check"></i><b>2.2</b> Data preparation</a></li>
<li class="chapter" data-level="2.3" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#quick-start"><i class="fa fa-check"></i><b>2.3</b> Quick start</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#retrieve-bulk-tf-chip-seq-binding-sites"><i class="fa fa-check"></i><b>2.3.1</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="2.3.2" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#link-atac-seq-peaks-to-target-genes"><i class="fa fa-check"></i><b>2.3.2</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="2.3.3" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#add-tf-motif-binding-to-peaks"><i class="fa fa-check"></i><b>2.3.3</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="2.3.4" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#generate-regulons"><i class="fa fa-check"></i><b>2.3.4</b> Generate regulons</a></li>
<li class="chapter" data-level="2.3.5" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#network-pruning-highly-recommended"><i class="fa fa-check"></i><b>2.3.5</b> Network pruning (highly recommended)</a></li>
<li class="chapter" data-level="2.3.6" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#add-weights"><i class="fa fa-check"></i><b>2.3.6</b> Add Weights</a></li>
<li class="chapter" data-level="2.3.7" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#optional-annotate-with-tf-motifs"><i class="fa fa-check"></i><b>2.3.7</b> (Optional) Annotate with TF motifs</a></li>
<li class="chapter" data-level="2.3.8" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#calculate-tf-activity"><i class="fa fa-check"></i><b>2.3.8</b> Calculate TF activity</a></li>
<li class="chapter" data-level="2.3.9" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#perform-differential-activity"><i class="fa fa-check"></i><b>2.3.9</b> Perform differential activity</a></li>
<li class="chapter" data-level="2.3.10" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#visualize-the-results"><i class="fa fa-check"></i><b>2.3.10</b> Visualize the results</a></li>
<li class="chapter" data-level="2.3.11" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#geneset-enrichment"><i class="fa fa-check"></i><b>2.3.11</b> Geneset enrichment</a></li>
<li class="chapter" data-level="2.3.12" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#network-analysis"><i class="fa fa-check"></i><b>2.3.12</b> Network analysis</a></li>
<li class="chapter" data-level="2.3.13" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#differential-networks"><i class="fa fa-check"></i><b>2.3.13</b> Differential networks</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#session-info"><i class="fa fa-check"></i><b>2.4</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html"><i class="fa fa-check"></i><b>3</b> Hematopoeisis tutorial - MAE</a>
<ul>
<li class="chapter" data-level="3.1" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#installation-2"><i class="fa fa-check"></i><b>3.1</b> Installation</a></li>
<li class="chapter" data-level="3.2" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#data-preparation-1"><i class="fa fa-check"></i><b>3.2</b> Data preparation</a></li>
<li class="chapter" data-level="3.3" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#quick-start-1"><i class="fa fa-check"></i><b>3.3</b> Quick start</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#retrieve-bulk-tf-chip-seq-binding-sites-1"><i class="fa fa-check"></i><b>3.3.1</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="3.3.2" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#link-atacseq-peaks-to-target-genes"><i class="fa fa-check"></i><b>3.3.2</b> Link ATACseq peaks to target genes</a></li>
<li class="chapter" data-level="3.3.3" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#add-tf-motif-binding-to-peaks-1"><i class="fa fa-check"></i><b>3.3.3</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="3.3.4" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#generate-regulons-1"><i class="fa fa-check"></i><b>3.3.4</b> Generate regulons</a></li>
<li class="chapter" data-level="3.3.5" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#prune-network"><i class="fa fa-check"></i><b>3.3.5</b> Prune network</a></li>
<li class="chapter" data-level="3.3.6" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#add-weights-1"><i class="fa fa-check"></i><b>3.3.6</b> Add Weights</a></li>
<li class="chapter" data-level="3.3.7" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#calculate-tf-activity-1"><i class="fa fa-check"></i><b>3.3.7</b> Calculate TF activity</a></li>
<li class="chapter" data-level="3.3.8" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#differential-tf-activity-test"><i class="fa fa-check"></i><b>3.3.8</b> Differential TF activity test</a></li>
<li class="chapter" data-level="3.3.9" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#visualizing-tf-activities"><i class="fa fa-check"></i><b>3.3.9</b> Visualizing TF activities</a></li>
<li class="chapter" data-level="3.3.10" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#geneset-enrichment-1"><i class="fa fa-check"></i><b>3.3.10</b> Geneset enrichment</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#differential-network-analysis"><i class="fa fa-check"></i><b>3.4</b> Differential Network analysis</a></li>
<li class="chapter" data-level="3.5" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#session-info-1"><i class="fa fa-check"></i><b>3.5</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html"><i class="fa fa-check"></i><b>4</b> Prostate cancer cells</a>
<ul>
<li class="chapter" data-level="4.1" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#data-preparation-2"><i class="fa fa-check"></i><b>4.1</b> Data preparation</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#load-archr-project"><i class="fa fa-check"></i><b>4.1.1</b> Load ArchR project</a></li>
<li class="chapter" data-level="4.1.2" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#retrieve-matrices-from-archr-project"><i class="fa fa-check"></i><b>4.1.2</b> Retrieve matrices from ArchR project</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#quick-start-2"><i class="fa fa-check"></i><b>4.2</b> Quick start</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#retrieve-bulk-tf-chip-seq-binding-sites-2"><i class="fa fa-check"></i><b>4.2.1</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="4.2.2" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#link-atac-seq-peaks-to-target-genes-1"><i class="fa fa-check"></i><b>4.2.2</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="4.2.3" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#add-tf-motif-binding-to-peaks-2"><i class="fa fa-check"></i><b>4.2.3</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="4.2.4" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#generate-regulons-2"><i class="fa fa-check"></i><b>4.2.4</b> Generate regulons</a></li>
<li class="chapter" data-level="4.2.5" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#calculate-tf-activity-2"><i class="fa fa-check"></i><b>4.2.5</b> Calculate TF activity</a></li>
<li class="chapter" data-level="4.2.6" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#perform-differential-activity-1"><i class="fa fa-check"></i><b>4.2.6</b> Perform differential activity</a></li>
<li class="chapter" data-level="4.2.7" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#visualize-the-results-1"><i class="fa fa-check"></i><b>4.2.7</b> Visualize the results</a></li>
<li class="chapter" data-level="4.2.8" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#geneset-enrichment-2"><i class="fa fa-check"></i><b>4.2.8</b> Geneset enrichment</a></li>
<li class="chapter" data-level="4.2.9" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#network-analysis-1"><i class="fa fa-check"></i><b>4.2.9</b> Network analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#session-info-2"><i class="fa fa-check"></i><b>4.3</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html"><i class="fa fa-check"></i><b>5</b> Single modality: gene expression</a>
<ul>
<li class="chapter" data-level="5.1" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#load-regulon"><i class="fa fa-check"></i><b>5.1</b> Load regulon</a></li>
<li class="chapter" data-level="5.2" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#load-scrna-seq-data"><i class="fa fa-check"></i><b>5.2</b> Load scRNA-seq data</a></li>
<li class="chapter" data-level="5.3" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#calculate-activity"><i class="fa fa-check"></i><b>5.3</b> Calculate activity</a></li>
<li class="chapter" data-level="5.4" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#perform-differential-activity-2"><i class="fa fa-check"></i><b>5.4</b> Perform differential activity</a></li>
<li class="chapter" data-level="5.5" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#visualize-activity"><i class="fa fa-check"></i><b>5.5</b> Visualize activity</a></li>
<li class="chapter" data-level="5.6" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#session-info-3"><i class="fa fa-check"></i><b>5.6</b> Session Info</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Epiregulon documentation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="single-modality-gene-expression" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Single modality: gene expression<a href="single-modality-gene-expression.html#single-modality-gene-expression" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Epiregulon also supports transcription factor activity inference when users only have scRNA-seq. After all, multiome or scATAC-seq data is still relatively rare. To enable TF activity inference on scRNA-seq, users can supply a pre-constructed gene regulatory network. <a href="https://saezlab.github.io/dorothea/articles/dorothea.html">Dorothea</a> provides both human and mouse pre-constructed gene regulatory networks based on curated experimental and computational data. In this vignette, we bypass the regulon construction step and go straight to calculate TF activity from a Dorothea GRN.</p>
<div id="load-regulon" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Load regulon<a href="single-modality-gene-expression.html#load-regulon" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Dorothea assigns confidence level to its regulons with A being the most confident (i.e. supported by multiple lines of evidence) and E being the least confident.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="single-modality-gene-expression.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dorothea)</span>
<span id="cb1-2"><a href="single-modality-gene-expression.html#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(dorothea_mm, <span class="at">package =</span> <span class="st">&quot;dorothea&quot;</span>)</span>
<span id="cb1-3"><a href="single-modality-gene-expression.html#cb1-3" aria-hidden="true" tabindex="-1"></a>regulon <span class="ot">&lt;-</span> dorothea_mm </span>
<span id="cb1-4"><a href="single-modality-gene-expression.html#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="single-modality-gene-expression.html#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#known tfs</span></span>
<span id="cb1-6"><a href="single-modality-gene-expression.html#cb1-6" aria-hidden="true" tabindex="-1"></a>genes_to_plot <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Foxa1&quot;</span>, <span class="st">&quot;Neurod1&quot;</span>,<span class="st">&quot;Pdx1&quot;</span>,<span class="st">&quot;Arx&quot;</span>)</span></code></pre></div>
</div>
<div id="load-scrna-seq-data" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Load scRNA-seq data<a href="single-modality-gene-expression.html#load-scrna-seq-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We download the raw counts of a mouse pancreas data set from <a href="https://bioconductor.org/packages/release/data/experiment/html/scRNAseq.html">scRNAseq</a>. We add normalized logcounts, perform dimension reduction and visualize the embeddings using <a href="https://bioconductor.org/packages/release/bioc/html/scater.html">scater</a>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="single-modality-gene-expression.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scRNAseq)</span>
<span id="cb2-2"><a href="single-modality-gene-expression.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb2-3"><a href="single-modality-gene-expression.html#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="single-modality-gene-expression.html#cb2-4" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">BaronPancreasData</span>(<span class="st">&#39;mouse&#39;</span>)</span>
<span id="cb2-5"><a href="single-modality-gene-expression.html#cb2-5" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">logNormCounts</span>(sce)</span>
<span id="cb2-6"><a href="single-modality-gene-expression.html#cb2-6" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(sce)</span>
<span id="cb2-7"><a href="single-modality-gene-expression.html#cb2-7" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(sce)</span>
<span id="cb2-8"><a href="single-modality-gene-expression.html#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="single-modality-gene-expression.html#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUMAP</span>(sce, <span class="at">colour_by =</span> <span class="st">&quot;label&quot;</span>, <span class="at">text_by =</span> <span class="st">&quot;label&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/load_scRNAseq_singlemodality-1.png" width="672" /></p>
</div>
<div id="calculate-activity" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Calculate activity<a href="single-modality-gene-expression.html#calculate-activity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Even though Dorothea provides weights under the mor column, we achieved superior performance if we recompute the weights based on the correlation between tf and target gene expression based on our own data. We performed 2 steps, the first step is to add weights to the Dorothea regulons and the second step is to estimate the TF activity by taking the weighted average of the target gene expression.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="single-modality-gene-expression.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(epiregulon)</span>
<span id="cb3-2"><a href="single-modality-gene-expression.html#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="single-modality-gene-expression.html#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="single-modality-gene-expression.html#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Add weights to regulon</span></span>
<span id="cb3-5"><a href="single-modality-gene-expression.html#cb3-5" aria-hidden="true" tabindex="-1"></a>regulon.ms <span class="ot">&lt;-</span> <span class="fu">addWeights</span>(<span class="at">regulon =</span> regulon,</span>
<span id="cb3-6"><a href="single-modality-gene-expression.html#cb3-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">expMatrix =</span> sce,</span>
<span id="cb3-7"><a href="single-modality-gene-expression.html#cb3-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">clusters =</span> sce<span class="sc">$</span>label,</span>
<span id="cb3-8"><a href="single-modality-gene-expression.html#cb3-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">BPPARAM =</span> BiocParallel<span class="sc">::</span><span class="fu">MulticoreParam</span>())</span>
<span id="cb3-9"><a href="single-modality-gene-expression.html#cb3-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-10"><a href="single-modality-gene-expression.html#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate activity</span></span>
<span id="cb3-11"><a href="single-modality-gene-expression.html#cb3-11" aria-hidden="true" tabindex="-1"></a>score.combine <span class="ot">&lt;-</span> <span class="fu">calculateActivity</span>(sce, </span>
<span id="cb3-12"><a href="single-modality-gene-expression.html#cb3-12" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">regulon =</span> regulon.ms, </span>
<span id="cb3-13"><a href="single-modality-gene-expression.html#cb3-13" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">mode =</span> <span class="st">&quot;weight&quot;</span>, </span>
<span id="cb3-14"><a href="single-modality-gene-expression.html#cb3-14" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">&quot;weightedMean&quot;</span>)</span></code></pre></div>
</div>
<div id="perform-differential-activity-2" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Perform differential activity<a href="single-modality-gene-expression.html#perform-differential-activity-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="single-modality-gene-expression.html#cb4-1" aria-hidden="true" tabindex="-1"></a>markers <span class="ot">&lt;-</span> <span class="fu">findDifferentialActivity</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb4-2"><a href="single-modality-gene-expression.html#cb4-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">groups =</span> sce<span class="sc">$</span>label, </span>
<span id="cb4-3"><a href="single-modality-gene-expression.html#cb4-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">pval.type =</span> <span class="st">&quot;some&quot;</span>, </span>
<span id="cb4-4"><a href="single-modality-gene-expression.html#cb4-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">direction =</span> <span class="st">&quot;up&quot;</span>, </span>
<span id="cb4-5"><a href="single-modality-gene-expression.html#cb4-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">test.type =</span> <span class="st">&quot;t&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: The `groups` argument of `findDifferentialActivity()` is deprecated as of
## epiregulon 1.0.0.
## ℹ Please use the `clusters` argument instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.</code></pre>
<p>Take the top TFs</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="single-modality-gene-expression.html#cb6-1" aria-hidden="true" tabindex="-1"></a>markers.sig <span class="ot">&lt;-</span> <span class="fu">getSigGenes</span>(markers, <span class="at">topgenes =</span> <span class="dv">5</span> )</span></code></pre></div>
<pre><code>## Using a logFC cutoff of 0.1 for class B_cell</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class T_cell</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class activated_stellate</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class alpha</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class beta</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class delta</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class ductal</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class endothelial</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class gamma</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class immune_other</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class macrophage</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class quiescent_stellate</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class schwann</code></pre>
</div>
<div id="visualize-activity" class="section level2 hasAnchor" number="5.5">
<h2><span class="header-section-number">5.5</span> Visualize activity<a href="single-modality-gene-expression.html#visualize-activity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Finally we visualize the TF activity by either UMAP, violin plots or bubble plots.
We confirm the activity of known lineage factors Pdx1 and Neurod1 in beta cells,
Arx in alpha cells and Foxa1 in ductal cells.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="single-modality-gene-expression.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot umap</span></span>
<span id="cb20-2"><a href="single-modality-gene-expression.html#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotActivityDim</span>(<span class="at">sce =</span> sce, </span>
<span id="cb20-3"><a href="single-modality-gene-expression.html#cb20-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">activity_matrix =</span> score.combine,</span>
<span id="cb20-4"><a href="single-modality-gene-expression.html#cb20-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">tf =</span> genes_to_plot, </span>
<span id="cb20-5"><a href="single-modality-gene-expression.html#cb20-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">legend.label =</span> <span class="st">&quot;score&quot;</span>,</span>
<span id="cb20-6"><a href="single-modality-gene-expression.html#cb20-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">point_size =</span> <span class="fl">0.1</span>,</span>
<span id="cb20-7"><a href="single-modality-gene-expression.html#cb20-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">dimtype =</span> <span class="st">&quot;UMAP&quot;</span>, </span>
<span id="cb20-8"><a href="single-modality-gene-expression.html#cb20-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="st">&quot;label&quot;</span>, </span>
<span id="cb20-9"><a href="single-modality-gene-expression.html#cb20-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb20-10"><a href="single-modality-gene-expression.html#cb20-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">text_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/visualization_singlemodality-1.png" width="672" /></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="single-modality-gene-expression.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot violin plot</span></span>
<span id="cb21-2"><a href="single-modality-gene-expression.html#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotActivityViolin</span>(score.combine, </span>
<span id="cb21-3"><a href="single-modality-gene-expression.html#cb21-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">tf =</span> genes_to_plot,</span>
<span id="cb21-4"><a href="single-modality-gene-expression.html#cb21-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">clusters =</span> sce<span class="sc">$</span>label)</span></code></pre></div>
<p><img src="_main_files/figure-html/visualization_singlemodality-2.png" width="672" /></p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="single-modality-gene-expression.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot bubble plot</span></span>
<span id="cb22-2"><a href="single-modality-gene-expression.html#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotBubble</span>(score.combine, </span>
<span id="cb22-3"><a href="single-modality-gene-expression.html#cb22-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">tf =</span> genes_to_plot, </span>
<span id="cb22-4"><a href="single-modality-gene-expression.html#cb22-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">clusters =</span> sce<span class="sc">$</span>label)</span></code></pre></div>
<p><img src="_main_files/figure-html/visualization_singlemodality-3.png" width="672" />
Plot bubble plot of differential TFs</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="single-modality-gene-expression.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotBubble</span>(score.combine, </span>
<span id="cb23-2"><a href="single-modality-gene-expression.html#cb23-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">tf =</span> markers.sig<span class="sc">$</span>tf, </span>
<span id="cb23-3"><a href="single-modality-gene-expression.html#cb23-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">clusters =</span> sce<span class="sc">$</span>label)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-2-1.png" width="672" />
We can adapt the epiregulon package to plot gene expression. When compared against TF activity, gene expression of Foxa1 and Arx has noisy signals and high dropout rates.
Epiregulon enhances the signal to noise ratio of TF activity and better resolves lineage differences.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="single-modality-gene-expression.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot umap</span></span>
<span id="cb24-2"><a href="single-modality-gene-expression.html#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotActivityDim</span>(<span class="at">sce =</span> sce, </span>
<span id="cb24-3"><a href="single-modality-gene-expression.html#cb24-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">activity_matrix =</span> <span class="fu">logcounts</span>(sce),</span>
<span id="cb24-4"><a href="single-modality-gene-expression.html#cb24-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">tf =</span> genes_to_plot, </span>
<span id="cb24-5"><a href="single-modality-gene-expression.html#cb24-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">legend.label =</span> <span class="st">&quot;score&quot;</span>,</span>
<span id="cb24-6"><a href="single-modality-gene-expression.html#cb24-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">point_size =</span> <span class="fl">0.1</span>,</span>
<span id="cb24-7"><a href="single-modality-gene-expression.html#cb24-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">dimtype =</span> <span class="st">&quot;UMAP&quot;</span>, </span>
<span id="cb24-8"><a href="single-modality-gene-expression.html#cb24-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="st">&quot;label&quot;</span>, </span>
<span id="cb24-9"><a href="single-modality-gene-expression.html#cb24-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">combine =</span> <span class="cn">TRUE</span>,</span>
<span id="cb24-10"><a href="single-modality-gene-expression.html#cb24-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">text_size =</span> <span class="dv">2</span>,</span>
<span id="cb24-11"><a href="single-modality-gene-expression.html#cb24-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&quot;gray&quot;</span>,<span class="st">&quot;blue&quot;</span>),</span>
<span id="cb24-12"><a href="single-modality-gene-expression.html#cb24-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">limit =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">2</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/scRNAseq-1.png" width="672" /></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="single-modality-gene-expression.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot violin plot</span></span>
<span id="cb25-2"><a href="single-modality-gene-expression.html#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotActivityViolin</span>(<span class="fu">logcounts</span>(sce), </span>
<span id="cb25-3"><a href="single-modality-gene-expression.html#cb25-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">tf =</span> genes_to_plot,</span>
<span id="cb25-4"><a href="single-modality-gene-expression.html#cb25-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">clusters =</span> sce<span class="sc">$</span>label)</span></code></pre></div>
<p><img src="_main_files/figure-html/scRNAseq-2.png" width="672" /></p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="single-modality-gene-expression.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot Bubble plot</span></span>
<span id="cb26-2"><a href="single-modality-gene-expression.html#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotBubble</span>(<span class="fu">logcounts</span>(sce), </span>
<span id="cb26-3"><a href="single-modality-gene-expression.html#cb26-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">tf =</span> markers.sig<span class="sc">$</span>tf, </span>
<span id="cb26-4"><a href="single-modality-gene-expression.html#cb26-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">clusters =</span> sce<span class="sc">$</span>label)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<p>We can visualize the target genes for transcription factors of interest</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="single-modality-gene-expression.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmapRegulon</span>(<span class="at">sce=</span>sce,</span>
<span id="cb27-2"><a href="single-modality-gene-expression.html#cb27-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">tfs=</span>genes_to_plot,</span>
<span id="cb27-3"><a href="single-modality-gene-expression.html#cb27-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">regulon=</span>regulon.ms,</span>
<span id="cb27-4"><a href="single-modality-gene-expression.html#cb27-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">regulon_cutoff=</span><span class="fl">0.5</span>,</span>
<span id="cb27-5"><a href="single-modality-gene-expression.html#cb27-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">downsample=</span><span class="dv">1000</span>,</span>
<span id="cb27-6"><a href="single-modality-gene-expression.html#cb27-6" aria-hidden="true" tabindex="-1"></a>                   <span class="at">cell_attributes=</span><span class="st">&quot;label&quot;</span>,</span>
<span id="cb27-7"><a href="single-modality-gene-expression.html#cb27-7" aria-hidden="true" tabindex="-1"></a>                   <span class="at">col_gap=</span><span class="st">&quot;label&quot;</span>,</span>
<span id="cb27-8"><a href="single-modality-gene-expression.html#cb27-8" aria-hidden="true" tabindex="-1"></a>                   <span class="at">exprs_values=</span><span class="st">&quot;logcounts&quot;</span>,</span>
<span id="cb27-9"><a href="single-modality-gene-expression.html#cb27-9" aria-hidden="true" tabindex="-1"></a>                   <span class="at">name=</span><span class="st">&quot;regulon heatmap&quot;</span>,</span>
<span id="cb27-10"><a href="single-modality-gene-expression.html#cb27-10" aria-hidden="true" tabindex="-1"></a>                   <span class="at">column_title_rot =</span> <span class="dv">45</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-4-1.png" width="1152" /></p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="single-modality-gene-expression.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmapActivity</span>(<span class="at">activity_matrix =</span> score.combine,</span>
<span id="cb28-2"><a href="single-modality-gene-expression.html#cb28-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">sce=</span>sce,</span>
<span id="cb28-3"><a href="single-modality-gene-expression.html#cb28-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">tfs=</span>genes_to_plot,</span>
<span id="cb28-4"><a href="single-modality-gene-expression.html#cb28-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">downsample=</span><span class="dv">1000</span>,</span>
<span id="cb28-5"><a href="single-modality-gene-expression.html#cb28-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">cell_attributes=</span><span class="st">&quot;label&quot;</span>,</span>
<span id="cb28-6"><a href="single-modality-gene-expression.html#cb28-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">col_gap=</span><span class="st">&quot;label&quot;</span>,</span>
<span id="cb28-7"><a href="single-modality-gene-expression.html#cb28-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">name=</span><span class="st">&quot;regulon heatmap&quot;</span>,</span>
<span id="cb28-8"><a href="single-modality-gene-expression.html#cb28-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">column_title_rot =</span> <span class="dv">45</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-4-2.png" width="1152" />
## Pathway enrichment</p>
<p>Sometimes it is useful to understand what pathways are enriched in the regulons. We take the highly correlated target genes of a regulon and perform geneset enrichment using the enricher function from <a href="http://yulab-smu.top/biomedical-knowledge-mining-book/universal-api.html">clusterProfiler</a>.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="single-modality-gene-expression.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co">#retrieve genesets</span></span>
<span id="cb29-2"><a href="single-modality-gene-expression.html#cb29-2" aria-hidden="true" tabindex="-1"></a>H <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">getGenesets</span>(<span class="at">org =</span> <span class="st">&quot;mmu&quot;</span>, </span>
<span id="cb29-3"><a href="single-modality-gene-expression.html#cb29-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">db =</span> <span class="st">&quot;msigdb&quot;</span>, </span>
<span id="cb29-4"><a href="single-modality-gene-expression.html#cb29-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">cat =</span> <span class="st">&quot;H&quot;</span>, </span>
<span id="cb29-5"><a href="single-modality-gene-expression.html#cb29-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">gene.id.type =</span> <span class="st">&quot;SYMBOL&quot;</span>, </span>
<span id="cb29-6"><a href="single-modality-gene-expression.html#cb29-6" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">cache =</span> <span class="cn">FALSE</span>)</span>
<span id="cb29-7"><a href="single-modality-gene-expression.html#cb29-7" aria-hidden="true" tabindex="-1"></a>C6 <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">getGenesets</span>(<span class="at">org =</span> <span class="st">&quot;mmu&quot;</span>, </span>
<span id="cb29-8"><a href="single-modality-gene-expression.html#cb29-8" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">db =</span> <span class="st">&quot;msigdb&quot;</span>, </span>
<span id="cb29-9"><a href="single-modality-gene-expression.html#cb29-9" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">cat =</span> <span class="st">&quot;C6&quot;</span>, </span>
<span id="cb29-10"><a href="single-modality-gene-expression.html#cb29-10" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">gene.id.type =</span> <span class="st">&quot;SYMBOL&quot;</span>,  </span>
<span id="cb29-11"><a href="single-modality-gene-expression.html#cb29-11" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">cache =</span> <span class="cn">FALSE</span>)</span>
<span id="cb29-12"><a href="single-modality-gene-expression.html#cb29-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-13"><a href="single-modality-gene-expression.html#cb29-13" aria-hidden="true" tabindex="-1"></a><span class="co">#combine genesets and convert genesets to be compatible with enricher</span></span>
<span id="cb29-14"><a href="single-modality-gene-expression.html#cb29-14" aria-hidden="true" tabindex="-1"></a>gs <span class="ot">&lt;-</span> <span class="fu">c</span>(H,C6)</span>
<span id="cb29-15"><a href="single-modality-gene-expression.html#cb29-15" aria-hidden="true" tabindex="-1"></a>gs.list <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind,<span class="fu">lapply</span>(<span class="fu">names</span>(gs), <span class="cf">function</span>(x) </span>
<span id="cb29-16"><a href="single-modality-gene-expression.html#cb29-16" aria-hidden="true" tabindex="-1"></a>  {<span class="fu">data.frame</span>(<span class="at">gs =</span> x, <span class="at">genes =</span> gs[[x]])}))</span>
<span id="cb29-17"><a href="single-modality-gene-expression.html#cb29-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-18"><a href="single-modality-gene-expression.html#cb29-18" aria-hidden="true" tabindex="-1"></a>enrichresults <span class="ot">&lt;-</span> <span class="fu">regulonEnrich</span>(genes_to_plot, </span>
<span id="cb29-19"><a href="single-modality-gene-expression.html#cb29-19" aria-hidden="true" tabindex="-1"></a>                               <span class="at">regulon =</span> regulon.ms, </span>
<span id="cb29-20"><a href="single-modality-gene-expression.html#cb29-20" aria-hidden="true" tabindex="-1"></a>                               <span class="at">weight =</span> <span class="st">&quot;weight&quot;</span>,</span>
<span id="cb29-21"><a href="single-modality-gene-expression.html#cb29-21" aria-hidden="true" tabindex="-1"></a>                               <span class="at">weight_cutoff =</span> <span class="fl">0.5</span>, </span>
<span id="cb29-22"><a href="single-modality-gene-expression.html#cb29-22" aria-hidden="true" tabindex="-1"></a>                               <span class="at">genesets =</span> gs.list)</span></code></pre></div>
<pre><code>## Foxa1</code></pre>
<pre><code>## </code></pre>
<pre><code>## Neurod1</code></pre>
<pre><code>## Pdx1</code></pre>
<pre><code>## Arx</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="single-modality-gene-expression.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co">#plot results</span></span>
<span id="cb35-2"><a href="single-modality-gene-expression.html#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">enrichPlot</span>(<span class="at">results =</span> enrichresults, <span class="at">ncol =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/enrichment_singlemodality-1.png" width="672" /></p>
</div>
<div id="session-info-3" class="section level2 hasAnchor" number="5.6">
<h2><span class="header-section-number">5.6</span> Session Info<a href="single-modality-gene-expression.html#session-info-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="single-modality-gene-expression.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.3.0 (2023-04-21)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 18.04.6 LTS
## 
## Matrix products: default
## BLAS:   /usr/local/lib/R/lib/libRblas.so 
## LAPACK: /usr/local/lib/R/lib/libRlapack.so;  LAPACK version 3.11.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
##  [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
##  [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
## [10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 
## 
## time zone: Etc/UTC
## tzcode source: system (glibc)
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] org.Mm.eg.db_3.18.0         AnnotationDbi_1.63.2       
##  [3] msigdbr_7.5.1               epiregulon_1.0.35          
##  [5] Matrix_1.6-1.1              scater_1.29.4              
##  [7] ggplot2_3.4.4               scuttle_1.11.3             
##  [9] scRNAseq_2.15.0             SingleCellExperiment_1.23.0
## [11] SummarizedExperiment_1.31.1 Biobase_2.61.0             
## [13] GenomicRanges_1.53.3        GenomeInfoDb_1.37.7        
## [15] IRanges_2.35.3              S4Vectors_0.39.3           
## [17] BiocGenerics_0.47.1         MatrixGenerics_1.13.2      
## [19] matrixStats_1.0.0           dorothea_1.13.0            
## 
## loaded via a namespace (and not attached):
##   [1] fs_1.6.2                           ProtGenerics_1.33.1               
##   [3] GSVA_1.49.8                        bitops_1.0-7                      
##   [5] enrichplot_1.21.3                  HDO.db_0.99.1                     
##   [7] httr_1.4.6                         RColorBrewer_1.1-3                
##   [9] doParallel_1.0.17                  Rgraphviz_2.45.0                  
##  [11] tools_4.3.0                        backports_1.4.1                   
##  [13] utf8_1.2.3                         R6_2.5.1                          
##  [15] HDF5Array_1.29.3                   lazyeval_0.2.2                    
##  [17] uwot_0.1.16                        rhdf5filters_1.13.5               
##  [19] GetoptLong_1.0.5                   withr_2.5.0                       
##  [21] prettyunits_1.1.1                  gridExtra_2.3                     
##  [23] cli_3.6.1                          Cairo_1.6-1                       
##  [25] scatterpie_0.2.1                   labeling_0.4.3                    
##  [27] sass_0.4.6                         KEGGgraph_1.61.0                  
##  [29] yulab.utils_0.1.0                  Rsamtools_2.17.0                  
##  [31] gson_0.1.0                         DOSE_3.27.3                       
##  [33] bcellViper_1.37.0                  BSgenome_1.69.1                   
##  [35] limma_3.57.11                      rstudioapi_0.15.0                 
##  [37] RSQLite_2.3.1                      FNN_1.1.3.2                       
##  [39] gridGraphics_0.5-1                 generics_0.1.3                    
##  [41] shape_1.4.6                        BiocIO_1.11.0                     
##  [43] dplyr_1.1.3                        GO.db_3.18.0                      
##  [45] ggbeeswarm_0.7.2                   fansi_1.0.4                       
##  [47] abind_1.4-5                        lifecycle_1.0.3                   
##  [49] yaml_2.3.7                         edgeR_3.99.6                      
##  [51] qvalue_2.33.0                      rhdf5_2.45.1                      
##  [53] SparseArray_1.1.12                 BiocFileCache_2.9.1               
##  [55] grid_4.3.0                         blob_1.2.4                        
##  [57] promises_1.2.0.1                   dqrng_0.3.1                       
##  [59] ExperimentHub_2.9.1                crayon_1.5.2                      
##  [61] lattice_0.22-5                     beachmat_2.17.17                  
##  [63] cowplot_1.1.1                      GenomicFeatures_1.53.3            
##  [65] annotate_1.79.0                    KEGGREST_1.41.4                   
##  [67] magick_2.8.1                       pillar_1.9.0                      
##  [69] knitr_1.44                         ComplexHeatmap_2.17.0             
##  [71] metapod_1.9.0                      fgsea_1.27.1                      
##  [73] rjson_0.2.21                       codetools_0.2-19                  
##  [75] fastmatch_1.1-4                    glue_1.6.2                        
##  [77] ggfun_0.1.3                        data.table_1.14.8                 
##  [79] MultiAssayExperiment_1.27.5        treeio_1.25.4                     
##  [81] vctrs_0.6.3                        png_0.1-8                         
##  [83] gtable_0.3.4                       cachem_1.0.8                      
##  [85] xfun_0.39                          S4Arrays_1.1.6                    
##  [87] mime_0.12                          tidygraph_1.2.3                   
##  [89] iterators_1.0.14                   statmod_1.5.0                     
##  [91] bluster_1.11.4                     interactiveDisplayBase_1.39.0     
##  [93] ellipsis_0.3.2                     nlme_3.1-163                      
##  [95] ArchR_1.0.3                        ggtree_3.9.1                      
##  [97] bit64_4.0.5                        progress_1.2.2                    
##  [99] filelock_1.0.2                     bslib_0.5.1                       
## [101] irlba_2.3.5.1                      vipor_0.4.5                       
## [103] colorspace_2.1-0                   DBI_1.1.3                         
## [105] tidyselect_1.2.0                   bit_4.0.5                         
## [107] compiler_4.3.0                     curl_5.0.0                        
## [109] graph_1.79.4                       BiocNeighbors_1.19.0              
## [111] BSgenome.Mmusculus.UCSC.mm10_1.4.3 xml2_1.3.5                        
## [113] DelayedArray_0.27.10               shadowtext_0.1.2                  
## [115] bookdown_0.36                      rtracklayer_1.61.2                
## [117] checkmate_2.2.0                    scales_1.2.1                      
## [119] rappdirs_0.3.3                     stringr_1.5.0                     
## [121] digest_0.6.31                      rmarkdown_2.25                    
## [123] BSgenome.Hsapiens.UCSC.hg19_1.4.3  XVector_0.41.2                    
## [125] htmltools_0.5.5                    pkgconfig_2.0.3                   
## [127] sparseMatrixStats_1.13.4           dbplyr_2.3.4                      
## [129] fastmap_1.1.1                      ensembldb_2.25.1                  
## [131] rlang_1.1.1                        GlobalOptions_0.1.2               
## [133] shiny_1.7.5.1                      DelayedMatrixStats_1.23.9         
## [135] farver_2.1.1                       jquerylib_0.1.4                   
## [137] jsonlite_1.8.7                     BiocParallel_1.35.4               
## [139] GOSemSim_2.27.3                    BiocSingular_1.17.1               
## [141] RCurl_1.98-1.12                    magrittr_2.0.3                    
## [143] ggplotify_0.1.2                    GenomeInfoDbData_1.2.11           
## [145] patchwork_1.1.3                    Rhdf5lib_1.23.2                   
## [147] munsell_0.5.0                      Rcpp_1.0.11                       
## [149] ape_5.7-1                          babelgene_22.9                    
## [151] viridis_0.6.4                      EnrichmentBrowser_2.31.5          
## [153] stringi_1.7.12                     ggraph_2.1.0                      
## [155] MASS_7.3-60                        zlibbioc_1.47.0                   
## [157] AnnotationHub_3.9.2                plyr_1.8.9                        
## [159] HPO.db_0.99.2                      parallel_4.3.0                    
## [161] ggrepel_0.9.4                      graphlayouts_1.0.1                
## [163] splines_4.3.0                      Biostrings_2.69.2                 
## [165] hms_1.1.3                          circlize_0.4.15                   
## [167] locfit_1.5-9.8                     BSgenome.Hsapiens.UCSC.hg38_1.4.5 
## [169] igraph_1.5.1                       reshape2_1.4.4                    
## [171] biomaRt_2.57.1                     ScaledMatrix_1.9.1                
## [173] BiocVersion_3.18.0                 XML_3.99-0.14                     
## [175] evaluate_0.21                      scran_1.29.3                      
## [177] BiocManager_1.30.22                tweenr_2.0.2                      
## [179] foreach_1.5.2                      httpuv_1.6.11                     
## [181] polyclip_1.10-6                    tidyr_1.3.0                       
## [183] purrr_1.0.2                        clue_0.3-65                       
## [185] ggforce_0.4.1                      rsvd_1.0.5                        
## [187] xtable_1.8-4                       restfulr_0.0.15                   
## [189] AnnotationFilter_1.25.0            tidytree_0.4.5                    
## [191] MPO.db_0.99.7                      later_1.3.1                       
## [193] viridisLite_0.4.2                  tibble_3.2.1                      
## [195] aplot_0.2.2                        clusterProfiler_4.9.5             
## [197] memoise_2.0.1                      beeswarm_0.4.0                    
## [199] GenomicAlignments_1.37.0           cluster_2.1.4                     
## [201] GSEABase_1.63.0</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="prostate-cancer-cells.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/05-scRNAseq.only.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
