<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Advanced workflow with MultiAssayExperiment | Epiregulon documentation</title>
  <meta name="description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="generator" content="bookdown 0.38 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Advanced workflow with MultiAssayExperiment | Epiregulon documentation" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Advanced workflow with MultiAssayExperiment | Epiregulon documentation" />
  
  <meta name="twitter:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  

<meta name="author" content="Xiaosai Yao, Tomasz Włodarczyk" />


<meta name="date" content="2024-03-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="basic-workflow-with-multiassayexperiment.html"/>
<link rel="next" href="archr-workflow-and-different-weight-methods.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Epiregulon documentation</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>1</b> Installation</a></li>
<li class="chapter" data-level="2" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>2</b> Basic workflow with MultiAssayExperiment</a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#data-preparation"><i class="fa fa-check"></i><b>2.1</b> Data preparation</a></li>
<li class="chapter" data-level="2.2" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#quick-start"><i class="fa fa-check"></i><b>2.2</b> Quick start</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#retrieve-bulk-tf-chip-seq-binding-sites"><i class="fa fa-check"></i><b>2.2.1</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="2.2.2" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#link-atac-seq-peaks-to-target-genes"><i class="fa fa-check"></i><b>2.2.2</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="2.2.3" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#add-tf-motif-binding-to-peaks"><i class="fa fa-check"></i><b>2.2.3</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="2.2.4" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#generate-regulons"><i class="fa fa-check"></i><b>2.2.4</b> Generate regulons</a></li>
<li class="chapter" data-level="2.2.5" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#network-pruning-highly-recommended"><i class="fa fa-check"></i><b>2.2.5</b> Network pruning (highly recommended)</a></li>
<li class="chapter" data-level="2.2.6" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#add-weights"><i class="fa fa-check"></i><b>2.2.6</b> Add Weights</a></li>
<li class="chapter" data-level="2.2.7" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#optional-annotate-with-tf-motifs"><i class="fa fa-check"></i><b>2.2.7</b> (Optional) Annotate with TF motifs</a></li>
<li class="chapter" data-level="2.2.8" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#calculate-tf-activity"><i class="fa fa-check"></i><b>2.2.8</b> Calculate TF activity</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#session-info"><i class="fa fa-check"></i><b>2.3</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>3</b> Advanced workflow with MultiAssayExperiment</a>
<ul>
<li class="chapter" data-level="3.1" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#data-preparation-1"><i class="fa fa-check"></i><b>3.1</b> Data preparation</a></li>
<li class="chapter" data-level="3.2" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#retrieve-bulk-tf-chip-seq-binding-sites-1"><i class="fa fa-check"></i><b>3.2</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="3.3" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#link-atacseq-peaks-to-target-genes"><i class="fa fa-check"></i><b>3.3</b> Link ATACseq peaks to target genes</a></li>
<li class="chapter" data-level="3.4" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#add-tf-motif-binding-to-peaks-1"><i class="fa fa-check"></i><b>3.4</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="3.5" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#generate-regulons-1"><i class="fa fa-check"></i><b>3.5</b> Generate regulons</a></li>
<li class="chapter" data-level="3.6" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#prune-network"><i class="fa fa-check"></i><b>3.6</b> Prune network</a></li>
<li class="chapter" data-level="3.7" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#add-weights-1"><i class="fa fa-check"></i><b>3.7</b> Add Weights</a></li>
<li class="chapter" data-level="3.8" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#calculate-tf-activity-1"><i class="fa fa-check"></i><b>3.8</b> Calculate TF activity</a></li>
<li class="chapter" data-level="3.9" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#differential-tf-activity-test"><i class="fa fa-check"></i><b>3.9</b> Differential TF activity test</a></li>
<li class="chapter" data-level="3.10" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#visualizing-tf-activities"><i class="fa fa-check"></i><b>3.10</b> Visualizing TF activities</a></li>
<li class="chapter" data-level="3.11" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#geneset-enrichment"><i class="fa fa-check"></i><b>3.11</b> Geneset enrichment</a></li>
<li class="chapter" data-level="3.12" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#differential-network-analysis"><i class="fa fa-check"></i><b>3.12</b> Differential Network analysis</a></li>
<li class="chapter" data-level="3.13" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#session-info-1"><i class="fa fa-check"></i><b>3.13</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html"><i class="fa fa-check"></i><b>4</b> ArchR workflow and different weight methods</a>
<ul>
<li class="chapter" data-level="4.1" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#data-preparation-2"><i class="fa fa-check"></i><b>4.1</b> Data preparation</a></li>
<li class="chapter" data-level="4.2" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#load-archr-project"><i class="fa fa-check"></i><b>4.2</b> Load ArchR project</a></li>
<li class="chapter" data-level="4.3" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#retrieve-matrices-from-archr-project"><i class="fa fa-check"></i><b>4.3</b> Retrieve matrices from ArchR project</a></li>
<li class="chapter" data-level="4.4" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#retrieve-bulk-tf-chip-seq-binding-sites-2"><i class="fa fa-check"></i><b>4.4</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="4.5" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#link-atac-seq-peaks-to-target-genes-1"><i class="fa fa-check"></i><b>4.5</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="4.6" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#add-tf-motif-binding-to-peaks-2"><i class="fa fa-check"></i><b>4.6</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="4.7" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#generate-regulons-2"><i class="fa fa-check"></i><b>4.7</b> Generate regulons</a></li>
<li class="chapter" data-level="4.8" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#optional-annotate-with-tf-motifs-1"><i class="fa fa-check"></i><b>4.8</b> (Optional) Annotate with TF motifs</a></li>
<li class="chapter" data-level="4.9" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#prune-network-1"><i class="fa fa-check"></i><b>4.9</b> Prune network</a></li>
<li class="chapter" data-level="4.10" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#add-weights-2"><i class="fa fa-check"></i><b>4.10</b> Add Weights</a></li>
<li class="chapter" data-level="4.11" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#calculate-tf-activity-2"><i class="fa fa-check"></i><b>4.11</b> Calculate TF activity</a></li>
<li class="chapter" data-level="4.12" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#perform-differential-activity"><i class="fa fa-check"></i><b>4.12</b> Perform differential activity</a></li>
<li class="chapter" data-level="4.13" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#visualize-the-results"><i class="fa fa-check"></i><b>4.13</b> Visualize the results</a></li>
<li class="chapter" data-level="4.14" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#differential-network-analysis-1"><i class="fa fa-check"></i><b>4.14</b> Differential network analysis</a></li>
<li class="chapter" data-level="4.15" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#session-info-2"><i class="fa fa-check"></i><b>4.15</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html"><i class="fa fa-check"></i><b>5</b> Single modality: scRNA-seq only</a>
<ul>
<li class="chapter" data-level="5.1" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#load-regulon"><i class="fa fa-check"></i><b>5.1</b> Load regulon</a></li>
<li class="chapter" data-level="5.2" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#load-scrna-seq-data"><i class="fa fa-check"></i><b>5.2</b> Load scRNA-seq data</a></li>
<li class="chapter" data-level="5.3" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#calculate-activity"><i class="fa fa-check"></i><b>5.3</b> Calculate activity</a></li>
<li class="chapter" data-level="5.4" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#perform-differential-activity-1"><i class="fa fa-check"></i><b>5.4</b> Perform differential activity</a></li>
<li class="chapter" data-level="5.5" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#visualize-activity"><i class="fa fa-check"></i><b>5.5</b> Visualize activity</a></li>
<li class="chapter" data-level="5.6" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#pathway-enrichment"><i class="fa fa-check"></i><b>5.6</b> Pathway enrichment</a></li>
<li class="chapter" data-level="5.7" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#session-info-3"><i class="fa fa-check"></i><b>5.7</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="contact.html"><a href="contact.html"><i class="fa fa-check"></i><b>6</b> Contact</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Epiregulon documentation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="advanced-workflow-with-multiassayexperiment" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> Advanced workflow with MultiAssayExperiment<a href="advanced-workflow-with-multiassayexperiment.html#advanced-workflow-with-multiassayexperiment" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In this chapter, we illustrate the epiregulon workflow starting from data in the form of <code>SingleCellExperiment</code> objects using the Wilcoxon weight estimation method.</p>
<p>This is a dataset of hematopoiesis from the ArchR <a href="https://www.archrproject.com/bookdown/index.html">tutorial</a>. Prior to using epiregulon, this dataset has been fully preprocessed in ArchR, and converted to a <code>MultiAssayExperiment</code> using <code>epireglon.archr::archr2MAE</code>. The MAE object was uploaded to <code>scMultiome</code> for full reproducibility. In this dataset, scRNAseq and scATACseq were unpaired and integrated by the <code>ArchR::addGeneIntegrationMatrix</code> function.</p>
<div id="data-preparation-1" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Data preparation<a href="advanced-workflow-with-multiassayexperiment.html#data-preparation-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Download the example dataset from <em><a href="https://bioconductor.org/packages/3.19/scMultiome">scMultiome</a></em> package</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-1" tabindex="-1"></a>mae <span class="ot">&lt;-</span> scMultiome<span class="sc">::</span><span class="fu">hematopoiesis</span>()</span>
<span id="cb68-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-2" tabindex="-1"></a></span>
<span id="cb68-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-3" tabindex="-1"></a><span class="co"># Load peak matrix</span></span>
<span id="cb68-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-4" tabindex="-1"></a>PeakMatrix <span class="ot">&lt;-</span> mae[[<span class="st">&quot;PeakMatrix&quot;</span>]]</span>
<span id="cb68-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-5" tabindex="-1"></a></span>
<span id="cb68-6"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-6" tabindex="-1"></a><span class="co"># Load expression matrix</span></span>
<span id="cb68-7"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-7" tabindex="-1"></a>GeneExpressionMatrix <span class="ot">&lt;-</span> mae[[<span class="st">&quot;GeneIntegrationMatrix&quot;</span>]]</span>
<span id="cb68-8"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-8" tabindex="-1"></a></span>
<span id="cb68-9"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-9" tabindex="-1"></a><span class="co"># Add gene symbols to rownames</span></span>
<span id="cb68-10"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-10" tabindex="-1"></a><span class="fu">rownames</span>(GeneExpressionMatrix) <span class="ot">&lt;-</span> <span class="fu">rowData</span>(GeneExpressionMatrix)<span class="sc">$</span>name</span>
<span id="cb68-11"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-11" tabindex="-1"></a></span>
<span id="cb68-12"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-12" tabindex="-1"></a><span class="co"># Transfer dimensionality reduction matrix to GeneExpression</span></span>
<span id="cb68-13"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-13" tabindex="-1"></a><span class="fu">reducedDim</span>(GeneExpressionMatrix, <span class="st">&quot;IterativeLSI&quot;</span>) <span class="ot">&lt;-</span> </span>
<span id="cb68-14"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-14" tabindex="-1"></a>  <span class="fu">reducedDim</span>(mae[[<span class="st">&#39;TileMatrix500&#39;</span>]], <span class="st">&quot;IterativeLSI&quot;</span>)</span>
<span id="cb68-15"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-15" tabindex="-1"></a><span class="fu">reducedDim</span>(GeneExpressionMatrix, <span class="st">&quot;UMAP&quot;</span>) <span class="ot">&lt;-</span> </span>
<span id="cb68-16"><a href="advanced-workflow-with-multiassayexperiment.html#cb68-16" tabindex="-1"></a>  <span class="fu">reducedDim</span>(mae[[<span class="st">&#39;TileMatrix500&#39;</span>]], <span class="st">&quot;UMAP&quot;</span>)</span></code></pre></div>
<p>Visualize the data</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb69-1" tabindex="-1"></a>scater<span class="sc">::</span><span class="fu">plotReducedDim</span>(GeneExpressionMatrix, </span>
<span id="cb69-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb69-2" tabindex="-1"></a>                       <span class="at">dimred =</span> <span class="st">&quot;UMAP&quot;</span>, </span>
<span id="cb69-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb69-3" tabindex="-1"></a>                       <span class="at">text_by =</span> <span class="st">&quot;Clusters2&quot;</span>, </span>
<span id="cb69-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb69-4" tabindex="-1"></a>                       <span class="at">colour_by =</span> <span class="st">&quot;Clusters2&quot;</span>,</span>
<span id="cb69-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb69-5" tabindex="-1"></a>                       <span class="at">point_size =</span> <span class="fl">0.3</span>,</span>
<span id="cb69-6"><a href="advanced-workflow-with-multiassayexperiment.html#cb69-6" tabindex="-1"></a>                       <span class="at">point_alpha =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
</div>
<div id="retrieve-bulk-tf-chip-seq-binding-sites-1" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Retrieve bulk TF ChIP-seq binding sites<a href="advanced-workflow-with-multiassayexperiment.html#retrieve-bulk-tf-chip-seq-binding-sites-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First, we retrieve the information of TF binding sites collected from Cistrome and ENCODE ChIP-seq. Currently, human genomes hg19 and hg38 and mouse genome mm10 are available</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb70-1" tabindex="-1"></a><span class="fu">library</span>(epiregulon)</span>
<span id="cb70-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb70-2" tabindex="-1"></a>grl <span class="ot">&lt;-</span> <span class="fu">getTFMotifInfo</span>(<span class="at">genome =</span> <span class="st">&quot;hg19&quot;</span>)</span></code></pre></div>
<pre><code>## snapshotDate(): 2024-03-11</code></pre>
<pre><code>## see ?scMultiome and browseVignettes(&#39;scMultiome&#39;) for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb74-1" tabindex="-1"></a>grl</span></code></pre></div>
<pre><code>## GRangesList object of length 1558:
## $AEBP2
## GRanges object with 2761 ranges and 0 metadata columns:
##          seqnames            ranges strand
##             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;
##      [1]     chr1       10001-10446      *
##      [2]     chr1     877485-877780      *
##      [3]     chr1     919866-920161      *
##      [4]     chr1   2985496-2985846      *
##      [5]     chr1   2985975-2986514      *
##      ...      ...               ...    ...
##   [2757]     chrY   8333302-8333771      *
##   [2758]     chrY 13842450-13842966      *
##   [2759]     chrY 13868154-13868670      *
##   [2760]     chrY 21464547-21465020      *
##   [2761]     chrY 22147548-22147868      *
##   -------
##   seqinfo: 25 sequences from an unspecified genome; no seqlengths
## 
## ...
## &lt;1557 more elements&gt;</code></pre>
</div>
<div id="link-atacseq-peaks-to-target-genes" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Link ATACseq peaks to target genes<a href="advanced-workflow-with-multiassayexperiment.html#link-atacseq-peaks-to-target-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next, we compute peak to gene correlations using a custom algorithm that has similar performance to ArchR’s P2G function. Wherever possible, use a multidimensional dimensionality reduction matrix such as LSI or PCA instead of UMAP or TSNE since the former provides a more accurate estimate of cell similarity.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb76-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1010</span>)</span>
<span id="cb76-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb76-2" tabindex="-1"></a>p2g <span class="ot">&lt;-</span> <span class="fu">calculateP2G</span>(<span class="at">peakMatrix =</span> PeakMatrix, </span>
<span id="cb76-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb76-3" tabindex="-1"></a>                    <span class="at">expMatrix =</span> GeneExpressionMatrix, </span>
<span id="cb76-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb76-4" tabindex="-1"></a>                    <span class="at">exp_assay =</span> <span class="st">&quot;normalizedCounts&quot;</span>,</span>
<span id="cb76-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb76-5" tabindex="-1"></a>                    <span class="at">reducedDim =</span> <span class="fu">reducedDim</span>(GeneExpressionMatrix, <span class="st">&quot;IterativeLSI&quot;</span>))</span></code></pre></div>
<pre><code>## Using epiregulon to compute peak to gene links...</code></pre>
<pre><code>## performing k means clustering to form metacells</code></pre>
<pre><code>## Computing correlation</code></pre>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb80-1" tabindex="-1"></a>p2g</span></code></pre></div>
<pre><code>## DataFrame with 11694 rows and 8 columns
##         idxATAC         chr     start       end    idxRNA    target Correlation  distance
##       &lt;integer&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;   &lt;array&gt;    &lt;matrix&gt; &lt;integer&gt;
## 1             7        chr1    801002    801502         2 LINC00115    0.627327     36099
## 2            24        chr1    894453    894953         6    KLHL17    0.617991         0
## 3            25        chr1    894960    895460         6    KLHL17    0.564360         0
## 4            25        chr1    894960    895460         9     ISG15    0.524174     51386
## 5            37        chr1    935289    935789         8      HES4    0.509850         0
## ...         ...         ...       ...       ...       ...       ...         ...       ...
## 11690    146390       chr22  50980758  50981258     12088  MAPK8IP2    0.600164     55872
## 11691    146393       chr22  50983942  50984442     12079    NCAPH2    0.532094     37097
## 11692    146403       chr22  51021154  51021654     12078      LMF2    0.502373     73018
## 11693    146403       chr22  51021154  51021654     12089      ARSA    0.577122     44747
## 11694    146412       chr22  51110826  51111326     12090    SHANK3    0.532273         0</code></pre>
</div>
<div id="add-tf-motif-binding-to-peaks-1" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Add TF motif binding to peaks<a href="advanced-workflow-with-multiassayexperiment.html#add-tf-motif-binding-to-peaks-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The next step is to add the TF motif binding information by overlapping the regions of the peak matrix with the bulk chip-seq database.</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb82-1" tabindex="-1"></a>overlap <span class="ot">&lt;-</span> <span class="fu">addTFMotifInfo</span>(<span class="at">grl =</span> grl, <span class="at">p2g =</span> p2g, <span class="at">peakMatrix =</span> PeakMatrix)</span></code></pre></div>
<pre><code>## Computing overlap...</code></pre>
<pre><code>## Success!</code></pre>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb85-1" tabindex="-1"></a><span class="fu">head</span>(overlap)</span></code></pre></div>
<pre><code>##      idxATAC idxTF      tf
## 1013       7    13    ARNT
## 1014       7    17    ATF2
## 1015       7    21    ATF7
## 1016       7    28    BCL6
## 1017       7    30    BCOR
## 1018       7    32 BHLHE40</code></pre>
</div>
<div id="generate-regulons-1" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Generate regulons<a href="advanced-workflow-with-multiassayexperiment.html#generate-regulons-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A long format dataframe, representing the inferred regulons, is then generated. The dataframe consists of three columns:</p>
<ul>
<li>tf (transcription factor)</li>
<li>target gene</li>
<li>peak to gene correlation between tf and target gene</li>
</ul>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb87-1" tabindex="-1"></a>regulon <span class="ot">&lt;-</span> <span class="fu">getRegulon</span>(p2g, overlap, <span class="at">aggregate=</span><span class="cn">FALSE</span>)</span>
<span id="cb87-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb87-2" tabindex="-1"></a>regulon</span></code></pre></div>
<pre><code>## DataFrame with 1712540 rows and 10 columns
##           idxATAC         chr     start       end    idxRNA      target  distance     idxTF              tf     corr
##         &lt;integer&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt;     &lt;character&gt; &lt;matrix&gt;
## 1               7        chr1    801002    801502         2   LINC00115     36099        13            ARNT 0.627327
## 2               7        chr1    801002    801502         2   LINC00115     36099        17            ATF2 0.627327
## 3               7        chr1    801002    801502         2   LINC00115     36099        21            ATF7 0.627327
## 4               7        chr1    801002    801502         2   LINC00115     36099        28            BCL6 0.627327
## 5               7        chr1    801002    801502         2   LINC00115     36099        30            BCOR 0.627327
## ...           ...         ...       ...       ...       ...         ...       ...       ...             ...      ...
## 1712536    146412       chr22  51110826  51111326     12090      SHANK3         0      1172 POLR2AphosphoS5 0.532273
## 1712537    146412       chr22  51110826  51111326     12090      SHANK3         0      1312           ZNF16 0.532273
## 1712538    146412       chr22  51110826  51111326     12090      SHANK3         0      1461          ZNF600 0.532273
## 1712539    146412       chr22  51110826  51111326     12090      SHANK3         0      1490          ZNF687 0.532273
## 1712540    146412       chr22  51110826  51111326     12090      SHANK3         0      1520          ZNF777 0.532273</code></pre>
</div>
<div id="prune-network" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> Prune network<a href="advanced-workflow-with-multiassayexperiment.html#prune-network" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Epiregulon prunes the network by performing tests of independence on the observed number of cells jointly expressing transcription factor (TF), regulatory element (RE) and target gene (TG) vs the expected number of cells if TF/RE and TG are independently expressed. We implement two tests, the binomial test and the chi-square test. In the binomial test, the expected probability is P(TF, RE) * P(TG), and the number of trials is the total number of cells, and the observed successes is the number of cells jointly expressing all three elements. In the chi-square test, the expected probability for having all 3 elements active is also P(TF, RE) * P(TG) and the probability otherwise is 1- P(TF, RE) * P(TG). The observed cell count for the active category is the number of cells jointly expressing all three elements, and the cell count for the inactive category is n - n_triple.</p>
<p>We calculate cluster-specific p-values if users supply cluster labels. This is useful if we are interested in cluster-specific networks. The pruned regulons can then be used to visualize differential networks for transcription factors of interest. See section on differential networks.</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb89-1" tabindex="-1"></a>pruned.regulon <span class="ot">&lt;-</span> <span class="fu">pruneRegulon</span>(<span class="at">expMatrix =</span> GeneExpressionMatrix,</span>
<span id="cb89-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb89-2" tabindex="-1"></a>                               <span class="at">exp_assay =</span> <span class="st">&quot;normalizedCounts&quot;</span>,</span>
<span id="cb89-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb89-3" tabindex="-1"></a>                               <span class="at">peakMatrix =</span> PeakMatrix,</span>
<span id="cb89-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb89-4" tabindex="-1"></a>                               <span class="at">peak_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb89-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb89-5" tabindex="-1"></a>                               <span class="at">regulon =</span> regulon,</span>
<span id="cb89-6"><a href="advanced-workflow-with-multiassayexperiment.html#cb89-6" tabindex="-1"></a>                               <span class="at">prune_value =</span> <span class="st">&quot;pval&quot;</span>,</span>
<span id="cb89-7"><a href="advanced-workflow-with-multiassayexperiment.html#cb89-7" tabindex="-1"></a>                               <span class="at">regulon_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb89-8"><a href="advanced-workflow-with-multiassayexperiment.html#cb89-8" tabindex="-1"></a>                               <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>Clusters2)</span></code></pre></div>
<pre><code>## pruning network with chi.sq tests using a regulon cutoff of pval&lt;0.05</code></pre>
<pre><code>## pruning regulons</code></pre>
</div>
<div id="add-weights-1" class="section level2 hasAnchor" number="3.7">
<h2><span class="header-section-number">3.7</span> Add Weights<a href="advanced-workflow-with-multiassayexperiment.html#add-weights-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>While the <code>pruneRegulon</code> function provides statistics on the joint occurrence of TF-RE-TG, we would like to further estimate the strength of regulation. Biologically, this can be interpreted as the magnitude of gene expression changes induced by transcription factor activity. Epiregulon estimates the regulatory potential using one of the three measures: 1) correlation between TG and TF or between TG and the product of TF and RE, 2) mutual information between TG and TF expression or between TG and the product of TF and RE, or 3) Wilcoxon test statistics of target gene expression in cells jointly expressing all 3 elements vs cells that do not.</p>
<p>Two measures (correlation and Wilcoxon) give both the magnitude and directionality of changes whereas mutational information is always positive. The correlation and mutual information statistics are computed on grouped pseudobulks by user-supplied cluster labels and yield a single weight across all clusters per each TF-RE-target triplet. In contrast, the Wilcoxon method group cells based on the joint expression of TF, RE and TG in each single cell or in cell aggregates. If cell labels are provided, we calculate cluster-specific weights in addition to estimating weights from all the cells. Cell aggregation uses a default value of 10 cells and can help overcome sparsity and speed up computation. If cluster labels are provided, we can obtain weights of individual clusters and all cells combined.</p>
<p>In this example, we apply Wilcoxon test on cell aggregates of 10 cells. We use the Wilcoxon weight method because we are interested in computing cell type-specific weights.</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb92-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1010</span>)</span>
<span id="cb92-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb92-2" tabindex="-1"></a>regulon.w <span class="ot">&lt;-</span> <span class="fu">addWeights</span>(<span class="at">regulon =</span> pruned.regulon,</span>
<span id="cb92-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb92-3" tabindex="-1"></a>                        <span class="at">expMatrix  =</span> GeneExpressionMatrix,</span>
<span id="cb92-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb92-4" tabindex="-1"></a>                        <span class="at">exp_assay  =</span> <span class="st">&quot;normalizedCounts&quot;</span>,</span>
<span id="cb92-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb92-5" tabindex="-1"></a>                        <span class="at">peakMatrix =</span> PeakMatrix,</span>
<span id="cb92-6"><a href="advanced-workflow-with-multiassayexperiment.html#cb92-6" tabindex="-1"></a>                        <span class="at">peak_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb92-7"><a href="advanced-workflow-with-multiassayexperiment.html#cb92-7" tabindex="-1"></a>                        <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>Clusters2,</span>
<span id="cb92-8"><a href="advanced-workflow-with-multiassayexperiment.html#cb92-8" tabindex="-1"></a>                        <span class="at">aggregateCells =</span> <span class="cn">TRUE</span>,</span>
<span id="cb92-9"><a href="advanced-workflow-with-multiassayexperiment.html#cb92-9" tabindex="-1"></a>                        <span class="at">method =</span> <span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb92-10"><a href="advanced-workflow-with-multiassayexperiment.html#cb92-10" tabindex="-1"></a>                        <span class="at">useDim =</span> <span class="st">&quot;IterativeLSI&quot;</span>)</span></code></pre></div>
<pre><code>## adding weights using wilcoxon...</code></pre>
<pre><code>## performing pseudobulk using an average of 10 cells</code></pre>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb95-1" tabindex="-1"></a>regulon.w</span></code></pre></div>
<pre><code>## DataFrame with 343577 rows and 14 columns
##          idxATAC         chr     start       end    idxRNA      target  distance     idxTF          tf     corr
##        &lt;integer&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt; &lt;matrix&gt;
## 1            891        chr1   9223922   9224422       107        H6PD     68440      1030        ADNP 0.566311
## 2           1154        chr1  11724524  11725024       135       FBXO6       174      1030        ADNP 0.505591
## 3           1476        chr1  16003338  16003838       181        DDI2     59185      1030        ADNP 0.672807
## 4           2329        chr1  25237599  25238099       295       RUNX3     53313      1030        ADNP 0.865375
## 5           2333        chr1  25241815  25242315       295       RUNX3     49097      1030        ADNP 0.751697
## ...          ...         ...       ...       ...       ...         ...       ...       ...         ...      ...
## 343573    131524       chr19  12895246  12895746      8765        KLF1    102071       473     ZSCAN29 0.734430
## 343574    134071       chr19  41768840  41769340      9199       TGFB1     90291       473     ZSCAN29 0.635403
## 343575    135998       chr19  54711800  54712300      9580      MBOAT7     16066       473     ZSCAN29 0.793423
## 343576    144131       chr22  30838237  30838737     11838       SF3A1     83323       473     ZSCAN29 0.502352
## 343577    144587       chr22  37297198  37297698     11901      CSF2RB      9976       473     ZSCAN29 0.672100
##                             pval                     stats                qval                             weight
##                         &lt;matrix&gt;                  &lt;matrix&gt;            &lt;matrix&gt;                           &lt;matrix&gt;
## 1      2.42572e-53:1:1.00000:... 236.3764:0:0.00000000:... 3.77430e-47:1:1:...   0.352108:-0.0572865:0.218160:...
## 2      1.70871e-07:1:0.96773:...  27.3374:0:0.00163667:... 2.42454e-01:1:1:...   0.163694: 0.1313720:0.236942:...
## 3      2.50344e-04:1:1.00000:...  13.4096:0:0.00000000:... 1.00000e+00:1:1:...   0.177100: 0.0000000:0.000000:...
## 4      7.39278e-14:1:1.00000:...  55.9609:0:0.00000000:... 1.08689e-07:1:1:...   0.427978: 0.0000000:0.281650:...
## 5      2.20925e-10:1:1.00000:...  40.2722:0:0.00000000:... 3.19614e-04:1:1:...   0.313465: 0.0000000:0.229459:...
## ...                          ...                       ...                 ...                                ...
## 343573       5.96086e-04:1:1:...         11.788160:0:0:...           1:1:1:... 0.174968:-0.0612248: 0.0000000:...
## 343574       7.23903e-01:1:1:...          0.124784:0:0:...           1:1:1:... 0.157035: 0.1168639:-0.1527121:...
## 343575       4.26510e-02:1:1:...          4.109182:0:0:...           1:1:1:... 0.227676:-0.0954814: 0.0000000:...
## 343576       2.76179e-05:1:1:...         17.575117:0:0:...           1:1:1:... 0.255186:-0.0791276:-0.0727201:...
## 343577       1.12773e-03:1:1:...         10.605155:0:0:...           1:1:1:... 0.253897:-0.1591292: 0.0000000:...</code></pre>
</div>
<div id="calculate-tf-activity-1" class="section level2 hasAnchor" number="3.8">
<h2><span class="header-section-number">3.8</span> Calculate TF activity<a href="advanced-workflow-with-multiassayexperiment.html#calculate-tf-activity-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Finally, the activities for a specific TF in each cell are computed by averaging the weighted expressions of target genes linked to the TF weighted.
<span class="math display">\[y=\frac{1}{n}\sum_{i=1}^{n} x_i * weight_i\]</span>
where <span class="math inline">\(y\)</span> is the activity of a TF for a cell
<span class="math inline">\(n\)</span> is the total number of targets for a TF
<span class="math inline">\(x_i\)</span> is the log count expression of target i where i in {1,2,…,n}
<span class="math inline">\(weight_i\)</span> is the weight of TF and target i</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb97-1" tabindex="-1"></a>score.combine <span class="ot">&lt;-</span> <span class="fu">calculateActivity</span>(<span class="at">expMatrix =</span> GeneExpressionMatrix, </span>
<span id="cb97-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb97-2" tabindex="-1"></a>                                   <span class="at">regulon =</span> regulon.w, </span>
<span id="cb97-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb97-3" tabindex="-1"></a>                                   <span class="at">mode =</span> <span class="st">&quot;weight&quot;</span>, </span>
<span id="cb97-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb97-4" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">&quot;weightedMean&quot;</span>, </span>
<span id="cb97-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb97-5" tabindex="-1"></a>                                   <span class="at">exp_assay =</span> <span class="st">&quot;normalizedCounts&quot;</span>)</span></code></pre></div>
<pre><code>## calculating TF activity from regulon using weightedmean</code></pre>
<pre><code>## Warning in calculateActivity(expMatrix = GeneExpressionMatrix, regulon = regulon.w, : The weight column contains multiple
## subcolumns but no cluster information was provided. Using first column to compute activity...</code></pre>
<pre><code>## aggregating regulons...</code></pre>
<pre><code>## creating weight matrix...</code></pre>
<pre><code>## calculating activity scores...</code></pre>
<pre><code>## normalize by the number of targets...</code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb104-1" tabindex="-1"></a><span class="fu">head</span>(score.combine[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code></pre></div>
<pre><code>## 5 x 5 sparse Matrix of class &quot;dgCMatrix&quot;
##      scATAC_BMMC_R1#TTATGTCAGTGATTAG-1 scATAC_BMMC_R1#AAGATAGTCACCGCGA-1 scATAC_BMMC_R1#GCATTGAAGATTCCGT-1
## ADNP                         0.1441244                         0.2181751                         0.1829715
## AFF1                         0.1812722                         0.1908101                         0.5706316
## AFF4                         0.1226549                         0.3673765                         0.1364362
## AGO1                         0.1818927                         0.2179344                         0.1032600
## AGO2                         0.0791438                         0.4689620                         0.1131031
##      scATAC_BMMC_R1#TATGTTCAGGGTTCCC-1 scATAC_BMMC_R1#AGTTACGAGAACGTCG-1
## ADNP                         0.1636268                        0.15758333
## AFF1                         0.1928402                        0.19996029
## AFF4                         0.2080821                        0.13736470
## AGO1                         0.1678979                        0.22520235
## AGO2                         0.2267187                        0.06771202</code></pre>
</div>
<div id="differential-tf-activity-test" class="section level2 hasAnchor" number="3.9">
<h2><span class="header-section-number">3.9</span> Differential TF activity test<a href="advanced-workflow-with-multiassayexperiment.html#differential-tf-activity-test" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can next determine which TFs exhibit differential activities across cell clusters/groups via the <code>findDifferentialActivity</code> function. This function depends on <code>findMarkers</code> function from <em><a href="https://bioconductor.org/packages/3.19/scran">scran</a></em> package.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb106-1" tabindex="-1"></a><span class="fu">library</span>(epiregulon.extra)</span>
<span id="cb106-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb106-2" tabindex="-1"></a>markers  <span class="ot">&lt;-</span> <span class="fu">findDifferentialActivity</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb106-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb106-3" tabindex="-1"></a>                                    <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>Clusters2, </span>
<span id="cb106-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb106-4" tabindex="-1"></a>                                    <span class="at">pval.type =</span> <span class="st">&quot;some&quot;</span>, </span>
<span id="cb106-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb106-5" tabindex="-1"></a>                                    <span class="at">direction =</span> <span class="st">&quot;up&quot;</span>, </span>
<span id="cb106-6"><a href="advanced-workflow-with-multiassayexperiment.html#cb106-6" tabindex="-1"></a>                                    <span class="at">test.type =</span> <span class="st">&quot;t&quot;</span>)</span></code></pre></div>
<p>getSigGenes compiles the different test results into a single dataframe and enables user to supply their desired cutoffs for significance and variable to order by.</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb107-1" tabindex="-1"></a>markers.sig <span class="ot">&lt;-</span> <span class="fu">getSigGenes</span>(markers, <span class="at">topgenes =</span> <span class="dv">3</span> )</span></code></pre></div>
<pre><code>## Using a logFC cutoff of 0.2 for class B for direction equal to any</code></pre>
<pre><code>## Using a logFC cutoff of 0.5 for class CD4.M for direction equal to any</code></pre>
<pre><code>## Using a logFC cutoff of 0.4 for class CD4.N for direction equal to any</code></pre>
<pre><code>## Using a logFC cutoff of 0.3 for class CLP for direction equal to any</code></pre>
<pre><code>## Using a logFC cutoff of 0.4 for class Erythroid for direction equal to any</code></pre>
<pre><code>## Using a logFC cutoff of 0.2 for class GMP for direction equal to any</code></pre>
<pre><code>## Using a logFC cutoff of 0.9 for class Mono for direction equal to any</code></pre>
<pre><code>## Using a logFC cutoff of 0.5 for class NK for direction equal to any</code></pre>
<pre><code>## Using a logFC cutoff of 0.2 for class pDC for direction equal to any</code></pre>
<pre><code>## Using a logFC cutoff of 0.3 for class PreB for direction equal to any</code></pre>
<pre><code>## Using a logFC cutoff of 0.2 for class Progenitor for direction equal to any</code></pre>
</div>
<div id="visualizing-tf-activities" class="section level2 hasAnchor" number="3.10">
<h2><span class="header-section-number">3.10</span> Visualizing TF activities<a href="advanced-workflow-with-multiassayexperiment.html#visualizing-tf-activities" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Epiregulon also provides multiple options for visualizing the inferred TF activities by reduced dimensional space</p>
<p>tSNE or UMAP plots:</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb119-1" tabindex="-1"></a><span class="fu">options</span>(<span class="at">ggrastr.default.dpi=</span><span class="dv">300</span>)</span>
<span id="cb119-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb119-2" tabindex="-1"></a>tfs_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;EBF1&quot;</span>,<span class="st">&quot;PAX5&quot;</span>, <span class="st">&quot;GATA3&quot;</span>,<span class="st">&quot;SPI1&quot;</span>)</span>
<span id="cb119-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb119-3" tabindex="-1"></a><span class="fu">plotActivityDim</span>(<span class="at">sce =</span> GeneExpressionMatrix,</span>
<span id="cb119-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb119-4" tabindex="-1"></a>                <span class="at">activity_matrix =</span> score.combine[tfs_interest,], </span>
<span id="cb119-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb119-5" tabindex="-1"></a>                <span class="at">tf =</span> tfs_interest, </span>
<span id="cb119-6"><a href="advanced-workflow-with-multiassayexperiment.html#cb119-6" tabindex="-1"></a>                <span class="at">dimtype =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb119-7"><a href="advanced-workflow-with-multiassayexperiment.html#cb119-7" tabindex="-1"></a>                <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb119-8"><a href="advanced-workflow-with-multiassayexperiment.html#cb119-8" tabindex="-1"></a>                <span class="at">ncol=</span><span class="dv">2</span>,</span>
<span id="cb119-9"><a href="advanced-workflow-with-multiassayexperiment.html#cb119-9" tabindex="-1"></a>                <span class="at">point_size=</span><span class="fl">0.1</span>,</span>
<span id="cb119-10"><a href="advanced-workflow-with-multiassayexperiment.html#cb119-10" tabindex="-1"></a>                <span class="at">rasterise =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<p>We can compare the activity with gene expression of the same TFs.</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb120-1" tabindex="-1"></a><span class="fu">plotActivityDim</span>(<span class="at">sce =</span> GeneExpressionMatrix, </span>
<span id="cb120-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb120-2" tabindex="-1"></a>                <span class="at">activity_matrix =</span> <span class="fu">assay</span>(GeneExpressionMatrix, <span class="st">&quot;normalizedCounts&quot;</span>)[tfs_interest,], </span>
<span id="cb120-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb120-3" tabindex="-1"></a>                <span class="at">tf =</span> tfs_interest, </span>
<span id="cb120-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb120-4" tabindex="-1"></a>                <span class="at">dimtype =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb120-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb120-5" tabindex="-1"></a>                <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb120-6"><a href="advanced-workflow-with-multiassayexperiment.html#cb120-6" tabindex="-1"></a>                <span class="at">ncol=</span><span class="dv">2</span>,</span>
<span id="cb120-7"><a href="advanced-workflow-with-multiassayexperiment.html#cb120-7" tabindex="-1"></a>                <span class="at">legend.label =</span> <span class="st">&quot;Gex&quot;</span>,</span>
<span id="cb120-8"><a href="advanced-workflow-with-multiassayexperiment.html#cb120-8" tabindex="-1"></a>                <span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&quot;grey&quot;</span>,<span class="st">&quot;blue&quot;</span>),</span>
<span id="cb120-9"><a href="advanced-workflow-with-multiassayexperiment.html#cb120-9" tabindex="-1"></a>                <span class="at">point_size=</span><span class="fl">0.1</span>,</span>
<span id="cb120-10"><a href="advanced-workflow-with-multiassayexperiment.html#cb120-10" tabindex="-1"></a>                <span class="at">rasterise =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<p>We can also plot violin plot to visualize TF activity.</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb121-1" tabindex="-1"></a><span class="fu">plotActivityViolin</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb121-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb121-2" tabindex="-1"></a>                   <span class="at">tf =</span> tfs_interest, </span>
<span id="cb121-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb121-3" tabindex="-1"></a>                   <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>Clusters2,</span>
<span id="cb121-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb121-4" tabindex="-1"></a>                   <span class="at">legend.label =</span> <span class="st">&quot;Gex&quot;</span>,</span>
<span id="cb121-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb121-5" tabindex="-1"></a>                   <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb121-6"><a href="advanced-workflow-with-multiassayexperiment.html#cb121-6" tabindex="-1"></a>                   <span class="at">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
<p>We plot violin plot to visualize TF gene expression.</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb122-1" tabindex="-1"></a><span class="fu">plotActivityViolin</span>(<span class="at">activity_matrix =</span> <span class="fu">assay</span>(GeneExpressionMatrix, <span class="st">&quot;normalizedCounts&quot;</span>)[tfs_interest,],  </span>
<span id="cb122-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb122-2" tabindex="-1"></a>                   <span class="at">tf =</span> tfs_interest, </span>
<span id="cb122-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb122-3" tabindex="-1"></a>                   <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>Clusters2,</span>
<span id="cb122-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb122-4" tabindex="-1"></a>                   <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb122-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb122-5" tabindex="-1"></a>                   <span class="at">ncol=</span><span class="dv">2</span>,</span>
<span id="cb122-6"><a href="advanced-workflow-with-multiassayexperiment.html#cb122-6" tabindex="-1"></a>                   <span class="at">legend.label =</span> <span class="st">&quot;gene expression&quot;</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<p>We can visualize the different TFs in a bubble plot:</p>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb123-1" tabindex="-1"></a><span class="fu">plotBubble</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb123-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb123-2" tabindex="-1"></a>           <span class="at">tf =</span> tfs_interest, </span>
<span id="cb123-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb123-3" tabindex="-1"></a>           GeneExpressionMatrix<span class="sc">$</span>Clusters2, </span>
<span id="cb123-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb123-4" tabindex="-1"></a>           <span class="at">bubblesize =</span> <span class="st">&quot;FDR&quot;</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<p>We visualize the top differential TFs based on activity.</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb124-1" tabindex="-1"></a><span class="fu">plotBubble</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb124-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb124-2" tabindex="-1"></a>           <span class="at">tf =</span> markers.sig<span class="sc">$</span>tf, </span>
<span id="cb124-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb124-3" tabindex="-1"></a>           GeneExpressionMatrix<span class="sc">$</span>Clusters2, </span>
<span id="cb124-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb124-4" tabindex="-1"></a>           <span class="at">bubblesize =</span> <span class="st">&quot;FDR&quot;</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
</div>
<div id="geneset-enrichment" class="section level2 hasAnchor" number="3.11">
<h2><span class="header-section-number">3.11</span> Geneset enrichment<a href="advanced-workflow-with-multiassayexperiment.html#geneset-enrichment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Sometimes we are interested to know what pathways are enriched in the regulon of a particular TF. We can perform geneset enrichment using the enricher function from <a href="http://yulab-smu.top/biomedical-knowledge-mining-book/universal-api.html">clusterProfiler</a>.</p>
<p>Here we first download Hallmark and C2 signatures from hallmark and then perform gene set enrichment of the known lineage factors. As expected, EBF1 is consistent with a B cell lineage factor, GATA3 and RUNX3 with lymphoid lineage and SPI1 with myeloid lineage.</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb125-1" tabindex="-1"></a><span class="co">#retrieve genesets</span></span>
<span id="cb125-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb125-2" tabindex="-1"></a>H <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">getGenesets</span>(<span class="at">org =</span> <span class="st">&quot;hsa&quot;</span>, </span>
<span id="cb125-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb125-3" tabindex="-1"></a>                                    <span class="at">db =</span> <span class="st">&quot;msigdb&quot;</span>, </span>
<span id="cb125-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb125-4" tabindex="-1"></a>                                    <span class="at">cat =</span> <span class="st">&quot;H&quot;</span>, </span>
<span id="cb125-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb125-5" tabindex="-1"></a>                                    <span class="at">gene.id.type =</span> <span class="st">&quot;SYMBOL&quot;</span> )</span></code></pre></div>
<pre><code>## Using cached version from 2024-03-19 13:20:14</code></pre>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb127-1" tabindex="-1"></a>C2 <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">getGenesets</span>(<span class="at">org =</span> <span class="st">&quot;hsa&quot;</span>, </span>
<span id="cb127-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb127-2" tabindex="-1"></a>                                     <span class="at">db =</span> <span class="st">&quot;msigdb&quot;</span>, </span>
<span id="cb127-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb127-3" tabindex="-1"></a>                                     <span class="at">cat =</span> <span class="st">&quot;C2&quot;</span>,</span>
<span id="cb127-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb127-4" tabindex="-1"></a>                                     <span class="at">gene.id.type =</span> <span class="st">&quot;SYMBOL&quot;</span> )</span></code></pre></div>
<pre><code>## Using cached version from 2024-03-19 11:49:25</code></pre>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb129-1" tabindex="-1"></a><span class="co">#combine genesets and convert genesets to be compatible with enricher</span></span>
<span id="cb129-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb129-2" tabindex="-1"></a>gs <span class="ot">&lt;-</span> <span class="fu">c</span>(H, C2)</span>
<span id="cb129-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb129-3" tabindex="-1"></a>gs.list <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind,<span class="fu">lapply</span>(<span class="fu">names</span>(gs), <span class="cf">function</span>(x) </span>
<span id="cb129-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb129-4" tabindex="-1"></a>  {<span class="fu">data.frame</span>(<span class="at">gs=</span>x, <span class="at">genes=</span>gs[[x]])}))</span>
<span id="cb129-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb129-5" tabindex="-1"></a></span>
<span id="cb129-6"><a href="advanced-workflow-with-multiassayexperiment.html#cb129-6" tabindex="-1"></a>enrichresults <span class="ot">&lt;-</span> <span class="fu">regulonEnrich</span>(<span class="at">TF =</span> tfs_interest, </span>
<span id="cb129-7"><a href="advanced-workflow-with-multiassayexperiment.html#cb129-7" tabindex="-1"></a>                               <span class="at">regulon =</span> regulon.w, </span>
<span id="cb129-8"><a href="advanced-workflow-with-multiassayexperiment.html#cb129-8" tabindex="-1"></a>                               <span class="at">weight =</span> <span class="st">&quot;weight&quot;</span>,</span>
<span id="cb129-9"><a href="advanced-workflow-with-multiassayexperiment.html#cb129-9" tabindex="-1"></a>                               <span class="at">weight_cutoff =</span> <span class="dv">0</span>, </span>
<span id="cb129-10"><a href="advanced-workflow-with-multiassayexperiment.html#cb129-10" tabindex="-1"></a>                               <span class="at">genesets =</span> gs.list)</span></code></pre></div>
<pre><code>## EBF1</code></pre>
<pre><code>## PAX5</code></pre>
<pre><code>## GATA3</code></pre>
<pre><code>## SPI1</code></pre>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb134-1" tabindex="-1"></a><span class="co">#plot results</span></span>
<span id="cb134-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb134-2" tabindex="-1"></a><span class="fu">enrichPlot</span>(<span class="at">results =</span> enrichresults, <span class="at">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/enrichment_hematopoeisis-1.png" width="1152" /></p>
</div>
<div id="differential-network-analysis" class="section level2 hasAnchor" number="3.12">
<h2><span class="header-section-number">3.12</span> Differential Network analysis<a href="advanced-workflow-with-multiassayexperiment.html#differential-network-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In addition to looking at the summed TF activity, a second approach to investigate differential TF activity is to compare and contrast target genes or network topology. In this example, we know that EBF1 is a B cell lineage factor. If we plot the differential network of EBF1 using the regulon with cluster-specific weights, we can see that EBF1 has many more targets in PreB cells than it has in CD4 memory T cells.</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb135-1" tabindex="-1"></a><span class="fu">plotDiffNetwork</span>(regulon.w,</span>
<span id="cb135-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb135-2" tabindex="-1"></a>                <span class="at">cutoff =</span> <span class="dv">0</span>,</span>
<span id="cb135-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb135-3" tabindex="-1"></a>                <span class="at">tf =</span> <span class="fu">c</span>(<span class="st">&quot;EBF1&quot;</span>),</span>
<span id="cb135-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb135-4" tabindex="-1"></a>                <span class="at">weight =</span> <span class="st">&quot;weight&quot;</span>,</span>
<span id="cb135-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb135-5" tabindex="-1"></a>                <span class="at">clusters =</span> <span class="fu">c</span>(<span class="st">&quot;PreB&quot;</span>,<span class="st">&quot;CD4.M&quot;</span>),</span>
<span id="cb135-6"><a href="advanced-workflow-with-multiassayexperiment.html#cb135-6" tabindex="-1"></a>                <span class="at">layout =</span> <span class="st">&quot;stress&quot;</span>)</span></code></pre></div>
<pre><code>## Replacement of na values for weights with 0</code></pre>
<pre><code>## Building graph using weight as edge weights</code></pre>
<p><img src="Epiregulon-documentation_files/figure-html/differential%20networks%20hematopoeisis-1.png" width="672" /></p>
<p>Sometimes, we are interested to identify interaction partners of the TFs of interest. This can be achieved by comparing the overlap of the targets genes for all the TFs and identify the most similar TFs by Jaccard similarity. To illustrate this function, we take a look at the top most similar 20 TFs to EBF1, and we successfully identify PAX5 as the most similar TF. Both PAX5 and EBF1 are important factors for B cell development (<a href="https://www.nature.com/articles/ni.2641" class="uri">https://www.nature.com/articles/ni.2641</a>).</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb138-1" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb138-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb138-2" tabindex="-1"></a><span class="co"># construct a graph of the preB cells</span></span>
<span id="cb138-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb138-3" tabindex="-1"></a>preB_network <span class="ot">&lt;-</span> <span class="fu">buildGraph</span>(regulon.w, <span class="at">weights =</span> <span class="st">&quot;weight&quot;</span>, <span class="at">cluster=</span><span class="st">&quot;PreB&quot;</span>)</span></code></pre></div>
<pre><code>## Building graph using weight as edge weights</code></pre>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-1" tabindex="-1"></a><span class="co"># compute a similarity matrix of all TFs</span></span>
<span id="cb140-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-2" tabindex="-1"></a>similarity_score <span class="ot">&lt;-</span> <span class="fu">calculateJaccardSimilarity</span>(preB_network)</span>
<span id="cb140-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-3" tabindex="-1"></a></span>
<span id="cb140-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-4" tabindex="-1"></a><span class="co"># Focus on EBF1</span></span>
<span id="cb140-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-5" tabindex="-1"></a>similarity_score_EBF1 <span class="ot">&lt;-</span> similarity_score[, <span class="st">&quot;EBF1&quot;</span>]</span>
<span id="cb140-6"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-6" tabindex="-1"></a>similarity_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">similarity =</span> <span class="fu">head</span>(<span class="fu">sort</span>(similarity_score_EBF1, </span>
<span id="cb140-7"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-7" tabindex="-1"></a>                                                   <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>),</span>
<span id="cb140-8"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-8" tabindex="-1"></a>                            <span class="at">TF =</span> <span class="fu">names</span>(<span class="fu">head</span>(<span class="fu">sort</span>(similarity_score_EBF1,</span>
<span id="cb140-9"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-9" tabindex="-1"></a>                                                 <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>)))</span>
<span id="cb140-10"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-10" tabindex="-1"></a></span>
<span id="cb140-11"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-11" tabindex="-1"></a>similarity_df<span class="sc">$</span>TF <span class="ot">&lt;-</span> <span class="fu">factor</span>(similarity_df<span class="sc">$</span>TF, <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(similarity_df<span class="sc">$</span>TF)))</span>
<span id="cb140-12"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-12" tabindex="-1"></a></span>
<span id="cb140-13"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-13" tabindex="-1"></a><span class="co"># plot top TFs most similar to EBF1</span></span>
<span id="cb140-14"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-14" tabindex="-1"></a>topTFplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(similarity_df, <span class="fu">aes</span>(<span class="at">x=</span>TF, <span class="at">y=</span>similarity)) <span class="sc">+</span></span>
<span id="cb140-15"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-15" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb140-16"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-16" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb140-17"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-17" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;EBF1 similarity&quot;</span>) <span class="sc">+</span></span>
<span id="cb140-18"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-18" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb140-19"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-19" tabindex="-1"></a></span>
<span id="cb140-20"><a href="advanced-workflow-with-multiassayexperiment.html#cb140-20" tabindex="-1"></a><span class="fu">print</span>(topTFplot)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
<p>In order to convince ourselves that our differential network is statistically significant, we permute the edges and obtain a background graph from averaging many iterations. Here, we plot the differential network graph subtracted by permuted graphs.</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-1" tabindex="-1"></a><span class="co"># create a permuted graph by rewiring the edges 100 times</span></span>
<span id="cb141-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-2" tabindex="-1"></a>permute_matrix <span class="ot">&lt;-</span> <span class="fu">permuteGraph</span>(preB_network, <span class="st">&quot;EBF1&quot;</span>, <span class="dv">100</span>, <span class="at">p=</span><span class="dv">1</span>)</span>
<span id="cb141-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-3" tabindex="-1"></a>permute_matrix <span class="ot">&lt;-</span> permute_matrix[<span class="fu">names</span>(similarity_score_EBF1),]</span>
<span id="cb141-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-4" tabindex="-1"></a>diff_matrix <span class="ot">&lt;-</span> similarity_score_EBF1<span class="sc">-</span><span class="fu">rowMeans</span>(permute_matrix)</span>
<span id="cb141-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-5" tabindex="-1"></a></span>
<span id="cb141-6"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-6" tabindex="-1"></a>diff_matrix_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">similarity =</span> <span class="fu">head</span>(<span class="fu">sort</span>(diff_matrix, </span>
<span id="cb141-7"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-7" tabindex="-1"></a>                                                    <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>),</span>
<span id="cb141-8"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-8" tabindex="-1"></a>                            <span class="at">TF =</span> <span class="fu">names</span>(<span class="fu">head</span>(<span class="fu">sort</span>(diff_matrix,</span>
<span id="cb141-9"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-9" tabindex="-1"></a>                                                 <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>)))</span>
<span id="cb141-10"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-10" tabindex="-1"></a></span>
<span id="cb141-11"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-11" tabindex="-1"></a>diff_matrix_df<span class="sc">$</span>TF <span class="ot">&lt;-</span> <span class="fu">factor</span>(diff_matrix_df<span class="sc">$</span>TF, <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(diff_matrix_df<span class="sc">$</span>TF)))</span>
<span id="cb141-12"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-12" tabindex="-1"></a></span>
<span id="cb141-13"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-13" tabindex="-1"></a><span class="co"># plot top TFs most similar to EBF1</span></span>
<span id="cb141-14"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-14" tabindex="-1"></a>topTFplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(diff_matrix_df, <span class="fu">aes</span>(<span class="at">x=</span>TF, <span class="at">y=</span>similarity)) <span class="sc">+</span></span>
<span id="cb141-15"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-15" tabindex="-1"></a>            <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb141-16"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-16" tabindex="-1"></a>            <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb141-17"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-17" tabindex="-1"></a>            <span class="fu">ggtitle</span>(<span class="st">&quot;background subtracted EBF1 similarity &quot;</span>) <span class="sc">+</span></span>
<span id="cb141-18"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-18" tabindex="-1"></a>            <span class="fu">theme_classic</span>()</span>
<span id="cb141-19"><a href="advanced-workflow-with-multiassayexperiment.html#cb141-19" tabindex="-1"></a><span class="fu">print</span>(topTFplot)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb142-1" tabindex="-1"></a><span class="co"># obtain empirical p-values</span></span>
<span id="cb142-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb142-2" tabindex="-1"></a>p_matrix <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">apply</span>(permute_matrix, <span class="dv">2</span>, <span class="cf">function</span>(x) {x <span class="sc">&gt;</span> similarity_score_EBF1}))</span>
<span id="cb142-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb142-3" tabindex="-1"></a>p_matrix[<span class="fu">names</span>(<span class="fu">head</span>(<span class="fu">sort</span>(diff_matrix,<span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>))]</span></code></pre></div>
<pre><code>##   PAX5   IRF4   TCF3  TCF12   RAG2  NIPBL POU2F2   TCF4   RAG1    RB1    MYB TRIM22   EZH2  EP300  MEF2A   CDK9    ERG 
##      0      0      0      0      0      0      0      0      0      0      0      0      0      0      0      0      0 
##  FOXP1 CREBBP BCL11A 
##      0      0      0</code></pre>
<p>Next, we are interested to compare the networks of two cell types, in this case, CD4 memory T cells (CD4.M) vs Monocytes (mono) cells. We build an edge subtracted graph and then calculate the degree centrality of the subtracted graph. We normalize centrality using the default square root function. The top 5 most positive TFs represent lineage factors more active in NK cells whereas the bottom 5 TFs present lineage factors enriched in CD4. We successfully identified the myeloid factor SPI1 to be associated with monocytes and Th1 factor TBX21 to be associated with CD4 T cells.</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb144-1" tabindex="-1"></a><span class="co">#regulon.w.2 &lt;- regulon.w</span></span>
<span id="cb144-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb144-2" tabindex="-1"></a><span class="co">#regulon.w &lt;- readRDS(&quot;/gstore/project/lineage/manuscript/epiregulon/OUTPUT/regulon.w.rds&quot;)</span></span>
<span id="cb144-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb144-3" tabindex="-1"></a><span class="co"># construct a graph of the CD4.M and NK cells respectively</span></span>
<span id="cb144-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb144-4" tabindex="-1"></a>CD4.M_network <span class="ot">&lt;-</span> <span class="fu">buildGraph</span>(regulon.w, <span class="at">weights =</span> <span class="st">&quot;weight&quot;</span>, <span class="at">cluster=</span><span class="st">&quot;CD4.M&quot;</span>)</span></code></pre></div>
<pre><code>## Building graph using weight as edge weights</code></pre>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb146-1" tabindex="-1"></a>Mono_network <span class="ot">&lt;-</span> <span class="fu">buildGraph</span>(regulon.w, <span class="at">weights =</span> <span class="st">&quot;weight&quot;</span>, <span class="at">cluster=</span><span class="st">&quot;Mono&quot;</span>)</span></code></pre></div>
<pre><code>## Building graph using weight as edge weights</code></pre>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb148-1" tabindex="-1"></a><span class="co"># construct a difference graph</span></span>
<span id="cb148-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb148-2" tabindex="-1"></a>diff_graph <span class="ot">&lt;-</span> <span class="fu">buildDiffGraph</span>(Mono_network,CD4.M_network, <span class="at">abs_diff =</span> <span class="cn">FALSE</span>)</span>
<span id="cb148-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb148-3" tabindex="-1"></a>diff_graph <span class="ot">&lt;-</span> <span class="fu">addCentrality</span>(diff_graph)</span>
<span id="cb148-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb148-4" tabindex="-1"></a>diff_graph <span class="ot">&lt;-</span> <span class="fu">normalizeCentrality</span>(diff_graph)</span>
<span id="cb148-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb148-5" tabindex="-1"></a>rank_table <span class="ot">&lt;-</span> <span class="fu">rankTfs</span>(diff_graph)</span>
<span id="cb148-6"><a href="advanced-workflow-with-multiassayexperiment.html#cb148-6" tabindex="-1"></a></span>
<span id="cb148-7"><a href="advanced-workflow-with-multiassayexperiment.html#cb148-7" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb148-8"><a href="advanced-workflow-with-multiassayexperiment.html#cb148-8" tabindex="-1"></a><span class="fu">ggplot</span>(rank_table, <span class="fu">aes</span>(<span class="at">x =</span> rank, <span class="at">y =</span> centrality)) <span class="sc">+</span></span>
<span id="cb148-9"><a href="advanced-workflow-with-multiassayexperiment.html#cb148-9" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb148-10"><a href="advanced-workflow-with-multiassayexperiment.html#cb148-10" tabindex="-1"></a>    ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span> <span class="fu">rbind</span>(<span class="fu">head</span>(rank_table, <span class="dv">10</span>), </span>
<span id="cb148-11"><a href="advanced-workflow-with-multiassayexperiment.html#cb148-11" tabindex="-1"></a>                                          <span class="fu">tail</span>(rank_table, <span class="dv">10</span>)), </span>
<span id="cb148-12"><a href="advanced-workflow-with-multiassayexperiment.html#cb148-12" tabindex="-1"></a>                             <span class="fu">aes</span>(<span class="at">label =</span> tf), </span>
<span id="cb148-13"><a href="advanced-workflow-with-multiassayexperiment.html#cb148-13" tabindex="-1"></a>                             <span class="at">nudge_x =</span> <span class="dv">0</span>, <span class="at">nudge_y =</span> <span class="dv">0</span>, <span class="at">box.padding =</span> <span class="fl">0.5</span>, <span class="at">max.overlaps =</span> <span class="cn">Inf</span>) <span class="sc">+</span></span>
<span id="cb148-14"><a href="advanced-workflow-with-multiassayexperiment.html#cb148-14" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span> <span class="fu">ggtitle</span> (<span class="st">&quot;differential TFs (Mono-CD4.M) ranked by degree centrality&quot;</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
<p>We can further explore interacting factors with the myeloid factor SPI1 using the same Jaccard similarity approach. We found CEBPA as the most similar TF as SPI1. SPI1 and CEBPA are known to be important for differentiation into myeloid cells (<a href="https://www.cell.com/cell-reports/pdfExtended/S2211-1247(18)30745-9" class="uri">https://www.cell.com/cell-reports/pdfExtended/S2211-1247(18)30745-9</a>).</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb149-1" tabindex="-1"></a><span class="fu">library</span>(igraph)</span></code></pre></div>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-1" tabindex="-1"></a>diff_graph_filter <span class="ot">&lt;-</span> <span class="fu">subgraph.edges</span>(diff_graph, </span>
<span id="cb150-2"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-2" tabindex="-1"></a>                                    <span class="fu">E</span>(diff_graph)[<span class="fu">E</span>(diff_graph)<span class="sc">$</span>weight<span class="sc">&gt;</span><span class="dv">0</span>], </span>
<span id="cb150-3"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-3" tabindex="-1"></a>                                    <span class="at">del=</span><span class="cn">TRUE</span>)</span>
<span id="cb150-4"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-4" tabindex="-1"></a></span>
<span id="cb150-5"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-5" tabindex="-1"></a></span>
<span id="cb150-6"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-6" tabindex="-1"></a><span class="co"># compute a similarity matrix of all TFs</span></span>
<span id="cb150-7"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-7" tabindex="-1"></a>similarity_score <span class="ot">&lt;-</span> <span class="fu">calculateJaccardSimilarity</span>(diff_graph_filter)</span>
<span id="cb150-8"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-8" tabindex="-1"></a></span>
<span id="cb150-9"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-9" tabindex="-1"></a><span class="co"># Focus on SPI1</span></span>
<span id="cb150-10"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-10" tabindex="-1"></a>similarity_score_SPI1 <span class="ot">&lt;-</span> similarity_score[, <span class="st">&quot;SPI1&quot;</span>]</span>
<span id="cb150-11"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-11" tabindex="-1"></a>similarity_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">similarity =</span> <span class="fu">head</span>(<span class="fu">sort</span>(similarity_score_SPI1, </span>
<span id="cb150-12"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-12" tabindex="-1"></a>                                                   <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>),</span>
<span id="cb150-13"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-13" tabindex="-1"></a>                            <span class="at">TF =</span> <span class="fu">names</span>(<span class="fu">head</span>(<span class="fu">sort</span>(similarity_score_SPI1,</span>
<span id="cb150-14"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-14" tabindex="-1"></a>                                                 <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>)))</span>
<span id="cb150-15"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-15" tabindex="-1"></a></span>
<span id="cb150-16"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-16" tabindex="-1"></a>similarity_df<span class="sc">$</span>TF <span class="ot">&lt;-</span> <span class="fu">factor</span>(similarity_df<span class="sc">$</span>TF, </span>
<span id="cb150-17"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-17" tabindex="-1"></a>                           <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(similarity_df<span class="sc">$</span>TF)))</span>
<span id="cb150-18"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-18" tabindex="-1"></a></span>
<span id="cb150-19"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-19" tabindex="-1"></a><span class="co"># plot top TFs most similar to SPI1</span></span>
<span id="cb150-20"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-20" tabindex="-1"></a>topTFplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(similarity_df, <span class="fu">aes</span>(<span class="at">x=</span>TF, <span class="at">y=</span>similarity)) <span class="sc">+</span></span>
<span id="cb150-21"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-21" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb150-22"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-22" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb150-23"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-23" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;SPI1 similarity&quot;</span>) <span class="sc">+</span></span>
<span id="cb150-24"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-24" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb150-25"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-25" tabindex="-1"></a></span>
<span id="cb150-26"><a href="advanced-workflow-with-multiassayexperiment.html#cb150-26" tabindex="-1"></a><span class="fu">print</span>(topTFplot)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
</div>
<div id="session-info-1" class="section level2 hasAnchor" number="3.13">
<h2><span class="header-section-number">3.13</span> Session Info<a href="advanced-workflow-with-multiassayexperiment.html#session-info-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="advanced-workflow-with-multiassayexperiment.html#cb151-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R Under development (unstable) (2023-12-04 r85659)
## Platform: x86_64-pc-linux-gnu
## Running under: Ubuntu 18.04.6 LTS
## 
## Matrix products: default
## BLAS:   /usr/local/lib/R/lib/libRblas.so 
## LAPACK: /usr/local/lib/R/lib/libRlapack.so;  LAPACK version 3.11.0
## 
## Random number generation:
##  RNG:     L&#39;Ecuyer-CMRG 
##  Normal:  Inversion 
##  Sample:  Rejection 
##  
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## time zone: Etc/UTC
## tzcode source: system (glibc)
## 
## attached base packages:
##  [1] parallel  stats4    grid      stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] igraph_2.0.3                      msigdbr_7.5.1                     epiregulon.extra_0.99.5          
##  [4] BSgenome.Hsapiens.UCSC.hg38_1.4.5 BSgenome_1.71.2                   rtracklayer_1.63.1               
##  [7] BiocIO_1.13.0                     Biostrings_2.71.4                 XVector_0.43.1                   
## [10] epiregulon_0.99.5                 scMultiome_1.3.0                  SingleCellExperiment_1.25.0      
## [13] MultiAssayExperiment_1.29.1       ExperimentHub_2.11.1              AnnotationHub_3.11.1             
## [16] BiocFileCache_2.11.1              dbplyr_2.4.0                      BiocStyle_2.31.0                 
## [19] rhdf5_2.47.6                      SummarizedExperiment_1.33.3       Biobase_2.63.0                   
## [22] RcppArmadillo_0.12.8.1.0          Rcpp_1.0.12                       Matrix_1.6-4                     
## [25] GenomicRanges_1.55.3              GenomeInfoDb_1.39.9               IRanges_2.37.1                   
## [28] S4Vectors_0.41.4                  BiocGenerics_0.49.1               sparseMatrixStats_1.15.0         
## [31] MatrixGenerics_1.15.0             matrixStats_1.2.0                 data.table_1.15.2                
## [34] stringr_1.5.1                     plyr_1.8.9                        magrittr_2.0.3                   
## [37] ggplot2_3.5.0                     gtable_0.3.4                      gtools_3.9.5                     
## [40] gridExtra_2.3                     devtools_2.4.5                    usethis_2.2.3                    
## [43] ArchR_1.0.3                      
## 
## loaded via a namespace (and not attached):
##   [1] fs_1.6.3                    bitops_1.0-7                enrichplot_1.23.1           DirichletMultinomial_1.45.0
##   [5] TFBSTools_1.41.0            RColorBrewer_1.1-3          HDO.db_0.99.1               httr_1.4.7                 
##   [9] Rgraphviz_2.47.0            profvis_0.3.8               tools_4.4.0                 backports_1.4.1            
##  [13] utf8_1.2.4                  R6_2.5.1                    HDF5Array_1.31.6            lazyeval_0.2.2             
##  [17] rhdf5filters_1.15.4         urlchecker_1.0.1            withr_3.0.0                 cli_3.6.2                  
##  [21] Cairo_1.6-2                 scatterpie_0.2.1            labeling_0.4.3              sass_0.4.9                 
##  [25] KEGGgraph_1.63.0            readr_2.1.5                 yulab.utils_0.1.4           Rsamtools_2.19.3           
##  [29] gson_0.1.0                  DOSE_3.29.2                 R.utils_2.12.3              scater_1.31.2              
##  [33] sessioninfo_1.2.2           limma_3.59.6                rstudioapi_0.15.0           RSQLite_2.3.5              
##  [37] gridGraphics_0.5-1          generics_0.1.3              dplyr_1.1.4                 GO.db_3.18.0               
##  [41] ggbeeswarm_0.7.2            fansi_1.0.6                 abind_1.4-5                 R.methodsS3_1.8.2          
##  [45] lifecycle_1.0.4             yaml_2.3.8                  edgeR_4.1.18                qvalue_2.35.0              
##  [49] SparseArray_1.3.4           blob_1.2.4                  promises_1.2.1              dqrng_0.3.2                
##  [53] crayon_1.5.2                miniUI_0.1.1.1              lattice_0.22-5              beachmat_2.19.1            
##  [57] cowplot_1.1.3               annotate_1.81.2             KEGGREST_1.43.0             pillar_1.9.0               
##  [61] knitr_1.45                  metapod_1.11.1              fgsea_1.29.0                rjson_0.2.21               
##  [65] codetools_0.2-19            fastmatch_1.1-4             glue_1.7.0                  ggfun_0.1.4                
##  [69] remotes_2.5.0               treeio_1.27.0               vctrs_0.6.5                 png_0.1-8                  
##  [73] poweRlaw_0.80.0             cachem_1.0.8                xfun_0.42                   S4Arrays_1.3.6             
##  [77] mime_0.12                   tidygraph_1.3.1             pracma_2.4.4                statmod_1.5.0              
##  [81] bluster_1.13.0              ellipsis_0.3.2              nlme_3.1-164                ggtree_3.11.1              
##  [85] bit64_4.0.5                 filelock_1.0.3              bslib_0.6.1                 irlba_2.3.5.1              
##  [89] vipor_0.4.7                 colorspace_2.1-0            seqLogo_1.69.0              DBI_1.2.2                  
##  [93] ggrastr_1.0.2               tidyselect_1.2.1            bit_4.0.5                   compiler_4.4.0             
##  [97] curl_5.2.1                  graph_1.81.0                BiocNeighbors_1.21.2        DelayedArray_0.29.9        
## [101] shadowtext_0.1.3            bookdown_0.38               checkmate_2.3.1             scales_1.3.0               
## [105] caTools_1.18.2              rappdirs_0.3.3              digest_0.6.35               motifmatchr_1.25.0         
## [109] rmarkdown_2.26              htmltools_0.5.7             pkgconfig_2.0.3             highr_0.10                 
## [113] fastmap_1.1.1               rlang_1.1.3                 htmlwidgets_1.6.4           shiny_1.8.0                
## [117] DelayedMatrixStats_1.25.1   farver_2.1.1                jquerylib_0.1.4             jsonlite_1.8.8             
## [121] BiocParallel_1.37.1         GOSemSim_2.29.1             R.oo_1.26.0                 BiocSingular_1.19.0        
## [125] RCurl_1.98-1.14             ggplotify_0.1.2             scuttle_1.13.1              GenomeInfoDbData_1.2.11    
## [129] patchwork_1.2.0             Rhdf5lib_1.25.1             munsell_0.5.0               ape_5.7-1                  
## [133] babelgene_22.9              viridis_0.6.5               EnrichmentBrowser_2.33.1    stringi_1.8.3              
## [137] ggraph_2.2.1                MASS_7.3-60.1               zlibbioc_1.49.3             pkgbuild_1.4.4             
## [141] ggrepel_0.9.5               CNEr_1.39.0                 graphlayouts_1.1.1          splines_4.4.0              
## [145] hms_1.1.3                   locfit_1.5-9.9              reshape2_1.4.4              ScaledMatrix_1.11.1        
## [149] pkgload_1.3.4               TFMPvalue_0.0.9             BiocVersion_3.19.1          XML_3.99-0.16.1            
## [153] evaluate_0.23               scran_1.31.3                BiocManager_1.30.22         tweenr_2.0.3               
## [157] tzdb_0.4.0                  httpuv_1.6.14               polyclip_1.10-6             tidyr_1.3.1                
## [161] purrr_1.0.2                 ggforce_0.4.2               rsvd_1.0.5                  xtable_1.8-4               
## [165] restfulr_0.0.15             tidytree_0.4.6              later_1.3.2                 viridisLite_0.4.2          
## [169] tibble_3.2.1                aplot_0.2.2                 clusterProfiler_4.11.0      memoise_2.0.1              
## [173] beeswarm_0.4.0              AnnotationDbi_1.65.2        GenomicAlignments_1.39.4    cluster_2.1.6              
## [177] GSEABase_1.65.1</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="basic-workflow-with-multiassayexperiment.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="archr-workflow-and-different-weight-methods.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/hematopoeisis.mae.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Epiregulon documentation.pdf", "Epiregulon documentation.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
