<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Basic workflow with MultiAssayExperiment | Epiregulon documentation</title>
  <meta name="description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Basic workflow with MultiAssayExperiment | Epiregulon documentation" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Basic workflow with MultiAssayExperiment | Epiregulon documentation" />
  
  <meta name="twitter:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  

<meta name="author" content="Xiaosai Yao, Tomasz Włodarczyk" />


<meta name="date" content="2025-01-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-preparation.html"/>
<link rel="next" href="advanced-workflow-with-multiassayexperiment.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Epiregulon documentation</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>1</b> Installation</a>
<ul>
<li class="chapter" data-level="1.1" data-path="installation.html"><a href="installation.html#installing-from-github"><i class="fa fa-check"></i><b>1.1</b> Installing from github</a></li>
<li class="chapter" data-level="1.2" data-path="installation.html"><a href="installation.html#installing-from-bioconductor"><i class="fa fa-check"></i><b>1.2</b> Installing from Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-preparation.html"><a href="data-preparation.html"><i class="fa fa-check"></i><b>2</b> Data preparation</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-preparation.html"><a href="data-preparation.html#constructing-a-singlecellexperiment-object"><i class="fa fa-check"></i><b>2.1</b> Constructing a SingleCellExperiment object</a></li>
<li class="chapter" data-level="2.2" data-path="data-preparation.html"><a href="data-preparation.html#starting-from-an-archr-object"><i class="fa fa-check"></i><b>2.2</b> Starting from an ArchR object</a></li>
<li class="chapter" data-level="2.3" data-path="data-preparation.html"><a href="data-preparation.html#starting-from-a-seuratsignac-object"><i class="fa fa-check"></i><b>2.3</b> Starting from a Seurat/Signac object</a></li>
<li class="chapter" data-level="2.4" data-path="data-preparation.html"><a href="data-preparation.html#starting-from-anndata"><i class="fa fa-check"></i><b>2.4</b> Starting from AnnData</a></li>
<li class="chapter" data-level="2.5" data-path="data-preparation.html"><a href="data-preparation.html#starting-from-10x-data-formats"><i class="fa fa-check"></i><b>2.5</b> Starting from 10x data formats</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="data-preparation.html"><a href="data-preparation.html#reading-from-.h5-file"><i class="fa fa-check"></i><b>2.5.1</b> Reading from .h5 file</a></li>
<li class="chapter" data-level="2.5.2" data-path="data-preparation.html"><a href="data-preparation.html#reading-directly-from-a-10x-directory"><i class="fa fa-check"></i><b>2.5.2</b> Reading directly from a 10x directory</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="data-preparation.html"><a href="data-preparation.html#reading-directly-from-csv"><i class="fa fa-check"></i><b>2.6</b> Reading directly from CSV</a></li>
<li class="chapter" data-level="2.7" data-path="data-preparation.html"><a href="data-preparation.html#important-points"><i class="fa fa-check"></i><b>2.7</b> Important points</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>3</b> Basic workflow with MultiAssayExperiment</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#data-preparation-1"><i class="fa fa-check"></i><b>3.1</b> Data preparation</a></li>
<li class="chapter" data-level="3.2" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#quick-start"><i class="fa fa-check"></i><b>3.2</b> Quick start</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#retrieve-bulk-tf-chip-seq-binding-sites"><i class="fa fa-check"></i><b>3.2.1</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="3.2.2" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#link-atac-seq-peaks-to-target-genes"><i class="fa fa-check"></i><b>3.2.2</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="3.2.3" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#add-tf-motif-binding-to-peaks"><i class="fa fa-check"></i><b>3.2.3</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="3.2.4" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#generate-regulons"><i class="fa fa-check"></i><b>3.2.4</b> Generate regulons</a></li>
<li class="chapter" data-level="3.2.5" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#network-pruning-highly-recommended"><i class="fa fa-check"></i><b>3.2.5</b> Network pruning (highly recommended)</a></li>
<li class="chapter" data-level="3.2.6" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#add-weights"><i class="fa fa-check"></i><b>3.2.6</b> Add Weights</a></li>
<li class="chapter" data-level="3.2.7" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#optional-annotate-with-tf-motifs"><i class="fa fa-check"></i><b>3.2.7</b> (Optional) Annotate with TF motifs</a></li>
<li class="chapter" data-level="3.2.8" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#calculate-tf-activity"><i class="fa fa-check"></i><b>3.2.8</b> Calculate TF activity</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#session-info"><i class="fa fa-check"></i><b>3.3</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>4</b> Advanced workflow with MultiAssayExperiment</a>
<ul>
<li class="chapter" data-level="4.1" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#data-preparation-2"><i class="fa fa-check"></i><b>4.1</b> Data preparation</a></li>
<li class="chapter" data-level="4.2" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#retrieve-bulk-tf-chip-seq-binding-sites-1"><i class="fa fa-check"></i><b>4.2</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="4.3" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#link-atacseq-peaks-to-target-genes"><i class="fa fa-check"></i><b>4.3</b> Link ATACseq peaks to target genes</a></li>
<li class="chapter" data-level="4.4" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#add-tf-motif-binding-to-peaks-1"><i class="fa fa-check"></i><b>4.4</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="4.5" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#generate-regulons-1"><i class="fa fa-check"></i><b>4.5</b> Generate regulons</a></li>
<li class="chapter" data-level="4.6" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#prune-network"><i class="fa fa-check"></i><b>4.6</b> Prune network</a></li>
<li class="chapter" data-level="4.7" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#add-weights-1"><i class="fa fa-check"></i><b>4.7</b> Add Weights</a></li>
<li class="chapter" data-level="4.8" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#calculate-tf-activity-1"><i class="fa fa-check"></i><b>4.8</b> Calculate TF activity</a></li>
<li class="chapter" data-level="4.9" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#differential-tf-activity-test"><i class="fa fa-check"></i><b>4.9</b> Differential TF activity test</a></li>
<li class="chapter" data-level="4.10" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#visualizing-tf-activities"><i class="fa fa-check"></i><b>4.10</b> Visualizing TF activities</a></li>
<li class="chapter" data-level="4.11" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#geneset-enrichment"><i class="fa fa-check"></i><b>4.11</b> Geneset enrichment</a></li>
<li class="chapter" data-level="4.12" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#differential-network-analysis"><i class="fa fa-check"></i><b>4.12</b> Differential Network analysis</a></li>
<li class="chapter" data-level="4.13" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#session-info-1"><i class="fa fa-check"></i><b>4.13</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html"><i class="fa fa-check"></i><b>5</b> ArchR workflow and different weight methods</a>
<ul>
<li class="chapter" data-level="5.1" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#data-preparation-3"><i class="fa fa-check"></i><b>5.1</b> Data preparation</a></li>
<li class="chapter" data-level="5.2" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#load-archr-project"><i class="fa fa-check"></i><b>5.2</b> Load ArchR project</a></li>
<li class="chapter" data-level="5.3" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#retrieve-matrices-from-archr-project"><i class="fa fa-check"></i><b>5.3</b> Retrieve matrices from ArchR project</a></li>
<li class="chapter" data-level="5.4" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#retrieve-bulk-tf-chip-seq-binding-sites-2"><i class="fa fa-check"></i><b>5.4</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="5.5" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#link-atac-seq-peaks-to-target-genes-1"><i class="fa fa-check"></i><b>5.5</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="5.6" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#add-tf-motif-binding-to-peaks-2"><i class="fa fa-check"></i><b>5.6</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="5.7" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#generate-regulons-2"><i class="fa fa-check"></i><b>5.7</b> Generate regulons</a></li>
<li class="chapter" data-level="5.8" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#optional-annotate-with-tf-motifs-1"><i class="fa fa-check"></i><b>5.8</b> (Optional) Annotate with TF motifs</a></li>
<li class="chapter" data-level="5.9" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#prune-network-1"><i class="fa fa-check"></i><b>5.9</b> Prune network</a></li>
<li class="chapter" data-level="5.10" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#add-weights-2"><i class="fa fa-check"></i><b>5.10</b> Add Weights</a></li>
<li class="chapter" data-level="5.11" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#calculate-tf-activity-2"><i class="fa fa-check"></i><b>5.11</b> Calculate TF activity</a></li>
<li class="chapter" data-level="5.12" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#perform-differential-activity"><i class="fa fa-check"></i><b>5.12</b> Perform differential activity</a></li>
<li class="chapter" data-level="5.13" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#visualize-the-results"><i class="fa fa-check"></i><b>5.13</b> Visualize the results</a></li>
<li class="chapter" data-level="5.14" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#differential-network-analysis-1"><i class="fa fa-check"></i><b>5.14</b> Differential network analysis</a></li>
<li class="chapter" data-level="5.15" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#session-info-2"><i class="fa fa-check"></i><b>5.15</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html"><i class="fa fa-check"></i><b>6</b> Single modality: scRNA-seq only</a>
<ul>
<li class="chapter" data-level="6.1" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#load-regulon"><i class="fa fa-check"></i><b>6.1</b> Load regulon</a></li>
<li class="chapter" data-level="6.2" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#load-scrna-seq-data"><i class="fa fa-check"></i><b>6.2</b> Load scRNA-seq data</a></li>
<li class="chapter" data-level="6.3" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#calculate-activity"><i class="fa fa-check"></i><b>6.3</b> Calculate activity</a></li>
<li class="chapter" data-level="6.4" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#perform-differential-activity-1"><i class="fa fa-check"></i><b>6.4</b> Perform differential activity</a></li>
<li class="chapter" data-level="6.5" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#visualize-activity"><i class="fa fa-check"></i><b>6.5</b> Visualize activity</a></li>
<li class="chapter" data-level="6.6" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#pathway-enrichment"><i class="fa fa-check"></i><b>6.6</b> Pathway enrichment</a></li>
<li class="chapter" data-level="6.7" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#session-info-3"><i class="fa fa-check"></i><b>6.7</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="contact.html"><a href="contact.html"><i class="fa fa-check"></i><b>7</b> Contact</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Epiregulon documentation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="basic-workflow-with-multiassayexperiment" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> Basic workflow with MultiAssayExperiment<a href="basic-workflow-with-multiassayexperiment.html#basic-workflow-with-multiassayexperiment" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In this chapter, we illustrate the epiregulon workflow starting from data in the form of <code>SingleCellExperiment</code> objects using the correlation weight estimation method.</p>
<p>This tutorial demonstrates the basic functions of <code>epiregulon</code>, using the reprogram-seq dataset which can be downloaded from the <em><a href="https://bioconductor.org/packages/3.20/scMultiome">scMultiome</a></em> package. In this example, prostate cancer cells (LNCaP) were infected in separate wells with viruses encoding 4 transcription factors (NKX2-1, GATA6, FOXA1 and FOXA2) and a positive control (mNeonGreen) before pooling. The identity of the infected transcription factors was tracked through cell hashing (available in the field <code>hash_assignment</code> of the <code>colData</code>) and serves as the ground truth.</p>
<div id="data-preparation-1" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Data preparation<a href="basic-workflow-with-multiassayexperiment.html#data-preparation-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Prior to using <code>epiregulon</code>, single cell preprocessing needs to performed by user’s favorite methods. The following components are required: <br>
1. Peak matrix from scATAC-seq containing the chromatin accessibility information <br>
2. Gene expression matrix from either paired or unpaired scRNA-seq. RNA-seq integration needs to be performed for unpaired dataset. <br>
3. Dimensionality reduction matrix from with either single modalities or joint scRNA-seq and scATAC-seq <br></p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="basic-workflow-with-multiassayexperiment.html#cb113-1" tabindex="-1"></a><span class="co"># load the MAE object</span></span>
<span id="cb113-2"><a href="basic-workflow-with-multiassayexperiment.html#cb113-2" tabindex="-1"></a><span class="fu">library</span>(scMultiome)</span>
<span id="cb113-3"><a href="basic-workflow-with-multiassayexperiment.html#cb113-3" tabindex="-1"></a><span class="fu">library</span>(epiregulon)</span>
<span id="cb113-4"><a href="basic-workflow-with-multiassayexperiment.html#cb113-4" tabindex="-1"></a></span>
<span id="cb113-5"><a href="basic-workflow-with-multiassayexperiment.html#cb113-5" tabindex="-1"></a>mae <span class="ot">&lt;-</span> scMultiome<span class="sc">::</span><span class="fu">reprogramSeq</span>()</span>
<span id="cb113-6"><a href="basic-workflow-with-multiassayexperiment.html#cb113-6" tabindex="-1"></a></span>
<span id="cb113-7"><a href="basic-workflow-with-multiassayexperiment.html#cb113-7" tabindex="-1"></a><span class="co"># extract peak matrix</span></span>
<span id="cb113-8"><a href="basic-workflow-with-multiassayexperiment.html#cb113-8" tabindex="-1"></a>PeakMatrix <span class="ot">&lt;-</span> mae[[<span class="st">&quot;PeakMatrix&quot;</span>]]</span>
<span id="cb113-9"><a href="basic-workflow-with-multiassayexperiment.html#cb113-9" tabindex="-1"></a></span>
<span id="cb113-10"><a href="basic-workflow-with-multiassayexperiment.html#cb113-10" tabindex="-1"></a><span class="co"># extract expression matrix</span></span>
<span id="cb113-11"><a href="basic-workflow-with-multiassayexperiment.html#cb113-11" tabindex="-1"></a>GeneExpressionMatrix <span class="ot">&lt;-</span> mae[[<span class="st">&quot;GeneExpressionMatrix&quot;</span>]]</span>
<span id="cb113-12"><a href="basic-workflow-with-multiassayexperiment.html#cb113-12" tabindex="-1"></a><span class="fu">rownames</span>(GeneExpressionMatrix) <span class="ot">&lt;-</span> <span class="fu">rowData</span>(GeneExpressionMatrix)<span class="sc">$</span>name</span>
<span id="cb113-13"><a href="basic-workflow-with-multiassayexperiment.html#cb113-13" tabindex="-1"></a></span>
<span id="cb113-14"><a href="basic-workflow-with-multiassayexperiment.html#cb113-14" tabindex="-1"></a><span class="co"># define the order of hash_assigment</span></span>
<span id="cb113-15"><a href="basic-workflow-with-multiassayexperiment.html#cb113-15" tabindex="-1"></a>GeneExpressionMatrix<span class="sc">$</span>hash_assignment <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(GeneExpressionMatrix<span class="sc">$</span>hash_assignment),</span>
<span id="cb113-16"><a href="basic-workflow-with-multiassayexperiment.html#cb113-16" tabindex="-1"></a>                                               <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;HTO10_GATA6_UTR&quot;</span>, <span class="st">&quot;HTO2_GATA6_v2&quot;</span>, <span class="st">&quot;HTO8_NKX2.1_UTR&quot;</span>, <span class="st">&quot;HTO3_NKX2.1_v2&quot;</span>, </span>
<span id="cb113-17"><a href="basic-workflow-with-multiassayexperiment.html#cb113-17" tabindex="-1"></a>                                                          <span class="st">&quot;HTO1_FOXA2_v2&quot;</span>, <span class="st">&quot;HTO4_mFOXA1_v2&quot;</span>, <span class="st">&quot;HTO6_hFOXA1_UTR&quot;</span>, <span class="st">&quot;HTO5_NeonG_v2&quot;</span>))</span>
<span id="cb113-18"><a href="basic-workflow-with-multiassayexperiment.html#cb113-18" tabindex="-1"></a><span class="co"># extract dimensional reduction matrix</span></span>
<span id="cb113-19"><a href="basic-workflow-with-multiassayexperiment.html#cb113-19" tabindex="-1"></a>reducedDimMatrix <span class="ot">&lt;-</span> <span class="fu">reducedDim</span>(mae[[<span class="st">&#39;TileMatrix500&#39;</span>]], <span class="st">&quot;LSI_ATAC&quot;</span>)</span>
<span id="cb113-20"><a href="basic-workflow-with-multiassayexperiment.html#cb113-20" tabindex="-1"></a></span>
<span id="cb113-21"><a href="basic-workflow-with-multiassayexperiment.html#cb113-21" tabindex="-1"></a><span class="co"># transfer UMAP_combined from TileMatrix to GeneExpressionMatrix</span></span>
<span id="cb113-22"><a href="basic-workflow-with-multiassayexperiment.html#cb113-22" tabindex="-1"></a><span class="fu">reducedDim</span>(GeneExpressionMatrix, <span class="st">&quot;UMAP_Combined&quot;</span>) <span class="ot">&lt;-</span> <span class="fu">reducedDim</span>(mae[[<span class="st">&#39;TileMatrix500&#39;</span>]], <span class="st">&quot;UMAP_Combined&quot;</span>)</span></code></pre></div>
<p>Visualize singleCellExperiment by UMAP</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="basic-workflow-with-multiassayexperiment.html#cb114-1" tabindex="-1"></a>scater<span class="sc">::</span><span class="fu">plotReducedDim</span>(GeneExpressionMatrix, </span>
<span id="cb114-2"><a href="basic-workflow-with-multiassayexperiment.html#cb114-2" tabindex="-1"></a>                       <span class="at">dimred =</span> <span class="st">&quot;UMAP_Combined&quot;</span>, </span>
<span id="cb114-3"><a href="basic-workflow-with-multiassayexperiment.html#cb114-3" tabindex="-1"></a>                       <span class="at">text_by =</span> <span class="st">&quot;Clusters&quot;</span>, </span>
<span id="cb114-4"><a href="basic-workflow-with-multiassayexperiment.html#cb114-4" tabindex="-1"></a>                       <span class="at">colour_by =</span> <span class="st">&quot;Clusters&quot;</span>)</span></code></pre></div>
<p><img src="multiome.mae_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
</div>
<div id="quick-start" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Quick start<a href="basic-workflow-with-multiassayexperiment.html#quick-start" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="retrieve-bulk-tf-chip-seq-binding-sites" class="section level3 hasAnchor" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Retrieve bulk TF ChIP-seq binding sites<a href="basic-workflow-with-multiassayexperiment.html#retrieve-bulk-tf-chip-seq-binding-sites" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First, we retrieve a <code>GRangesList</code> object containing the binding sites of all the transcription factors and co-regulators. These binding sites are derived from bulk ChIP-seq data in the <a href="https://chip-atlas.org/">ChIP-Atlas</a> and <a href="https://www.encodeproject.org/">ENCODE</a> databases. For the same transcription factor, multiple ChIP-seq files from different cell lines or tissues are merged. For further information on how these peaks are derived, please refer to <code>?epiregulon::getTFMotifInfo</code>. Currently, human genomes hg19 and hg38 and mouse mm10 are supported.</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="basic-workflow-with-multiassayexperiment.html#cb115-1" tabindex="-1"></a>grl <span class="ot">&lt;-</span> <span class="fu">getTFMotifInfo</span>(<span class="at">genome =</span> <span class="st">&quot;hg38&quot;</span>)</span></code></pre></div>
<pre><code>## see ?scMultiome and browseVignettes(&#39;scMultiome&#39;) for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="basic-workflow-with-multiassayexperiment.html#cb118-1" tabindex="-1"></a>grl</span></code></pre></div>
<pre><code>## GRangesList object of length 1558:
## $AEBP2
## GRanges object with 2700 ranges and 0 metadata columns:
##          seqnames            ranges strand
##             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;
##      [1]     chr1        9792-10446      *
##      [2]     chr1     942105-942400      *
##      [3]     chr1     984486-984781      *
##      [4]     chr1   3068932-3069282      *
##      [5]     chr1   3069411-3069950      *
##      ...      ...               ...    ...
##   [2696]     chrY   8465261-8465730      *
##   [2697]     chrY 11721744-11722260      *
##   [2698]     chrY 11747448-11747964      *
##   [2699]     chrY 19302661-19303134      *
##   [2700]     chrY 19985662-19985982      *
##   -------
##   seqinfo: 25 sequences from an unspecified genome; no seqlengths
## 
## ...
## &lt;1557 more elements&gt;</code></pre>
</div>
<div id="link-atac-seq-peaks-to-target-genes" class="section level3 hasAnchor" number="3.2.2">
<h3><span class="header-section-number">3.2.2</span> Link ATAC-seq peaks to target genes<a href="basic-workflow-with-multiassayexperiment.html#link-atac-seq-peaks-to-target-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Next, we try to link ATAC-seq peaks to their putative target genes. We assign a peak to a gene within a size window (default ±250kb) if the chromatin accessibility of the peak and expression of the target genes are highly correlated (default threshold 0.5). To compute correlations, we first create cell aggregates by performing k-means clustering on the reduced dimensionality matrix. Then we aggregate the counts of the gene expression and peak matrix and average across the number of cells. Correlations are computed on the averaged gene expression and chromatin accessibility.</p>
<p>If cluster labels are provided, peak-to-gene correlations are computed on all the cells and for each cluster. Peak-to-gene links are retained as long as any of the correlations pass the threshold; the longer list of peak-to-gene links capture both inter- and intra-cluster variations.</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="basic-workflow-with-multiassayexperiment.html#cb120-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1010</span>)</span>
<span id="cb120-2"><a href="basic-workflow-with-multiassayexperiment.html#cb120-2" tabindex="-1"></a>p2g <span class="ot">&lt;-</span> <span class="fu">calculateP2G</span>(<span class="at">peakMatrix =</span> PeakMatrix, </span>
<span id="cb120-3"><a href="basic-workflow-with-multiassayexperiment.html#cb120-3" tabindex="-1"></a>                    <span class="at">expMatrix =</span> GeneExpressionMatrix, </span>
<span id="cb120-4"><a href="basic-workflow-with-multiassayexperiment.html#cb120-4" tabindex="-1"></a>                    <span class="at">reducedDim =</span> reducedDimMatrix,</span>
<span id="cb120-5"><a href="basic-workflow-with-multiassayexperiment.html#cb120-5" tabindex="-1"></a>                    <span class="at">exp_assay =</span> <span class="st">&quot;normalizedCounts&quot;</span>,</span>
<span id="cb120-6"><a href="basic-workflow-with-multiassayexperiment.html#cb120-6" tabindex="-1"></a>                    <span class="at">peak_assay =</span> <span class="st">&quot;counts&quot;</span>)</span></code></pre></div>
<pre><code>## Using epiregulon to compute peak to gene links...</code></pre>
<pre><code>## performing k means clustering to form metacells</code></pre>
<pre><code>## Computing correlation</code></pre>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="basic-workflow-with-multiassayexperiment.html#cb124-1" tabindex="-1"></a>p2g</span></code></pre></div>
<pre><code>## DataFrame with 7892 rows and 8 columns
##        idxATAC         chr     start       end    idxRNA     target Correlation
##      &lt;integer&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;    &lt;array&gt;    &lt;matrix&gt;
## 1           37        chr1   1020509   1021009        22 AL645608.4    0.592745
## 2           39        chr1   1039811   1040311        33       AGRN    0.523777
## 3           42        chr1   1053517   1054017        33       AGRN    0.521471
## 4           86        chr1   1290083   1290583        58       DVL1    0.524540
## 5          104        chr1   1343943   1344443        66 AL391244.1    0.515194
## ...        ...         ...       ...       ...       ...        ...         ...
## 7888    126535        chrX 154368842 154369342     36389       FLNA    0.852869
## 7889    126536        chrX 154369373 154369873     36389       FLNA    0.828660
## 7890    126537        chrX 154370531 154371031     36389       FLNA    0.573066
## 7891    126541        chrX 154374945 154375445     36389       FLNA    0.572236
## 7892    126590        chrX 155228844 155229344     36426      CLIC2    0.527470
##       distance
##      &lt;integer&gt;
## 1       106338
## 2        19691
## 3        33397
## 4        58527
## 5        75800
## ...        ...
## 7888      1878
## 7889      1347
## 7890       189
## 7891      3723
## 7892    105268</code></pre>
</div>
<div id="add-tf-motif-binding-to-peaks" class="section level3 hasAnchor" number="3.2.3">
<h3><span class="header-section-number">3.2.3</span> Add TF motif binding to peaks<a href="basic-workflow-with-multiassayexperiment.html#add-tf-motif-binding-to-peaks" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The next step is to add the TF binding information by overlapping regions of the peak matrix with the bulk chip-seq database. The output is a data frame object with three columns:</p>
<ol style="list-style-type: decimal">
<li><code>idxATAC</code> - index of the peak in the peak matrix</li>
<li><code>idxTF</code> - index in the gene expression matrix corresponding to the transcription factor</li>
<li><code>tf</code> - name of the transcription factor</li>
</ol>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="basic-workflow-with-multiassayexperiment.html#cb126-1" tabindex="-1"></a>overlap <span class="ot">&lt;-</span> <span class="fu">addTFMotifInfo</span>(<span class="at">grl =</span> grl, <span class="at">p2g =</span> p2g, <span class="at">peakMatrix =</span> PeakMatrix)</span></code></pre></div>
<pre><code>## Computing overlap...</code></pre>
<pre><code>## Success!</code></pre>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="basic-workflow-with-multiassayexperiment.html#cb129-1" tabindex="-1"></a><span class="fu">head</span>(overlap)</span></code></pre></div>
<pre><code>##      idxATAC idxTF     tf
## 9312      37     4   AGO1
## 9313      37     7 ARID1B
## 9314      37     8  ARID2
## 9315      37    18   ATF3
## 9316      37    21   ATF7
## 9317      37    23  BACH1</code></pre>
</div>
<div id="generate-regulons" class="section level3 hasAnchor" number="3.2.4">
<h3><span class="header-section-number">3.2.4</span> Generate regulons<a href="basic-workflow-with-multiassayexperiment.html#generate-regulons" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A DataFrame, representing the inferred regulons, is then generated. The DataFrame consists of ten columns:</p>
<ol style="list-style-type: decimal">
<li><code>idxATAC</code> - index of the peak in the peak matrix</li>
<li><code>chr</code> - chromosome number</li>
<li><code>start</code> - start position of the peak</li>
<li><code>end</code> - end position of the peak</li>
<li><code>idxRNA</code> - index in the gene expression matrix corresponding to the target gene</li>
<li><code>target</code> - name of the target gene</li>
<li><code>distance</code> - distance between the transcription start site of the target gene and the middle of the peak</li>
<li><code>idxTF</code> - index in the gene expression matrix corresponding to the transcription factor</li>
<li><code>tf</code> - name of the transcription factor</li>
<li><code>corr</code> - correlation between target gene expression and the chromatin accessibility at the peak. if cluster labels are provided,
this field is a matrix with columns names corresponding to correlation across all cells and for each of the clusters.</li>
</ol>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="basic-workflow-with-multiassayexperiment.html#cb131-1" tabindex="-1"></a>regulon <span class="ot">&lt;-</span> <span class="fu">getRegulon</span>(<span class="at">p2g =</span> p2g, <span class="at">overlap =</span> overlap, <span class="at">aggregate =</span> <span class="cn">FALSE</span>)</span>
<span id="cb131-2"><a href="basic-workflow-with-multiassayexperiment.html#cb131-2" tabindex="-1"></a>regulon</span></code></pre></div>
<pre><code>## DataFrame with 743656 rows and 10 columns
##          idxATAC         chr     start       end    idxRNA      target
##        &lt;integer&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt;
## 1             37        chr1   1020509   1021009        22  AL645608.4
## 2             37        chr1   1020509   1021009        22  AL645608.4
## 3             37        chr1   1020509   1021009        22  AL645608.4
## 4             37        chr1   1020509   1021009        22  AL645608.4
## 5             37        chr1   1020509   1021009        22  AL645608.4
## ...          ...         ...       ...       ...       ...         ...
## 743652    126541        chrX 154374945 154375445     36389        FLNA
## 743653    126541        chrX 154374945 154375445     36389        FLNA
## 743654    126590        chrX 155228844 155229344     36426       CLIC2
## 743655    126590        chrX 155228844 155229344     36426       CLIC2
## 743656    126590        chrX 155228844 155229344     36426       CLIC2
##         distance     idxTF          tf     corr
##        &lt;integer&gt; &lt;integer&gt; &lt;character&gt; &lt;matrix&gt;
## 1         106338         4        AGO1 0.592745
## 2         106338         7      ARID1B 0.592745
## 3         106338         8       ARID2 0.592745
## 4         106338        18        ATF3 0.592745
## 5         106338        21        ATF7 0.592745
## ...          ...       ...         ...      ...
## 743652      3723      1492      ZNF692 0.572236
## 743653      3723      1557        ZXDB 0.572236
## 743654    105268       114       FOXA1 0.527470
## 743655    105268       128       GATA2 0.527470
## 743656    105268       952       SUMO2 0.527470</code></pre>
</div>
<div id="network-pruning-highly-recommended" class="section level3 hasAnchor" number="3.2.5">
<h3><span class="header-section-number">3.2.5</span> Network pruning (highly recommended)<a href="basic-workflow-with-multiassayexperiment.html#network-pruning-highly-recommended" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Epiregulon prunes the network by performing tests of independence on the observed number of cells jointly expressing transcription factor (TF), regulatory element (RE) and target gene (TG) vs the expected number of cells if TF/RE and TG are independently expressed. The users can choose between two tests, the binomial test and the chi-square test. In the binomial test, the expected probability is P(TF, RE) * P(TG), and the number of trials is the total number of cells, and the observed successes is the number of cells jointly expressing all three elements. In the chi-square test, the expected probability for having all 3 elements active is also P(TF, RE) * P(TG) and the probability otherwise is 1- P(TF, RE) * P(TG). The observed cell count for the category of “active TF” is the number of cells jointly expressing all three elements, and the cell count for the inactive category is n - n_triple.</p>
<p>We calculate cluster-specific p-values if users supply cluster labels. This is useful if we are interested in cluster-specific networks. The pruned regulons can then be used to visualize differential networks for transcription factors of interest.</p>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="basic-workflow-with-multiassayexperiment.html#cb133-1" tabindex="-1"></a>pruned.regulon <span class="ot">&lt;-</span> <span class="fu">pruneRegulon</span>(<span class="at">expMatrix =</span> GeneExpressionMatrix,</span>
<span id="cb133-2"><a href="basic-workflow-with-multiassayexperiment.html#cb133-2" tabindex="-1"></a>                               <span class="at">exp_assay =</span> <span class="st">&quot;normalizedCounts&quot;</span>,</span>
<span id="cb133-3"><a href="basic-workflow-with-multiassayexperiment.html#cb133-3" tabindex="-1"></a>                               <span class="at">peakMatrix =</span> PeakMatrix,</span>
<span id="cb133-4"><a href="basic-workflow-with-multiassayexperiment.html#cb133-4" tabindex="-1"></a>                               <span class="at">peak_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb133-5"><a href="basic-workflow-with-multiassayexperiment.html#cb133-5" tabindex="-1"></a>                               <span class="at">test =</span> <span class="st">&quot;chi.sq&quot;</span>,</span>
<span id="cb133-6"><a href="basic-workflow-with-multiassayexperiment.html#cb133-6" tabindex="-1"></a>                               regulon[regulon<span class="sc">$</span>tf <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;NKX2-1&quot;</span>,<span class="st">&quot;GATA6&quot;</span>,<span class="st">&quot;FOXA1&quot;</span>,<span class="st">&quot;FOXA2&quot;</span>, <span class="st">&quot;AR&quot;</span>),],</span>
<span id="cb133-7"><a href="basic-workflow-with-multiassayexperiment.html#cb133-7" tabindex="-1"></a>                               <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>Clusters,</span>
<span id="cb133-8"><a href="basic-workflow-with-multiassayexperiment.html#cb133-8" tabindex="-1"></a>                               <span class="at">prune_value =</span> <span class="st">&quot;pval&quot;</span>,</span>
<span id="cb133-9"><a href="basic-workflow-with-multiassayexperiment.html#cb133-9" tabindex="-1"></a>                               <span class="at">regulon_cutoff =</span> <span class="fl">0.05</span></span>
<span id="cb133-10"><a href="basic-workflow-with-multiassayexperiment.html#cb133-10" tabindex="-1"></a>                               )</span>
<span id="cb133-11"><a href="basic-workflow-with-multiassayexperiment.html#cb133-11" tabindex="-1"></a></span>
<span id="cb133-12"><a href="basic-workflow-with-multiassayexperiment.html#cb133-12" tabindex="-1"></a>pruned.regulon</span></code></pre></div>
</div>
<div id="add-weights" class="section level3 hasAnchor" number="3.2.6">
<h3><span class="header-section-number">3.2.6</span> Add Weights<a href="basic-workflow-with-multiassayexperiment.html#add-weights" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>While the <code>pruneRegulon</code> function provides statistics on the joint occurrence of TF-RE-TG, we would like to further estimate the strength of regulation. Biologically, this can be interpreted as the magnitude of gene expression changes induced by transcription factor activity. Epiregulon estimates the regulatory potential using one of the three measures: 1) correlation between TF and target gene expression, 2) mutual information between the TF and target gene expression and 3) Wilcoxon test statistics of target gene expression in cells jointly expressing all 3 elements vs cells that do not.</p>
<p>Two of the measures (correlation and Wilcoxon statistics) give both the magnitude and directionality of changes whereas mutual information is always positive. The correlation and mutual information statistics are computed on pseudobulks aggregated by user-supplied cluster labels, whereas the Wilcoxon method groups cells into two categories: 1) the active category of cells jointly expressing TF, RE and TG and 2) the inactive category consisting of the remaining cells.</p>
<p>We calculate cluster-specific weights if users supply cluster labels.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="basic-workflow-with-multiassayexperiment.html#cb134-1" tabindex="-1"></a>regulon.w <span class="ot">&lt;-</span> <span class="fu">addWeights</span>(<span class="at">regulon =</span> pruned.regulon,</span>
<span id="cb134-2"><a href="basic-workflow-with-multiassayexperiment.html#cb134-2" tabindex="-1"></a>                        <span class="at">expMatrix  =</span> GeneExpressionMatrix,</span>
<span id="cb134-3"><a href="basic-workflow-with-multiassayexperiment.html#cb134-3" tabindex="-1"></a>                        <span class="at">exp_assay  =</span> <span class="st">&quot;normalizedCounts&quot;</span>,</span>
<span id="cb134-4"><a href="basic-workflow-with-multiassayexperiment.html#cb134-4" tabindex="-1"></a>                        <span class="at">peakMatrix =</span> PeakMatrix,</span>
<span id="cb134-5"><a href="basic-workflow-with-multiassayexperiment.html#cb134-5" tabindex="-1"></a>                        <span class="at">peak_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb134-6"><a href="basic-workflow-with-multiassayexperiment.html#cb134-6" tabindex="-1"></a>                        <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>Clusters,</span>
<span id="cb134-7"><a href="basic-workflow-with-multiassayexperiment.html#cb134-7" tabindex="-1"></a>                        <span class="at">method =</span> <span class="st">&quot;wilcox&quot;</span>)</span>
<span id="cb134-8"><a href="basic-workflow-with-multiassayexperiment.html#cb134-8" tabindex="-1"></a></span>
<span id="cb134-9"><a href="basic-workflow-with-multiassayexperiment.html#cb134-9" tabindex="-1"></a>regulon.w</span></code></pre></div>
</div>
<div id="optional-annotate-with-tf-motifs" class="section level3 hasAnchor" number="3.2.7">
<h3><span class="header-section-number">3.2.7</span> (Optional) Annotate with TF motifs<a href="basic-workflow-with-multiassayexperiment.html#optional-annotate-with-tf-motifs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>So far the gene regulatory network was constructed from TF ChIP-seq exclusively. Some users would prefer to further annotate regulatory elements with the presence of motifs. We provide an option to annotate peaks with motifs from the Cisbp database. If no motifs are present for this particular factor (as in the case of co-factors or chromatin modifiers), we return NA. If motifs are available for a factor and the RE contains a motif, we return 1. If motifs are available and the RE does not contain a motif, we return 0.</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="basic-workflow-with-multiassayexperiment.html#cb135-1" tabindex="-1"></a>regulon.w.motif <span class="ot">&lt;-</span> <span class="fu">addMotifScore</span>(<span class="at">regulon =</span> regulon.w,</span>
<span id="cb135-2"><a href="basic-workflow-with-multiassayexperiment.html#cb135-2" tabindex="-1"></a>                                 <span class="at">peaks =</span> <span class="fu">rowRanges</span>(PeakMatrix),</span>
<span id="cb135-3"><a href="basic-workflow-with-multiassayexperiment.html#cb135-3" tabindex="-1"></a>                                 <span class="at">species =</span> <span class="st">&quot;human&quot;</span>,</span>
<span id="cb135-4"><a href="basic-workflow-with-multiassayexperiment.html#cb135-4" tabindex="-1"></a>                                 <span class="at">genome =</span> <span class="st">&quot;hg38&quot;</span>)</span>
<span id="cb135-5"><a href="basic-workflow-with-multiassayexperiment.html#cb135-5" tabindex="-1"></a></span>
<span id="cb135-6"><a href="basic-workflow-with-multiassayexperiment.html#cb135-6" tabindex="-1"></a><span class="co"># if desired, set weight to 0 if no motif is found</span></span>
<span id="cb135-7"><a href="basic-workflow-with-multiassayexperiment.html#cb135-7" tabindex="-1"></a>regulon.w.motif<span class="sc">$</span>weight[regulon.w.motif<span class="sc">$</span>motif <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb135-8"><a href="basic-workflow-with-multiassayexperiment.html#cb135-8" tabindex="-1"></a></span>
<span id="cb135-9"><a href="basic-workflow-with-multiassayexperiment.html#cb135-9" tabindex="-1"></a>regulon.w.motif</span></code></pre></div>
<pre><code>## DataFrame with 3290 rows and 15 columns
##        idxATAC         chr     start       end    idxRNA      target  distance
##      &lt;integer&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt; &lt;integer&gt;
## 1           86        chr1   1290083   1290583        58        DVL1     58527
## 2          288        chr1   2526365   2526865       108  AL139246.1     33107
## 3          640        chr1   7988538   7989038       208     TNFRSF9     47672
## 4          654        chr1   8040536   8041036       212      ERRFI1     24903
## 5          787        chr1   9290024   9290524       229       SPSB1      2368
## ...        ...         ...       ...       ...       ...         ...       ...
## 3286    120296        chr9  36118633  36119133     34311        CLTA     71721
## 3287    121835        chr9  98192900  98193400     34702      CORO2A     20016
## 3288    122254        chr9 108991440 108991940     34784     CTNNAL1     27885
## 3289    122461        chr9 113174265 113174765     34825     SLC31A1     46777
## 3290    123380        chr9 128635245 128635745     35034      SPTAN1     82687
##          idxTF          tf     corr                                   pval
##      &lt;integer&gt; &lt;character&gt; &lt;matrix&gt;                               &lt;matrix&gt;
## 1          490          AR 0.524540 1.91265e-03:0.50585953:0.000249278:...
## 2          490          AR 0.602193 5.02384e-02:0.00249757:1.000000000:...
## 3          490          AR 0.875387 7.53206e-09:0.15249790:1.000000000:...
## 4          490          AR 0.550906 9.18676e-02:0.76535176:1.000000000:...
## 5          490          AR 0.891738 9.64012e-13:0.51358560:1.000000000:...
## ...        ...         ...      ...                                    ...
## 3286       807      NKX2-1 0.522644                    0.035831331:1:1:...
## 3287       807      NKX2-1 0.541195                    0.000419607:1:1:...
## 3288       807      NKX2-1 0.872888                    0.013099814:1:1:...
## 3289       807      NKX2-1 0.683659                    0.045120208:1:1:...
## 3290       807      NKX2-1 0.745530                    0.045120208:1:1:...
##                               stats                qval            weight
##                            &lt;matrix&gt;            &lt;matrix&gt;          &lt;matrix&gt;
## 1     9.63153:0.4426222:13.4176:... 1.00000e+00:1:1:...         0:0:0:...
## 2     3.83348:9.1423696: 0.0000:... 1.00000e+00:1:1:...         0:0:0:...
## 3    33.39232:2.0470852: 0.0000:... 1.24761e-04:1:1:...         0:0:0:...
## 4     2.84134:0.0890786: 0.0000:... 1.00000e+00:1:1:...         0:0:0:...
## 5    50.91607:0.4267549: 0.0000:... 1.60836e-08:1:1:...         0:0:0:...
## ...                             ...                 ...               ...
## 3286                4.40511:0:0:...           1:1:1:... 0.0343753:0:0:...
## 3287               12.44280:0:0:...           1:1:1:... 0.0000000:0:0:...
## 3288                6.15558:0:0:...           1:1:1:... 0.0000000:0:0:...
## 3289                4.01414:0:0:...           1:1:1:... 0.0000000:0:0:...
## 3290                4.01414:0:0:...           1:1:1:... 0.0000000:0:0:...
##          motif
##      &lt;numeric&gt;
## 1            0
## 2            0
## 3            0
## 4            0
## 5            0
## ...        ...
## 3286         1
## 3287         0
## 3288         0
## 3289         0
## 3290         0</code></pre>
</div>
<div id="calculate-tf-activity" class="section level3 hasAnchor" number="3.2.8">
<h3><span class="header-section-number">3.2.8</span> Calculate TF activity<a href="basic-workflow-with-multiassayexperiment.html#calculate-tf-activity" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Finally, the activities for a specific TF in each cell are computed by averaging expressions of target genes linked to the TF weighted by the test statistics of choice, chosen from either correlation, mutual information or the Wilcoxon test statistics.
<span class="math display">\[y=\frac{1}{n}\sum_{i=1}^{n} x_i * weights_i\]</span>
where <span class="math inline">\(y\)</span> is the activity of a TF for a cell,
<span class="math inline">\(n\)</span> is the total number of targets for a TF,
<span class="math inline">\(x_i\)</span> is the log count expression of target <span class="math inline">\(i\)</span> where <span class="math inline">\(i\)</span> in {1,2,…,n} and
<span class="math inline">\(weights_i\)</span> is the weight of TF - target <span class="math inline">\(i\)</span></p>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="basic-workflow-with-multiassayexperiment.html#cb137-1" tabindex="-1"></a>score.combine <span class="ot">&lt;-</span> <span class="fu">calculateActivity</span>(<span class="at">expMatrix =</span> GeneExpressionMatrix,</span>
<span id="cb137-2"><a href="basic-workflow-with-multiassayexperiment.html#cb137-2" tabindex="-1"></a>                                   <span class="at">regulon =</span> regulon.w, </span>
<span id="cb137-3"><a href="basic-workflow-with-multiassayexperiment.html#cb137-3" tabindex="-1"></a>                                   <span class="at">mode =</span> <span class="st">&quot;weight&quot;</span>, </span>
<span id="cb137-4"><a href="basic-workflow-with-multiassayexperiment.html#cb137-4" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">&quot;weightedMean&quot;</span>, </span>
<span id="cb137-5"><a href="basic-workflow-with-multiassayexperiment.html#cb137-5" tabindex="-1"></a>                                   <span class="at">exp_assay =</span> <span class="st">&quot;normalizedCounts&quot;</span>,</span>
<span id="cb137-6"><a href="basic-workflow-with-multiassayexperiment.html#cb137-6" tabindex="-1"></a>                                   <span class="at">normalize =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## calculating TF activity from regulon using weightedmean</code></pre>
<pre><code>## Warning in calculateActivity(expMatrix = GeneExpressionMatrix, regulon =
## regulon.w, : The weight column contains multiple subcolumns but no cluster
## information was provided. Using first column to compute activity...</code></pre>
<pre><code>## aggregating regulons...</code></pre>
<pre><code>## creating weight matrix...</code></pre>
<pre><code>## calculating activity scores...</code></pre>
<pre><code>## normalize by the number of targets...</code></pre>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="basic-workflow-with-multiassayexperiment.html#cb144-1" tabindex="-1"></a>score.combine[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>## 5 x 5 sparse Matrix of class &quot;dgCMatrix&quot;
##        reprogram#TTAGGAACAAGGTACG-1 reprogram#GAGCGGTCAACCTGGT-1
## AR                      0.043408741                   0.06010198
## FOXA1                   0.035260181                   0.04408534
## FOXA2                   0.008868738                   0.04479888
## GATA6                   0.033885727                   0.13969050
## NKX2-1                  0.065298024                   0.02791934
##        reprogram#TTATAGCCACCCTCAC-1 reprogram#TGGTGATTCCTGTTCA-1
## AR                       0.02734512                   0.03422265
## FOXA1                    0.01708987                   0.01827346
## FOXA2                    0.01153884                   0.01998275
## GATA6                    0.01667011                   0.01904236
## NKX2-1                   0.01410221                   0.01023911
##        reprogram#TCGGTTCTCACTAGGT-1
## AR                       0.04019227
## FOXA1                    0.03426071
## FOXA2                    0.01368200
## GATA6                    0.02905760
## NKX2-1                   0.05365239</code></pre>
</div>
</div>
<div id="session-info" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Session Info<a href="basic-workflow-with-multiassayexperiment.html#session-info" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="basic-workflow-with-multiassayexperiment.html#cb146-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.4.0 (2024-04-24)
## Platform: x86_64-pc-linux-gnu
## Running under: Red Hat Enterprise Linux 8.6 (Ootpa)
## 
## Matrix products: default
## BLAS/LAPACK: /apps/rocs/2020.08/cascadelake/software/OpenBLAS/0.3.9-GCC-9.3.0/lib/libopenblas_skylakexp-r0.3.9.so;  LAPACK version 3.9.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## time zone: America/Los_Angeles
## tzcode source: system (glibc)
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] BSgenome.Hsapiens.UCSC.hg38_1.4.5 BSgenome_1.73.1                  
##  [3] rtracklayer_1.65.0                BiocIO_1.15.2                    
##  [5] Biostrings_2.73.2                 XVector_0.45.0                   
##  [7] epiregulon_1.3.4                  scMultiome_1.5.7                 
##  [9] SingleCellExperiment_1.27.2       MultiAssayExperiment_1.31.5      
## [11] SummarizedExperiment_1.35.4       Biobase_2.65.1                   
## [13] GenomicRanges_1.57.2              GenomeInfoDb_1.41.2              
## [15] IRanges_2.39.2                    S4Vectors_0.43.2                 
## [17] MatrixGenerics_1.17.0             matrixStats_1.4.1                
## [19] ExperimentHub_2.13.1              AnnotationHub_3.13.3             
## [21] BiocFileCache_2.13.2              dbplyr_2.5.0                     
## [23] BiocGenerics_0.51.3               BiocStyle_2.33.1                 
## 
## loaded via a namespace (and not attached):
##   [1] later_1.3.2                 bitops_1.0-9               
##   [3] filelock_1.0.3              R.oo_1.26.0                
##   [5] tibble_3.2.1                XML_3.99-0.17              
##   [7] DirichletMultinomial_1.47.0 lifecycle_1.0.4            
##   [9] pwalign_1.1.0               edgeR_4.3.19               
##  [11] processx_3.8.4              lattice_0.22-6             
##  [13] backports_1.5.0             magrittr_2.0.3             
##  [15] limma_3.61.12               sass_0.4.9                 
##  [17] rmarkdown_2.28              jquerylib_0.1.4            
##  [19] yaml_2.3.10                 metapod_1.13.0             
##  [21] cowplot_1.1.3               chromote_0.3.1             
##  [23] DBI_1.2.3                   CNEr_1.41.0                
##  [25] abind_1.4-8                 zlibbioc_1.51.1            
##  [27] R.utils_2.12.3              purrr_1.0.2                
##  [29] RCurl_1.98-1.16             pracma_2.4.4               
##  [31] rappdirs_0.3.3              GenomeInfoDbData_1.2.13    
##  [33] ggrepel_0.9.6               irlba_2.3.5.1              
##  [35] seqLogo_1.71.0              annotate_1.83.0            
##  [37] dqrng_0.4.1                 codetools_0.2-20           
##  [39] DelayedArray_0.31.14        scuttle_1.15.4             
##  [41] tidyselect_1.2.1            UCSC.utils_1.1.0           
##  [43] farver_2.1.2                ScaledMatrix_1.13.0        
##  [45] viridis_0.6.5               GenomicAlignments_1.41.0   
##  [47] jsonlite_1.8.9              BiocNeighbors_1.99.2       
##  [49] motifmatchr_1.27.0          scater_1.33.4              
##  [51] tools_4.4.0                 TFMPvalue_0.0.9            
##  [53] Rcpp_1.0.13                 glue_1.8.0                 
##  [55] gridExtra_2.3               SparseArray_1.5.44         
##  [57] xfun_0.48                   websocket_1.4.2            
##  [59] dplyr_1.1.4                 HDF5Array_1.33.8           
##  [61] withr_3.0.1                 BiocManager_1.30.25        
##  [63] fastmap_1.2.0               rhdf5filters_1.17.0        
##  [65] bluster_1.15.1              fansi_1.0.6                
##  [67] caTools_1.18.3              digest_0.6.37              
##  [69] rsvd_1.0.5                  R6_2.5.1                   
##  [71] mime_0.12                   colorspace_2.1-1           
##  [73] GO.db_3.20.0                poweRlaw_0.80.0            
##  [75] gtools_3.9.5                RSQLite_2.3.7              
##  [77] R.methodsS3_1.8.2           utf8_1.2.4                 
##  [79] generics_0.1.3              httr_1.4.7                 
##  [81] S4Arrays_1.5.10             TFBSTools_1.43.0           
##  [83] pkgconfig_2.0.3             gtable_0.3.5               
##  [85] blob_1.2.4                  htmltools_0.5.8.1          
##  [87] bookdown_0.40               scales_1.3.0               
##  [89] png_0.1-8                   scran_1.33.2               
##  [91] knitr_1.48                  rstudioapi_0.16.0          
##  [93] tzdb_0.4.0                  reshape2_1.4.4             
##  [95] rjson_0.2.23                checkmate_2.3.2            
##  [97] curl_5.2.3                  cachem_1.1.0               
##  [99] rhdf5_2.49.0                stringr_1.5.1              
## [101] BiocVersion_3.20.0          parallel_4.4.0             
## [103] vipor_0.4.7                 AnnotationDbi_1.67.0       
## [105] restfulr_0.0.15             pillar_1.9.0               
## [107] grid_4.4.0                  vctrs_0.6.5                
## [109] promises_1.3.0              BiocSingular_1.21.4        
## [111] beachmat_2.21.6             xtable_1.8-4               
## [113] cluster_2.1.6               beeswarm_0.4.0             
## [115] evaluate_1.0.1              readr_2.1.5                
## [117] cli_3.6.3                   locfit_1.5-9.10            
## [119] compiler_4.4.0              Rsamtools_2.21.2           
## [121] rlang_1.1.4                 crayon_1.5.3               
## [123] labeling_0.4.3              ps_1.8.0                   
## [125] plyr_1.8.9                  ggbeeswarm_0.7.2           
## [127] stringi_1.8.4               viridisLite_0.4.2          
## [129] BiocParallel_1.39.0         munsell_0.5.1              
## [131] Matrix_1.7-0                hms_1.1.3                  
## [133] bit64_4.5.2                 ggplot2_3.5.1              
## [135] Rhdf5lib_1.27.0             KEGGREST_1.45.1            
## [137] statmod_1.5.0               highr_0.11                 
## [139] beachmat.hdf5_1.3.3         igraph_2.0.3               
## [141] memoise_2.0.1               bslib_0.8.0                
## [143] bit_4.5.0</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-preparation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="advanced-workflow-with-multiassayexperiment.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/multiome.mae.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Epiregulon documentation.pdf", "Epiregulon documentation.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
