<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Finding patterns of transcription factor activity | Epiregulon documentation</title>
  <meta name="description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="generator" content="bookdown 0.45 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Finding patterns of transcription factor activity | Epiregulon documentation" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Finding patterns of transcription factor activity | Epiregulon documentation" />
  
  <meta name="twitter:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  

<meta name="author" content="Xiaosai Yao, Tomasz WÅ‚odarczyk" />


<meta name="date" content="2025-12-02" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="calculation-of-the-tf-activity.html"/>
<link rel="next" href="data-preparation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Epiregulon documentation</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>1</b> Installation</a>
<ul>
<li class="chapter" data-level="1.1" data-path="installation.html"><a href="installation.html#installing-from-github"><i class="fa fa-check"></i><b>1.1</b> Installing from github</a></li>
<li class="chapter" data-level="1.2" data-path="installation.html"><a href="installation.html#installing-from-bioconductor"><i class="fa fa-check"></i><b>1.2</b> Installing from Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="construction.html"><a href="construction.html"><i class="fa fa-check"></i><b>2</b> Construction of the gene regulatory network</a>
<ul>
<li class="chapter" data-level="2.1" data-path="construction.html"><a href="construction.html#downloading-and-formatting-data-set"><i class="fa fa-check"></i><b>2.1</b> Downloading and formatting data set</a></li>
<li class="chapter" data-level="2.2" data-path="construction.html"><a href="construction.html#retrieve-tf-binding-sites"><i class="fa fa-check"></i><b>2.2</b> Retrieve TF binding sites</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="construction.html"><a href="construction.html#chip-seq-data"><i class="fa fa-check"></i><b>2.2.1</b> ChIP-seq data</a></li>
<li class="chapter" data-level="2.2.2" data-path="construction.html"><a href="construction.html#motif-information"><i class="fa fa-check"></i><b>2.2.2</b> Motif information</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="construction.html"><a href="construction.html#link-atacseq-peaks-to-target-genes"><i class="fa fa-check"></i><b>2.3</b> Link ATACseq peaks to target genes</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="construction.html"><a href="construction.html#optimization-of-the-number-of-metacells-optional"><i class="fa fa-check"></i><b>2.3.1</b> Optimization of the number of metacells (optional)</a></li>
<li class="chapter" data-level="2.3.2" data-path="construction.html"><a href="construction.html#distance_window"><i class="fa fa-check"></i><b>2.3.2</b> Finding peak-target gene links within a defined distance</a></li>
<li class="chapter" data-level="2.3.3" data-path="construction.html"><a href="construction.html#linking-peaks-to-the-nearest-genes"><i class="fa fa-check"></i><b>2.3.3</b> Linking peaks to the nearest genes</a></li>
<li class="chapter" data-level="2.3.4" data-path="construction.html"><a href="construction.html#filtering"><i class="fa fa-check"></i><b>2.3.4</b> Filtering</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="construction.html"><a href="construction.html#add-tf-binding-sites-to-peaks"><i class="fa fa-check"></i><b>2.4</b> Add TF binding sites to peaks</a></li>
<li class="chapter" data-level="2.5" data-path="construction.html"><a href="construction.html#generate-regulons"><i class="fa fa-check"></i><b>2.5</b> Generate regulons</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="refinement.html"><a href="refinement.html"><i class="fa fa-check"></i><b>3</b> Gene regulatory network refinement</a>
<ul>
<li class="chapter" data-level="3.1" data-path="refinement.html"><a href="refinement.html#pruning"><i class="fa fa-check"></i><b>3.1</b> Network pruning</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="refinement.html"><a href="refinement.html#cell_aggregation"><i class="fa fa-check"></i><b>3.1.1</b> Cell aggregation</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="refinement.html"><a href="refinement.html#annotation-with-tf-motifs"><i class="fa fa-check"></i><b>3.2</b> Annotation with TF motifs</a></li>
<li class="chapter" data-level="3.3" data-path="refinement.html"><a href="refinement.html#annotate-with-log-fold-changes"><i class="fa fa-check"></i><b>3.3</b> Annotate with log fold changes</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="calculation-of-the-tf-activity.html"><a href="calculation-of-the-tf-activity.html"><i class="fa fa-check"></i><b>4</b> Calculation of the TF activity</a>
<ul>
<li class="chapter" data-level="4.1" data-path="calculation-of-the-tf-activity.html"><a href="calculation-of-the-tf-activity.html#add-weights"><i class="fa fa-check"></i><b>4.1</b> Add Weights</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="calculation-of-the-tf-activity.html"><a href="calculation-of-the-tf-activity.html#cell-aggregation"><i class="fa fa-check"></i><b>4.1.1</b> Cell aggregation</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="calculation-of-the-tf-activity.html"><a href="calculation-of-the-tf-activity.html#calculate-tf-activity"><i class="fa fa-check"></i><b>4.2</b> Calculate TF activity</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="finding-patterns-of-transcription-factor-activity.html"><a href="finding-patterns-of-transcription-factor-activity.html"><i class="fa fa-check"></i><b>5</b> Finding patterns of transcription factor activity</a>
<ul>
<li class="chapter" data-level="5.1" data-path="finding-patterns-of-transcription-factor-activity.html"><a href="finding-patterns-of-transcription-factor-activity.html#differential-tf-activity-test"><i class="fa fa-check"></i><b>5.1</b> Differential TF activity test</a></li>
<li class="chapter" data-level="5.2" data-path="finding-patterns-of-transcription-factor-activity.html"><a href="finding-patterns-of-transcription-factor-activity.html#visualizing-tf-activities"><i class="fa fa-check"></i><b>5.2</b> Visualizing TF activities</a></li>
<li class="chapter" data-level="5.3" data-path="finding-patterns-of-transcription-factor-activity.html"><a href="finding-patterns-of-transcription-factor-activity.html#geneset-enrichment"><i class="fa fa-check"></i><b>5.3</b> Geneset enrichment</a></li>
<li class="chapter" data-level="5.4" data-path="finding-patterns-of-transcription-factor-activity.html"><a href="finding-patterns-of-transcription-factor-activity.html#differential-network-analysis"><i class="fa fa-check"></i><b>5.4</b> Differential Network analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-preparation.html"><a href="data-preparation.html"><i class="fa fa-check"></i><b>6</b> Data preparation</a>
<ul>
<li class="chapter" data-level="6.1" data-path="data-preparation.html"><a href="data-preparation.html#constructing-a-singlecellexperiment-object"><i class="fa fa-check"></i><b>6.1</b> Constructing a SingleCellExperiment object</a></li>
<li class="chapter" data-level="6.2" data-path="data-preparation.html"><a href="data-preparation.html#starting-from-an-archr-object"><i class="fa fa-check"></i><b>6.2</b> Starting from an ArchR object</a></li>
<li class="chapter" data-level="6.3" data-path="data-preparation.html"><a href="data-preparation.html#starting-from-a-seuratsignac-object"><i class="fa fa-check"></i><b>6.3</b> Starting from a Seurat/Signac object</a></li>
<li class="chapter" data-level="6.4" data-path="data-preparation.html"><a href="data-preparation.html#starting-from-anndata"><i class="fa fa-check"></i><b>6.4</b> Starting from AnnData</a></li>
<li class="chapter" data-level="6.5" data-path="data-preparation.html"><a href="data-preparation.html#starting-from-10x-data-formats"><i class="fa fa-check"></i><b>6.5</b> Starting from 10x data formats</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="data-preparation.html"><a href="data-preparation.html#reading-from-.h5-file"><i class="fa fa-check"></i><b>6.5.1</b> Reading from .h5 file</a></li>
<li class="chapter" data-level="6.5.2" data-path="data-preparation.html"><a href="data-preparation.html#reading-directly-from-a-10x-directory"><i class="fa fa-check"></i><b>6.5.2</b> Reading directly from a 10x directory</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="data-preparation.html"><a href="data-preparation.html#reading-directly-from-csv"><i class="fa fa-check"></i><b>6.6</b> Reading directly from CSV</a></li>
<li class="chapter" data-level="6.7" data-path="data-preparation.html"><a href="data-preparation.html#important-points"><i class="fa fa-check"></i><b>6.7</b> Important points</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>7</b> Basic workflow with MultiAssayExperiment</a>
<ul>
<li class="chapter" data-level="7.1" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#data-preparation-1"><i class="fa fa-check"></i><b>7.1</b> Data preparation</a></li>
<li class="chapter" data-level="7.2" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#quick-start"><i class="fa fa-check"></i><b>7.2</b> Quick start</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#retrieve-bulk-tf-chip-seq-binding-sites"><i class="fa fa-check"></i><b>7.2.1</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="7.2.2" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#link-atac-seq-peaks-to-target-genes"><i class="fa fa-check"></i><b>7.2.2</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="7.2.3" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#add-tf-motif-binding-to-peaks"><i class="fa fa-check"></i><b>7.2.3</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="7.2.4" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#generate-regulons-1"><i class="fa fa-check"></i><b>7.2.4</b> Generate regulons</a></li>
<li class="chapter" data-level="7.2.5" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#network-pruning-highly-recommended"><i class="fa fa-check"></i><b>7.2.5</b> Network pruning (highly recommended)</a></li>
<li class="chapter" data-level="7.2.6" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#add-weights-1"><i class="fa fa-check"></i><b>7.2.6</b> Add Weights</a></li>
<li class="chapter" data-level="7.2.7" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#optional-annotate-with-tf-motifs"><i class="fa fa-check"></i><b>7.2.7</b> (Optional) Annotate with TF motifs</a></li>
<li class="chapter" data-level="7.2.8" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#calculate-tf-activity-1"><i class="fa fa-check"></i><b>7.2.8</b> Calculate TF activity</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="basic-workflow-with-multiassayexperiment.html"><a href="basic-workflow-with-multiassayexperiment.html#session-info"><i class="fa fa-check"></i><b>7.3</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>8</b> Advanced workflow with MultiAssayExperiment</a>
<ul>
<li class="chapter" data-level="8.1" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#data-preparation-2"><i class="fa fa-check"></i><b>8.1</b> Data preparation</a></li>
<li class="chapter" data-level="8.2" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#retrieve-bulk-tf-chip-seq-binding-sites-1"><i class="fa fa-check"></i><b>8.2</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="8.3" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#link-atacseq-peaks-to-target-genes-1"><i class="fa fa-check"></i><b>8.3</b> Link ATACseq peaks to target genes</a></li>
<li class="chapter" data-level="8.4" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#add-tf-motif-binding-to-peaks-1"><i class="fa fa-check"></i><b>8.4</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="8.5" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#generate-regulons-2"><i class="fa fa-check"></i><b>8.5</b> Generate regulons</a></li>
<li class="chapter" data-level="8.6" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#prune-network"><i class="fa fa-check"></i><b>8.6</b> Prune network</a></li>
<li class="chapter" data-level="8.7" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#add-weights-2"><i class="fa fa-check"></i><b>8.7</b> Add Weights</a></li>
<li class="chapter" data-level="8.8" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#calculate-tf-activity-2"><i class="fa fa-check"></i><b>8.8</b> Calculate TF activity</a></li>
<li class="chapter" data-level="8.9" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#differential-tf-activity-test-1"><i class="fa fa-check"></i><b>8.9</b> Differential TF activity test</a></li>
<li class="chapter" data-level="8.10" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#visualizing-tf-activities-1"><i class="fa fa-check"></i><b>8.10</b> Visualizing TF activities</a></li>
<li class="chapter" data-level="8.11" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#geneset-enrichment-1"><i class="fa fa-check"></i><b>8.11</b> Geneset enrichment</a></li>
<li class="chapter" data-level="8.12" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#differential-network-analysis-1"><i class="fa fa-check"></i><b>8.12</b> Differential Network analysis</a></li>
<li class="chapter" data-level="8.13" data-path="advanced-workflow-with-multiassayexperiment.html"><a href="advanced-workflow-with-multiassayexperiment.html#session-info-1"><i class="fa fa-check"></i><b>8.13</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html"><i class="fa fa-check"></i><b>9</b> ArchR workflow and different weight methods</a>
<ul>
<li class="chapter" data-level="9.1" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#data-preparation-3"><i class="fa fa-check"></i><b>9.1</b> Data preparation</a></li>
<li class="chapter" data-level="9.2" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#load-archr-project"><i class="fa fa-check"></i><b>9.2</b> Load ArchR project</a></li>
<li class="chapter" data-level="9.3" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#retrieve-matrices-from-archr-project"><i class="fa fa-check"></i><b>9.3</b> Retrieve matrices from ArchR project</a></li>
<li class="chapter" data-level="9.4" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#retrieve-bulk-tf-chip-seq-binding-sites-2"><i class="fa fa-check"></i><b>9.4</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="9.5" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#link-atac-seq-peaks-to-target-genes-1"><i class="fa fa-check"></i><b>9.5</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="9.6" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#add-tf-motif-binding-to-peaks-2"><i class="fa fa-check"></i><b>9.6</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="9.7" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#generate-regulons-3"><i class="fa fa-check"></i><b>9.7</b> Generate regulons</a></li>
<li class="chapter" data-level="9.8" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#optional-annotate-with-tf-motifs-1"><i class="fa fa-check"></i><b>9.8</b> (Optional) Annotate with TF motifs</a></li>
<li class="chapter" data-level="9.9" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#prune-network-1"><i class="fa fa-check"></i><b>9.9</b> Prune network</a></li>
<li class="chapter" data-level="9.10" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#weights_prostate"><i class="fa fa-check"></i><b>9.10</b> Add Weights</a></li>
<li class="chapter" data-level="9.11" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#calculate-tf-activity-3"><i class="fa fa-check"></i><b>9.11</b> Calculate TF activity</a></li>
<li class="chapter" data-level="9.12" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#perform-differential-activity"><i class="fa fa-check"></i><b>9.12</b> Perform differential activity</a></li>
<li class="chapter" data-level="9.13" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#visualize-the-results"><i class="fa fa-check"></i><b>9.13</b> Visualize the results</a></li>
<li class="chapter" data-level="9.14" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#differential-network-analysis-2"><i class="fa fa-check"></i><b>9.14</b> Differential network analysis</a></li>
<li class="chapter" data-level="9.15" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#session-info-2"><i class="fa fa-check"></i><b>9.15</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html"><i class="fa fa-check"></i><b>10</b> Single modality: scRNA-seq only</a>
<ul>
<li class="chapter" data-level="10.1" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#load-regulon"><i class="fa fa-check"></i><b>10.1</b> Load regulon</a></li>
<li class="chapter" data-level="10.2" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#load-scrna-seq-data"><i class="fa fa-check"></i><b>10.2</b> Load scRNA-seq data</a></li>
<li class="chapter" data-level="10.3" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#calculate-activity"><i class="fa fa-check"></i><b>10.3</b> Calculate activity</a></li>
<li class="chapter" data-level="10.4" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#perform-differential-activity-1"><i class="fa fa-check"></i><b>10.4</b> Perform differential activity</a></li>
<li class="chapter" data-level="10.5" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#visualize-activity"><i class="fa fa-check"></i><b>10.5</b> Visualize activity</a></li>
<li class="chapter" data-level="10.6" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#pathway-enrichment"><i class="fa fa-check"></i><b>10.6</b> Pathway enrichment</a></li>
<li class="chapter" data-level="10.7" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#session-info-3"><i class="fa fa-check"></i><b>10.7</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="contact.html"><a href="contact.html"><i class="fa fa-check"></i><b>11</b> Contact</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Epiregulon documentation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="finding-patterns-of-transcription-factor-activity" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">5</span> Finding patterns of transcription factor activity<a href="finding-patterns-of-transcription-factor-activity.html#finding-patterns-of-transcription-factor-activity" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The <code>epiregulon.extra</code> package provides tools for convenient downstream analysis of the transcription factor activity matrix. Users can easily identify transcription factors that differ in activity across cell groups, visualize these differences, construct differential graphs highlighting differences between cell groups in GRN topology, and find potential interaction partners by searching for transcription factors with overlapping regulons.</p>
<div id="differential-tf-activity-test" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Differential TF activity test<a href="finding-patterns-of-transcription-factor-activity.html#differential-tf-activity-test" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can next determine which TFs exhibit differential activities across cell clusters/groups via the <code>findDifferentialActivity</code> function. This function depends on <code>findMarkers</code> function from <em><a href="https://bioconductor.org/packages/3.22/scran">scran</a></em> package.</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb87-1" tabindex="-1"></a><span class="fu">library</span>(epiregulon.extra)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;epiregulon.extra&#39;</code></pre>
<pre><code>## The following object is masked _by_ &#39;.GlobalEnv&#39;:
## 
##     regulon</code></pre>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb90-1" tabindex="-1"></a>markers  <span class="ot">&lt;-</span> <span class="fu">findDifferentialActivity</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb90-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb90-2" tabindex="-1"></a>                                    <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>cell_type, </span>
<span id="cb90-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb90-3" tabindex="-1"></a>                                    <span class="at">pval.type =</span> <span class="st">&quot;some&quot;</span>, </span>
<span id="cb90-4"><a href="finding-patterns-of-transcription-factor-activity.html#cb90-4" tabindex="-1"></a>                                    <span class="at">direction =</span> <span class="st">&quot;up&quot;</span>, </span>
<span id="cb90-5"><a href="finding-patterns-of-transcription-factor-activity.html#cb90-5" tabindex="-1"></a>                                    <span class="at">logvalues =</span> <span class="cn">FALSE</span>)</span>
<span id="cb90-6"><a href="finding-patterns-of-transcription-factor-activity.html#cb90-6" tabindex="-1"></a>markers</span></code></pre></div>
<pre><code>## List of length 11
## names(11): B CD14+ Mono DC FCGR3A+ Mono ... NA Naive CD4+ T Naive CD8+ T NK</code></pre>
<p>getSigGenes compiles the different test results into a single dataframe and enables user to supply their desired cutoffs for significance and variable to order by.</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb92-1" tabindex="-1"></a>markers.sig <span class="ot">&lt;-</span> <span class="fu">getSigGenes</span>(markers, <span class="at">topgenes =</span> <span class="dv">3</span> )</span></code></pre></div>
<pre><code>## Using a cutoff of 0.01 for class B for direction equal to any</code></pre>
<pre><code>## Using a cutoff of 0.04 for class CD14+ Mono for direction equal to any</code></pre>
<pre><code>## Using a cutoff of 0.018 for class DC for direction equal to any</code></pre>
<pre><code>## Using a cutoff of 0.035 for class FCGR3A+ Mono for direction equal to any</code></pre>
<pre><code>## Using a cutoff of 0.016 for class Memory CD4+ T for direction equal to any</code></pre>
<pre><code>## Using a cutoff of 0.006 for class Memory CD8+ T for direction equal to any</code></pre>
<pre><code>## Using a cutoff of 0.025 for class Monocytes for direction equal to any</code></pre>
<pre><code>## Using a cutoff of 0.0093 for class NA for direction equal to any</code></pre>
<pre><code>## Using a cutoff of 0.023 for class Naive CD4+ T for direction equal to any</code></pre>
<pre><code>## Using a cutoff of 0.023 for class Naive CD8+ T for direction equal to any</code></pre>
<pre><code>## Using a cutoff of 0.0089 for class NK for direction equal to any</code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb104-1" tabindex="-1"></a>markers.sig </span></code></pre></div>
<pre><code>##           p.value           FDR summary.diff         class      tf
## 6    0.000000e+00  0.000000e+00  0.193325737             B    CD74
## 37   0.000000e+00  0.000000e+00  0.100447985             B POU2AF1
## 17   0.000000e+00  0.000000e+00  0.039562050             B    IRF8
## 22   0.000000e+00  0.000000e+00  0.120908613    CD14+ Mono   MAML3
## 8    0.000000e+00  0.000000e+00  0.114698914    CD14+ Mono   CREB5
## 41   0.000000e+00  0.000000e+00  0.099610316    CD14+ Mono   SRSF9
## 38  1.128666e-114 2.778776e-112  0.018734362            DC    RXRA
## 43  5.428903e-114 1.113830e-111  0.021602710            DC  TCF7L2
## 52  1.236842e-112 1.903191e-110  0.030075002            DC   ZFP36
## 21   0.000000e+00  0.000000e+00  0.080942070  FCGR3A+ Mono    IRF8
## 9    0.000000e+00  0.000000e+00  0.078922634  FCGR3A+ Mono   CREB5
## 61   0.000000e+00  0.000000e+00  0.077188563  FCGR3A+ Mono    CD74
## 5   4.244677e-195 2.612599e-192  0.019021906 Memory CD4+ T     MAF
## 10  5.182685e-134 5.799896e-132  0.041957212 Memory CD4+ T   STAT4
## 1   2.465418e-132 2.529108e-130  0.055545225 Memory CD4+ T  BCL11B
## 23   0.000000e+00  0.000000e+00  0.007724666 Memory CD8+ T   TBX21
## 4    0.000000e+00 7.806237e-322  0.017725797 Memory CD8+ T   EOMES
## 101 1.668341e-284 4.107457e-282  0.008820261 Memory CD8+ T    LDB2
## 62  1.155957e-170 3.557458e-168  0.028284948     Monocytes  BCL11A
## 60  1.088991e-159 1.340548e-157  0.028093577     Monocytes  ZNF703
## 231 2.934533e-156 2.124947e-154  0.074502531     Monocytes    IRF8
## 2    6.643574e-13  8.178239e-10  0.038336649            NA  BCL11B
## 3    1.870194e-12  1.151105e-09  0.013042475            NA    ETS1
## 51   7.780093e-12  3.192431e-09  0.023982082            NA    LEF1
## 11  1.598013e-250 6.557181e-248  0.023665835  Naive CD4+ T    LEF1
## 42  9.272474e-152 6.714362e-150  0.047429424  Naive CD4+ T   STAT4
## 24  4.146310e-129 2.126712e-127  0.023443747  Naive CD4+ T  NFATC2
## 12  9.053450e-262 1.857466e-259  0.024415366  Naive CD8+ T    LEF1
## 44  6.883051e-146 3.258860e-144  0.047021830  Naive CD8+ T   STAT4
## 26  1.456341e-101 3.382558e-100  0.046648772  Naive CD8+ T   PRKCQ
## 16  4.467607e-251 5.499625e-248  0.028244464            NK   EOMES
## 381 2.554555e-228 1.572329e-225  0.013529838            NK   TBX21
## 25  1.349155e-226 5.536033e-224  0.022278908            NK    LDB2</code></pre>
</div>
<div id="visualizing-tf-activities" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Visualizing TF activities<a href="finding-patterns-of-transcription-factor-activity.html#visualizing-tf-activities" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Epiregulon also provides the option for visualizing the inferred TF activities in reduced dimensions.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb106-1" tabindex="-1"></a><span class="fu">options</span>(<span class="at">ggrastr.default.dpi=</span><span class="dv">300</span>)</span>
<span id="cb106-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb106-2" tabindex="-1"></a>tfs_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;BCL11B&quot;</span>, <span class="st">&quot;KLF4&quot;</span>, <span class="st">&quot;IRF8&quot;</span>, <span class="st">&quot;EBF1&quot;</span>)</span>
<span id="cb106-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb106-3" tabindex="-1"></a><span class="fu">plotActivityDim</span>(<span class="at">sce =</span> GeneExpressionMatrix,</span>
<span id="cb106-4"><a href="finding-patterns-of-transcription-factor-activity.html#cb106-4" tabindex="-1"></a>                <span class="at">activity_matrix =</span> score.combine[tfs_interest,], </span>
<span id="cb106-5"><a href="finding-patterns-of-transcription-factor-activity.html#cb106-5" tabindex="-1"></a>                <span class="at">tf =</span> tfs_interest, </span>
<span id="cb106-6"><a href="finding-patterns-of-transcription-factor-activity.html#cb106-6" tabindex="-1"></a>                <span class="at">dimtype =</span> <span class="st">&quot;UMAP_RNA&quot;</span>,</span>
<span id="cb106-7"><a href="finding-patterns-of-transcription-factor-activity.html#cb106-7" tabindex="-1"></a>                <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb106-8"><a href="finding-patterns-of-transcription-factor-activity.html#cb106-8" tabindex="-1"></a>                <span class="at">ncol=</span><span class="dv">2</span>,</span>
<span id="cb106-9"><a href="finding-patterns-of-transcription-factor-activity.html#cb106-9" tabindex="-1"></a>                <span class="at">point_size=</span><span class="fl">0.1</span>,</span>
<span id="cb106-10"><a href="finding-patterns-of-transcription-factor-activity.html#cb106-10" tabindex="-1"></a>                <span class="at">rasterise =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-33-1.png" width="672" /></p>
<p>We can compare the activity with gene expression of the same TFs.</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb107-1" tabindex="-1"></a><span class="fu">plotActivityDim</span>(<span class="at">sce =</span> GeneExpressionMatrix, </span>
<span id="cb107-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb107-2" tabindex="-1"></a>                <span class="at">activity_matrix =</span> <span class="fu">assay</span>(GeneExpressionMatrix, <span class="st">&quot;normalizedCounts&quot;</span>)[tfs_interest,], </span>
<span id="cb107-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb107-3" tabindex="-1"></a>                <span class="at">tf =</span> tfs_interest, </span>
<span id="cb107-4"><a href="finding-patterns-of-transcription-factor-activity.html#cb107-4" tabindex="-1"></a>                <span class="at">dimtype =</span> <span class="st">&quot;UMAP_RNA&quot;</span>,</span>
<span id="cb107-5"><a href="finding-patterns-of-transcription-factor-activity.html#cb107-5" tabindex="-1"></a>                <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb107-6"><a href="finding-patterns-of-transcription-factor-activity.html#cb107-6" tabindex="-1"></a>                <span class="at">ncol=</span><span class="dv">2</span>,</span>
<span id="cb107-7"><a href="finding-patterns-of-transcription-factor-activity.html#cb107-7" tabindex="-1"></a>                <span class="at">legend.label =</span> <span class="st">&quot;Gex&quot;</span>,</span>
<span id="cb107-8"><a href="finding-patterns-of-transcription-factor-activity.html#cb107-8" tabindex="-1"></a>                <span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&quot;grey&quot;</span>,<span class="st">&quot;blue&quot;</span>),</span>
<span id="cb107-9"><a href="finding-patterns-of-transcription-factor-activity.html#cb107-9" tabindex="-1"></a>                <span class="at">point_size=</span><span class="fl">0.1</span>,</span>
<span id="cb107-10"><a href="finding-patterns-of-transcription-factor-activity.html#cb107-10" tabindex="-1"></a>                <span class="at">rasterise =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-34-1.png" width="672" /></p>
<p>We can also plot violin plot to visualize TF activity.</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb108-1" tabindex="-1"></a><span class="fu">plotActivityViolin</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb108-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb108-2" tabindex="-1"></a>                   <span class="at">tf =</span> tfs_interest, </span>
<span id="cb108-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb108-3" tabindex="-1"></a>                   <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>cell_type,</span>
<span id="cb108-4"><a href="finding-patterns-of-transcription-factor-activity.html#cb108-4" tabindex="-1"></a>                   <span class="at">legend.label =</span> <span class="st">&quot;Gex&quot;</span>,</span>
<span id="cb108-5"><a href="finding-patterns-of-transcription-factor-activity.html#cb108-5" tabindex="-1"></a>                   <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb108-6"><a href="finding-patterns-of-transcription-factor-activity.html#cb108-6" tabindex="-1"></a>                   <span class="at">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-35-1.png" width="672" /></p>
<p>We plot violin plot to visualize TF gene expression.</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb109-1" tabindex="-1"></a><span class="fu">plotActivityViolin</span>(<span class="at">activity_matrix =</span> <span class="fu">assay</span>(GeneExpressionMatrix, <span class="st">&quot;normalizedCounts&quot;</span>)[tfs_interest,],  </span>
<span id="cb109-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb109-2" tabindex="-1"></a>                   <span class="at">tf =</span> tfs_interest, </span>
<span id="cb109-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb109-3" tabindex="-1"></a>                   <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>cell_type,</span>
<span id="cb109-4"><a href="finding-patterns-of-transcription-factor-activity.html#cb109-4" tabindex="-1"></a>                   <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb109-5"><a href="finding-patterns-of-transcription-factor-activity.html#cb109-5" tabindex="-1"></a>                   <span class="at">ncol=</span><span class="dv">2</span>,</span>
<span id="cb109-6"><a href="finding-patterns-of-transcription-factor-activity.html#cb109-6" tabindex="-1"></a>                   <span class="at">legend.label =</span> <span class="st">&quot;gene expression&quot;</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-36-1.png" width="672" /></p>
<p>We can visualize the different TFs in a bubble plot:</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb110-1" tabindex="-1"></a><span class="fu">plotBubble</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb110-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb110-2" tabindex="-1"></a>           <span class="at">tf =</span> tfs_interest,</span>
<span id="cb110-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb110-3" tabindex="-1"></a>           GeneExpressionMatrix<span class="sc">$</span>cell_type,</span>
<span id="cb110-4"><a href="finding-patterns-of-transcription-factor-activity.html#cb110-4" tabindex="-1"></a>           <span class="at">pval.type =</span> <span class="st">&quot;some&quot;</span>, </span>
<span id="cb110-5"><a href="finding-patterns-of-transcription-factor-activity.html#cb110-5" tabindex="-1"></a>           <span class="at">direction =</span> <span class="st">&quot;up&quot;</span>, </span>
<span id="cb110-6"><a href="finding-patterns-of-transcription-factor-activity.html#cb110-6" tabindex="-1"></a>           <span class="at">logvalues =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-37-1.png" width="672" /></p>
<p>We visualize the top differential TFs based on activity.</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb111-1" tabindex="-1"></a><span class="fu">plotBubble</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb111-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb111-2" tabindex="-1"></a>           <span class="at">tf =</span> markers.sig<span class="sc">$</span>tf, </span>
<span id="cb111-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb111-3" tabindex="-1"></a>           GeneExpressionMatrix<span class="sc">$</span>cell_type,</span>
<span id="cb111-4"><a href="finding-patterns-of-transcription-factor-activity.html#cb111-4" tabindex="-1"></a>           <span class="at">pval.type =</span> <span class="st">&quot;some&quot;</span>, </span>
<span id="cb111-5"><a href="finding-patterns-of-transcription-factor-activity.html#cb111-5" tabindex="-1"></a>           <span class="at">direction =</span> <span class="st">&quot;up&quot;</span>, </span>
<span id="cb111-6"><a href="finding-patterns-of-transcription-factor-activity.html#cb111-6" tabindex="-1"></a>           <span class="at">logvalues =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-38-1.png" width="672" /></p>
</div>
<div id="geneset-enrichment" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Geneset enrichment<a href="finding-patterns-of-transcription-factor-activity.html#geneset-enrichment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Sometimes we are interested to know what pathways are enriched in the regulon of a particular TF. We can perform geneset enrichment using the enricher function from <a href="http://yulab-smu.top/biomedical-knowledge-mining-book/universal-api.html">clusterProfiler</a>.</p>
<p>Here we first download Hallmark and C2 signatures from hallmark and then perform gene set enrichment of the known lineage factors. As expected, EBF1 is consistent with a B cell lineage factor, GATA3 and RUNX3 with lymphoid lineage and SPI1 with myeloid lineage.</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-1" tabindex="-1"></a><span class="co">#retrieve genesets</span></span>
<span id="cb112-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-2" tabindex="-1"></a>H <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">getGenesets</span>(<span class="at">org =</span> <span class="st">&quot;hsa&quot;</span>, </span>
<span id="cb112-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-3" tabindex="-1"></a>                                    <span class="at">db =</span> <span class="st">&quot;msigdb&quot;</span>, </span>
<span id="cb112-4"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-4" tabindex="-1"></a>                                    <span class="at">cat =</span> <span class="st">&quot;H&quot;</span>, </span>
<span id="cb112-5"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-5" tabindex="-1"></a>                                    <span class="at">gene.id.type =</span> <span class="st">&quot;SYMBOL&quot;</span> )</span>
<span id="cb112-6"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-6" tabindex="-1"></a>C2 <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">getGenesets</span>(<span class="at">org =</span> <span class="st">&quot;hsa&quot;</span>, </span>
<span id="cb112-7"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-7" tabindex="-1"></a>                                     <span class="at">db =</span> <span class="st">&quot;msigdb&quot;</span>, </span>
<span id="cb112-8"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-8" tabindex="-1"></a>                                     <span class="at">cat =</span> <span class="st">&quot;C2&quot;</span>,</span>
<span id="cb112-9"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-9" tabindex="-1"></a>                                     <span class="at">gene.id.type =</span> <span class="st">&quot;SYMBOL&quot;</span> )</span>
<span id="cb112-10"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-10" tabindex="-1"></a></span>
<span id="cb112-11"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-11" tabindex="-1"></a></span>
<span id="cb112-12"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-12" tabindex="-1"></a><span class="co">#combine genesets and convert genesets to be compatible with enricher</span></span>
<span id="cb112-13"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-13" tabindex="-1"></a>gs <span class="ot">&lt;-</span> <span class="fu">c</span>(H, C2)</span>
<span id="cb112-14"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-14" tabindex="-1"></a>gs.list <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind,<span class="fu">lapply</span>(<span class="fu">names</span>(gs), <span class="cf">function</span>(x) </span>
<span id="cb112-15"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-15" tabindex="-1"></a>  {<span class="fu">data.frame</span>(<span class="at">gs=</span>x, <span class="at">genes=</span>gs[[x]])}))</span>
<span id="cb112-16"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-16" tabindex="-1"></a></span>
<span id="cb112-17"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-17" tabindex="-1"></a>enrichresults <span class="ot">&lt;-</span> <span class="fu">regulonEnrich</span>(<span class="at">TF =</span> tfs_interest, </span>
<span id="cb112-18"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-18" tabindex="-1"></a>                               <span class="at">regulon =</span> regulon.w, </span>
<span id="cb112-19"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-19" tabindex="-1"></a>                               <span class="at">weight =</span> <span class="st">&quot;weight&quot;</span>,</span>
<span id="cb112-20"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-20" tabindex="-1"></a>                               <span class="at">weight_cutoff =</span> <span class="dv">0</span>, </span>
<span id="cb112-21"><a href="finding-patterns-of-transcription-factor-activity.html#cb112-21" tabindex="-1"></a>                               <span class="at">genesets =</span> gs.list)</span></code></pre></div>
<pre><code>## BCL11B</code></pre>
<pre><code>## </code></pre>
<pre><code>## KLF4</code></pre>
<pre><code>## IRF8</code></pre>
<pre><code>## EBF1</code></pre>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb118-1" tabindex="-1"></a><span class="co">#plot results</span></span>
<span id="cb118-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb118-2" tabindex="-1"></a><span class="fu">enrichPlot</span>(<span class="at">results =</span> enrichresults, <span class="at">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-39-1.png" width="1152" /></p>
</div>
<div id="differential-network-analysis" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Differential Network analysis<a href="finding-patterns-of-transcription-factor-activity.html#differential-network-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In addition to looking at the summed TF activity, a second approach to investigate differential TF activity is to compare and contrast target genes or network topology. In this example, we know that EBF1 is a B cell lineage factor. If we plot the differential network of EBF1 using the regulon with cluster-specific weights, we can see that EBF1 has many more targets in B cells than it has in CD4 memory T cells.</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb119-1" tabindex="-1"></a><span class="fu">plotDiffNetwork</span>(regulon.w,</span>
<span id="cb119-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb119-2" tabindex="-1"></a>                <span class="at">cutoff =</span> <span class="dv">0</span>,</span>
<span id="cb119-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb119-3" tabindex="-1"></a>                <span class="at">tf =</span> <span class="fu">c</span>(<span class="st">&quot;EBF1&quot;</span>),</span>
<span id="cb119-4"><a href="finding-patterns-of-transcription-factor-activity.html#cb119-4" tabindex="-1"></a>                <span class="at">weight =</span> <span class="st">&quot;weight&quot;</span>,</span>
<span id="cb119-5"><a href="finding-patterns-of-transcription-factor-activity.html#cb119-5" tabindex="-1"></a>                <span class="at">clusters =</span> <span class="fu">c</span>(<span class="st">&quot;Memory CD4+ T&quot;</span>,<span class="st">&quot;B&quot;</span>),</span>
<span id="cb119-6"><a href="finding-patterns-of-transcription-factor-activity.html#cb119-6" tabindex="-1"></a>                <span class="at">layout =</span> <span class="st">&quot;stress&quot;</span>)</span></code></pre></div>
<pre><code>## Building graph using weight as edge weights</code></pre>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-40-1.png" width="672" /></p>
<p>Sometimes, we are interested to identify interaction partners of the TFs of interest. This can be achieved by comparing the overlap of the targets genes for all the TFs and identify the most similar TFs by Jaccard similarity. To illustrate this function, we take a look at the top most similar 20 TFs to EBF1, and we successfully identify PAX5 as the most similar TF. Both PAX5 and EBF1 are important factors for B cell development (<a href="https://www.nature.com/articles/ni.2641" class="uri">https://www.nature.com/articles/ni.2641</a>).</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb121-1" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb121-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb121-2" tabindex="-1"></a><span class="co"># construct a graph of the preB cells</span></span>
<span id="cb121-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb121-3" tabindex="-1"></a>B_network <span class="ot">&lt;-</span> <span class="fu">buildGraph</span>(regulon.w, <span class="at">weights =</span> <span class="st">&quot;weight&quot;</span>, <span class="at">cluster=</span><span class="st">&quot;B&quot;</span>)</span></code></pre></div>
<pre><code>## Building graph using weight as edge weights</code></pre>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-1" tabindex="-1"></a><span class="co"># compute a similarity matrix of all TFs</span></span>
<span id="cb123-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-2" tabindex="-1"></a>similarity_score <span class="ot">&lt;-</span> <span class="fu">calculateJaccardSimilarity</span>(B_network)</span>
<span id="cb123-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-3" tabindex="-1"></a></span>
<span id="cb123-4"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-4" tabindex="-1"></a><span class="co"># Focus on EBF1</span></span>
<span id="cb123-5"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-5" tabindex="-1"></a>similarity_score_EBF1 <span class="ot">&lt;-</span> similarity_score[, <span class="st">&quot;EBF1&quot;</span>]</span>
<span id="cb123-6"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-6" tabindex="-1"></a>similarity_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">similarity =</span> <span class="fu">head</span>(<span class="fu">sort</span>(similarity_score_EBF1, </span>
<span id="cb123-7"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-7" tabindex="-1"></a>                                                   <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>),</span>
<span id="cb123-8"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-8" tabindex="-1"></a>                            <span class="at">TF =</span> <span class="fu">names</span>(<span class="fu">head</span>(<span class="fu">sort</span>(similarity_score_EBF1,</span>
<span id="cb123-9"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-9" tabindex="-1"></a>                                                 <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>)))</span>
<span id="cb123-10"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-10" tabindex="-1"></a></span>
<span id="cb123-11"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-11" tabindex="-1"></a>similarity_df<span class="sc">$</span>TF <span class="ot">&lt;-</span> <span class="fu">factor</span>(similarity_df<span class="sc">$</span>TF, <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(similarity_df<span class="sc">$</span>TF)))</span>
<span id="cb123-12"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-12" tabindex="-1"></a></span>
<span id="cb123-13"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-13" tabindex="-1"></a><span class="co"># plot top TFs most similar to EBF1</span></span>
<span id="cb123-14"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-14" tabindex="-1"></a>topTFplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(similarity_df, <span class="fu">aes</span>(<span class="at">x=</span>TF, <span class="at">y=</span>similarity)) <span class="sc">+</span></span>
<span id="cb123-15"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-15" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb123-16"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-16" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb123-17"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-17" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;EBF1 similarity&quot;</span>) <span class="sc">+</span></span>
<span id="cb123-18"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-18" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb123-19"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-19" tabindex="-1"></a></span>
<span id="cb123-20"><a href="finding-patterns-of-transcription-factor-activity.html#cb123-20" tabindex="-1"></a><span class="fu">print</span>(topTFplot)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-41-1.png" width="672" /></p>
<p>In order to convince ourselves that our differential network is statistically significant, we permute the edges and obtain a background graph from averaging many iterations. Here, we plot the differential network graph subtracted by permuted graphs.</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-1" tabindex="-1"></a><span class="co"># create a permuted graph by rewiring the edges 100 times</span></span>
<span id="cb124-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-2" tabindex="-1"></a>permute_matrix <span class="ot">&lt;-</span> <span class="fu">permuteGraph</span>(B_network, <span class="st">&quot;EBF1&quot;</span>, <span class="dv">100</span>, <span class="at">p=</span><span class="dv">1</span>)</span>
<span id="cb124-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-3" tabindex="-1"></a>permute_matrix <span class="ot">&lt;-</span> permute_matrix[<span class="fu">names</span>(similarity_score_EBF1),]</span>
<span id="cb124-4"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-4" tabindex="-1"></a>diff_matrix <span class="ot">&lt;-</span> similarity_score_EBF1<span class="sc">-</span><span class="fu">rowMeans</span>(permute_matrix)</span>
<span id="cb124-5"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-5" tabindex="-1"></a></span>
<span id="cb124-6"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-6" tabindex="-1"></a>diff_matrix_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">similarity =</span> <span class="fu">head</span>(<span class="fu">sort</span>(diff_matrix, </span>
<span id="cb124-7"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-7" tabindex="-1"></a>                                                    <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>),</span>
<span id="cb124-8"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-8" tabindex="-1"></a>                            <span class="at">TF =</span> <span class="fu">names</span>(<span class="fu">head</span>(<span class="fu">sort</span>(diff_matrix,</span>
<span id="cb124-9"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-9" tabindex="-1"></a>                                                 <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>)))</span>
<span id="cb124-10"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-10" tabindex="-1"></a></span>
<span id="cb124-11"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-11" tabindex="-1"></a>diff_matrix_df<span class="sc">$</span>TF <span class="ot">&lt;-</span> <span class="fu">factor</span>(diff_matrix_df<span class="sc">$</span>TF, <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(diff_matrix_df<span class="sc">$</span>TF)))</span>
<span id="cb124-12"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-12" tabindex="-1"></a></span>
<span id="cb124-13"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-13" tabindex="-1"></a><span class="co"># plot top TFs most similar to EBF1</span></span>
<span id="cb124-14"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-14" tabindex="-1"></a>topTFplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(diff_matrix_df, <span class="fu">aes</span>(<span class="at">x=</span>TF, <span class="at">y=</span>similarity)) <span class="sc">+</span></span>
<span id="cb124-15"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-15" tabindex="-1"></a>            <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb124-16"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-16" tabindex="-1"></a>            <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb124-17"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-17" tabindex="-1"></a>            <span class="fu">ggtitle</span>(<span class="st">&quot;background subtracted EBF1 similarity &quot;</span>) <span class="sc">+</span></span>
<span id="cb124-18"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-18" tabindex="-1"></a>            <span class="fu">theme_classic</span>()</span>
<span id="cb124-19"><a href="finding-patterns-of-transcription-factor-activity.html#cb124-19" tabindex="-1"></a><span class="fu">print</span>(topTFplot)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-42-1.png" width="672" /></p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb125-1" tabindex="-1"></a><span class="co"># obtain empirical p-values</span></span>
<span id="cb125-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb125-2" tabindex="-1"></a>p_matrix <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">apply</span>(permute_matrix, <span class="dv">2</span>, <span class="cf">function</span>(x) {x <span class="sc">&gt;</span> similarity_score_EBF1}))</span>
<span id="cb125-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb125-3" tabindex="-1"></a>p_matrix[<span class="fu">names</span>(<span class="fu">head</span>(<span class="fu">sort</span>(diff_matrix,<span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>))]</span></code></pre></div>
<pre><code>##    PAX5  POU2F2    RELB  BCL11A    TCF4     NBN     EED  BCLAF1   MEF2A   RUNX3 
##       0       0       0       0       0       0       0       0       0       0 
##    TCF3 BHLHE40    MTA3    IRF4    MTA2     PML    CHD2 TBL1XR1    BCOR   KMT2D 
##       0       0       0       0       0       0       0       0       0       0</code></pre>
<p>Next, we are interested to compare the networks of two cell types, in this case, CD4 memory T cells vs monocytes. We build an edge subtracted graph and then calculate the degree centrality of the subtracted graph. We normalize centrality using the default square root function. The top 5 most positive TFs represent lineage factors more active in monocytes whereas the bottom 5 TFs present lineage factors enriched in CD4. We successfully identified the myeloid factor SPI1 to be associated with monocytes and Th1 factor TBX21 to be associated with CD4 T cells.</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb127-1" tabindex="-1"></a><span class="co"># construct a graph of the CD4.M and NK cells respectively</span></span>
<span id="cb127-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb127-2" tabindex="-1"></a>CD4.M_network <span class="ot">&lt;-</span> <span class="fu">buildGraph</span>(regulon.w, <span class="at">weights =</span> <span class="st">&quot;weight&quot;</span>, <span class="at">cluster=</span><span class="st">&quot;Memory CD4+ T&quot;</span>)</span></code></pre></div>
<pre><code>## Building graph using weight as edge weights</code></pre>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb129-1" tabindex="-1"></a>Mono_network <span class="ot">&lt;-</span> <span class="fu">buildGraph</span>(regulon.w, <span class="at">weights =</span> <span class="st">&quot;weight&quot;</span>, <span class="at">cluster=</span><span class="st">&quot;CD14+ Mono&quot;</span>)</span></code></pre></div>
<pre><code>## Building graph using weight as edge weights</code></pre>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb131-1" tabindex="-1"></a><span class="co"># construct a difference graph</span></span>
<span id="cb131-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb131-2" tabindex="-1"></a>diff_graph <span class="ot">&lt;-</span> <span class="fu">buildDiffGraph</span>(Mono_network,CD4.M_network, <span class="at">abs_diff =</span> <span class="cn">FALSE</span>)</span>
<span id="cb131-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb131-3" tabindex="-1"></a>diff_graph <span class="ot">&lt;-</span> <span class="fu">addCentrality</span>(diff_graph)</span>
<span id="cb131-4"><a href="finding-patterns-of-transcription-factor-activity.html#cb131-4" tabindex="-1"></a>diff_graph <span class="ot">&lt;-</span> <span class="fu">normalizeCentrality</span>(diff_graph)</span>
<span id="cb131-5"><a href="finding-patterns-of-transcription-factor-activity.html#cb131-5" tabindex="-1"></a>rank_table <span class="ot">&lt;-</span> <span class="fu">rankTfs</span>(diff_graph)</span>
<span id="cb131-6"><a href="finding-patterns-of-transcription-factor-activity.html#cb131-6" tabindex="-1"></a></span>
<span id="cb131-7"><a href="finding-patterns-of-transcription-factor-activity.html#cb131-7" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb131-8"><a href="finding-patterns-of-transcription-factor-activity.html#cb131-8" tabindex="-1"></a><span class="fu">ggplot</span>(rank_table, <span class="fu">aes</span>(<span class="at">x =</span> rank, <span class="at">y =</span> centrality)) <span class="sc">+</span></span>
<span id="cb131-9"><a href="finding-patterns-of-transcription-factor-activity.html#cb131-9" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb131-10"><a href="finding-patterns-of-transcription-factor-activity.html#cb131-10" tabindex="-1"></a>    ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span> <span class="fu">rbind</span>(<span class="fu">head</span>(rank_table, <span class="dv">10</span>), </span>
<span id="cb131-11"><a href="finding-patterns-of-transcription-factor-activity.html#cb131-11" tabindex="-1"></a>                                          <span class="fu">tail</span>(rank_table, <span class="dv">10</span>)), </span>
<span id="cb131-12"><a href="finding-patterns-of-transcription-factor-activity.html#cb131-12" tabindex="-1"></a>                             <span class="fu">aes</span>(<span class="at">label =</span> tf), </span>
<span id="cb131-13"><a href="finding-patterns-of-transcription-factor-activity.html#cb131-13" tabindex="-1"></a>                             <span class="at">nudge_x =</span> <span class="dv">0</span>, <span class="at">nudge_y =</span> <span class="dv">0</span>, <span class="at">box.padding =</span> <span class="fl">0.5</span>, <span class="at">max.overlaps =</span> <span class="cn">Inf</span>) <span class="sc">+</span></span>
<span id="cb131-14"><a href="finding-patterns-of-transcription-factor-activity.html#cb131-14" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span> <span class="fu">ggtitle</span> (<span class="st">&quot;differential TFs (Mono-CD4.M) ranked by degree centrality&quot;</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-43-1.png" width="672" /></p>
<p>We can further explore interacting factors with the myeloid factor SPI1 using the same Jaccard similarity approach. We found CEBPA as the most similar TF as SPI1. SPI1 and CEBPA are known to be important for differentiation into myeloid cells (<a href="https://www.cell.com/cell-reports/pdfExtended/S2211-1247(18)30745-9" class="uri">https://www.cell.com/cell-reports/pdfExtended/S2211-1247(18)30745-9</a>).</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb132-1" tabindex="-1"></a><span class="fu">library</span>(igraph)</span></code></pre></div>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb133-1" tabindex="-1"></a>diff_graph_filter <span class="ot">&lt;-</span> <span class="fu">subgraph.edges</span>(diff_graph, </span>
<span id="cb133-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb133-2" tabindex="-1"></a>                                    <span class="fu">E</span>(diff_graph)[<span class="fu">E</span>(diff_graph)<span class="sc">$</span>weight<span class="sc">&gt;</span><span class="dv">0</span>], </span>
<span id="cb133-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb133-3" tabindex="-1"></a>                                    <span class="at">del=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Warning: `subgraph.edges()` was deprecated in igraph 2.1.0.
## â„¹ Please use `subgraph_from_edges()` instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.</code></pre>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-1" tabindex="-1"></a><span class="co"># compute a similarity matrix of all TFs</span></span>
<span id="cb135-2"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-2" tabindex="-1"></a>similarity_score <span class="ot">&lt;-</span> <span class="fu">calculateJaccardSimilarity</span>(diff_graph_filter)</span>
<span id="cb135-3"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-3" tabindex="-1"></a></span>
<span id="cb135-4"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-4" tabindex="-1"></a><span class="co"># Focus on SPI1</span></span>
<span id="cb135-5"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-5" tabindex="-1"></a>similarity_score_SPI1 <span class="ot">&lt;-</span> similarity_score[, <span class="st">&quot;SPI1&quot;</span>]</span>
<span id="cb135-6"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-6" tabindex="-1"></a>similarity_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">similarity =</span> <span class="fu">head</span>(<span class="fu">sort</span>(similarity_score_SPI1, </span>
<span id="cb135-7"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-7" tabindex="-1"></a>                                                   <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>),</span>
<span id="cb135-8"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-8" tabindex="-1"></a>                            <span class="at">TF =</span> <span class="fu">names</span>(<span class="fu">head</span>(<span class="fu">sort</span>(similarity_score_SPI1,</span>
<span id="cb135-9"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-9" tabindex="-1"></a>                                                 <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>)))</span>
<span id="cb135-10"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-10" tabindex="-1"></a></span>
<span id="cb135-11"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-11" tabindex="-1"></a>similarity_df<span class="sc">$</span>TF <span class="ot">&lt;-</span> <span class="fu">factor</span>(similarity_df<span class="sc">$</span>TF, </span>
<span id="cb135-12"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-12" tabindex="-1"></a>                           <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(similarity_df<span class="sc">$</span>TF)))</span>
<span id="cb135-13"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-13" tabindex="-1"></a></span>
<span id="cb135-14"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-14" tabindex="-1"></a><span class="co"># plot top TFs most similar to SPI1</span></span>
<span id="cb135-15"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-15" tabindex="-1"></a>topTFplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(similarity_df, <span class="fu">aes</span>(<span class="at">x=</span>TF, <span class="at">y=</span>similarity)) <span class="sc">+</span></span>
<span id="cb135-16"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-16" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb135-17"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-17" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb135-18"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-18" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;SPI1 similarity&quot;</span>) <span class="sc">+</span></span>
<span id="cb135-19"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-19" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb135-20"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-20" tabindex="-1"></a></span>
<span id="cb135-21"><a href="finding-patterns-of-transcription-factor-activity.html#cb135-21" tabindex="-1"></a><span class="fu">print</span>(topTFplot)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-45-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="calculation-of-the-tf-activity.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="data-preparation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/Graph_construction.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["Epiregulon documentation.pdf", "Epiregulon documentation.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
