<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Hematopoeisis tutorial - MAE | Epiregulon documentation</title>
  <meta name="description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Hematopoeisis tutorial - MAE | Epiregulon documentation" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Hematopoeisis tutorial - MAE | Epiregulon documentation" />
  
  <meta name="twitter:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  

<meta name="author" content="Xiaosai Yao, Tomasz Włodarczyk" />


<meta name="date" content="2023-11-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="multiome---archr-workflow.html"/>
<link rel="next" href="prostate-cancer-cells.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Epiregulon documentation</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>1</b> Installation</a></li>
<li class="chapter" data-level="2" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html"><i class="fa fa-check"></i><b>2</b> Multiome - ArchR workflow</a>
<ul>
<li class="chapter" data-level="2.1" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#data-preparation"><i class="fa fa-check"></i><b>2.1</b> Data preparation</a></li>
<li class="chapter" data-level="2.2" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#quick-start"><i class="fa fa-check"></i><b>2.2</b> Quick start</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#retrieve-bulk-tf-chip-seq-binding-sites"><i class="fa fa-check"></i><b>2.2.1</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="2.2.2" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#link-atac-seq-peaks-to-target-genes"><i class="fa fa-check"></i><b>2.2.2</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="2.2.3" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#add-tf-motif-binding-to-peaks"><i class="fa fa-check"></i><b>2.2.3</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="2.2.4" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#generate-regulons"><i class="fa fa-check"></i><b>2.2.4</b> Generate regulons</a></li>
<li class="chapter" data-level="2.2.5" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#network-pruning-highly-recommended"><i class="fa fa-check"></i><b>2.2.5</b> Network pruning (highly recommended)</a></li>
<li class="chapter" data-level="2.2.6" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#add-weights"><i class="fa fa-check"></i><b>2.2.6</b> Add Weights</a></li>
<li class="chapter" data-level="2.2.7" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#optional-annotate-with-tf-motifs"><i class="fa fa-check"></i><b>2.2.7</b> (Optional) Annotate with TF motifs</a></li>
<li class="chapter" data-level="2.2.8" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#calculate-tf-activity"><i class="fa fa-check"></i><b>2.2.8</b> Calculate TF activity</a></li>
<li class="chapter" data-level="2.2.9" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#perform-differential-activity"><i class="fa fa-check"></i><b>2.2.9</b> Perform differential activity</a></li>
<li class="chapter" data-level="2.2.10" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#visualize-the-results"><i class="fa fa-check"></i><b>2.2.10</b> Visualize the results</a></li>
<li class="chapter" data-level="2.2.11" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#geneset-enrichment"><i class="fa fa-check"></i><b>2.2.11</b> Geneset enrichment</a></li>
<li class="chapter" data-level="2.2.12" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#network-analysis"><i class="fa fa-check"></i><b>2.2.12</b> Network analysis</a></li>
<li class="chapter" data-level="2.2.13" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#differential-networks"><i class="fa fa-check"></i><b>2.2.13</b> Differential networks</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="multiome---archr-workflow.html"><a href="multiome---archr-workflow.html#session-info"><i class="fa fa-check"></i><b>2.3</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html"><i class="fa fa-check"></i><b>3</b> Hematopoeisis tutorial - MAE</a>
<ul>
<li class="chapter" data-level="3.1" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#data-preparation-1"><i class="fa fa-check"></i><b>3.1</b> Data preparation</a></li>
<li class="chapter" data-level="3.2" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#quick-start-1"><i class="fa fa-check"></i><b>3.2</b> Quick start</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#retrieve-bulk-tf-chip-seq-binding-sites-1"><i class="fa fa-check"></i><b>3.2.1</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="3.2.2" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#link-atacseq-peaks-to-target-genes"><i class="fa fa-check"></i><b>3.2.2</b> Link ATACseq peaks to target genes</a></li>
<li class="chapter" data-level="3.2.3" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#add-tf-motif-binding-to-peaks-1"><i class="fa fa-check"></i><b>3.2.3</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="3.2.4" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#generate-regulons-1"><i class="fa fa-check"></i><b>3.2.4</b> Generate regulons</a></li>
<li class="chapter" data-level="3.2.5" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#prune-network"><i class="fa fa-check"></i><b>3.2.5</b> Prune network</a></li>
<li class="chapter" data-level="3.2.6" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#add-weights-1"><i class="fa fa-check"></i><b>3.2.6</b> Add Weights</a></li>
<li class="chapter" data-level="3.2.7" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#calculate-tf-activity-1"><i class="fa fa-check"></i><b>3.2.7</b> Calculate TF activity</a></li>
<li class="chapter" data-level="3.2.8" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#differential-tf-activity-test"><i class="fa fa-check"></i><b>3.2.8</b> Differential TF activity test</a></li>
<li class="chapter" data-level="3.2.9" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#visualizing-tf-activities"><i class="fa fa-check"></i><b>3.2.9</b> Visualizing TF activities</a></li>
<li class="chapter" data-level="3.2.10" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#geneset-enrichment-1"><i class="fa fa-check"></i><b>3.2.10</b> Geneset enrichment</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="hematopoeisis-tutorial---mae.html"><a href="hematopoeisis-tutorial---mae.html#session-info-1"><i class="fa fa-check"></i><b>3.3</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html"><i class="fa fa-check"></i><b>4</b> Prostate cancer cells</a>
<ul>
<li class="chapter" data-level="4.1" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#data-preparation-2"><i class="fa fa-check"></i><b>4.1</b> Data preparation</a></li>
<li class="chapter" data-level="4.2" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#quick-start-2"><i class="fa fa-check"></i><b>4.2</b> Quick start</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#retrieve-bulk-tf-chip-seq-binding-sites-2"><i class="fa fa-check"></i><b>4.2.1</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="4.2.2" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#link-atac-seq-peaks-to-target-genes-1"><i class="fa fa-check"></i><b>4.2.2</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="4.2.3" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#add-tf-motif-binding-to-peaks-2"><i class="fa fa-check"></i><b>4.2.3</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="4.2.4" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#generate-regulons-2"><i class="fa fa-check"></i><b>4.2.4</b> Generate regulons</a></li>
<li class="chapter" data-level="4.2.5" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#calculate-tf-activity-2"><i class="fa fa-check"></i><b>4.2.5</b> Calculate TF activity</a></li>
<li class="chapter" data-level="4.2.6" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#perform-differential-activity-1"><i class="fa fa-check"></i><b>4.2.6</b> Perform differential activity</a></li>
<li class="chapter" data-level="4.2.7" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#visualize-the-results-1"><i class="fa fa-check"></i><b>4.2.7</b> Visualize the results</a></li>
<li class="chapter" data-level="4.2.8" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#geneset-enrichment-2"><i class="fa fa-check"></i><b>4.2.8</b> Geneset enrichment</a></li>
<li class="chapter" data-level="4.2.9" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#network-analysis-1"><i class="fa fa-check"></i><b>4.2.9</b> Network analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="prostate-cancer-cells.html"><a href="prostate-cancer-cells.html#session-info-2"><i class="fa fa-check"></i><b>4.3</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html"><i class="fa fa-check"></i><b>5</b> Single modality: gene expression</a>
<ul>
<li class="chapter" data-level="5.1" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#load-regulon"><i class="fa fa-check"></i><b>5.1</b> Load regulon</a></li>
<li class="chapter" data-level="5.2" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#load-scrna-seq-data"><i class="fa fa-check"></i><b>5.2</b> Load scRNA-seq data</a></li>
<li class="chapter" data-level="5.3" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#calculate-activity"><i class="fa fa-check"></i><b>5.3</b> Calculate activity</a></li>
<li class="chapter" data-level="5.4" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#perform-differential-activity-2"><i class="fa fa-check"></i><b>5.4</b> Perform differential activity</a></li>
<li class="chapter" data-level="5.5" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#visualize-activity"><i class="fa fa-check"></i><b>5.5</b> Visualize activity</a></li>
<li class="chapter" data-level="5.6" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#pathway-enrichment"><i class="fa fa-check"></i><b>5.6</b> Pathway enrichment</a></li>
<li class="chapter" data-level="5.7" data-path="single-modality-gene-expression.html"><a href="single-modality-gene-expression.html#session-info-3"><i class="fa fa-check"></i><b>5.7</b> Session Info</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Epiregulon documentation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="hematopoeisis-tutorial---mae" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> Hematopoeisis tutorial - MAE<a href="hematopoeisis-tutorial---mae.html#hematopoeisis-tutorial---mae" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In this chapter, we used a dataset from the ArchR <a href="https://www.archrproject.com/bookdown/index.html">tutorial</a>. Prior to using epiregulon, this dataset has been fully preprocessed in ArchR, and converted to a <code>MultiAssayExperiment</code> using <code>epireglon::archr2MAE</code>. The MAE object was uploaded to <code>scMultiome</code> for full reproducibility. In this dataset, scRNAseq and scATACseq were unpaired and integrated by the <code>ArchR::addGeneIntegrationMatrix</code> function.</p>
<div id="data-preparation-1" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Data preparation<a href="hematopoeisis-tutorial---mae.html#data-preparation-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Download the example dataset from <em><a href="https://bioconductor.org/packages/3.18/scMultiome">scMultiome</a></em> package</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="hematopoeisis-tutorial---mae.html#cb3-1" aria-hidden="true" tabindex="-1"></a>mae <span class="ot">&lt;-</span> scMultiome<span class="sc">::</span><span class="fu">hematopoiesis</span>()</span></code></pre></div>
<pre><code>## snapshotDate(): 2023-10-24</code></pre>
<pre><code>## see ?scMultiome and browseVignettes(&#39;scMultiome&#39;) for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="hematopoeisis-tutorial---mae.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load peak matrix</span></span>
<span id="cb7-2"><a href="hematopoeisis-tutorial---mae.html#cb7-2" aria-hidden="true" tabindex="-1"></a>PeakMatrix <span class="ot">&lt;-</span> mae[[<span class="st">&quot;PeakMatrix&quot;</span>]]</span>
<span id="cb7-3"><a href="hematopoeisis-tutorial---mae.html#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="hematopoeisis-tutorial---mae.html#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Load expression matrix</span></span>
<span id="cb7-5"><a href="hematopoeisis-tutorial---mae.html#cb7-5" aria-hidden="true" tabindex="-1"></a>GeneExpressionMatrix <span class="ot">&lt;-</span> mae[[<span class="st">&quot;GeneIntegrationMatrix&quot;</span>]]</span>
<span id="cb7-6"><a href="hematopoeisis-tutorial---mae.html#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="hematopoeisis-tutorial---mae.html#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Add gene symbols to rownames</span></span>
<span id="cb7-8"><a href="hematopoeisis-tutorial---mae.html#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(GeneExpressionMatrix) <span class="ot">&lt;-</span> <span class="fu">rowData</span>(GeneExpressionMatrix)<span class="sc">$</span>name</span>
<span id="cb7-9"><a href="hematopoeisis-tutorial---mae.html#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="hematopoeisis-tutorial---mae.html#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Transfer dimensionality reduction matrix to GeneExpression</span></span>
<span id="cb7-11"><a href="hematopoeisis-tutorial---mae.html#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="fu">reducedDim</span>(GeneExpressionMatrix, <span class="st">&quot;IterativeLSI&quot;</span>) <span class="ot">&lt;-</span> </span>
<span id="cb7-12"><a href="hematopoeisis-tutorial---mae.html#cb7-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reducedDim</span>(mae[[<span class="st">&#39;TileMatrix500&#39;</span>]], <span class="st">&quot;IterativeLSI&quot;</span>)</span>
<span id="cb7-13"><a href="hematopoeisis-tutorial---mae.html#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="fu">reducedDim</span>(GeneExpressionMatrix, <span class="st">&quot;UMAP&quot;</span>) <span class="ot">&lt;-</span> </span>
<span id="cb7-14"><a href="hematopoeisis-tutorial---mae.html#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reducedDim</span>(mae[[<span class="st">&#39;TileMatrix500&#39;</span>]], <span class="st">&quot;UMAP&quot;</span>)</span></code></pre></div>
<p>Visualize the data</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="hematopoeisis-tutorial---mae.html#cb8-1" aria-hidden="true" tabindex="-1"></a>scater<span class="sc">::</span><span class="fu">plotReducedDim</span>(GeneExpressionMatrix, </span>
<span id="cb8-2"><a href="hematopoeisis-tutorial---mae.html#cb8-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">dimred =</span> <span class="st">&quot;UMAP&quot;</span>, </span>
<span id="cb8-3"><a href="hematopoeisis-tutorial---mae.html#cb8-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">text_by =</span> <span class="st">&quot;Clusters2&quot;</span>, </span>
<span id="cb8-4"><a href="hematopoeisis-tutorial---mae.html#cb8-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">colour_by =</span> <span class="st">&quot;Clusters2&quot;</span>,</span>
<span id="cb8-5"><a href="hematopoeisis-tutorial---mae.html#cb8-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">point_size =</span> <span class="fl">0.3</span>,</span>
<span id="cb8-6"><a href="hematopoeisis-tutorial---mae.html#cb8-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">point_alpha =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
</div>
<div id="quick-start-1" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Quick start<a href="hematopoeisis-tutorial---mae.html#quick-start-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="retrieve-bulk-tf-chip-seq-binding-sites-1" class="section level3 hasAnchor" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Retrieve bulk TF ChIP-seq binding sites<a href="hematopoeisis-tutorial---mae.html#retrieve-bulk-tf-chip-seq-binding-sites-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First, we retrieve the information of TF binding sites collected from Cistrome and ENCODE ChIP-seq. Currently, human genomes hg19 and hg38 and mouse genome mm10 are available</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="hematopoeisis-tutorial---mae.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(epiregulon)</span>
<span id="cb9-2"><a href="hematopoeisis-tutorial---mae.html#cb9-2" aria-hidden="true" tabindex="-1"></a>grl <span class="ot">&lt;-</span> <span class="fu">getTFMotifInfo</span>(<span class="at">genome =</span> <span class="st">&quot;hg19&quot;</span>)</span></code></pre></div>
<pre><code>## snapshotDate(): 2023-10-24</code></pre>
<pre><code>## see ?scMultiome and browseVignettes(&#39;scMultiome&#39;) for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="hematopoeisis-tutorial---mae.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(grl)</span></code></pre></div>
<pre><code>## GRangesList object of length 6:
## $`5-hmC`
## GRanges object with 22860 ranges and 0 metadata columns:
##           seqnames        ranges strand
##              &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt;
##       [1]     chr1   10001-10685      *
##       [2]     chr1   13362-13694      *
##       [3]     chr1   29631-29989      *
##       [4]     chr1   40454-40754      *
##       [5]     chr1 135395-135871      *
##       ...      ...           ...    ...
##   [22856]     chrM   15303-15326      *
##   [22857]     chrM   15328-16172      *
##   [22858]     chrM   16174-16183      *
##   [22859]     chrM   16186-16224      *
##   [22860]     chrM   16226-16492      *
##   -------
##   seqinfo: 25 sequences from an unspecified genome; no seqlengths
## 
## ...
## &lt;5 more elements&gt;</code></pre>
</div>
<div id="link-atacseq-peaks-to-target-genes" class="section level3 hasAnchor" number="3.2.2">
<h3><span class="header-section-number">3.2.2</span> Link ATACseq peaks to target genes<a href="hematopoeisis-tutorial---mae.html#link-atacseq-peaks-to-target-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Next, we compute peak to gene correlations using a custom algorithm that has similar performance to ArchR’s P2G function. Wherever possible, use a multidimensional dimensionality reduction matrix such as LSI or PCA instead of UMAP or TSNE since the former provides a more accurate estimate of cell similarity.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="hematopoeisis-tutorial---mae.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1010</span>)</span>
<span id="cb15-2"><a href="hematopoeisis-tutorial---mae.html#cb15-2" aria-hidden="true" tabindex="-1"></a>p2g <span class="ot">&lt;-</span> <span class="fu">calculateP2G</span>(<span class="at">peakMatrix =</span> PeakMatrix, </span>
<span id="cb15-3"><a href="hematopoeisis-tutorial---mae.html#cb15-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">expMatrix =</span> GeneExpressionMatrix, </span>
<span id="cb15-4"><a href="hematopoeisis-tutorial---mae.html#cb15-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">exp_assay =</span> <span class="st">&quot;normalizedCounts&quot;</span>,</span>
<span id="cb15-5"><a href="hematopoeisis-tutorial---mae.html#cb15-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">reducedDim =</span> <span class="fu">reducedDim</span>(GeneExpressionMatrix, <span class="st">&quot;IterativeLSI&quot;</span>))</span></code></pre></div>
<pre><code>## Using epiregulon to compute peak to gene links...</code></pre>
<pre><code>## performing k means clustering to form metacells</code></pre>
<pre><code>## Computing correlation</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="hematopoeisis-tutorial---mae.html#cb19-1" aria-hidden="true" tabindex="-1"></a>p2g</span></code></pre></div>
<pre><code>## DataFrame with 22004 rows and 8 columns
##         idxATAC         chr     start       end    idxRNA    target Correlation  distance
##       &lt;integer&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;   &lt;array&gt;    &lt;matrix&gt; &lt;integer&gt;
## 1             7        chr1    801002    801502         2 LINC00115    0.895735     36099
## 2             8        chr1    805039    805539         6    KLHL17    0.743856     88427
## 3            24        chr1    894453    894953         6    KLHL17    0.766909         0
## 4            24        chr1    894453    894953         9     ISG15    0.634559     51893
## 5            25        chr1    894960    895460         6    KLHL17    0.604835         0
## ...         ...         ...       ...       ...       ...       ...         ...       ...
## 22000    146402       chr22  51020398  51020898     12089      ARSA    0.522949     45503
## 22001    146403       chr22  51021154  51021654     12078      LMF2    0.633179     73018
## 22002    146403       chr22  51021154  51021654     12089      ARSA    0.675409     44747
## 22003    146412       chr22  51110826  51111326     12090    SHANK3    0.541264         0
## 22004    146421       chr22  51213512  51214012     12091       ACR    0.510150     36660</code></pre>
</div>
<div id="add-tf-motif-binding-to-peaks-1" class="section level3 hasAnchor" number="3.2.3">
<h3><span class="header-section-number">3.2.3</span> Add TF motif binding to peaks<a href="hematopoeisis-tutorial---mae.html#add-tf-motif-binding-to-peaks-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The next step is to add the TF motif binding information by overlapping the regions of the peak matrix with the bulk chip-seq database.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="hematopoeisis-tutorial---mae.html#cb21-1" aria-hidden="true" tabindex="-1"></a>overlap <span class="ot">&lt;-</span> <span class="fu">addTFMotifInfo</span>(<span class="at">grl =</span> grl, <span class="at">p2g =</span> p2g, <span class="at">peakMatrix =</span> PeakMatrix)</span></code></pre></div>
<pre><code>## Computing overlap...</code></pre>
<pre><code>## Success!</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="hematopoeisis-tutorial---mae.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(overlap)</span></code></pre></div>
<pre><code>##      idxATAC idxTF      tf
## 1018       7    35    ARNT
## 1019       7    50    ATF2
## 1020       7    55    ATF7
## 1021       7    76    BCL6
## 1022       7    80    BCOR
## 1023       7    82 BHLHE40</code></pre>
</div>
<div id="generate-regulons-1" class="section level3 hasAnchor" number="3.2.4">
<h3><span class="header-section-number">3.2.4</span> Generate regulons<a href="hematopoeisis-tutorial---mae.html#generate-regulons-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A long format dataframe, representing the inferred regulons, is then generated. The dataframe consists of three columns:</p>
<ul>
<li>tf (transcription factor)</li>
<li>target gene</li>
<li>peak to gene correlation between tf and target gene</li>
</ul>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="hematopoeisis-tutorial---mae.html#cb26-1" aria-hidden="true" tabindex="-1"></a>regulon <span class="ot">&lt;-</span> <span class="fu">getRegulon</span>(p2g, overlap, <span class="at">aggregate=</span><span class="cn">FALSE</span>)</span>
<span id="cb26-2"><a href="hematopoeisis-tutorial---mae.html#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(regulon)</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 10 columns
##     idxATAC         chr     start       end    idxRNA      target  distance     idxTF          tf     corr
##   &lt;integer&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt; &lt;matrix&gt;
## 1         7        chr1    801002    801502         2   LINC00115     36099        35        ARNT 0.895735
## 2         7        chr1    801002    801502         2   LINC00115     36099        50        ATF2 0.895735
## 3         7        chr1    801002    801502         2   LINC00115     36099        55        ATF7 0.895735
## 4         7        chr1    801002    801502         2   LINC00115     36099        76        BCL6 0.895735
## 5         7        chr1    801002    801502         2   LINC00115     36099        80        BCOR 0.895735
## 6         7        chr1    801002    801502         2   LINC00115     36099        82     BHLHE40 0.895735</code></pre>
</div>
<div id="prune-network" class="section level3 hasAnchor" number="3.2.5">
<h3><span class="header-section-number">3.2.5</span> Prune network<a href="hematopoeisis-tutorial---mae.html#prune-network" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Epiregulon prunes the network by performing tests of independence on the observed number of cells jointly expressing transcription factor (TF), regulatory element (RE) and target gene (TG) vs the expected number of cells if TF/RE and TG are independently expressed. We implement two tests, the binomial test and the chi-square test. In the binomial test, the expected probability is P(TF, RE) * P(TG), and the number of trials is the total number of cells, and the observed successes is the number of cells jointly expressing all three elements. In the chi-square test, the expected probability for having all 3 elements active is also P(TF, RE) * P(TG) and the probability otherwise is 1- P(TF, RE) * P(TG). The observed cell count for the active category is the number of cells jointly expressing all three elements, and the cell count for the inactive category is n - n_triple.</p>
<p>We calculate cluster-specific p-values if users supply cluster labels. This is useful if we are interested in cluster-specific networks. The pruned regulons can then be used to visualize differential networks for transcription factors of interest. See section on differential networks.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="hematopoeisis-tutorial---mae.html#cb28-1" aria-hidden="true" tabindex="-1"></a>pruned.regulon <span class="ot">&lt;-</span> <span class="fu">pruneRegulon</span>(<span class="at">expMatrix =</span> GeneExpressionMatrix,</span>
<span id="cb28-2"><a href="hematopoeisis-tutorial---mae.html#cb28-2" aria-hidden="true" tabindex="-1"></a>                               <span class="at">exp_assay =</span> <span class="st">&quot;normalizedCounts&quot;</span>,</span>
<span id="cb28-3"><a href="hematopoeisis-tutorial---mae.html#cb28-3" aria-hidden="true" tabindex="-1"></a>                               <span class="at">peakMatrix =</span> PeakMatrix,</span>
<span id="cb28-4"><a href="hematopoeisis-tutorial---mae.html#cb28-4" aria-hidden="true" tabindex="-1"></a>                               <span class="at">peak_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb28-5"><a href="hematopoeisis-tutorial---mae.html#cb28-5" aria-hidden="true" tabindex="-1"></a>                               <span class="at">regulon =</span> regulon,</span>
<span id="cb28-6"><a href="hematopoeisis-tutorial---mae.html#cb28-6" aria-hidden="true" tabindex="-1"></a>                               <span class="at">prune_value =</span> <span class="st">&quot;pval&quot;</span>,</span>
<span id="cb28-7"><a href="hematopoeisis-tutorial---mae.html#cb28-7" aria-hidden="true" tabindex="-1"></a>                               <span class="at">regulon_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb28-8"><a href="hematopoeisis-tutorial---mae.html#cb28-8" aria-hidden="true" tabindex="-1"></a>                               <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>Clusters2,</span>
<span id="cb28-9"><a href="hematopoeisis-tutorial---mae.html#cb28-9" aria-hidden="true" tabindex="-1"></a>                               <span class="at">exp_cutoff =</span> <span class="cn">NULL</span>)</span></code></pre></div>
<pre><code>## pruning network with chi.sq tests using a regulon cutoff of pval&lt;0.05</code></pre>
<pre><code>## pruning regulons</code></pre>
</div>
<div id="add-weights-1" class="section level3 hasAnchor" number="3.2.6">
<h3><span class="header-section-number">3.2.6</span> Add Weights<a href="hematopoeisis-tutorial---mae.html#add-weights-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>While the `pruneRegulon’ function provides statistics on the joint occurrence of TF-RE-TG, we would like to further estimate the strength of regulation. Biologically, this can be interpreted as the magnitude of gene expression changes induced by transcription factor activity. Epiregulon estimates the regulatory potential using one of the four measures: 1) correlation between TF and target gene expression, 2) mutual information between the TF and target gene expression, 3) Wilcoxon test statistics of target gene expression in cells jointly expressing all 3 elements vs cells that do not, or 4) log 2 fold difference of target gene expression in cells jointly expressing all 3 elements vs cells that do not.</p>
<p>Three measures (correlation, Wilcoxon statistics and log 2 fold difference) give both the magnitude and directionality of changes whereas mutational information is always positive. The correlation and mutual information statistics are computed on grouped pseudobulks by user-supplied cluster labels and yield a single weight across all clusters per each TF-RE-target triplet. In contrast, Wilcoxon and log fold change methods group cells based on the joint expression of TF, RE and TG in each single cell or in cell aggregates. Cell aggregation uses a default value of 10 cells and can help overcome sparsity and speed up computation. If cluster labels are provided, we can obtain weights of individual clusters and all cells combined. In this example, we apply Wilcoxon test on cell aggregates of 10 cells.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="hematopoeisis-tutorial---mae.html#cb31-1" aria-hidden="true" tabindex="-1"></a>regulon.w <span class="ot">&lt;-</span> <span class="fu">addWeights</span>(<span class="at">regulon =</span> pruned.regulon,</span>
<span id="cb31-2"><a href="hematopoeisis-tutorial---mae.html#cb31-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">expMatrix  =</span> GeneExpressionMatrix,</span>
<span id="cb31-3"><a href="hematopoeisis-tutorial---mae.html#cb31-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">exp_assay  =</span> <span class="st">&quot;normalizedCounts&quot;</span>,</span>
<span id="cb31-4"><a href="hematopoeisis-tutorial---mae.html#cb31-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">peakMatrix =</span> PeakMatrix,</span>
<span id="cb31-5"><a href="hematopoeisis-tutorial---mae.html#cb31-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">peak_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb31-6"><a href="hematopoeisis-tutorial---mae.html#cb31-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>Clusters2,</span>
<span id="cb31-7"><a href="hematopoeisis-tutorial---mae.html#cb31-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">aggregateCells =</span> <span class="cn">TRUE</span>,</span>
<span id="cb31-8"><a href="hematopoeisis-tutorial---mae.html#cb31-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">method =</span> <span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb31-9"><a href="hematopoeisis-tutorial---mae.html#cb31-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">useDim =</span> <span class="st">&quot;IterativeLSI&quot;</span>)</span></code></pre></div>
<pre><code>## adding weights using wilcoxon...</code></pre>
<pre><code>## performing pseudobulk using an average of 10 cells</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="hematopoeisis-tutorial---mae.html#cb34-1" aria-hidden="true" tabindex="-1"></a>regulon.w</span></code></pre></div>
<pre><code>## DataFrame with 2292760 rows and 14 columns
##           idxATAC         chr     start       end    idxRNA      target  distance     idxTF          tf     corr
##         &lt;integer&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt; &lt;character&gt; &lt;matrix&gt;
## 1              26        chr1    895689    896189         6      KLHL17         0         5        ADNP 0.595722
## 2              26        chr1    895689    896189         9       ISG15     50657         5        ADNP 0.569188
## 3             463        chr1   3709928   3710428        66       SMIM1     20394         5        ADNP 0.610804
## 4             732        chr1   8021367   8021867        95        UTS2     46072         5        ADNP 0.593762
## 5             733        chr1   8021996   8022496        95        UTS2     46701         5        ADNP 0.591618
## ...           ...         ...       ...       ...       ...         ...       ...       ...         ...      ...
## 2292756    145986       chr22  46645960  46646460     12045       PPARA     99261      1557        ZXDC 0.549096
## 2292757    145987       chr22  46646470  46646970     12045       PPARA     99771      1557        ZXDC 0.554923
## 2292758    146374       chr22  50963929  50964429     12078        LMF2     15793      1557        ZXDC 0.689212
## 2292759    146403       chr22  51021154  51021654     12078        LMF2     73018      1557        ZXDC 0.633179
## 2292760    146403       chr22  51021154  51021654     12089        ARSA     44747      1557        ZXDC 0.675409
##                                          pval                              stats                qval
##                                      &lt;matrix&gt;                           &lt;matrix&gt;            &lt;matrix&gt;
## 1       2.12324e-01:2.36663e-01:0.6405757:...      1.55551: 1.40036:0.217990:... 1.00000e+00:1:1:...
## 2       9.52210e-05:2.84547e-02:0.4447283:...     15.22917: 4.80030:0.584052:... 1.00000e+00:1:1:...
## 3       2.88106e-04:1.00000e+00:0.4032360:...     13.14618: 0.00000:0.698655:... 1.00000e+00:1:1:...
## 4       2.71837e-29:3.79230e-06:0.0117417:...    126.24453:21.36706:6.349467:... 8.01807e-23:1:1:...
## 5       4.45015e-06:6.06719e-03:0.2144717:...     21.06047: 7.53024:1.540982:... 1.00000e+00:1:1:...
## ...                                       ...                                ...                 ...
## 2292756     0.852623645:0.424559:0.337005:...  0.0345112:0.6376622:0.9217923:...           1:1:1:...
## 2292757     0.242104230:0.800758:0.304168:...  1.3682997:0.0636888:1.0558312:...           1:1:1:...
## 2292758     0.833286943:0.390888:0.725549:...  0.0443051:0.7361788:0.1232378:...           1:1:1:...
## 2292759     0.000372059:0.308149:0.868956:... 12.6675377:1.0385922:0.0272198:...           1:1:1:...
## 2292760     0.026174461:0.017739:0.698380:...  4.9444569:5.6217414:0.1501623:...           1:1:1:...
##                                       weight
##                                     &lt;matrix&gt;
## 1          0.107149: 0.4231457:0.2368892:...
## 2          0.180432: 0.3966991:0.2125929:...
## 3          0.123088:-0.0827606:0.0872641:...
## 4          0.150402: 0.2869753:0.3338592:...
## 5          0.136803: 0.0152085:0.2251617:...
## ...                                      ...
## 2292756 0.000726069:-0.0161671:0.0000000:...
## 2292757 0.043667547: 0.0986710:0.0000000:...
## 2292758 0.103316818: 0.3103894:0.0000000:...
## 2292759 0.104718509: 0.3815615:0.0727201:...
## 2292760 0.087638797:-0.1719992:0.0000000:...</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="hematopoeisis-tutorial---mae.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co">#saveRDS(regulon, &quot;/gstore/project/lineage/manuscript/epiregulon/OUTPUT/regulon.rds&quot;)</span></span>
<span id="cb36-2"><a href="hematopoeisis-tutorial---mae.html#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="co">#saveRDS(regulon.w, &quot;/gstore/project/lineage/manuscript/epiregulon/OUTPUT/regulon.w.rds&quot;)</span></span></code></pre></div>
</div>
<div id="calculate-tf-activity-1" class="section level3 hasAnchor" number="3.2.7">
<h3><span class="header-section-number">3.2.7</span> Calculate TF activity<a href="hematopoeisis-tutorial---mae.html#calculate-tf-activity-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Finally, the activities for a specific TF in each cell are computed by averaging the weighted expressions of target genes linked to the TF weighted.
<span class="math display">\[y=\frac{1}{n}\sum_{i=1}^{n} x_i * weight_i\]</span>
where <span class="math inline">\(y\)</span> is the activity of a TF for a cell
<span class="math inline">\(n\)</span> is the total number of targets for a TF
<span class="math inline">\(x_i\)</span> is the log count expression of target i where i in {1,2,…,n}
<span class="math inline">\(weight_i\)</span> is the weight of TF and target i</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="hematopoeisis-tutorial---mae.html#cb37-1" aria-hidden="true" tabindex="-1"></a>score.combine <span class="ot">&lt;-</span> <span class="fu">calculateActivity</span>(<span class="at">expMatrix =</span> GeneExpressionMatrix, </span>
<span id="cb37-2"><a href="hematopoeisis-tutorial---mae.html#cb37-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">regulon =</span> regulon.w, </span>
<span id="cb37-3"><a href="hematopoeisis-tutorial---mae.html#cb37-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">mode =</span> <span class="st">&quot;weight&quot;</span>, </span>
<span id="cb37-4"><a href="hematopoeisis-tutorial---mae.html#cb37-4" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">&quot;weightedMean&quot;</span>, </span>
<span id="cb37-5"><a href="hematopoeisis-tutorial---mae.html#cb37-5" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">exp_assay =</span> <span class="st">&quot;normalizedCounts&quot;</span>)</span></code></pre></div>
<pre><code>## calculating TF activity from regulon using weightedmean</code></pre>
<pre><code>## Warning in calculateActivity(expMatrix = GeneExpressionMatrix, regulon = regulon.w, : The weight column contains
## multiple subcolumns but no cluster information was provided. Using first column to compute activity...</code></pre>
<pre><code>## aggregating regulons...</code></pre>
<pre><code>## creating weight matrix...</code></pre>
<pre><code>## calculating activity scores...</code></pre>
<pre><code>## normalize by the number of targets...</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="hematopoeisis-tutorial---mae.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(score.combine[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code></pre></div>
<pre><code>## 5 x 5 sparse Matrix of class &quot;dgCMatrix&quot;
##       scATAC_BMMC_R1#TTATGTCAGTGATTAG-1 scATAC_BMMC_R1#AAGATAGTCACCGCGA-1 scATAC_BMMC_R1#GCATTGAAGATTCCGT-1
## ADNP                          0.3247975                         0.3566263                         0.2619925
## AEBP2                         0.1181665                         0.3533748                         0.2303255
## AFF1                          0.2554757                         0.2769220                         0.2502943
## AFF4                          0.3477904                         0.4799658                         0.3094043
## AGO1                          0.2357290                         0.3934343                         0.2176925
##       scATAC_BMMC_R1#TATGTTCAGGGTTCCC-1 scATAC_BMMC_R1#AGTTACGAGAACGTCG-1
## ADNP                          0.3009295                         0.3406015
## AEBP2                         0.1783639                         0.1254906
## AFF1                          0.2478131                         0.2491399
## AFF4                          0.3708383                         0.3498188
## AGO1                          0.2753658                         0.2441136</code></pre>
</div>
<div id="differential-tf-activity-test" class="section level3 hasAnchor" number="3.2.8">
<h3><span class="header-section-number">3.2.8</span> Differential TF activity test<a href="hematopoeisis-tutorial---mae.html#differential-tf-activity-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can next determine which TFs exhibit differential activities across cell clusters/groups via the <code>findDifferentialActivity</code> function. This function depends on <code>findMarkers</code> function from <em><a href="https://bioconductor.org/packages/3.18/scran">scran</a></em> package.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="hematopoeisis-tutorial---mae.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(epiregulon.extra)</span>
<span id="cb46-2"><a href="hematopoeisis-tutorial---mae.html#cb46-2" aria-hidden="true" tabindex="-1"></a>markers  <span class="ot">&lt;-</span> <span class="fu">findDifferentialActivity</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb46-3"><a href="hematopoeisis-tutorial---mae.html#cb46-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">groups =</span> GeneExpressionMatrix<span class="sc">$</span>Clusters2, </span>
<span id="cb46-4"><a href="hematopoeisis-tutorial---mae.html#cb46-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">pval.type =</span> <span class="st">&quot;some&quot;</span>, </span>
<span id="cb46-5"><a href="hematopoeisis-tutorial---mae.html#cb46-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">direction =</span> <span class="st">&quot;up&quot;</span>, </span>
<span id="cb46-6"><a href="hematopoeisis-tutorial---mae.html#cb46-6" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">test.type =</span> <span class="st">&quot;t&quot;</span>)</span></code></pre></div>
<p>getSigGenes compiles the different test results into a single dataframe and enables user to supply their desired cutoffs for significance and variable to order by.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="hematopoeisis-tutorial---mae.html#cb47-1" aria-hidden="true" tabindex="-1"></a>markers.sig <span class="ot">&lt;-</span> <span class="fu">getSigGenes</span>(markers, <span class="at">topgenes =</span> <span class="dv">3</span> )</span></code></pre></div>
<pre><code>## Using a logFC cutoff of 0.1 for class B</code></pre>
<pre><code>## Using a logFC cutoff of 0.4 for class CD4.M</code></pre>
<pre><code>## Using a logFC cutoff of 0.2 for class CD4.N</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class CLP</code></pre>
<pre><code>## Using a logFC cutoff of 0.2 for class Erythroid</code></pre>
<pre><code>## Using a logFC cutoff of 0.2 for class GMP</code></pre>
<pre><code>## Using a logFC cutoff of 0.4 for class Mono</code></pre>
<pre><code>## Using a logFC cutoff of 0.4 for class NK</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class PreB</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class Progenitor</code></pre>
<pre><code>## Using a logFC cutoff of 0.1 for class pDC</code></pre>
</div>
<div id="visualizing-tf-activities" class="section level3 hasAnchor" number="3.2.9">
<h3><span class="header-section-number">3.2.9</span> Visualizing TF activities<a href="hematopoeisis-tutorial---mae.html#visualizing-tf-activities" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Epiregulon also provides multiple options for visualizing the inferred TF activities by reduced dimensional space</p>
<p>tSNE or UMAP plots:</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="hematopoeisis-tutorial---mae.html#cb59-1" aria-hidden="true" tabindex="-1"></a>tfs_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;EBF1&quot;</span>,<span class="st">&quot;GATA3&quot;</span>, <span class="st">&quot;SPI1&quot;</span>, <span class="st">&quot;GATA2&quot;</span>)</span>
<span id="cb59-2"><a href="hematopoeisis-tutorial---mae.html#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotActivityDim</span>(<span class="at">sce =</span> GeneExpressionMatrix, </span>
<span id="cb59-3"><a href="hematopoeisis-tutorial---mae.html#cb59-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">activity_matrix =</span> score.combine[tfs_interest,], </span>
<span id="cb59-4"><a href="hematopoeisis-tutorial---mae.html#cb59-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">tf =</span> tfs_interest, </span>
<span id="cb59-5"><a href="hematopoeisis-tutorial---mae.html#cb59-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">dimtype =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb59-6"><a href="hematopoeisis-tutorial---mae.html#cb59-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb59-7"><a href="hematopoeisis-tutorial---mae.html#cb59-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">ncol=</span><span class="dv">2</span>,</span>
<span id="cb59-8"><a href="hematopoeisis-tutorial---mae.html#cb59-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">point_size=</span><span class="fl">0.1</span>,</span>
<span id="cb59-9"><a href="hematopoeisis-tutorial---mae.html#cb59-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">rasterise =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-13-1.png" width="672" />
We can compare the activity with gene expression of the same TFs.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="hematopoeisis-tutorial---mae.html#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co">#options(ggrastr.default.dpi=300)</span></span>
<span id="cb60-2"><a href="hematopoeisis-tutorial---mae.html#cb60-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotActivityDim</span>(<span class="at">sce =</span> GeneExpressionMatrix, </span>
<span id="cb60-3"><a href="hematopoeisis-tutorial---mae.html#cb60-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">activity_matrix =</span> <span class="fu">assay</span>(GeneExpressionMatrix, <span class="st">&quot;normalizedCounts&quot;</span>)[tfs_interest,], </span>
<span id="cb60-4"><a href="hematopoeisis-tutorial---mae.html#cb60-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">tf =</span> tfs_interest, </span>
<span id="cb60-5"><a href="hematopoeisis-tutorial---mae.html#cb60-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">dimtype =</span> <span class="st">&quot;UMAP&quot;</span>,</span>
<span id="cb60-6"><a href="hematopoeisis-tutorial---mae.html#cb60-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb60-7"><a href="hematopoeisis-tutorial---mae.html#cb60-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">ncol=</span><span class="dv">2</span>,</span>
<span id="cb60-8"><a href="hematopoeisis-tutorial---mae.html#cb60-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">legend.label =</span> <span class="st">&quot;Gex&quot;</span>,</span>
<span id="cb60-9"><a href="hematopoeisis-tutorial---mae.html#cb60-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&quot;grey&quot;</span>,<span class="st">&quot;blue&quot;</span>),</span>
<span id="cb60-10"><a href="hematopoeisis-tutorial---mae.html#cb60-10" aria-hidden="true" tabindex="-1"></a>                <span class="co">#limit = c(0,2),</span></span>
<span id="cb60-11"><a href="hematopoeisis-tutorial---mae.html#cb60-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">point_size=</span><span class="fl">0.1</span>,</span>
<span id="cb60-12"><a href="hematopoeisis-tutorial---mae.html#cb60-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">rasterise =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p>We can also plot violin plot to visualize TF activity.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="hematopoeisis-tutorial---mae.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotActivityViolin</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb61-2"><a href="hematopoeisis-tutorial---mae.html#cb61-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">tf =</span> tfs_interest, </span>
<span id="cb61-3"><a href="hematopoeisis-tutorial---mae.html#cb61-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>Clusters2,</span>
<span id="cb61-4"><a href="hematopoeisis-tutorial---mae.html#cb61-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb61-5"><a href="hematopoeisis-tutorial---mae.html#cb61-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-15-1.png" width="672" />
We plot violin plot to visualize TF gene expression.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="hematopoeisis-tutorial---mae.html#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotActivityViolin</span>(<span class="at">activity_matrix =</span> <span class="fu">assay</span>(GeneExpressionMatrix, <span class="st">&quot;normalizedCounts&quot;</span>)[tfs_interest,],  </span>
<span id="cb62-2"><a href="hematopoeisis-tutorial---mae.html#cb62-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">tf =</span> tfs_interest, </span>
<span id="cb62-3"><a href="hematopoeisis-tutorial---mae.html#cb62-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>Clusters2,</span>
<span id="cb62-4"><a href="hematopoeisis-tutorial---mae.html#cb62-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb62-5"><a href="hematopoeisis-tutorial---mae.html#cb62-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">ncol=</span><span class="dv">2</span>,</span>
<span id="cb62-6"><a href="hematopoeisis-tutorial---mae.html#cb62-6" aria-hidden="true" tabindex="-1"></a>                   <span class="at">legend.label =</span> <span class="st">&quot;gene expression&quot;</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<p>We can visualize the different TFs in a bubble plot:</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="hematopoeisis-tutorial---mae.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotBubble</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb63-2"><a href="hematopoeisis-tutorial---mae.html#cb63-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">tf =</span> tfs_interest, </span>
<span id="cb63-3"><a href="hematopoeisis-tutorial---mae.html#cb63-3" aria-hidden="true" tabindex="-1"></a>           GeneExpressionMatrix<span class="sc">$</span>Clusters2, </span>
<span id="cb63-4"><a href="hematopoeisis-tutorial---mae.html#cb63-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">bubblesize =</span> <span class="st">&quot;FDR&quot;</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p>We visualize the top differential TFs based on activity.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="hematopoeisis-tutorial---mae.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotBubble</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb64-2"><a href="hematopoeisis-tutorial---mae.html#cb64-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">tf =</span> markers.sig<span class="sc">$</span>tf, </span>
<span id="cb64-3"><a href="hematopoeisis-tutorial---mae.html#cb64-3" aria-hidden="true" tabindex="-1"></a>           GeneExpressionMatrix<span class="sc">$</span>Clusters2, </span>
<span id="cb64-4"><a href="hematopoeisis-tutorial---mae.html#cb64-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">bubblesize =</span> <span class="st">&quot;FDR&quot;</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
</div>
<div id="geneset-enrichment-1" class="section level3 hasAnchor" number="3.2.10">
<h3><span class="header-section-number">3.2.10</span> Geneset enrichment<a href="hematopoeisis-tutorial---mae.html#geneset-enrichment-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Sometimes we are interested to know what pathways are enriched in the regulon of a particular TF. We can perform geneset enrichment using the enricher function from <a href="http://yulab-smu.top/biomedical-knowledge-mining-book/universal-api.html">clusterProfiler</a>.</p>
<p>Here we first download Hallmark and C2 signatures from hallmark and then perform gene set enrichment of the known lineage factors. As expected, EBF1 is consistent with a B cell lineage factor, GATA3 and RUNX3 with lymphoid lineage and SPI1 with myeloid lineage.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="hematopoeisis-tutorial---mae.html#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co">#retrieve genesets</span></span>
<span id="cb65-2"><a href="hematopoeisis-tutorial---mae.html#cb65-2" aria-hidden="true" tabindex="-1"></a>H <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">getGenesets</span>(<span class="at">org =</span> <span class="st">&quot;hsa&quot;</span>, </span>
<span id="cb65-3"><a href="hematopoeisis-tutorial---mae.html#cb65-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">db =</span> <span class="st">&quot;msigdb&quot;</span>, </span>
<span id="cb65-4"><a href="hematopoeisis-tutorial---mae.html#cb65-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">cat =</span> <span class="st">&quot;H&quot;</span>, </span>
<span id="cb65-5"><a href="hematopoeisis-tutorial---mae.html#cb65-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">gene.id.type =</span> <span class="st">&quot;SYMBOL&quot;</span> )</span></code></pre></div>
<pre><code>## Using cached version from 2023-11-17 18:29:09</code></pre>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="hematopoeisis-tutorial---mae.html#cb67-1" aria-hidden="true" tabindex="-1"></a>C2 <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">getGenesets</span>(<span class="at">org =</span> <span class="st">&quot;hsa&quot;</span>, </span>
<span id="cb67-2"><a href="hematopoeisis-tutorial---mae.html#cb67-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">db =</span> <span class="st">&quot;msigdb&quot;</span>, </span>
<span id="cb67-3"><a href="hematopoeisis-tutorial---mae.html#cb67-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">cat =</span> <span class="st">&quot;C2&quot;</span>,</span>
<span id="cb67-4"><a href="hematopoeisis-tutorial---mae.html#cb67-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">gene.id.type =</span> <span class="st">&quot;SYMBOL&quot;</span> )</span></code></pre></div>
<pre><code>## Using cached version from 2023-11-12 20:06:04</code></pre>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="hematopoeisis-tutorial---mae.html#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co">#combine genesets and convert genesets to be compatible with enricher</span></span>
<span id="cb69-2"><a href="hematopoeisis-tutorial---mae.html#cb69-2" aria-hidden="true" tabindex="-1"></a>gs <span class="ot">&lt;-</span> <span class="fu">c</span>(H, C2)</span>
<span id="cb69-3"><a href="hematopoeisis-tutorial---mae.html#cb69-3" aria-hidden="true" tabindex="-1"></a>gs.list <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind,<span class="fu">lapply</span>(<span class="fu">names</span>(gs), <span class="cf">function</span>(x) </span>
<span id="cb69-4"><a href="hematopoeisis-tutorial---mae.html#cb69-4" aria-hidden="true" tabindex="-1"></a>  {<span class="fu">data.frame</span>(<span class="at">gs=</span>x, <span class="at">genes=</span>gs[[x]])}))</span>
<span id="cb69-5"><a href="hematopoeisis-tutorial---mae.html#cb69-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-6"><a href="hematopoeisis-tutorial---mae.html#cb69-6" aria-hidden="true" tabindex="-1"></a>enrichresults <span class="ot">&lt;-</span> <span class="fu">regulonEnrich</span>(<span class="at">TF =</span> tfs_interest, </span>
<span id="cb69-7"><a href="hematopoeisis-tutorial---mae.html#cb69-7" aria-hidden="true" tabindex="-1"></a>                               <span class="at">regulon =</span> regulon.w, </span>
<span id="cb69-8"><a href="hematopoeisis-tutorial---mae.html#cb69-8" aria-hidden="true" tabindex="-1"></a>                               <span class="at">weight =</span> <span class="st">&quot;weight&quot;</span>,</span>
<span id="cb69-9"><a href="hematopoeisis-tutorial---mae.html#cb69-9" aria-hidden="true" tabindex="-1"></a>                               <span class="at">weight_cutoff =</span> <span class="dv">0</span>, </span>
<span id="cb69-10"><a href="hematopoeisis-tutorial---mae.html#cb69-10" aria-hidden="true" tabindex="-1"></a>                               <span class="at">genesets =</span> gs.list)</span></code></pre></div>
<pre><code>## EBF1</code></pre>
<pre><code>## GATA3</code></pre>
<pre><code>## SPI1</code></pre>
<pre><code>## GATA2</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="hematopoeisis-tutorial---mae.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co">#plot results</span></span>
<span id="cb74-2"><a href="hematopoeisis-tutorial---mae.html#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="fu">enrichPlot</span>(<span class="at">results =</span> enrichresults, <span class="at">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/enrichment_hematopoeisis-1.png" width="1152" />
## Differential Network analysis</p>
<p>In addition to looking at the summed TF activity, a second approach to investigate differential TF activity is to compare and contrast target genes or network topology. In this example, we know that EBF1 is a B cell lineage factor. If we plot the differential network of EBF1 using the regulon with cluster-specific weights, we can see that EBF1 has many more targets in PreB cells than it has in CD4 memory cells.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="hematopoeisis-tutorial---mae.html#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDiffNetwork</span>(regulon.w,</span>
<span id="cb75-2"><a href="hematopoeisis-tutorial---mae.html#cb75-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">cutoff =</span> <span class="dv">0</span>,</span>
<span id="cb75-3"><a href="hematopoeisis-tutorial---mae.html#cb75-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">tf =</span> <span class="fu">c</span>(<span class="st">&quot;EBF1&quot;</span>),</span>
<span id="cb75-4"><a href="hematopoeisis-tutorial---mae.html#cb75-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">weight =</span> <span class="st">&quot;weight&quot;</span>,</span>
<span id="cb75-5"><a href="hematopoeisis-tutorial---mae.html#cb75-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">clusters =</span> <span class="fu">c</span>(<span class="st">&quot;PreB&quot;</span>,<span class="st">&quot;CD4.M&quot;</span>),</span>
<span id="cb75-6"><a href="hematopoeisis-tutorial---mae.html#cb75-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">layout =</span> <span class="st">&quot;stress&quot;</span>)</span></code></pre></div>
<pre><code>## Replacement of na values for weights with 0</code></pre>
<pre><code>## Building graph using weight as edge weights</code></pre>
<p><img src="Epiregulon-documentation_files/figure-html/differential%20networks%20hematopoeisis-1.png" width="672" />
Sometimes, we are interested to identify interaction partners of the TFs of interest. This can be achieved by comparing the overlap of the targets genes for all the TFs and identify the most similar TFs by Jaccard similarity. To illustrate this function, we take a look at the top most similar 20 TFs to EBF1, and we successfully identify PAX5 as the most similar TF. Both PAX5 and EBF1 are important factors for B cell development (<a href="https://www.nature.com/articles/ni.2641" class="uri">https://www.nature.com/articles/ni.2641</a>).</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="hematopoeisis-tutorial---mae.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># construct a graph of the preB cells</span></span>
<span id="cb78-2"><a href="hematopoeisis-tutorial---mae.html#cb78-2" aria-hidden="true" tabindex="-1"></a>preB_network <span class="ot">&lt;-</span> <span class="fu">buildGraph</span>(regulon.w, <span class="at">weights =</span> <span class="st">&quot;weight&quot;</span>, <span class="at">cluster=</span><span class="st">&quot;PreB&quot;</span>)</span></code></pre></div>
<pre><code>## Building graph using weight as edge weights</code></pre>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="hematopoeisis-tutorial---mae.html#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># compute a similarity matrix of all TFs</span></span>
<span id="cb80-2"><a href="hematopoeisis-tutorial---mae.html#cb80-2" aria-hidden="true" tabindex="-1"></a>similarity_score <span class="ot">&lt;-</span> <span class="fu">calculateJaccardSimilarity</span>(preB_network)</span>
<span id="cb80-3"><a href="hematopoeisis-tutorial---mae.html#cb80-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-4"><a href="hematopoeisis-tutorial---mae.html#cb80-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Focus on EBF1</span></span>
<span id="cb80-5"><a href="hematopoeisis-tutorial---mae.html#cb80-5" aria-hidden="true" tabindex="-1"></a>similarity_score_EBF1 <span class="ot">&lt;-</span> similarity_score[, <span class="st">&quot;EBF1&quot;</span>]</span>
<span id="cb80-6"><a href="hematopoeisis-tutorial---mae.html#cb80-6" aria-hidden="true" tabindex="-1"></a>similarity_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">similarity =</span> <span class="fu">head</span>(<span class="fu">sort</span>(similarity_score_EBF1, </span>
<span id="cb80-7"><a href="hematopoeisis-tutorial---mae.html#cb80-7" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>),</span>
<span id="cb80-8"><a href="hematopoeisis-tutorial---mae.html#cb80-8" aria-hidden="true" tabindex="-1"></a>                            <span class="at">TF =</span> <span class="fu">names</span>(<span class="fu">head</span>(<span class="fu">sort</span>(similarity_score_EBF1,</span>
<span id="cb80-9"><a href="hematopoeisis-tutorial---mae.html#cb80-9" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>)))</span>
<span id="cb80-10"><a href="hematopoeisis-tutorial---mae.html#cb80-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-11"><a href="hematopoeisis-tutorial---mae.html#cb80-11" aria-hidden="true" tabindex="-1"></a>similarity_df<span class="sc">$</span>TF <span class="ot">&lt;-</span> <span class="fu">factor</span>(similarity_df<span class="sc">$</span>TF, <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(similarity_df<span class="sc">$</span>TF)))</span>
<span id="cb80-12"><a href="hematopoeisis-tutorial---mae.html#cb80-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-13"><a href="hematopoeisis-tutorial---mae.html#cb80-13" aria-hidden="true" tabindex="-1"></a><span class="co"># plot top TFs most similar to EBF1</span></span>
<span id="cb80-14"><a href="hematopoeisis-tutorial---mae.html#cb80-14" aria-hidden="true" tabindex="-1"></a>topTFplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(similarity_df, <span class="fu">aes</span>(<span class="at">x=</span>TF, <span class="at">y=</span>similarity)) <span class="sc">+</span></span>
<span id="cb80-15"><a href="hematopoeisis-tutorial---mae.html#cb80-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb80-16"><a href="hematopoeisis-tutorial---mae.html#cb80-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb80-17"><a href="hematopoeisis-tutorial---mae.html#cb80-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;EBF1 similarity&quot;</span>) <span class="sc">+</span></span>
<span id="cb80-18"><a href="hematopoeisis-tutorial---mae.html#cb80-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb80-19"><a href="hematopoeisis-tutorial---mae.html#cb80-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-20"><a href="hematopoeisis-tutorial---mae.html#cb80-20" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(topTFplot)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<p>In order to convince ourselves that our differential network is statistically significant, we permute the edges and obtain a background graph from averaging many iterations. Here, we plot the differential network graph subtracted by permuted graphs.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="hematopoeisis-tutorial---mae.html#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create a permuted graph by rewiring the edges 100 times</span></span>
<span id="cb81-2"><a href="hematopoeisis-tutorial---mae.html#cb81-2" aria-hidden="true" tabindex="-1"></a>permute_matrix <span class="ot">&lt;-</span> <span class="fu">permuteGraph</span>(preB_network, <span class="st">&quot;EBF1&quot;</span>, <span class="dv">100</span>, <span class="at">p=</span><span class="dv">1</span>)</span>
<span id="cb81-3"><a href="hematopoeisis-tutorial---mae.html#cb81-3" aria-hidden="true" tabindex="-1"></a>permute_matrix <span class="ot">&lt;-</span> permute_matrix[<span class="fu">names</span>(similarity_score_EBF1),]</span>
<span id="cb81-4"><a href="hematopoeisis-tutorial---mae.html#cb81-4" aria-hidden="true" tabindex="-1"></a>diff_matrix <span class="ot">&lt;-</span> similarity_score_EBF1<span class="sc">-</span><span class="fu">rowMeans</span>(permute_matrix)</span>
<span id="cb81-5"><a href="hematopoeisis-tutorial---mae.html#cb81-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-6"><a href="hematopoeisis-tutorial---mae.html#cb81-6" aria-hidden="true" tabindex="-1"></a>diff_matrix_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">similarity =</span> <span class="fu">head</span>(<span class="fu">sort</span>(diff_matrix, </span>
<span id="cb81-7"><a href="hematopoeisis-tutorial---mae.html#cb81-7" aria-hidden="true" tabindex="-1"></a>                                                    <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>),</span>
<span id="cb81-8"><a href="hematopoeisis-tutorial---mae.html#cb81-8" aria-hidden="true" tabindex="-1"></a>                            <span class="at">TF =</span> <span class="fu">names</span>(<span class="fu">head</span>(<span class="fu">sort</span>(diff_matrix,</span>
<span id="cb81-9"><a href="hematopoeisis-tutorial---mae.html#cb81-9" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>)))</span>
<span id="cb81-10"><a href="hematopoeisis-tutorial---mae.html#cb81-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-11"><a href="hematopoeisis-tutorial---mae.html#cb81-11" aria-hidden="true" tabindex="-1"></a>diff_matrix_df<span class="sc">$</span>TF <span class="ot">&lt;-</span> <span class="fu">factor</span>(diff_matrix_df<span class="sc">$</span>TF, <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(diff_matrix_df<span class="sc">$</span>TF)))</span>
<span id="cb81-12"><a href="hematopoeisis-tutorial---mae.html#cb81-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-13"><a href="hematopoeisis-tutorial---mae.html#cb81-13" aria-hidden="true" tabindex="-1"></a><span class="co"># plot top TFs most similar to EBF1</span></span>
<span id="cb81-14"><a href="hematopoeisis-tutorial---mae.html#cb81-14" aria-hidden="true" tabindex="-1"></a>topTFplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(diff_matrix_df, <span class="fu">aes</span>(<span class="at">x=</span>TF, <span class="at">y=</span>similarity)) <span class="sc">+</span></span>
<span id="cb81-15"><a href="hematopoeisis-tutorial---mae.html#cb81-15" aria-hidden="true" tabindex="-1"></a>            <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb81-16"><a href="hematopoeisis-tutorial---mae.html#cb81-16" aria-hidden="true" tabindex="-1"></a>            <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb81-17"><a href="hematopoeisis-tutorial---mae.html#cb81-17" aria-hidden="true" tabindex="-1"></a>            <span class="fu">ggtitle</span>(<span class="st">&quot;background subtracted EBF1 similarity &quot;</span>) <span class="sc">+</span></span>
<span id="cb81-18"><a href="hematopoeisis-tutorial---mae.html#cb81-18" aria-hidden="true" tabindex="-1"></a>            <span class="fu">theme_classic</span>()</span>
<span id="cb81-19"><a href="hematopoeisis-tutorial---mae.html#cb81-19" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(topTFplot)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="hematopoeisis-tutorial---mae.html#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># obtain empirical p-values</span></span>
<span id="cb82-2"><a href="hematopoeisis-tutorial---mae.html#cb82-2" aria-hidden="true" tabindex="-1"></a>p_matrix <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">apply</span>(permute_matrix, <span class="dv">2</span>, <span class="cf">function</span>(x) {x <span class="sc">&gt;</span> similarity_score_EBF1}))</span>
<span id="cb82-3"><a href="hematopoeisis-tutorial---mae.html#cb82-3" aria-hidden="true" tabindex="-1"></a>p_matrix[<span class="fu">names</span>(<span class="fu">head</span>(<span class="fu">sort</span>(diff_matrix,<span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>))]</span></code></pre></div>
<pre><code>##   PAX5   IRF4 POU2F2   RELB  IKZF2  FOXP1   BCL6  RUNX3  FOXO1   TCF4  NIPBL   RAG1 BCL11A BCLAF1  IKZF1 CREBBP 
##      0      0      0      0      0      0      0      0      0      0      0      0      0      0      0      0 
## NFATC1  KMT2D   CHD2   BCL3 
##      0      0      0      0</code></pre>
<p>Next, we are interested to compare the networks of two cell types, in this case, CD4 memory cells (CD4.M) vs Monocytes (mono) cells. We build an edge subtracted graph and then calculate the degree centrality of the subtracted graph. We normalize centrality using the default square root function. The top 5 most positive TFs represent lineage factors more active in NK cells whereas the bottom 5 TFs present lineage factors enriched in CD4. We successfully identified the myeloid factor SPI1 to be associated with monocytes and Th1 factor TBX21 to be associated with CD4 cells.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="hematopoeisis-tutorial---mae.html#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co"># construct a graph of the CD4.M and NK cells respectively</span></span>
<span id="cb84-2"><a href="hematopoeisis-tutorial---mae.html#cb84-2" aria-hidden="true" tabindex="-1"></a>CD4.M_network <span class="ot">&lt;-</span> <span class="fu">buildGraph</span>(regulon.w, <span class="at">weights =</span> <span class="st">&quot;weight&quot;</span>, <span class="at">cluster=</span><span class="st">&quot;CD4.M&quot;</span>)</span></code></pre></div>
<pre><code>## Building graph using weight as edge weights</code></pre>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="hematopoeisis-tutorial---mae.html#cb86-1" aria-hidden="true" tabindex="-1"></a>Mono_network <span class="ot">&lt;-</span> <span class="fu">buildGraph</span>(regulon.w, <span class="at">weights =</span> <span class="st">&quot;weight&quot;</span>, <span class="at">cluster=</span><span class="st">&quot;Mono&quot;</span>)</span></code></pre></div>
<pre><code>## Building graph using weight as edge weights</code></pre>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="hematopoeisis-tutorial---mae.html#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># construct a difference graph</span></span>
<span id="cb88-2"><a href="hematopoeisis-tutorial---mae.html#cb88-2" aria-hidden="true" tabindex="-1"></a>diff_graph <span class="ot">&lt;-</span> <span class="fu">buildDiffGraph</span>( Mono_network,CD4.M_network, <span class="at">abs_diff =</span> <span class="cn">FALSE</span>)</span>
<span id="cb88-3"><a href="hematopoeisis-tutorial---mae.html#cb88-3" aria-hidden="true" tabindex="-1"></a>diff_graph <span class="ot">&lt;-</span> <span class="fu">addCentrality</span>(diff_graph)</span>
<span id="cb88-4"><a href="hematopoeisis-tutorial---mae.html#cb88-4" aria-hidden="true" tabindex="-1"></a>diff_graph <span class="ot">&lt;-</span> <span class="fu">normalizeCentrality</span>(diff_graph)</span>
<span id="cb88-5"><a href="hematopoeisis-tutorial---mae.html#cb88-5" aria-hidden="true" tabindex="-1"></a>rank_table <span class="ot">&lt;-</span> <span class="fu">rankTfs</span>(diff_graph)</span>
<span id="cb88-6"><a href="hematopoeisis-tutorial---mae.html#cb88-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-7"><a href="hematopoeisis-tutorial---mae.html#cb88-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb88-8"><a href="hematopoeisis-tutorial---mae.html#cb88-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(rank_table, <span class="fu">aes</span>(<span class="at">x =</span> rank, <span class="at">y =</span> centrality)) <span class="sc">+</span></span>
<span id="cb88-9"><a href="hematopoeisis-tutorial---mae.html#cb88-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb88-10"><a href="hematopoeisis-tutorial---mae.html#cb88-10" aria-hidden="true" tabindex="-1"></a>    ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span> <span class="fu">rbind</span>(<span class="fu">head</span>(rank_table, <span class="dv">5</span>), </span>
<span id="cb88-11"><a href="hematopoeisis-tutorial---mae.html#cb88-11" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">tail</span>(rank_table,<span class="dv">5</span>)), </span>
<span id="cb88-12"><a href="hematopoeisis-tutorial---mae.html#cb88-12" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">aes</span>(<span class="at">label =</span> tf), </span>
<span id="cb88-13"><a href="hematopoeisis-tutorial---mae.html#cb88-13" aria-hidden="true" tabindex="-1"></a>                             <span class="at">nudge_x =</span> <span class="dv">0</span>, <span class="at">nudge_y =</span> <span class="dv">0</span>, <span class="at">box.padding =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb88-14"><a href="hematopoeisis-tutorial---mae.html#cb88-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span> <span class="fu">ggtitle</span> (<span class="st">&quot;differential TFs (Mono-CD4.M) ranked by degree centrality&quot;</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
<p>We can further explore interacting factors with the myeloid factor SPI1 using the same Jaccard similarity approach. We found CEBPA as the most similar TF as SPI1. SPI1 and CEBPA are known to be important for differentiation into myeloid cells (<a href="https://www.cell.com/cell-reports/pdfExtended/S2211-1247(18)30745-9" class="uri">https://www.cell.com/cell-reports/pdfExtended/S2211-1247(18)30745-9</a>).</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="hematopoeisis-tutorial---mae.html#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(igraph)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;igraph&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:rtracklayer&#39;:
## 
##     blocks, path</code></pre>
<pre><code>## The following object is masked from &#39;package:BiocIO&#39;:
## 
##     path</code></pre>
<pre><code>## The following object is masked from &#39;package:Biostrings&#39;:
## 
##     union</code></pre>
<pre><code>## The following object is masked from &#39;package:XVector&#39;:
## 
##     path</code></pre>
<pre><code>## The following objects are masked from &#39;package:DelayedArray&#39;:
## 
##     path, simplify</code></pre>
<pre><code>## The following object is masked from &#39;package:gtools&#39;:
## 
##     permute</code></pre>
<pre><code>## The following object is masked from &#39;package:HDF5Array&#39;:
## 
##     path</code></pre>
<pre><code>## The following object is masked from &#39;package:GenomicRanges&#39;:
## 
##     union</code></pre>
<pre><code>## The following object is masked from &#39;package:IRanges&#39;:
## 
##     union</code></pre>
<pre><code>## The following object is masked from &#39;package:S4Vectors&#39;:
## 
##     union</code></pre>
<pre><code>## The following objects are masked from &#39;package:BiocGenerics&#39;:
## 
##     normalize, path, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     decompose, spectrum</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     union</code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="hematopoeisis-tutorial---mae.html#cb104-1" aria-hidden="true" tabindex="-1"></a>diff_graph_filter <span class="ot">&lt;-</span> <span class="fu">subgraph.edges</span>(diff_graph, </span>
<span id="cb104-2"><a href="hematopoeisis-tutorial---mae.html#cb104-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">E</span>(diff_graph)[<span class="fu">E</span>(diff_graph)<span class="sc">$</span>weight<span class="sc">&gt;</span><span class="dv">0</span>], </span>
<span id="cb104-3"><a href="hematopoeisis-tutorial---mae.html#cb104-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">del=</span>T)</span>
<span id="cb104-4"><a href="hematopoeisis-tutorial---mae.html#cb104-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-5"><a href="hematopoeisis-tutorial---mae.html#cb104-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-6"><a href="hematopoeisis-tutorial---mae.html#cb104-6" aria-hidden="true" tabindex="-1"></a><span class="co"># compute a similarity matrix of all TFs</span></span>
<span id="cb104-7"><a href="hematopoeisis-tutorial---mae.html#cb104-7" aria-hidden="true" tabindex="-1"></a>similarity_score <span class="ot">&lt;-</span> <span class="fu">calculateJaccardSimilarity</span>(diff_graph_filter)</span>
<span id="cb104-8"><a href="hematopoeisis-tutorial---mae.html#cb104-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-9"><a href="hematopoeisis-tutorial---mae.html#cb104-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Focus on SPI1</span></span>
<span id="cb104-10"><a href="hematopoeisis-tutorial---mae.html#cb104-10" aria-hidden="true" tabindex="-1"></a>similarity_score_SPI1 <span class="ot">&lt;-</span> similarity_score[, <span class="st">&quot;SPI1&quot;</span>]</span>
<span id="cb104-11"><a href="hematopoeisis-tutorial---mae.html#cb104-11" aria-hidden="true" tabindex="-1"></a>similarity_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">similarity =</span> <span class="fu">head</span>(<span class="fu">sort</span>(similarity_score_SPI1, </span>
<span id="cb104-12"><a href="hematopoeisis-tutorial---mae.html#cb104-12" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>),</span>
<span id="cb104-13"><a href="hematopoeisis-tutorial---mae.html#cb104-13" aria-hidden="true" tabindex="-1"></a>                            <span class="at">TF =</span> <span class="fu">names</span>(<span class="fu">head</span>(<span class="fu">sort</span>(similarity_score_SPI1,</span>
<span id="cb104-14"><a href="hematopoeisis-tutorial---mae.html#cb104-14" aria-hidden="true" tabindex="-1"></a>                                                 <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>)))</span>
<span id="cb104-15"><a href="hematopoeisis-tutorial---mae.html#cb104-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-16"><a href="hematopoeisis-tutorial---mae.html#cb104-16" aria-hidden="true" tabindex="-1"></a>similarity_df<span class="sc">$</span>TF <span class="ot">&lt;-</span> <span class="fu">factor</span>(similarity_df<span class="sc">$</span>TF, </span>
<span id="cb104-17"><a href="hematopoeisis-tutorial---mae.html#cb104-17" aria-hidden="true" tabindex="-1"></a>                           <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(similarity_df<span class="sc">$</span>TF)))</span>
<span id="cb104-18"><a href="hematopoeisis-tutorial---mae.html#cb104-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-19"><a href="hematopoeisis-tutorial---mae.html#cb104-19" aria-hidden="true" tabindex="-1"></a><span class="co"># plot top TFs most similar to SPI1</span></span>
<span id="cb104-20"><a href="hematopoeisis-tutorial---mae.html#cb104-20" aria-hidden="true" tabindex="-1"></a>topTFplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(similarity_df, <span class="fu">aes</span>(<span class="at">x=</span>TF, <span class="at">y=</span>similarity)) <span class="sc">+</span></span>
<span id="cb104-21"><a href="hematopoeisis-tutorial---mae.html#cb104-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb104-22"><a href="hematopoeisis-tutorial---mae.html#cb104-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb104-23"><a href="hematopoeisis-tutorial---mae.html#cb104-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;SPI1 similarity&quot;</span>) <span class="sc">+</span></span>
<span id="cb104-24"><a href="hematopoeisis-tutorial---mae.html#cb104-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb104-25"><a href="hematopoeisis-tutorial---mae.html#cb104-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-26"><a href="hematopoeisis-tutorial---mae.html#cb104-26" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(topTFplot)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
</div>
</div>
<div id="session-info-1" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Session Info<a href="hematopoeisis-tutorial---mae.html#session-info-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="hematopoeisis-tutorial---mae.html#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.3.0 (2023-04-21)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 18.04.6 LTS
## 
## Matrix products: default
## BLAS:   /usr/local/lib/R/lib/libRblas.so 
## LAPACK: /usr/local/lib/R/lib/libRlapack.so;  LAPACK version 3.11.0
## 
## Random number generation:
##  RNG:     L&#39;Ecuyer-CMRG 
##  Normal:  Inversion 
##  Sample:  Rejection 
##  
## locale:
##  [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C            LC_COLLATE=C         LC_MONETARY=C       
##  [6] LC_MESSAGES=C        LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C         LC_TELEPHONE=C      
## [11] LC_MEASUREMENT=C     LC_IDENTIFICATION=C 
## 
## time zone: Etc/UTC
## tzcode source: system (glibc)
## 
## attached base packages:
## [1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] igraph_1.5.1                      BSgenome.Hsapiens.UCSC.hg38_1.4.5 BSgenome_1.69.1                  
##  [4] rtracklayer_1.61.2                BiocIO_1.11.0                     Biostrings_2.69.2                
##  [7] XVector_0.41.2                    org.Hs.eg.db_3.18.0               AnnotationDbi_1.63.2             
## [10] msigdbr_7.5.1                     DelayedArray_0.27.10              epiregulon.extra_0.99.0          
## [13] rhdf5_2.45.1                      RcppArmadillo_0.12.6.4.0          Rcpp_1.0.11                      
## [16] sparseMatrixStats_1.13.4          data.table_1.14.8                 stringr_1.5.0                    
## [19] plyr_1.8.9                        magrittr_2.0.3                    ggplot2_3.4.4                    
## [22] gtable_0.3.4                      gtools_3.9.4                      gridExtra_2.3                    
## [25] devtools_2.4.5                    usethis_2.1.6                     ArchR_1.0.3                      
## [28] Matrix_1.6-3                      HDF5Array_1.29.3                  scMultiome_1.1.2                 
## [31] MultiAssayExperiment_1.27.5       ExperimentHub_2.9.1               AnnotationHub_3.9.2              
## [34] BiocFileCache_2.9.1               dbplyr_2.3.4                      epiregulon_1.0.36                
## [37] SingleCellExperiment_1.23.0       SummarizedExperiment_1.31.1       Biobase_2.61.0                   
## [40] GenomicRanges_1.53.3              GenomeInfoDb_1.37.7               IRanges_2.35.3                   
## [43] S4Vectors_0.39.3                  BiocGenerics_0.47.1               MatrixGenerics_1.13.2            
## [46] matrixStats_1.0.0                 rmarkdown_2.25                    bookdown_0.36                    
## 
## loaded via a namespace (and not attached):
##   [1] R.methodsS3_1.8.2             GSEABase_1.63.0               urlchecker_1.0.1             
##   [4] poweRlaw_0.70.6               vctrs_0.6.3                   digest_0.6.31                
##   [7] png_0.1-8                     shape_1.4.6                   ggrepel_0.9.4                
##  [10] magick_2.8.1                  MASS_7.3-60                   reshape2_1.4.4               
##  [13] httpuv_1.6.11                 foreach_1.5.2                 qvalue_2.33.0                
##  [16] withr_2.5.0                   ggrastr_1.0.2                 xfun_0.39                    
##  [19] ggfun_0.1.3                   ellipsis_0.3.2                memoise_2.0.1                
##  [22] ggbeeswarm_0.7.2              clusterProfiler_4.9.5         gson_0.1.0                   
##  [25] profvis_0.3.8                 tidytree_0.4.5                GlobalOptions_0.1.2          
##  [28] KEGGgraph_1.61.0              R.oo_1.25.0                   prettyunits_1.1.1            
##  [31] KEGGREST_1.41.4               promises_1.2.0.1              httr_1.4.6                   
##  [34] restfulr_0.0.15               rhdf5filters_1.13.5           ps_1.7.5                     
##  [37] rstudioapi_0.15.0             miniUI_0.1.1.1                generics_0.1.3               
##  [40] DOSE_3.27.3                   processx_3.8.1                babelgene_22.9               
##  [43] curl_5.0.0                    zlibbioc_1.47.0               ScaledMatrix_1.9.1           
##  [46] ggraph_2.1.0                  polyclip_1.10-6               GenomeInfoDbData_1.2.11      
##  [49] SparseArray_1.1.12            interactiveDisplayBase_1.39.0 xtable_1.8-4                 
##  [52] pracma_2.4.2                  doParallel_1.0.17             evaluate_0.21                
##  [55] S4Arrays_1.1.6                hms_1.1.3                     irlba_2.3.5.1                
##  [58] colorspace_2.1-0              filelock_1.0.2                readr_2.1.4                  
##  [61] Rgraphviz_2.45.0              later_1.3.1                   viridis_0.6.4                
##  [64] ggtree_3.9.1                  lattice_0.22-5                XML_3.99-0.14                
##  [67] scuttle_1.11.3                shadowtext_0.1.2              cowplot_1.1.1                
##  [70] pillar_1.9.0                  nlme_3.1-163                  iterators_1.0.14             
##  [73] caTools_1.18.2                compiler_4.3.0                beachmat_2.17.17             
##  [76] stringi_1.7.12                GenomicAlignments_1.37.0      MPO.db_0.99.7                
##  [79] crayon_1.5.2                  abind_1.4-5                   scater_1.29.4                
##  [82] gridGraphics_0.5-1            locfit_1.5-9.8                graphlayouts_1.0.1           
##  [85] bit_4.0.5                     dplyr_1.1.3                   fastmatch_1.1-4              
##  [88] codetools_0.2-19              BiocSingular_1.17.1           bslib_0.5.1                  
##  [91] GetoptLong_1.0.5              mime_0.12                     splines_4.3.0                
##  [94] circlize_0.4.15               HDO.db_0.99.1                 EnrichmentBrowser_2.31.5     
##  [97] knitr_1.44                    blob_1.2.4                    utf8_1.2.3                   
## [100] clue_0.3-65                   BiocVersion_3.18.0            seqLogo_1.67.0               
## [103] fs_1.6.2                      checkmate_2.2.0               DelayedMatrixStats_1.23.9    
## [106] pkgbuild_1.4.0                ggplotify_0.1.2               tibble_3.2.1                 
## [109] callr_3.7.3                   statmod_1.5.0                 tzdb_0.4.0                   
## [112] tweenr_2.0.2                  pkgconfig_2.0.3               tools_4.3.0                  
## [115] cachem_1.0.8                  RSQLite_2.3.1                 viridisLite_0.4.2            
## [118] DBI_1.1.3                     fastmap_1.1.1                 scales_1.2.1                 
## [121] Rsamtools_2.17.0              sass_0.4.6                    patchwork_1.1.3              
## [124] BiocManager_1.30.22           graph_1.79.4                  farver_2.1.1                 
## [127] tidygraph_1.2.3               scatterpie_0.2.1              yaml_2.3.7                   
## [130] cli_3.6.1                     purrr_1.0.2                   motifmatchr_1.23.0           
## [133] lifecycle_1.0.3               bluster_1.11.4                sessioninfo_1.2.2            
## [136] backports_1.4.1               BiocParallel_1.35.4           annotate_1.79.0              
## [139] rjson_0.2.21                  parallel_4.3.0                ape_5.7-1                    
## [142] limma_3.57.11                 jsonlite_1.8.7                edgeR_3.99.6                 
## [145] TFBSTools_1.40.0              bitops_1.0-7                  HPO.db_0.99.2                
## [148] bit64_4.0.5                   yulab.utils_0.1.0             BiocNeighbors_1.19.0         
## [151] CNEr_1.37.0                   jquerylib_0.1.4               metapod_1.9.0                
## [154] GOSemSim_2.27.3               dqrng_0.3.1                   R.utils_2.12.2               
## [157] lazyeval_0.2.2                shiny_1.7.5.1                 htmltools_0.5.5              
## [160] enrichplot_1.21.3             GO.db_3.18.0                  rappdirs_0.3.3               
## [163] glue_1.6.2                    TFMPvalue_0.0.9               RCurl_1.98-1.12              
## [166] treeio_1.25.4                 scran_1.29.3                  R6_2.5.1                     
## [169] tidyr_1.3.0                   labeling_0.4.3                cluster_2.1.4                
## [172] pkgload_1.3.2                 Rhdf5lib_1.23.2               aplot_0.2.2                  
## [175] DirichletMultinomial_1.43.0   tidyselect_1.2.0              vipor_0.4.5                  
## [178] ggforce_0.4.1                 rsvd_1.0.5                    munsell_0.5.0                
## [181] BiocStyle_2.29.2              htmlwidgets_1.6.2             fgsea_1.27.1                 
## [184] ComplexHeatmap_2.17.0         RColorBrewer_1.1-3            rlang_1.1.1                  
## [187] remotes_2.4.2                 fansi_1.0.4                   Cairo_1.6-1                  
## [190] beeswarm_0.4.0</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="multiome---archr-workflow.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="prostate-cancer-cells.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/hematopoeisis.mae.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Epiregulon documentation.pdf", "Epiregulon documentation.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
