<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 Network analysis | Epiregulon documentation</title>
  <meta name="description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="generator" content="bookdown 0.46 and GitBook 2.6.7" />

  <meta property="og:title" content="7 Network analysis | Epiregulon documentation" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="github-repo" content="xiaosaiyao/epiregulon.book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 Network analysis | Epiregulon documentation" />
  
  <meta name="twitter:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  

<meta name="author" content="Xiaosai Yao, Tomasz WÅ‚odarczyk" />


<meta name="date" content="2025-12-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="calculation-of-the-tf-activity.html"/>
<link rel="next" href="archr-workflow-and-different-weight-methods.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Epiregulon documentation</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>1</b> Installation</a>
<ul>
<li class="chapter" data-level="1.1" data-path="installation.html"><a href="installation.html#install-from-github"><i class="fa fa-check"></i><b>1.1</b> Install from GitHub</a></li>
<li class="chapter" data-level="1.2" data-path="installation.html"><a href="installation.html#install-from-bioconductor"><i class="fa fa-check"></i><b>1.2</b> Install from Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="quick-start.html"><a href="quick-start.html"><i class="fa fa-check"></i><b>2</b> Quick start</a>
<ul>
<li class="chapter" data-level="2.1" data-path="quick-start.html"><a href="quick-start.html#prepare-data"><i class="fa fa-check"></i><b>2.1</b> Prepare data</a></li>
<li class="chapter" data-level="2.2" data-path="quick-start.html"><a href="quick-start.html#retrieve-bulk-tf-chip-seq-binding-sites"><i class="fa fa-check"></i><b>2.2</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="2.3" data-path="quick-start.html"><a href="quick-start.html#link-atac-seq-peaks-to-target-genes"><i class="fa fa-check"></i><b>2.3</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="2.4" data-path="quick-start.html"><a href="quick-start.html#add-tf-motif-binding-to-peaks"><i class="fa fa-check"></i><b>2.4</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="2.5" data-path="quick-start.html"><a href="quick-start.html#generate-regulons"><i class="fa fa-check"></i><b>2.5</b> Generate regulons</a></li>
<li class="chapter" data-level="2.6" data-path="quick-start.html"><a href="quick-start.html#prune-network-recommended"><i class="fa fa-check"></i><b>2.6</b> Prune network (recommended)</a></li>
<li class="chapter" data-level="2.7" data-path="quick-start.html"><a href="quick-start.html#annotate-regulon"><i class="fa fa-check"></i><b>2.7</b> Annotate regulon</a></li>
<li class="chapter" data-level="2.8" data-path="quick-start.html"><a href="quick-start.html#calculate-tf-activity"><i class="fa fa-check"></i><b>2.8</b> Calculate TF activity</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-preparation.html"><a href="data-preparation.html"><i class="fa fa-check"></i><b>3</b> Data preparation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-preparation.html"><a href="data-preparation.html#construct-a-singlecellexperiment-object"><i class="fa fa-check"></i><b>3.1</b> Construct a SingleCellExperiment object</a></li>
<li class="chapter" data-level="3.2" data-path="data-preparation.html"><a href="data-preparation.html#start-from-an-archr-object"><i class="fa fa-check"></i><b>3.2</b> Start from an ArchR object</a></li>
<li class="chapter" data-level="3.3" data-path="data-preparation.html"><a href="data-preparation.html#start-from-a-seuratsignac-object"><i class="fa fa-check"></i><b>3.3</b> Start from a Seurat/Signac object</a></li>
<li class="chapter" data-level="3.4" data-path="data-preparation.html"><a href="data-preparation.html#start-from-anndata"><i class="fa fa-check"></i><b>3.4</b> Start from AnnData</a></li>
<li class="chapter" data-level="3.5" data-path="data-preparation.html"><a href="data-preparation.html#start-from-10x-data-formats"><i class="fa fa-check"></i><b>3.5</b> Start from 10x data formats</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="data-preparation.html"><a href="data-preparation.html#read-from-.h5-file"><i class="fa fa-check"></i><b>3.5.1</b> Read from .h5 file</a></li>
<li class="chapter" data-level="3.5.2" data-path="data-preparation.html"><a href="data-preparation.html#read-directly-from-a-10x-directory"><i class="fa fa-check"></i><b>3.5.2</b> Read directly from a 10x directory</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="data-preparation.html"><a href="data-preparation.html#read-directly-from-csv"><i class="fa fa-check"></i><b>3.6</b> Read directly from CSV</a></li>
<li class="chapter" data-level="3.7" data-path="data-preparation.html"><a href="data-preparation.html#important-points"><i class="fa fa-check"></i><b>3.7</b> Important points</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="construction.html"><a href="construction.html"><i class="fa fa-check"></i><b>4</b> Construction of the gene regulatory network</a>
<ul>
<li class="chapter" data-level="4.1" data-path="construction.html"><a href="construction.html#download-and-format-dataset"><i class="fa fa-check"></i><b>4.1</b> Download and format dataset</a></li>
<li class="chapter" data-level="4.2" data-path="construction.html"><a href="construction.html#retrieve-tf-occupancy-data"><i class="fa fa-check"></i><b>4.2</b> Retrieve TF occupancy data</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="construction.html"><a href="construction.html#chip-seq-data"><i class="fa fa-check"></i><b>4.2.1</b> ChIP-seq data</a></li>
<li class="chapter" data-level="4.2.2" data-path="construction.html"><a href="construction.html#motif-information"><i class="fa fa-check"></i><b>4.2.2</b> Motif information</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="construction.html"><a href="construction.html#link-atacseq-peaks-to-target-genes"><i class="fa fa-check"></i><b>4.3</b> Link ATACseq peaks to target genes</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="construction.html"><a href="construction.html#optimization"><i class="fa fa-check"></i><b>4.3.1</b> Optimize the number of metacells (optional)</a></li>
<li class="chapter" data-level="4.3.2" data-path="construction.html"><a href="construction.html#distance_window"><i class="fa fa-check"></i><b>4.3.2</b> Find peak-target gene links within a defined distance</a></li>
<li class="chapter" data-level="4.3.3" data-path="construction.html"><a href="construction.html#link-peaks-to-the-nearest-genes"><i class="fa fa-check"></i><b>4.3.3</b> Link peaks to the nearest genes</a></li>
<li class="chapter" data-level="4.3.4" data-path="construction.html"><a href="construction.html#filter-peak-target-gene-links"><i class="fa fa-check"></i><b>4.3.4</b> Filter peak-target gene links</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="construction.html"><a href="construction.html#add-tf-occupancy-to-peaks"><i class="fa fa-check"></i><b>4.4</b> Add TF occupancy to peaks</a></li>
<li class="chapter" data-level="4.5" data-path="construction.html"><a href="construction.html#generate-regulons-1"><i class="fa fa-check"></i><b>4.5</b> Generate regulons</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="refinement.html"><a href="refinement.html"><i class="fa fa-check"></i><b>5</b> Refinement of gene regulatory network</a>
<ul>
<li class="chapter" data-level="5.1" data-path="refinement.html"><a href="refinement.html#pruning"><i class="fa fa-check"></i><b>5.1</b> Prune network</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="refinement.html"><a href="refinement.html#cell_aggregation"><i class="fa fa-check"></i><b>5.1.1</b> Aggregate cells</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="refinement.html"><a href="refinement.html#motif_score"><i class="fa fa-check"></i><b>5.2</b> Annotate with TF motifs</a></li>
<li class="chapter" data-level="5.3" data-path="refinement.html"><a href="refinement.html#annotate-with-log-fold-changes"><i class="fa fa-check"></i><b>5.3</b> Annotate with log fold changes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="calculation-of-the-tf-activity.html"><a href="calculation-of-the-tf-activity.html"><i class="fa fa-check"></i><b>6</b> Calculation of the TF activity</a>
<ul>
<li class="chapter" data-level="6.1" data-path="calculation-of-the-tf-activity.html"><a href="calculation-of-the-tf-activity.html#addWeights"><i class="fa fa-check"></i><b>6.1</b> Add weights</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="calculation-of-the-tf-activity.html"><a href="calculation-of-the-tf-activity.html#aggregate-cells"><i class="fa fa-check"></i><b>6.1.1</b> Aggregate cells</a></li>
<li class="chapter" data-level="6.1.2" data-path="calculation-of-the-tf-activity.html"><a href="calculation-of-the-tf-activity.html#using-chromatin-accessibility-data"><i class="fa fa-check"></i><b>6.1.2</b> Using chromatin accessibility data</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="calculation-of-the-tf-activity.html"><a href="calculation-of-the-tf-activity.html#calculate-tf-activity-1"><i class="fa fa-check"></i><b>6.2</b> Calculate TF activity</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="network-analysis.html"><a href="network-analysis.html"><i class="fa fa-check"></i><b>7</b> Network analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="network-analysis.html"><a href="network-analysis.html#perform-differential-tf-activity"><i class="fa fa-check"></i><b>7.1</b> Perform differential TF activity</a></li>
<li class="chapter" data-level="7.2" data-path="network-analysis.html"><a href="network-analysis.html#diff_activity_vis"><i class="fa fa-check"></i><b>7.2</b> Visualize TF activities</a></li>
<li class="chapter" data-level="7.3" data-path="network-analysis.html"><a href="network-analysis.html#perform-geneset-enrichment"><i class="fa fa-check"></i><b>7.3</b> Perform geneset enrichment</a></li>
<li class="chapter" data-level="7.4" data-path="network-analysis.html"><a href="network-analysis.html#diffNetwork"><i class="fa fa-check"></i><b>7.4</b> Perform differential network analysis</a></li>
<li class="chapter" data-level="7.5" data-path="network-analysis.html"><a href="network-analysis.html#find-interaction-partners"><i class="fa fa-check"></i><b>7.5</b> Find interaction partners</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html"><i class="fa fa-check"></i><b>8</b> ArchR workflow and different weight methods</a>
<ul>
<li class="chapter" data-level="8.1" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#prepare-data-1"><i class="fa fa-check"></i><b>8.1</b> Prepare data</a></li>
<li class="chapter" data-level="8.2" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#load-archr-project"><i class="fa fa-check"></i><b>8.2</b> Load ArchR project</a></li>
<li class="chapter" data-level="8.3" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#retrieve-matrices-from-archr-project"><i class="fa fa-check"></i><b>8.3</b> Retrieve matrices from ArchR project</a></li>
<li class="chapter" data-level="8.4" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#retrieve-bulk-tf-chip-seq-binding-sites-1"><i class="fa fa-check"></i><b>8.4</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="8.5" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#link-atac-seq-peaks-to-target-genes-1"><i class="fa fa-check"></i><b>8.5</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="8.6" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#add-tf-occupancy-data-to-peaks"><i class="fa fa-check"></i><b>8.6</b> Add TF occupancy data to peaks</a></li>
<li class="chapter" data-level="8.7" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#generate-regulons-2"><i class="fa fa-check"></i><b>8.7</b> Generate regulons</a></li>
<li class="chapter" data-level="8.8" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#optional-annotate-with-tf-motifs"><i class="fa fa-check"></i><b>8.8</b> (Optional) Annotate with TF motifs</a></li>
<li class="chapter" data-level="8.9" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#prune-network"><i class="fa fa-check"></i><b>8.9</b> Prune network</a></li>
<li class="chapter" data-level="8.10" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#weights_prostate"><i class="fa fa-check"></i><b>8.10</b> Add Weights</a></li>
<li class="chapter" data-level="8.11" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#calculate-tf-activity-2"><i class="fa fa-check"></i><b>8.11</b> Calculate TF activity</a></li>
<li class="chapter" data-level="8.12" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#perform-differential-activity"><i class="fa fa-check"></i><b>8.12</b> Perform differential activity</a></li>
<li class="chapter" data-level="8.13" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#visualize-the-results"><i class="fa fa-check"></i><b>8.13</b> Visualize the results</a></li>
<li class="chapter" data-level="8.14" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#geneset-enrichment"><i class="fa fa-check"></i><b>8.14</b> Geneset enrichment</a></li>
<li class="chapter" data-level="8.15" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#differential-network-analysis-diff_network"><i class="fa fa-check"></i><b>8.15</b> Differential network analysis {diff_network}</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html"><i class="fa fa-check"></i><b>9</b> Single modality: scRNA-seq only</a>
<ul>
<li class="chapter" data-level="9.1" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#load-regulons"><i class="fa fa-check"></i><b>9.1</b> Load regulons</a></li>
<li class="chapter" data-level="9.2" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#load-scrna-seq-data"><i class="fa fa-check"></i><b>9.2</b> Load scRNA-seq data</a></li>
<li class="chapter" data-level="9.3" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#calculate-activity"><i class="fa fa-check"></i><b>9.3</b> Calculate activity</a></li>
<li class="chapter" data-level="9.4" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#perform-differential-activity-1"><i class="fa fa-check"></i><b>9.4</b> Perform differential activity</a></li>
<li class="chapter" data-level="9.5" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#visualize-activity"><i class="fa fa-check"></i><b>9.5</b> Visualize activity</a></li>
<li class="chapter" data-level="9.6" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#perform-pathway-enrichment"><i class="fa fa-check"></i><b>9.6</b> Perform pathway enrichment</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>10</b> Session Info</a></li>
<li class="chapter" data-level="11" data-path="contact.html"><a href="contact.html"><i class="fa fa-check"></i><b>11</b> Contact</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Epiregulon documentation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="network-analysis" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">7</span> Network analysis<a href="network-analysis.html#network-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The <code>epiregulon.extra</code> package provides tools for downstream analysis of gene regulatory networks (GRNs) and TF activity. It enables users to identify differential TF activity, visualize these differences, construct differential graphs that highlight changes in GRN topology, and discover potential interaction partners by identifying overlapping regulons.</p>
<div id="perform-differential-tf-activity" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Perform differential TF activity<a href="network-analysis.html#perform-differential-tf-activity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can determine which TFs exhibit differential activities across cell clusters/groups via the <code>findDifferentialActivity</code> function. This function is a wrapper around the <code>findMarkers</code> function from <em><a href="https://bioconductor.org/packages/3.22/scran">scran</a></em> package.</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="network-analysis.html#cb138-1" tabindex="-1"></a><span class="fu">library</span>(epiregulon.extra)</span>
<span id="cb138-2"><a href="network-analysis.html#cb138-2" tabindex="-1"></a>markers  <span class="ot">&lt;-</span> <span class="fu">findDifferentialActivity</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb138-3"><a href="network-analysis.html#cb138-3" tabindex="-1"></a>                                    <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>cell_type, </span>
<span id="cb138-4"><a href="network-analysis.html#cb138-4" tabindex="-1"></a>                                    <span class="at">pval.type =</span> <span class="st">&quot;some&quot;</span>, </span>
<span id="cb138-5"><a href="network-analysis.html#cb138-5" tabindex="-1"></a>                                    <span class="at">direction =</span> <span class="st">&quot;up&quot;</span>, </span>
<span id="cb138-6"><a href="network-analysis.html#cb138-6" tabindex="-1"></a>                                    <span class="at">logvalues =</span> <span class="cn">FALSE</span>)</span>
<span id="cb138-7"><a href="network-analysis.html#cb138-7" tabindex="-1"></a>markers</span></code></pre></div>
<p><code>getSigGenes</code> compiles the group-wise tests into a single dataframe and enables users to select the top differential TFs based on their desired cutoffs.</p>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="network-analysis.html#cb139-1" tabindex="-1"></a>markers.sig <span class="ot">&lt;-</span> <span class="fu">getSigGenes</span>(markers, <span class="at">topgenes =</span> <span class="dv">5</span>, <span class="at">direction =</span> <span class="st">&quot;up&quot;</span> )</span>
<span id="cb139-2"><a href="network-analysis.html#cb139-2" tabindex="-1"></a>markers.sig </span></code></pre></div>
</div>
<div id="diff_activity_vis" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Visualize TF activities<a href="network-analysis.html#diff_activity_vis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Epiregulon also provides the option for visualizing the inferred TF activities.</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="network-analysis.html#cb140-1" tabindex="-1"></a><span class="fu">options</span>(<span class="at">ggrastr.default.dpi=</span><span class="dv">300</span>)</span>
<span id="cb140-2"><a href="network-analysis.html#cb140-2" tabindex="-1"></a>tfs_interest <span class="ot">&lt;-</span> <span class="fu">c</span>( <span class="st">&quot;EBF1&quot;</span>,<span class="st">&quot;TBX21&quot;</span>,<span class="st">&quot;EOMES&quot;</span>,<span class="st">&quot;SPI1&quot;</span>)</span>
<span id="cb140-3"><a href="network-analysis.html#cb140-3" tabindex="-1"></a><span class="fu">plotActivityDim</span>(<span class="at">sce =</span> GeneExpressionMatrix,</span>
<span id="cb140-4"><a href="network-analysis.html#cb140-4" tabindex="-1"></a>                <span class="at">activity_matrix =</span> score.combine[tfs_interest,], </span>
<span id="cb140-5"><a href="network-analysis.html#cb140-5" tabindex="-1"></a>                <span class="at">tf =</span> tfs_interest,</span>
<span id="cb140-6"><a href="network-analysis.html#cb140-6" tabindex="-1"></a>                <span class="at">dimtype =</span> <span class="st">&quot;UMAP_RNA&quot;</span>,</span>
<span id="cb140-7"><a href="network-analysis.html#cb140-7" tabindex="-1"></a>                <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb140-8"><a href="network-analysis.html#cb140-8" tabindex="-1"></a>                <span class="at">ncol=</span><span class="dv">2</span>,</span>
<span id="cb140-9"><a href="network-analysis.html#cb140-9" tabindex="-1"></a>                <span class="at">point_size=</span><span class="fl">0.1</span>,</span>
<span id="cb140-10"><a href="network-analysis.html#cb140-10" tabindex="-1"></a>                <span class="at">label =</span> <span class="st">&quot;cell_type&quot;</span>,</span>
<span id="cb140-11"><a href="network-analysis.html#cb140-11" tabindex="-1"></a>                <span class="at">rasterise =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-74-1.png" width="672" /></p>
<p>We can compare the activity with gene expression of the same TFs.</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="network-analysis.html#cb141-1" tabindex="-1"></a><span class="fu">plotActivityDim</span>(<span class="at">sce =</span> GeneExpressionMatrix, </span>
<span id="cb141-2"><a href="network-analysis.html#cb141-2" tabindex="-1"></a>                <span class="at">activity_matrix =</span> <span class="fu">assay</span>(GeneExpressionMatrix, <span class="st">&quot;normalizedCounts&quot;</span>)[tfs_interest,], </span>
<span id="cb141-3"><a href="network-analysis.html#cb141-3" tabindex="-1"></a>                <span class="at">tf =</span> tfs_interest, </span>
<span id="cb141-4"><a href="network-analysis.html#cb141-4" tabindex="-1"></a>                <span class="at">dimtype =</span> <span class="st">&quot;UMAP_RNA&quot;</span>,</span>
<span id="cb141-5"><a href="network-analysis.html#cb141-5" tabindex="-1"></a>                <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb141-6"><a href="network-analysis.html#cb141-6" tabindex="-1"></a>                <span class="at">ncol=</span><span class="dv">2</span>,</span>
<span id="cb141-7"><a href="network-analysis.html#cb141-7" tabindex="-1"></a>                <span class="at">legend.label =</span> <span class="st">&quot;Gex&quot;</span>,</span>
<span id="cb141-8"><a href="network-analysis.html#cb141-8" tabindex="-1"></a>                <span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&quot;grey&quot;</span>,<span class="st">&quot;blue&quot;</span>),</span>
<span id="cb141-9"><a href="network-analysis.html#cb141-9" tabindex="-1"></a>                <span class="at">point_size=</span><span class="fl">0.1</span>,</span>
<span id="cb141-10"><a href="network-analysis.html#cb141-10" tabindex="-1"></a>                <span class="at">rasterise =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-75-1.png" width="672" /></p>
<p>We can also generate violin plots to visualize TF activity.</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="network-analysis.html#cb142-1" tabindex="-1"></a><span class="fu">plotActivityViolin</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb142-2"><a href="network-analysis.html#cb142-2" tabindex="-1"></a>                   <span class="at">tf =</span> tfs_interest, </span>
<span id="cb142-3"><a href="network-analysis.html#cb142-3" tabindex="-1"></a>                   <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>cell_type,</span>
<span id="cb142-4"><a href="network-analysis.html#cb142-4" tabindex="-1"></a>                   <span class="at">legend.label =</span> <span class="st">&quot;Gex&quot;</span>,</span>
<span id="cb142-5"><a href="network-analysis.html#cb142-5" tabindex="-1"></a>                   <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb142-6"><a href="network-analysis.html#cb142-6" tabindex="-1"></a>                   <span class="at">ncol=</span><span class="dv">2</span>,</span>
<span id="cb142-7"><a href="network-analysis.html#cb142-7" tabindex="-1"></a>                   <span class="at">boxplot =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-76-1.png" width="672" /></p>
<p>We generate violin plots to visualize TF gene expression.</p>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="network-analysis.html#cb143-1" tabindex="-1"></a><span class="fu">plotActivityViolin</span>(<span class="at">activity_matrix =</span> <span class="fu">assay</span>(GeneExpressionMatrix, <span class="st">&quot;normalizedCounts&quot;</span>)[tfs_interest,],  </span>
<span id="cb143-2"><a href="network-analysis.html#cb143-2" tabindex="-1"></a>                   <span class="at">tf =</span> tfs_interest, </span>
<span id="cb143-3"><a href="network-analysis.html#cb143-3" tabindex="-1"></a>                   <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>cell_type,</span>
<span id="cb143-4"><a href="network-analysis.html#cb143-4" tabindex="-1"></a>                   <span class="at">nrow=</span><span class="dv">2</span>,</span>
<span id="cb143-5"><a href="network-analysis.html#cb143-5" tabindex="-1"></a>                   <span class="at">ncol=</span><span class="dv">2</span>,</span>
<span id="cb143-6"><a href="network-analysis.html#cb143-6" tabindex="-1"></a>                   <span class="at">legend.label =</span> <span class="st">&quot;gene expression&quot;</span>,</span>
<span id="cb143-7"><a href="network-analysis.html#cb143-7" tabindex="-1"></a>                   <span class="at">boxplot=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-77-1.png" width="672" /></p>
<p>We can visualize the different TFs in a bubble plot:</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="network-analysis.html#cb144-1" tabindex="-1"></a><span class="fu">plotBubble</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb144-2"><a href="network-analysis.html#cb144-2" tabindex="-1"></a>           <span class="at">tf =</span> tfs_interest,</span>
<span id="cb144-3"><a href="network-analysis.html#cb144-3" tabindex="-1"></a>           <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>cell_type,</span>
<span id="cb144-4"><a href="network-analysis.html#cb144-4" tabindex="-1"></a>           <span class="at">pval.type =</span> <span class="st">&quot;some&quot;</span>, </span>
<span id="cb144-5"><a href="network-analysis.html#cb144-5" tabindex="-1"></a>           <span class="at">direction =</span> <span class="st">&quot;up&quot;</span>, </span>
<span id="cb144-6"><a href="network-analysis.html#cb144-6" tabindex="-1"></a>           <span class="at">logvalues =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-78-1.png" width="672" /></p>
<p>We visualize the top differential TFs based on activity.</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="network-analysis.html#cb145-1" tabindex="-1"></a><span class="fu">plotBubble</span>(<span class="at">activity_matrix =</span> score.combine, </span>
<span id="cb145-2"><a href="network-analysis.html#cb145-2" tabindex="-1"></a>           <span class="at">tf =</span> markers.sig<span class="sc">$</span>tf, </span>
<span id="cb145-3"><a href="network-analysis.html#cb145-3" tabindex="-1"></a>           GeneExpressionMatrix<span class="sc">$</span>cell_type,</span>
<span id="cb145-4"><a href="network-analysis.html#cb145-4" tabindex="-1"></a>           <span class="at">pval.type =</span> <span class="st">&quot;some&quot;</span>, </span>
<span id="cb145-5"><a href="network-analysis.html#cb145-5" tabindex="-1"></a>           <span class="at">direction =</span> <span class="st">&quot;up&quot;</span>, </span>
<span id="cb145-6"><a href="network-analysis.html#cb145-6" tabindex="-1"></a>           <span class="at">logvalues =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-79-1.png" width="672" /></p>
</div>
<div id="perform-geneset-enrichment" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Perform geneset enrichment<a href="network-analysis.html#perform-geneset-enrichment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Sometimes we are interested in knowing what pathways are enriched in the regulon of a particular TF. We can perform gene set enrichment using the enricher function from <a href="https://yulab-smu.top/biomedical-knowledge-mining-book/index.html">clusterProfiler</a>.</p>
<p>Here we first download Hallmark and C2 signatures from MSigDB and then perform gene set enrichment of the known lineage factors. As expected, EBF1 shows enrichment for pathways associated with B cells, and TBX21 with T cells, EOMES with NK cells, and SPI1 with myeloid cells.</p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="network-analysis.html#cb146-1" tabindex="-1"></a><span class="co">#retrieve genesets</span></span>
<span id="cb146-2"><a href="network-analysis.html#cb146-2" tabindex="-1"></a>H <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">getGenesets</span>(<span class="at">org =</span> <span class="st">&quot;hsa&quot;</span>, </span>
<span id="cb146-3"><a href="network-analysis.html#cb146-3" tabindex="-1"></a>                                    <span class="at">db =</span> <span class="st">&quot;msigdb&quot;</span>, </span>
<span id="cb146-4"><a href="network-analysis.html#cb146-4" tabindex="-1"></a>                                    <span class="at">cat =</span> <span class="st">&quot;H&quot;</span>, </span>
<span id="cb146-5"><a href="network-analysis.html#cb146-5" tabindex="-1"></a>                                    <span class="at">gene.id.type =</span> <span class="st">&quot;SYMBOL&quot;</span> )</span>
<span id="cb146-6"><a href="network-analysis.html#cb146-6" tabindex="-1"></a>C2 <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">getGenesets</span>(<span class="at">org =</span> <span class="st">&quot;hsa&quot;</span>, </span>
<span id="cb146-7"><a href="network-analysis.html#cb146-7" tabindex="-1"></a>                                     <span class="at">db =</span> <span class="st">&quot;msigdb&quot;</span>, </span>
<span id="cb146-8"><a href="network-analysis.html#cb146-8" tabindex="-1"></a>                                     <span class="at">cat =</span> <span class="st">&quot;C2&quot;</span>,</span>
<span id="cb146-9"><a href="network-analysis.html#cb146-9" tabindex="-1"></a>                                     <span class="at">gene.id.type =</span> <span class="st">&quot;SYMBOL&quot;</span> )</span>
<span id="cb146-10"><a href="network-analysis.html#cb146-10" tabindex="-1"></a></span>
<span id="cb146-11"><a href="network-analysis.html#cb146-11" tabindex="-1"></a></span>
<span id="cb146-12"><a href="network-analysis.html#cb146-12" tabindex="-1"></a><span class="co">#combine genesets and convert genesets to be compatible with enricher</span></span>
<span id="cb146-13"><a href="network-analysis.html#cb146-13" tabindex="-1"></a>gs <span class="ot">&lt;-</span> <span class="fu">c</span>(H, C2)</span>
<span id="cb146-14"><a href="network-analysis.html#cb146-14" tabindex="-1"></a>gs.list <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind,<span class="fu">lapply</span>(<span class="fu">names</span>(gs), <span class="cf">function</span>(x) </span>
<span id="cb146-15"><a href="network-analysis.html#cb146-15" tabindex="-1"></a>  {<span class="fu">data.frame</span>(<span class="at">gs=</span>x, <span class="at">genes=</span>gs[[x]])}))</span>
<span id="cb146-16"><a href="network-analysis.html#cb146-16" tabindex="-1"></a></span>
<span id="cb146-17"><a href="network-analysis.html#cb146-17" tabindex="-1"></a>enrichresults <span class="ot">&lt;-</span> <span class="fu">regulonEnrich</span>(<span class="at">TF =</span> tfs_interest, </span>
<span id="cb146-18"><a href="network-analysis.html#cb146-18" tabindex="-1"></a>                               <span class="at">regulon =</span> regulon.w, </span>
<span id="cb146-19"><a href="network-analysis.html#cb146-19" tabindex="-1"></a>                               <span class="at">weight =</span> <span class="st">&quot;weight&quot;</span>,</span>
<span id="cb146-20"><a href="network-analysis.html#cb146-20" tabindex="-1"></a>                               <span class="at">weight_cutoff =</span> <span class="fl">0.1</span>, </span>
<span id="cb146-21"><a href="network-analysis.html#cb146-21" tabindex="-1"></a>                               <span class="at">genesets =</span> gs.list)</span>
<span id="cb146-22"><a href="network-analysis.html#cb146-22" tabindex="-1"></a></span>
<span id="cb146-23"><a href="network-analysis.html#cb146-23" tabindex="-1"></a><span class="co">#plot results</span></span>
<span id="cb146-24"><a href="network-analysis.html#cb146-24" tabindex="-1"></a><span class="fu">enrichPlot</span>(<span class="at">results =</span> enrichresults, <span class="at">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-80-1.png" width="1440" /></p>
</div>
<div id="diffNetwork" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Perform differential network analysis<a href="network-analysis.html#diffNetwork" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A second approach to investigate differential TF activity is to compare the weights of the target genes between groups. In this example, we are interested to compare the networks of B cells and CD4 memory T cells. We build an edge subtracted graph and calculate the degree centrality of the subtracted graph. We normalize centrality using the default square root function. The top 5 most positive TFs represent lineage factors more active in B cells whereas the bottom 5 TFs present lineage factors enriched in CD4+ memory T cells. We successfully identified the B factors EBF1, PAX5 and IRF4. The putative CD4 memory T cells factors are less understood and remain to be validated.</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="network-analysis.html#cb147-1" tabindex="-1"></a><span class="co"># construct a graph of the CD4.M and NK cells respectively</span></span>
<span id="cb147-2"><a href="network-analysis.html#cb147-2" tabindex="-1"></a>CD4.M_network <span class="ot">&lt;-</span> <span class="fu">buildGraph</span>(regulon.w, <span class="at">weights =</span> <span class="st">&quot;weight&quot;</span>, <span class="at">cluster=</span><span class="st">&quot;Memory CD4+ T&quot;</span>)</span>
<span id="cb147-3"><a href="network-analysis.html#cb147-3" tabindex="-1"></a>B_network <span class="ot">&lt;-</span> <span class="fu">buildGraph</span>(regulon.w, <span class="at">weights =</span> <span class="st">&quot;weight&quot;</span>, <span class="at">cluster=</span><span class="st">&quot;B&quot;</span>)</span>
<span id="cb147-4"><a href="network-analysis.html#cb147-4" tabindex="-1"></a></span>
<span id="cb147-5"><a href="network-analysis.html#cb147-5" tabindex="-1"></a><span class="co"># construct a difference graph</span></span>
<span id="cb147-6"><a href="network-analysis.html#cb147-6" tabindex="-1"></a>diff_graph <span class="ot">&lt;-</span> <span class="fu">buildDiffGraph</span>(B_network,CD4.M_network, <span class="at">abs_diff =</span> <span class="cn">FALSE</span>)</span>
<span id="cb147-7"><a href="network-analysis.html#cb147-7" tabindex="-1"></a>diff_graph <span class="ot">&lt;-</span> <span class="fu">addCentrality</span>(diff_graph)</span>
<span id="cb147-8"><a href="network-analysis.html#cb147-8" tabindex="-1"></a>diff_graph <span class="ot">&lt;-</span> <span class="fu">normalizeCentrality</span>(diff_graph)</span>
<span id="cb147-9"><a href="network-analysis.html#cb147-9" tabindex="-1"></a>rank_table <span class="ot">&lt;-</span> <span class="fu">rankTfs</span>(diff_graph)</span>
<span id="cb147-10"><a href="network-analysis.html#cb147-10" tabindex="-1"></a></span>
<span id="cb147-11"><a href="network-analysis.html#cb147-11" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb147-12"><a href="network-analysis.html#cb147-12" tabindex="-1"></a><span class="fu">ggplot</span>(rank_table, <span class="fu">aes</span>(<span class="at">x =</span> rank, <span class="at">y =</span> centrality)) <span class="sc">+</span></span>
<span id="cb147-13"><a href="network-analysis.html#cb147-13" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb147-14"><a href="network-analysis.html#cb147-14" tabindex="-1"></a>    ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span> <span class="fu">rbind</span>(<span class="fu">head</span>(rank_table, <span class="dv">10</span>), </span>
<span id="cb147-15"><a href="network-analysis.html#cb147-15" tabindex="-1"></a>                                          <span class="fu">tail</span>(rank_table, <span class="dv">10</span>)), </span>
<span id="cb147-16"><a href="network-analysis.html#cb147-16" tabindex="-1"></a>                             <span class="fu">aes</span>(<span class="at">label =</span> tf), </span>
<span id="cb147-17"><a href="network-analysis.html#cb147-17" tabindex="-1"></a>                             <span class="at">nudge_x =</span> <span class="dv">0</span>, <span class="at">nudge_y =</span> <span class="dv">0</span>, <span class="at">box.padding =</span> <span class="fl">0.5</span>, <span class="at">max.overlaps =</span> <span class="cn">Inf</span>) <span class="sc">+</span></span>
<span id="cb147-18"><a href="network-analysis.html#cb147-18" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span> <span class="fu">ggtitle</span> (<span class="st">&quot;differential TFs (B-CD4.M) ranked by degree centrality&quot;</span>)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-81-1.png" width="672" /></p>
</div>
<div id="find-interaction-partners" class="section level2 hasAnchor" number="7.5">
<h2><span class="header-section-number">7.5</span> Find interaction partners<a href="network-analysis.html#find-interaction-partners" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Sometimes, we are interested to identify interaction partners of the TFs of interest. This can be achieved by comparing the overlap of regulons and identify the most similar TFs by Jaccard similarity. To illustrate this function, we take a look at the top most similar 20 TFs to EBF1, and we successfully identify PAX5 as the most similar TF. Both PAX5 and EBF1 are important factors for B cell development (<a href="https://www.nature.com/articles/ni.2641" class="uri">https://www.nature.com/articles/ni.2641</a>).</p>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="network-analysis.html#cb148-1" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb148-2"><a href="network-analysis.html#cb148-2" tabindex="-1"></a><span class="co"># construct a graph of the preB cells</span></span>
<span id="cb148-3"><a href="network-analysis.html#cb148-3" tabindex="-1"></a>B_network <span class="ot">&lt;-</span> <span class="fu">buildGraph</span>(regulon.w, <span class="at">weights =</span> <span class="st">&quot;weight&quot;</span>, <span class="at">cluster=</span><span class="st">&quot;B&quot;</span>)</span>
<span id="cb148-4"><a href="network-analysis.html#cb148-4" tabindex="-1"></a></span>
<span id="cb148-5"><a href="network-analysis.html#cb148-5" tabindex="-1"></a><span class="co"># compute a similarity matrix of all TFs</span></span>
<span id="cb148-6"><a href="network-analysis.html#cb148-6" tabindex="-1"></a>similarity_score <span class="ot">&lt;-</span> <span class="fu">calculateJaccardSimilarity</span>(B_network)</span>
<span id="cb148-7"><a href="network-analysis.html#cb148-7" tabindex="-1"></a></span>
<span id="cb148-8"><a href="network-analysis.html#cb148-8" tabindex="-1"></a><span class="co"># Focus on EBF1</span></span>
<span id="cb148-9"><a href="network-analysis.html#cb148-9" tabindex="-1"></a>similarity_score_EBF1 <span class="ot">&lt;-</span> similarity_score[, <span class="st">&quot;EBF1&quot;</span>]</span>
<span id="cb148-10"><a href="network-analysis.html#cb148-10" tabindex="-1"></a>similarity_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">similarity =</span> <span class="fu">head</span>(<span class="fu">sort</span>(similarity_score_EBF1, <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>),</span>
<span id="cb148-11"><a href="network-analysis.html#cb148-11" tabindex="-1"></a>                            <span class="at">TF =</span> <span class="fu">names</span>(<span class="fu">head</span>(<span class="fu">sort</span>(similarity_score_EBF1, <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>)))</span>
<span id="cb148-12"><a href="network-analysis.html#cb148-12" tabindex="-1"></a></span>
<span id="cb148-13"><a href="network-analysis.html#cb148-13" tabindex="-1"></a>similarity_df<span class="sc">$</span>TF <span class="ot">&lt;-</span> <span class="fu">factor</span>(similarity_df<span class="sc">$</span>TF, <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(similarity_df<span class="sc">$</span>TF)))</span>
<span id="cb148-14"><a href="network-analysis.html#cb148-14" tabindex="-1"></a></span>
<span id="cb148-15"><a href="network-analysis.html#cb148-15" tabindex="-1"></a><span class="co"># plot top TFs most similar to EBF1</span></span>
<span id="cb148-16"><a href="network-analysis.html#cb148-16" tabindex="-1"></a>topTFplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(similarity_df, <span class="fu">aes</span>(<span class="at">x=</span>TF, <span class="at">y=</span>similarity)) <span class="sc">+</span></span>
<span id="cb148-17"><a href="network-analysis.html#cb148-17" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb148-18"><a href="network-analysis.html#cb148-18" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb148-19"><a href="network-analysis.html#cb148-19" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;EBF1 similarity&quot;</span>) <span class="sc">+</span></span>
<span id="cb148-20"><a href="network-analysis.html#cb148-20" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb148-21"><a href="network-analysis.html#cb148-21" tabindex="-1"></a></span>
<span id="cb148-22"><a href="network-analysis.html#cb148-22" tabindex="-1"></a><span class="fu">print</span>(topTFplot)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-82-1.png" width="672" /></p>
<p>In order to convince ourselves that our differential network is statistically significant, we permute the edges and obtain a background graph from averaging many iterations. Here, we plot the differential network graph subtracted by the average edge weights of the permuted graphs.</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="network-analysis.html#cb149-1" tabindex="-1"></a><span class="co"># create a permuted graph by rewiring the edges 100 times</span></span>
<span id="cb149-2"><a href="network-analysis.html#cb149-2" tabindex="-1"></a>permute_matrix <span class="ot">&lt;-</span> <span class="fu">permuteGraph</span>(B_network, <span class="st">&quot;EBF1&quot;</span>, <span class="dv">100</span>, <span class="at">p=</span><span class="dv">1</span>)</span>
<span id="cb149-3"><a href="network-analysis.html#cb149-3" tabindex="-1"></a>permute_matrix <span class="ot">&lt;-</span> permute_matrix[<span class="fu">names</span>(similarity_score_EBF1),]</span>
<span id="cb149-4"><a href="network-analysis.html#cb149-4" tabindex="-1"></a>diff_matrix <span class="ot">&lt;-</span> similarity_score_EBF1<span class="sc">-</span><span class="fu">rowMeans</span>(permute_matrix)</span>
<span id="cb149-5"><a href="network-analysis.html#cb149-5" tabindex="-1"></a></span>
<span id="cb149-6"><a href="network-analysis.html#cb149-6" tabindex="-1"></a>diff_matrix_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">similarity =</span> <span class="fu">head</span>(<span class="fu">sort</span>(diff_matrix, <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>),</span>
<span id="cb149-7"><a href="network-analysis.html#cb149-7" tabindex="-1"></a>                             <span class="at">TF =</span> <span class="fu">names</span>(<span class="fu">head</span>(<span class="fu">sort</span>(diff_matrix, <span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>)))</span>
<span id="cb149-8"><a href="network-analysis.html#cb149-8" tabindex="-1"></a></span>
<span id="cb149-9"><a href="network-analysis.html#cb149-9" tabindex="-1"></a>diff_matrix_df<span class="sc">$</span>TF <span class="ot">&lt;-</span> <span class="fu">factor</span>(diff_matrix_df<span class="sc">$</span>TF, <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(diff_matrix_df<span class="sc">$</span>TF)))</span>
<span id="cb149-10"><a href="network-analysis.html#cb149-10" tabindex="-1"></a></span>
<span id="cb149-11"><a href="network-analysis.html#cb149-11" tabindex="-1"></a><span class="co"># plot top TFs most similar to EBF1</span></span>
<span id="cb149-12"><a href="network-analysis.html#cb149-12" tabindex="-1"></a>topTFplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(diff_matrix_df, <span class="fu">aes</span>(<span class="at">x=</span>TF, <span class="at">y=</span>similarity)) <span class="sc">+</span></span>
<span id="cb149-13"><a href="network-analysis.html#cb149-13" tabindex="-1"></a>            <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb149-14"><a href="network-analysis.html#cb149-14" tabindex="-1"></a>            <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb149-15"><a href="network-analysis.html#cb149-15" tabindex="-1"></a>            <span class="fu">ggtitle</span>(<span class="st">&quot;background subtracted EBF1 similarity &quot;</span>) <span class="sc">+</span></span>
<span id="cb149-16"><a href="network-analysis.html#cb149-16" tabindex="-1"></a>            <span class="fu">theme_classic</span>()</span>
<span id="cb149-17"><a href="network-analysis.html#cb149-17" tabindex="-1"></a><span class="fu">print</span>(topTFplot)</span></code></pre></div>
<p><img src="Epiregulon-documentation_files/figure-html/unnamed-chunk-83-1.png" width="672" /></p>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="network-analysis.html#cb150-1" tabindex="-1"></a><span class="co"># obtain empirical p-values</span></span>
<span id="cb150-2"><a href="network-analysis.html#cb150-2" tabindex="-1"></a>p_matrix <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">apply</span>(permute_matrix, <span class="dv">2</span>, <span class="cf">function</span>(x) {x <span class="sc">&gt;</span> similarity_score_EBF1}))</span>
<span id="cb150-3"><a href="network-analysis.html#cb150-3" tabindex="-1"></a>p_matrix[<span class="fu">names</span>(<span class="fu">head</span>(<span class="fu">sort</span>(diff_matrix,<span class="at">decreasing =</span> <span class="cn">TRUE</span>),<span class="dv">20</span>))]</span></code></pre></div>
<pre><code>##   PAX5 BCL11A POU2F2   RELB   MTA3  MEF2A   TCF4   TCF3  KMT2D    NBN NFATC3   ETV6 TRIM22 BCLAF1  NCOR2   SKIL  FOXO1    EED 
##      0      0      0      0      0      0      0      0      0      0      0      0      0      0      0      0      0      0 
## ARID3A   NFIC 
##      0      0</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="calculation-of-the-tf-activity.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="archr-workflow-and-different-weight-methods.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/xiaosai/epiregulon.book/edit/master/Graph_construction.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["Epiregulon documentation.pdf", "Epiregulon documentation.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
