<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Refinement of gene regulatory network | Epiregulon documentation</title>
  <meta name="description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="generator" content="bookdown 0.46 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Refinement of gene regulatory network | Epiregulon documentation" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Refinement of gene regulatory network | Epiregulon documentation" />
  
  <meta name="twitter:description" content="<p>This book presents the Epiregulon package. We provide the tutorials showing different
scenarions in which the package can be usd to analyse single-cell data.</p>" />
  

<meta name="author" content="Xiaosai Yao, Tomasz Włodarczyk" />


<meta name="date" content="2025-12-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="construction.html"/>
<link rel="next" href="calculation-of-the-tf-activity.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Epiregulon documentation</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="installation.html"><a href="installation.html"><i class="fa fa-check"></i><b>1</b> Installation</a>
<ul>
<li class="chapter" data-level="1.1" data-path="installation.html"><a href="installation.html#install-from-github"><i class="fa fa-check"></i><b>1.1</b> Install from GitHub</a></li>
<li class="chapter" data-level="1.2" data-path="installation.html"><a href="installation.html#install-from-bioconductor"><i class="fa fa-check"></i><b>1.2</b> Install from Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="quick-start.html"><a href="quick-start.html"><i class="fa fa-check"></i><b>2</b> Quick start</a>
<ul>
<li class="chapter" data-level="2.1" data-path="quick-start.html"><a href="quick-start.html#prepare-data"><i class="fa fa-check"></i><b>2.1</b> Prepare data</a></li>
<li class="chapter" data-level="2.2" data-path="quick-start.html"><a href="quick-start.html#retrieve-bulk-tf-chip-seq-binding-sites"><i class="fa fa-check"></i><b>2.2</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="2.3" data-path="quick-start.html"><a href="quick-start.html#link-atac-seq-peaks-to-target-genes"><i class="fa fa-check"></i><b>2.3</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="2.4" data-path="quick-start.html"><a href="quick-start.html#add-tf-motif-binding-to-peaks"><i class="fa fa-check"></i><b>2.4</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="2.5" data-path="quick-start.html"><a href="quick-start.html#generate-regulons"><i class="fa fa-check"></i><b>2.5</b> Generate regulons</a></li>
<li class="chapter" data-level="2.6" data-path="quick-start.html"><a href="quick-start.html#prune-network-recommended"><i class="fa fa-check"></i><b>2.6</b> Prune network (recommended)</a></li>
<li class="chapter" data-level="2.7" data-path="quick-start.html"><a href="quick-start.html#annotate-regulon"><i class="fa fa-check"></i><b>2.7</b> Annotate regulon</a></li>
<li class="chapter" data-level="2.8" data-path="quick-start.html"><a href="quick-start.html#calculate-tf-activity"><i class="fa fa-check"></i><b>2.8</b> Calculate TF activity</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-preparation.html"><a href="data-preparation.html"><i class="fa fa-check"></i><b>3</b> Data preparation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-preparation.html"><a href="data-preparation.html#construct-a-singlecellexperiment-object"><i class="fa fa-check"></i><b>3.1</b> Construct a SingleCellExperiment object</a></li>
<li class="chapter" data-level="3.2" data-path="data-preparation.html"><a href="data-preparation.html#start-from-an-archr-object"><i class="fa fa-check"></i><b>3.2</b> Start from an ArchR object</a></li>
<li class="chapter" data-level="3.3" data-path="data-preparation.html"><a href="data-preparation.html#start-from-a-seuratsignac-object"><i class="fa fa-check"></i><b>3.3</b> Start from a Seurat/Signac object</a></li>
<li class="chapter" data-level="3.4" data-path="data-preparation.html"><a href="data-preparation.html#start-from-anndata"><i class="fa fa-check"></i><b>3.4</b> Start from AnnData</a></li>
<li class="chapter" data-level="3.5" data-path="data-preparation.html"><a href="data-preparation.html#start-from-10x-data-formats"><i class="fa fa-check"></i><b>3.5</b> Start from 10x data formats</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="data-preparation.html"><a href="data-preparation.html#read-from-.h5-file"><i class="fa fa-check"></i><b>3.5.1</b> Read from .h5 file</a></li>
<li class="chapter" data-level="3.5.2" data-path="data-preparation.html"><a href="data-preparation.html#read-directly-from-a-10x-directory"><i class="fa fa-check"></i><b>3.5.2</b> Read directly from a 10x directory</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="data-preparation.html"><a href="data-preparation.html#read-directly-from-csv"><i class="fa fa-check"></i><b>3.6</b> Read directly from CSV</a></li>
<li class="chapter" data-level="3.7" data-path="data-preparation.html"><a href="data-preparation.html#important-points"><i class="fa fa-check"></i><b>3.7</b> Important points</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="construction.html"><a href="construction.html"><i class="fa fa-check"></i><b>4</b> Construction of the gene regulatory network</a>
<ul>
<li class="chapter" data-level="4.1" data-path="construction.html"><a href="construction.html#download-and-format-dataset"><i class="fa fa-check"></i><b>4.1</b> Download and format dataset</a></li>
<li class="chapter" data-level="4.2" data-path="construction.html"><a href="construction.html#retrieve-tf-occupancy-data"><i class="fa fa-check"></i><b>4.2</b> Retrieve TF occupancy data</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="construction.html"><a href="construction.html#chip-seq-data"><i class="fa fa-check"></i><b>4.2.1</b> ChIP-seq data</a></li>
<li class="chapter" data-level="4.2.2" data-path="construction.html"><a href="construction.html#motif-information"><i class="fa fa-check"></i><b>4.2.2</b> Motif information</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="construction.html"><a href="construction.html#link-atacseq-peaks-to-target-genes"><i class="fa fa-check"></i><b>4.3</b> Link ATACseq peaks to target genes</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="construction.html"><a href="construction.html#optimization"><i class="fa fa-check"></i><b>4.3.1</b> Optimize the number of metacells (optional)</a></li>
<li class="chapter" data-level="4.3.2" data-path="construction.html"><a href="construction.html#distance_window"><i class="fa fa-check"></i><b>4.3.2</b> Find peak-target gene links within a defined distance</a></li>
<li class="chapter" data-level="4.3.3" data-path="construction.html"><a href="construction.html#link-peaks-to-the-nearest-genes"><i class="fa fa-check"></i><b>4.3.3</b> Link peaks to the nearest genes</a></li>
<li class="chapter" data-level="4.3.4" data-path="construction.html"><a href="construction.html#filter-peak-target-gene-links"><i class="fa fa-check"></i><b>4.3.4</b> Filter peak-target gene links</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="construction.html"><a href="construction.html#add-tf-occupancy-to-peaks"><i class="fa fa-check"></i><b>4.4</b> Add TF occupancy to peaks</a></li>
<li class="chapter" data-level="4.5" data-path="construction.html"><a href="construction.html#generate-regulons-1"><i class="fa fa-check"></i><b>4.5</b> Generate regulons</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="refinement.html"><a href="refinement.html"><i class="fa fa-check"></i><b>5</b> Refinement of gene regulatory network</a>
<ul>
<li class="chapter" data-level="5.1" data-path="refinement.html"><a href="refinement.html#pruning"><i class="fa fa-check"></i><b>5.1</b> Network pruning</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="refinement.html"><a href="refinement.html#cell_aggregation"><i class="fa fa-check"></i><b>5.1.1</b> Aggregate cells</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="refinement.html"><a href="refinement.html#motif_score"><i class="fa fa-check"></i><b>5.2</b> Annotate with TF motifs</a></li>
<li class="chapter" data-level="5.3" data-path="refinement.html"><a href="refinement.html#annotate-with-log-fold-changes"><i class="fa fa-check"></i><b>5.3</b> Annotate with log fold changes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="calculation-of-the-tf-activity.html"><a href="calculation-of-the-tf-activity.html"><i class="fa fa-check"></i><b>6</b> Calculation of the TF activity</a>
<ul>
<li class="chapter" data-level="6.1" data-path="calculation-of-the-tf-activity.html"><a href="calculation-of-the-tf-activity.html#addWeights"><i class="fa fa-check"></i><b>6.1</b> Add weights</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="calculation-of-the-tf-activity.html"><a href="calculation-of-the-tf-activity.html#aggregate-cells"><i class="fa fa-check"></i><b>6.1.1</b> Aggregate cells</a></li>
<li class="chapter" data-level="6.1.2" data-path="calculation-of-the-tf-activity.html"><a href="calculation-of-the-tf-activity.html#using-chromatin-accessibility-data"><i class="fa fa-check"></i><b>6.1.2</b> Using chromatin accessibility data</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="calculation-of-the-tf-activity.html"><a href="calculation-of-the-tf-activity.html#calculate-tf-activity-1"><i class="fa fa-check"></i><b>6.2</b> Calculate TF activity</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="network-analysis.html"><a href="network-analysis.html"><i class="fa fa-check"></i><b>7</b> Network analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="network-analysis.html"><a href="network-analysis.html#differential-tf-activity"><i class="fa fa-check"></i><b>7.1</b> Differential TF activity</a></li>
<li class="chapter" data-level="7.2" data-path="network-analysis.html"><a href="network-analysis.html#diff_activity_vis"><i class="fa fa-check"></i><b>7.2</b> Visualize TF activities</a></li>
<li class="chapter" data-level="7.3" data-path="network-analysis.html"><a href="network-analysis.html#geneset-enrichment"><i class="fa fa-check"></i><b>7.3</b> Geneset enrichment</a></li>
<li class="chapter" data-level="7.4" data-path="network-analysis.html"><a href="network-analysis.html#diffNetwork"><i class="fa fa-check"></i><b>7.4</b> Differential network analysis</a></li>
<li class="chapter" data-level="7.5" data-path="network-analysis.html"><a href="network-analysis.html#find-interaction-partners"><i class="fa fa-check"></i><b>7.5</b> Find interaction partners</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html"><i class="fa fa-check"></i><b>8</b> ArchR workflow and different weight methods</a>
<ul>
<li class="chapter" data-level="8.1" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#prepare-data-1"><i class="fa fa-check"></i><b>8.1</b> Prepare data</a></li>
<li class="chapter" data-level="8.2" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#load-archr-project"><i class="fa fa-check"></i><b>8.2</b> Load ArchR project</a></li>
<li class="chapter" data-level="8.3" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#retrieve-matrices-from-archr-project"><i class="fa fa-check"></i><b>8.3</b> Retrieve matrices from ArchR project</a></li>
<li class="chapter" data-level="8.4" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#retrieve-bulk-tf-chip-seq-binding-sites-1"><i class="fa fa-check"></i><b>8.4</b> Retrieve bulk TF ChIP-seq binding sites</a></li>
<li class="chapter" data-level="8.5" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#link-atac-seq-peaks-to-target-genes-1"><i class="fa fa-check"></i><b>8.5</b> Link ATAC-seq peaks to target genes</a></li>
<li class="chapter" data-level="8.6" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#add-tf-motif-binding-to-peaks-1"><i class="fa fa-check"></i><b>8.6</b> Add TF motif binding to peaks</a></li>
<li class="chapter" data-level="8.7" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#generate-regulons-2"><i class="fa fa-check"></i><b>8.7</b> Generate regulons</a></li>
<li class="chapter" data-level="8.8" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#optional-annotate-with-tf-motifs"><i class="fa fa-check"></i><b>8.8</b> (Optional) Annotate with TF motifs</a></li>
<li class="chapter" data-level="8.9" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#prune-network"><i class="fa fa-check"></i><b>8.9</b> Prune network</a></li>
<li class="chapter" data-level="8.10" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#weights_prostate"><i class="fa fa-check"></i><b>8.10</b> Add Weights</a></li>
<li class="chapter" data-level="8.11" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#calculate-tf-activity-2"><i class="fa fa-check"></i><b>8.11</b> Calculate TF activity</a></li>
<li class="chapter" data-level="8.12" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#perform-differential-activity"><i class="fa fa-check"></i><b>8.12</b> Perform differential activity</a></li>
<li class="chapter" data-level="8.13" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#visualize-the-results"><i class="fa fa-check"></i><b>8.13</b> Visualize the results</a></li>
<li class="chapter" data-level="8.14" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#geneset-enrichment-1"><i class="fa fa-check"></i><b>8.14</b> Geneset enrichment</a></li>
<li class="chapter" data-level="8.15" data-path="archr-workflow-and-different-weight-methods.html"><a href="archr-workflow-and-different-weight-methods.html#differential-network-analysis-diff_network"><i class="fa fa-check"></i><b>8.15</b> Differential network analysis {diff_network}</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html"><i class="fa fa-check"></i><b>9</b> Single modality: scRNA-seq only</a>
<ul>
<li class="chapter" data-level="9.1" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#load-regulon"><i class="fa fa-check"></i><b>9.1</b> Load regulon</a></li>
<li class="chapter" data-level="9.2" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#load-scrna-seq-data"><i class="fa fa-check"></i><b>9.2</b> Load scRNA-seq data</a></li>
<li class="chapter" data-level="9.3" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#calculate-activity"><i class="fa fa-check"></i><b>9.3</b> Calculate activity</a></li>
<li class="chapter" data-level="9.4" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#perform-differential-activity-1"><i class="fa fa-check"></i><b>9.4</b> Perform differential activity</a></li>
<li class="chapter" data-level="9.5" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#visualize-activity"><i class="fa fa-check"></i><b>9.5</b> Visualize activity</a></li>
<li class="chapter" data-level="9.6" data-path="single-modality-scrna-seq-only.html"><a href="single-modality-scrna-seq-only.html#pathway-enrichment"><i class="fa fa-check"></i><b>9.6</b> Pathway enrichment</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>10</b> Session Info</a></li>
<li class="chapter" data-level="11" data-path="contact.html"><a href="contact.html"><i class="fa fa-check"></i><b>11</b> Contact</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Epiregulon documentation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="refinement" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">5</span> Refinement of gene regulatory network<a href="refinement.html#refinement" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The <code>regulon</code> object constructed in the previous <a href="construction.html#construction">chapter</a> contains all possible links between transcription factors and target genes. As a result, the GRN can be large and may contain some false connections. To refine it, we need to apply filters that retain only those connections supported by our dataset. One such filter has already been applied in the <a href="construction.html#distance_window"><code>calculateP2G</code> function</a>, which not only searches for regulatory elements within a specified distance from each target gene, but also computes the empirical p-value of the correlation for each target gene–regulatory element pair. Pairs are retained if their p-value is below the threshold specified by the <code>cutoff_sig</code> argument.</p>
<p>In this chapter, we introduce three additional filters. The <code>pruneRegulon</code> function filters the GRN based on the co-occurrence of TF gene expression, RE chromatin accessibility and target gene expression. The <code>addMotifScore</code> function annotates regulons for the presence of motifs. Lastly, the <code>addLogFC</code> function calculates changes in gene expression between conditions, so that only target genes showing differential expression are selected.</p>
<div id="pruning" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Network pruning<a href="refinement.html#pruning" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Epiregulon prunes the network by performing tests of independence on the observed number of cells jointly expressing transcription factor (<span class="math inline">\(TF\)</span>), regulatory element (<span class="math inline">\(RE\)</span>) and target gene (<span class="math inline">\(TG\)</span>). The thresholds at which the features are considered expressed are defined by the <code>peak_cutoff</code> and <code>exp_cutoff</code> arguments. We implement two tests, the binomial test and the chi-square test. In the binomial test, the expected probability is <span class="math inline">\(P(TF, RE) \cdot P(TG)\)</span>, and the number of trials is the total number of cells (or metacells if we set <code>aggregateCells=TRUE</code>), and the observed successes is the number of cells jointly expressing all three elements. In the chi-square test, the expected probability for having all 3 elements active is also <span class="math inline">\(P(TF, RE) \cdot P(TG)\)</span>. The observed cell count for the active category is the number of cells jointly expressing all three elements (<span class="math inline">\(n_{triple}\)</span>).</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="refinement.html#cb153-1" tabindex="-1"></a>pruned.regulon <span class="ot">&lt;-</span> <span class="fu">pruneRegulon</span>(<span class="at">expMatrix =</span> GeneExpressionMatrix,</span>
<span id="cb153-2"><a href="refinement.html#cb153-2" tabindex="-1"></a>                               <span class="at">exp_assay =</span> <span class="st">&quot;normalizedCounts&quot;</span>,</span>
<span id="cb153-3"><a href="refinement.html#cb153-3" tabindex="-1"></a>                               <span class="at">peakMatrix =</span> PeakMatrix,</span>
<span id="cb153-4"><a href="refinement.html#cb153-4" tabindex="-1"></a>                               <span class="at">peak_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb153-5"><a href="refinement.html#cb153-5" tabindex="-1"></a>                               <span class="at">regulon =</span> regulon,</span>
<span id="cb153-6"><a href="refinement.html#cb153-6" tabindex="-1"></a>                               <span class="at">prune_value =</span> <span class="st">&quot;pval&quot;</span>,</span>
<span id="cb153-7"><a href="refinement.html#cb153-7" tabindex="-1"></a>                               <span class="at">regulon_cutoff =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>## pruning network with chi.sq tests using a regulon cutoff of pval&lt;0.05</code></pre>
<pre><code>## pruning regulons</code></pre>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="refinement.html#cb156-1" tabindex="-1"></a>pruned.regulon[,<span class="fu">c</span>(<span class="st">&quot;idxATAC&quot;</span>, <span class="st">&quot;target&quot;</span>, <span class="st">&quot;tf&quot;</span>, <span class="st">&quot;corr&quot;</span>, <span class="st">&quot;pval&quot;</span>, <span class="st">&quot;stats&quot;</span>)]</span></code></pre></div>
<pre><code>## DataFrame with 2687769 rows and 6 columns
##           idxATAC   target          tf      corr        pval    stats
##         &lt;integer&gt;  &lt;array&gt; &lt;character&gt;  &lt;matrix&gt;    &lt;matrix&gt; &lt;matrix&gt;
## 1             135 TNFRSF18        ADNP  0.754387 1.15983e-04 14.85689
## 2             266     NADK        ADNP  0.801851 2.14101e-07 26.90142
## 3             991      CA6        ADNP  0.888973 2.09927e-03  9.46065
## 4            1103 SLC25A33        ADNP -0.712694 1.01188e-03 10.80569
## 5            1204    KIF1B        ADNP  0.742701 3.18769e-21 89.42324
## ...           ...      ...         ...       ...         ...      ...
## 2687765    155577     SCO2        ZXDC -0.543323 3.06126e-02  4.67456
## 2687766    155577     TYMP        ZXDC -0.544313 4.48926e-04 12.31669
## 2687767    155577    ODF3B        ZXDC -0.546970 6.00511e-03  7.54877
## 2687768    157674     GCNA        ZXDC  0.725899 4.54914e-02  4.00033
## 2687769    158053     RADX        ZXDC  0.689188 3.72567e-06 21.40105</code></pre>
<div id="cell_aggregation" class="section level3 hasAnchor" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Aggregate cells<a href="refinement.html#cell_aggregation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <code>pruneRegulon</code> function also supports aggregation of cells into metacells. This can substantially speed up GRN refinement if the dataset is large (&gt;100K cells). Ideally, cell aggregation should also reduce noise while still maintaining the underlying biological signals. Since the data is less sparse after aggregation, users may want to adjust the feature cutoffs to better differentiate between cell aggregates with low and high expression. Setting <code>exp_cutoff</code> and <code>peak_cutoff</code> to <code>NULL</code> allows <code>pruneRegulon</code> to calculate the mean values for each feature and use these as the thresholds.</p>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="refinement.html#cb158-1" tabindex="-1"></a>pruned.regulon.aggr <span class="ot">&lt;-</span> <span class="fu">pruneRegulon</span>(<span class="at">expMatrix =</span> GeneExpressionMatrix,</span>
<span id="cb158-2"><a href="refinement.html#cb158-2" tabindex="-1"></a>                               <span class="at">exp_assay =</span> <span class="st">&quot;normalizedCounts&quot;</span>,</span>
<span id="cb158-3"><a href="refinement.html#cb158-3" tabindex="-1"></a>                               <span class="at">peakMatrix =</span> PeakMatrix,</span>
<span id="cb158-4"><a href="refinement.html#cb158-4" tabindex="-1"></a>                               <span class="at">peak_assay =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb158-5"><a href="refinement.html#cb158-5" tabindex="-1"></a>                               <span class="at">regulon =</span> regulon,</span>
<span id="cb158-6"><a href="refinement.html#cb158-6" tabindex="-1"></a>                               <span class="at">prune_value =</span> <span class="st">&quot;pval&quot;</span>,</span>
<span id="cb158-7"><a href="refinement.html#cb158-7" tabindex="-1"></a>                               <span class="at">regulon_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb158-8"><a href="refinement.html#cb158-8" tabindex="-1"></a>                               <span class="at">aggregateCells =</span> <span class="cn">TRUE</span>,</span>
<span id="cb158-9"><a href="refinement.html#cb158-9" tabindex="-1"></a>                               <span class="at">useDim =</span> <span class="st">&quot;LSI_RNA&quot;</span>,</span>
<span id="cb158-10"><a href="refinement.html#cb158-10" tabindex="-1"></a>                               <span class="at">cellNum =</span> <span class="dv">10</span>,</span>
<span id="cb158-11"><a href="refinement.html#cb158-11" tabindex="-1"></a>                               <span class="at">exp_cutoff =</span> <span class="cn">NULL</span>,</span>
<span id="cb158-12"><a href="refinement.html#cb158-12" tabindex="-1"></a>                               <span class="at">peak_cutoff =</span> <span class="cn">NULL</span>)</span></code></pre></div>
</div>
</div>
<div id="motif_score" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Annotate with TF motifs<a href="refinement.html#motif_score" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>So far, the gene regulatory network has been constructed exclusively from TF ChIP-seq data. Some users may want to further annotate regulatory elements with motifs. We provide an option to annotate peaks with motifs from the <a href="https://cisbp.ccbr.utoronto.ca/">CIS_BP</a> database. If no motifs are present for a particular factor (e.g., for co-factors or chromatin modifiers), <code>NA</code> is returned. If motifs exist for a factor and the regulatory element contains a motif, <code>1</code> is returned; if the regulatory element does not contain a motif, <code>0</code> is returned. Users can also provide their own motif annotation through the <code>pwms</code> argument.</p>
<p>It is important to note that filtering for the presence of motifs removes a large fraction of the target genes. Motifs are often present in a small subset of the ChIP-seq peaks (can be as low as 10%). Second, indirect TF binding, possibly through its interaction partners, may have a true biological function.</p>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="refinement.html#cb159-1" tabindex="-1"></a>pruned.regulon.motif <span class="ot">&lt;-</span> <span class="fu">addMotifScore</span>(<span class="at">regulon =</span> pruned.regulon,</span>
<span id="cb159-2"><a href="refinement.html#cb159-2" tabindex="-1"></a>                                      <span class="at">peaks =</span> <span class="fu">rowRanges</span>(PeakMatrix),</span>
<span id="cb159-3"><a href="refinement.html#cb159-3" tabindex="-1"></a>                                      <span class="at">species =</span> <span class="st">&quot;human&quot;</span>,</span>
<span id="cb159-4"><a href="refinement.html#cb159-4" tabindex="-1"></a>                                      <span class="at">genome =</span> <span class="st">&quot;hg38&quot;</span>)</span>
<span id="cb159-5"><a href="refinement.html#cb159-5" tabindex="-1"></a></span>
<span id="cb159-6"><a href="refinement.html#cb159-6" tabindex="-1"></a><span class="co"># filter out the rows where TF motif is not found in the peak region</span></span>
<span id="cb159-7"><a href="refinement.html#cb159-7" tabindex="-1"></a>pruned.regulon.motif <span class="ot">&lt;-</span> pruned.regulon.motif[<span class="fu">which</span>(pruned.regulon.motif<span class="sc">$</span>motif <span class="sc">==</span> <span class="dv">1</span>),]</span>
<span id="cb159-8"><a href="refinement.html#cb159-8" tabindex="-1"></a></span>
<span id="cb159-9"><a href="refinement.html#cb159-9" tabindex="-1"></a>pruned.regulon.motif[,<span class="fu">c</span>(<span class="st">&quot;idxATAC&quot;</span>, <span class="st">&quot;target&quot;</span>, <span class="st">&quot;tf&quot;</span>, <span class="st">&quot;corr&quot;</span>, <span class="st">&quot;p_val_peak_gene&quot;</span>, <span class="st">&quot;pval&quot;</span>, <span class="st">&quot;motif&quot;</span>)]</span></code></pre></div>
<pre><code>## DataFrame with 204629 rows and 7 columns
##          idxATAC   target          tf      corr p_val_peak_gene        pval
##        &lt;integer&gt;  &lt;array&gt; &lt;character&gt;  &lt;matrix&gt;        &lt;matrix&gt;    &lt;matrix&gt;
## 1            199    ACAP3         AHR -0.528286      0.03577411 2.44093e-02
## 2            723 TNFRSF25         AHR  0.753414      0.02278260 3.79686e-08
## 3           1100 SLC25A33         AHR -0.652737      0.00883763 2.41562e-08
## 4           3937      AK2         AHR -0.500420      0.04646013 2.82830e-03
## 5           3937   RNF19B         AHR -0.711628      0.00321543 1.14941e-02
## ...          ...      ...         ...       ...             ...         ...
## 204625    129706    HOXB2     ZSCAN16  0.653195      0.04945592 0.004321664
## 204626    137004     MIDN     ZSCAN16  0.721823      0.02935730 0.005802058
## 204627    157055      CFP     ZSCAN16  0.840835      0.00942513 0.009152565
## 204628     33086   GTPBP8      ZSCAN4  0.751812      0.02303227 0.000171032
## 204629    104541    ZNF84      ZSCAN4  0.711943      0.03174999 0.034884124
##            motif
##        &lt;numeric&gt;
## 1              1
## 2              1
## 3              1
## 4              1
## 5              1
## ...          ...
## 204625         1
## 204626         1
## 204627         1
## 204628         1
## 204629         1</code></pre>
</div>
<div id="annotate-with-log-fold-changes" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Annotate with log fold changes<a href="refinement.html#annotate-with-log-fold-changes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It is sometimes helpful to filter the regulons based on gene expression changes between two conditions. The <code>addLogFC</code> function is a wrapper around <code>scran::findMarkers</code> and adds extra columns of log changes to the regulon <code>DataFrame</code>. Users can specify the reference group in <code>logFC_ref</code> and comparison groups in <code>logFC_condition</code>. If these are not provided, log fold changes are calculated for every condition in the cluster labels against the rest of the conditions.</p>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="refinement.html#cb161-1" tabindex="-1"></a><span class="co"># create logcounts</span></span>
<span id="cb161-2"><a href="refinement.html#cb161-2" tabindex="-1"></a>GeneExpressionMatrix <span class="ot">&lt;-</span> scuttle<span class="sc">::</span><span class="fu">logNormCounts</span>(GeneExpressionMatrix)</span>
<span id="cb161-3"><a href="refinement.html#cb161-3" tabindex="-1"></a></span>
<span id="cb161-4"><a href="refinement.html#cb161-4" tabindex="-1"></a><span class="co"># add log fold changes which are calculated by taking the difference of the log counts</span></span>
<span id="cb161-5"><a href="refinement.html#cb161-5" tabindex="-1"></a>pruned.regulon <span class="ot">&lt;-</span> <span class="fu">addLogFC</span>(<span class="at">regulon =</span> pruned.regulon,</span>
<span id="cb161-6"><a href="refinement.html#cb161-6" tabindex="-1"></a>                           <span class="at">clusters =</span> GeneExpressionMatrix<span class="sc">$</span>cell_type,</span>
<span id="cb161-7"><a href="refinement.html#cb161-7" tabindex="-1"></a>                           <span class="at">expMatrix  =</span> GeneExpressionMatrix,</span>
<span id="cb161-8"><a href="refinement.html#cb161-8" tabindex="-1"></a>                           <span class="at">assay.type  =</span> <span class="st">&quot;logcounts&quot;</span>,</span>
<span id="cb161-9"><a href="refinement.html#cb161-9" tabindex="-1"></a>                           <span class="at">pval.type =</span> <span class="st">&quot;any&quot;</span>,</span>
<span id="cb161-10"><a href="refinement.html#cb161-10" tabindex="-1"></a>                           <span class="at">logFC_condition =</span> <span class="fu">unique</span>(GeneExpressionMatrix<span class="sc">$</span>cell_type))</span>
<span id="cb161-11"><a href="refinement.html#cb161-11" tabindex="-1"></a></span>
<span id="cb161-12"><a href="refinement.html#cb161-12" tabindex="-1"></a>pruned.regulon[,<span class="fu">c</span>(<span class="st">&quot;idxATAC&quot;</span>, <span class="st">&quot;target&quot;</span>, <span class="st">&quot;tf&quot;</span>, <span class="st">&quot;Memory CD8+ T.vs.rest.logFC&quot;</span>, <span class="st">&quot;B.vs.rest.FDR&quot;</span>)]</span></code></pre></div>
<pre><code>## DataFrame with 2687769 rows and 5 columns
##           idxATAC   target          tf Memory CD8+ T.vs.rest.logFC
##         &lt;integer&gt;  &lt;array&gt; &lt;character&gt;                   &lt;numeric&gt;
## 1             135 TNFRSF18        ADNP                  -0.0906689
## 2             266     NADK        ADNP                  -0.3582261
## 3             991      CA6        ADNP                  -0.3802831
## 4            1103 SLC25A33        ADNP                  -0.2311624
## 5            1204    KIF1B        ADNP                  -0.3791885
## ...           ...      ...         ...                         ...
## 2687765    155577     SCO2        ZXDC                  -0.6694588
## 2687766    155577     TYMP        ZXDC                  -2.3157151
## 2687767    155577    ODF3B        ZXDC                  -0.3380033
## 2687768    157674     GCNA        ZXDC                   0.0624762
## 2687769    158053     RADX        ZXDC                   0.0456558
##         B.vs.rest.FDR
##             &lt;numeric&gt;
## 1         2.12607e-23
## 2         1.86039e-71
## 3         2.38300e-91
## 4         5.77745e-26
## 5        2.15099e-133
## ...               ...
## 2687765  1.26531e-315
## 2687766   0.00000e+00
## 2687767  2.16867e-129
## 2687768   5.28466e-04
## 2687769   1.51853e-04</code></pre>
<p>TFs do not necessarily alter the expression of the genes next to where they bind, and thus filtering for target genes that do show differential expression can help refine the regulons. Since we do not have a defined experimental setup in this dataset, we retain target genes that show differential expression in any cell types vs. the rest.</p>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="refinement.html#cb163-1" tabindex="-1"></a>FDR <span class="ot">&lt;-</span> pruned.regulon[,<span class="fu">grep</span>(<span class="st">&quot;rest.FDR&quot;</span>, <span class="fu">colnames</span>(pruned.regulon ))]</span>
<span id="cb163-2"><a href="refinement.html#cb163-2" tabindex="-1"></a>FDR <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(FDR) <span class="sc">&lt;</span> <span class="fl">0.05</span></span>
<span id="cb163-3"><a href="refinement.html#cb163-3" tabindex="-1"></a>FDR <span class="ot">&lt;-</span> <span class="fu">apply</span>(FDR,<span class="dv">2</span>, as.numeric)</span>
<span id="cb163-4"><a href="refinement.html#cb163-4" tabindex="-1"></a></span>
<span id="cb163-5"><a href="refinement.html#cb163-5" tabindex="-1"></a>logFC <span class="ot">&lt;-</span> pruned.regulon[,<span class="fu">grep</span>(<span class="st">&quot;rest.logFC&quot;</span>, <span class="fu">colnames</span>(pruned.regulon ))]</span>
<span id="cb163-6"><a href="refinement.html#cb163-6" tabindex="-1"></a>logFC <span class="ot">&lt;-</span> <span class="fu">abs</span>(<span class="fu">as.matrix</span>(logFC)) <span class="sc">&gt;</span> <span class="fl">0.3</span></span>
<span id="cb163-7"><a href="refinement.html#cb163-7" tabindex="-1"></a>logFC <span class="ot">&lt;-</span> <span class="fu">apply</span>(logFC,<span class="dv">2</span>, as.numeric)</span>
<span id="cb163-8"><a href="refinement.html#cb163-8" tabindex="-1"></a></span>
<span id="cb163-9"><a href="refinement.html#cb163-9" tabindex="-1"></a>logFC_FDR <span class="ot">&lt;-</span> FDR<span class="sc">*</span>logFC</span></code></pre></div>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="refinement.html#cb164-1" tabindex="-1"></a>pruned.regulon.filtered <span class="ot">&lt;-</span> pruned.regulon[<span class="fu">which</span>(<span class="fu">rowSums</span>(logFC_FDR) <span class="sc">&gt;</span> <span class="dv">0</span>), ]</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="construction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="calculation-of-the-tf-activity.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/USERNAME/REPO/edit/BRANCH/Regulon_refinement.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["Epiregulon documentation.pdf", "Epiregulon documentation.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
